
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://GenomicsStandardsConsortium.github.io/mixs/0010009/">
      
      
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.18">
    
    
      
        <title>Checklist: MIMARKS specimen (MimarksC) - mixs</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.7e37652d.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#checklist-mimarks-specimen-mimarksc" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="mixs" class="md-header__button md-logo" aria-label="mixs" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            mixs
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Checklist: MIMARKS specimen (MimarksC)
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/GenomicsStandardsConsortium/mixs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="mixs" class="md-nav__button md-logo" aria-label="mixs" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    mixs
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GenomicsStandardsConsortium/mixs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Index
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../term_list/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Full term table
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../combinations/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Combinations
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../enumerations/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Enumerations
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#terms" class="md-nav__link">
    <span class="md-ellipsis">
      Terms
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#aliases" class="md-nav__link">
    <span class="md-ellipsis">
      Aliases
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#linkml-source" class="md-nav__link">
    <span class="md-ellipsis">
      LinkML Source
    </span>
  </a>
  
    <nav class="md-nav" aria-label="LinkML Source">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#direct" class="md-nav__link">
    <span class="md-ellipsis">
      Direct
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#induced" class="md-nav__link">
    <span class="md-ellipsis">
      Induced
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="checklist-mimarks-specimen-mimarksc">Checklist: MIMARKS specimen (MimarksC)</h1>
<p><em>Minimal Information about a Marker Sequence: specimen</em></p>
<h2 id="terms">Terms</h2>
<table>
<thead>
<tr>
<th>MIXS ID</th>
<th>Name</th>
<th>Cardinality and Range</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://w3id.org/mixs/0001107">MIXS:0001107</a></td>
<td><a href="../0001107/">samp_name</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>A local identifier or name that for the material sample used for extracting n...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000046">MIXS:0000046</a></td>
<td><a href="../0000046/">pcr_primers</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../String/">String</a></td>
<td>PCR primers that were used to amplify the sequence of the targeted gene, locu...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000038">MIXS:0000038</a></td>
<td><a href="../0000038/">nucl_acid_amp</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../String/">String</a></td>
<td>A link to a literature reference, electronic resource or a standard operating...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000045">MIXS:0000045</a></td>
<td><a href="../0000045/">target_subfragment</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../String/">String</a></td>
<td>Name of subfragment of a gene or locus</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000113">MIXS:0000113</a></td>
<td><a href="../0000113/">temp</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../String/">String</a></td>
<td>Temperature of the sample at the time of sampling</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000049">MIXS:0000049</a></td>
<td><a href="../0000049/">pcr_cond</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../String/">String</a></td>
<td>Description of reaction conditions and components of polymerase chain reactio...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000037">MIXS:0000037</a></td>
<td><a href="../0000037/">nucl_acid_ext</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../String/">String</a></td>
<td>A link to a literature reference, electronic resource or a standard operating...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000001">MIXS:0000001</a></td>
<td><a href="../0000001/">samp_size</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>The total amount or size (volume (ml), mass (g) or area (m2) ) of sample coll...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000003">MIXS:0000003</a></td>
<td><a href="../0000003/">isol_growth_condt</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>Publication reference in the form of pubmed ID (pmid), digital object identif...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000094">MIXS:0000094</a></td>
<td><a href="../0000094/">alt</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../String/">String</a></td>
<td>Heights of objects such as airplanes, space shuttles, rockets, atmospheric ba...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000026">MIXS:0000026</a></td>
<td><a href="../0000026/">source_mat_id</a></td>
<td>* <em>recommended</em> <br/> <a href="../String/">String</a></td>
<td>A unique identifier assigned to a material sample (as defined by <a href="http://rs">http://rs</a></td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000023">MIXS:0000023</a></td>
<td><a href="../0000023/">extrachrom_elements</a></td>
<td>0..1 <br/> <a href="../Integer/">Integer</a></td>
<td>Do plasmids exist of significant phenotypic consequence (e</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000111">MIXS:0000111</a></td>
<td><a href="../0000111/">samp_vol_we_dna_ext</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>Volume (ml) or mass (g) of total collected sample processed for DNA extractio...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000015">MIXS:0000015</a></td>
<td><a href="../0000015/">rel_to_oxygen</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../RelToOxygenEnum/">RelToOxygenEnum</a></td>
<td>Is this organism an aerobe, anaerobe? Please note that aerobic and anaerobic ...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000002">MIXS:0000002</a></td>
<td><a href="../0000002/">samp_collect_device</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>The device used to collect an environmental sample</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000028">MIXS:0000028</a></td>
<td><a href="../0000028/">biotic_relationship</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../BioticRelationshipEnum/">BioticRelationshipEnum</a></td>
<td>Description of relationship(s) between the subject organism and other organis...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000051">MIXS:0000051</a></td>
<td><a href="../0000051/">seq_quality_check</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../SeqQualityCheckEnum/">SeqQualityCheckEnum</a></td>
<td>Indicate if the sequence has been called by automatic systems (none) or under...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000092">MIXS:0000092</a></td>
<td><a href="../0000092/">project_name</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>Name of the project within which the sequencing was organized</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0001321">MIXS:0001321</a></td>
<td><a href="../0001321/">neg_cont_type</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../NegContTypeEnum/">NegContTypeEnum</a></td>
<td>The substance or equipment used as a negative control in an investigation</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000052">MIXS:0000052</a></td>
<td><a href="../0000052/">chimera_check</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../String/">String</a></td>
<td>Tool(s) used for chimera checking, including version number and parameters, t...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000032">MIXS:0000032</a></td>
<td><a href="../0000032/">trophic_level</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../TrophicLevelEnum/">TrophicLevelEnum</a></td>
<td>Trophic levels are the feeding position in a food chain</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0001322">MIXS:0001322</a></td>
<td><a href="../0001322/">pos_cont_type</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../String/">String</a></td>
<td>The substance, mixture, product, or apparatus used to verify that a process w...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000020">MIXS:0000020</a></td>
<td><a href="../0000020/">subspecf_gen_lin</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../String/">String</a></td>
<td>Information about the genetic distinctness of the sequenced organism below th...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000013">MIXS:0000013</a></td>
<td><a href="../0000013/">env_local_scale</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>Report the entity or entities which are in the sample or specimen s local vic...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000016">MIXS:0000016</a></td>
<td><a href="../0000016/">samp_mat_process</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../String/">String</a></td>
<td>A brief description of any processing applied to the sample during or after r...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000018">MIXS:0000018</a></td>
<td><a href="../0000018/">depth</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../String/">String</a></td>
<td>The vertical distance below local surface</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0001225">MIXS:0001225</a></td>
<td><a href="../0001225/">samp_collect_method</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>The method employed for collecting the sample</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000014">MIXS:0000014</a></td>
<td><a href="../0000014/">env_medium</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>Report the environmental material(s) immediately surrounding the sample or sp...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0001320">MIXS:0001320</a></td>
<td><a href="../0001320/">samp_taxon_id</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>NCBI taxon id of the sample</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000010">MIXS:0000010</a></td>
<td><a href="../0000010/">geo_loc_name</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>The geographical origin of the sample as defined by the country or sea name f...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000011">MIXS:0000011</a></td>
<td><a href="../0000011/">collection_date</a></td>
<td>1 <br/> <a href="../Datetime/">Datetime</a></td>
<td>The time of sampling, either as an instance (single point in time) or interva...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000050">MIXS:0000050</a></td>
<td><a href="../0000050/">seq_meth</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>Sequencing machine used</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000009">MIXS:0000009</a></td>
<td><a href="../0000009/">lat_lon</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>The geographical origin of the sample as defined by latitude and longitude</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000093">MIXS:0000093</a></td>
<td><a href="../0000093/">elev</a></td>
<td>0..1 <em>recommended</em> <br/> <a href="../String/">String</a></td>
<td>Elevation of the sampling site is its height above a fixed reference point, m...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000012">MIXS:0000012</a></td>
<td><a href="../0000012/">env_broad_scale</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>Report the major environmental system the sample or specimen came from</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000008">MIXS:0000008</a></td>
<td><a href="../0000008/">experimental_factor</a></td>
<td>* <br/> <a href="../String/">String</a></td>
<td>Variable aspects of an experiment design that can be used to describe an expe...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000044">MIXS:0000044</a></td>
<td><a href="../0000044/">target_gene</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>Targeted gene or locus name for marker gene studies</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000091">MIXS:0000091</a></td>
<td><a href="../0000091/">associated_resource</a></td>
<td>* <em>recommended</em> <br/> <a href="../String/">String</a></td>
<td>A related resource that is referenced, cited, or otherwise associated to the ...</td>
</tr>
<tr>
<td><a href="https://w3id.org/mixs/0000090">MIXS:0000090</a></td>
<td><a href="../0000090/">sop</a></td>
<td>* <em>recommended</em> <br/> <a href="../String/">String</a></td>
<td>Standard operating procedures used in assembly and/or annotation of genomes, ...</td>
</tr>
</tbody>
</table>
<h2 id="aliases">Aliases</h2>
<ul>
<li>mimarks_c</li>
<li>MIMARKS-SU</li>
<li>MIMARKS-specimen</li>
</ul>
<h2 id="linkml-source">LinkML Source</h2>
<!-- TODO: investigate https://stackoverflow.com/questions/37606292/how-to-create-tabbed-code-blocks-in-mkdocs-or-sphinx -->

<h3 id="direct">Direct</h3>
<details>

<pre><code class="language-yaml">name: MimarksC
description: 'Minimal Information about a Marker Sequence: specimen'
title: MIMARKS specimen
comments:
- for marker gene sequences from cultured or voucher-identifiable specimens
from_schema: https://w3id.org/mixs
aliases:
- mimarks_c
- MIMARKS-SU
- MIMARKS-specimen
is_a: Checklist
mixin: true
slots:
- samp_name
- pcr_primers
- nucl_acid_amp
- target_subfragment
- temp
- pcr_cond
- nucl_acid_ext
- samp_size
- isol_growth_condt
- alt
- source_mat_id
- extrachrom_elements
- samp_vol_we_dna_ext
- rel_to_oxygen
- samp_collect_device
- biotic_relationship
- seq_quality_check
- project_name
- neg_cont_type
- chimera_check
- trophic_level
- pos_cont_type
- subspecf_gen_lin
- env_local_scale
- samp_mat_process
- depth
- samp_collect_method
- env_medium
- samp_taxon_id
- geo_loc_name
- collection_date
- seq_meth
- lat_lon
- elev
- env_broad_scale
- experimental_factor
- target_gene
- associated_resource
- sop
slot_usage:
  alt:
    name: alt
    recommended: true
  biotic_relationship:
    name: biotic_relationship
    recommended: true
  chimera_check:
    name: chimera_check
    recommended: true
  depth:
    name: depth
    examples:
    - value: 10 meter
    recommended: true
  elev:
    name: elev
    recommended: true
  isol_growth_condt:
    name: isol_growth_condt
    required: true
  nucl_acid_amp:
    name: nucl_acid_amp
    recommended: true
  nucl_acid_ext:
    name: nucl_acid_ext
    recommended: true
  pcr_cond:
    name: pcr_cond
    recommended: true
  pcr_primers:
    name: pcr_primers
    recommended: true
  rel_to_oxygen:
    name: rel_to_oxygen
    recommended: true
  samp_collect_device:
    name: samp_collect_device
    examples:
    - value: swab, biopsy, niskin bottle, push core, drag swab [GENEPIO:0002713]
  samp_collect_method:
    name: samp_collect_method
    examples:
    - value: swabbing
  samp_mat_process:
    name: samp_mat_process
    recommended: true
  seq_quality_check:
    name: seq_quality_check
    recommended: true
  sop:
    name: sop
    recommended: true
  source_mat_id:
    name: source_mat_id
    recommended: true
  subspecf_gen_lin:
    name: subspecf_gen_lin
    recommended: true
  target_gene:
    name: target_gene
    required: true
  target_subfragment:
    name: target_subfragment
    recommended: true
  temp:
    name: temp
    recommended: true
  trophic_level:
    name: trophic_level
    recommended: true
class_uri: MIXS:0010009

</code></pre>

</details>

<h3 id="induced">Induced</h3>
<details>

<pre><code class="language-yaml">name: MimarksC
description: 'Minimal Information about a Marker Sequence: specimen'
title: MIMARKS specimen
comments:
- for marker gene sequences from cultured or voucher-identifiable specimens
from_schema: https://w3id.org/mixs
aliases:
- mimarks_c
- MIMARKS-SU
- MIMARKS-specimen
is_a: Checklist
mixin: true
slot_usage:
  alt:
    name: alt
    recommended: true
  biotic_relationship:
    name: biotic_relationship
    recommended: true
  chimera_check:
    name: chimera_check
    recommended: true
  depth:
    name: depth
    examples:
    - value: 10 meter
    recommended: true
  elev:
    name: elev
    recommended: true
  isol_growth_condt:
    name: isol_growth_condt
    required: true
  nucl_acid_amp:
    name: nucl_acid_amp
    recommended: true
  nucl_acid_ext:
    name: nucl_acid_ext
    recommended: true
  pcr_cond:
    name: pcr_cond
    recommended: true
  pcr_primers:
    name: pcr_primers
    recommended: true
  rel_to_oxygen:
    name: rel_to_oxygen
    recommended: true
  samp_collect_device:
    name: samp_collect_device
    examples:
    - value: swab, biopsy, niskin bottle, push core, drag swab [GENEPIO:0002713]
  samp_collect_method:
    name: samp_collect_method
    examples:
    - value: swabbing
  samp_mat_process:
    name: samp_mat_process
    recommended: true
  seq_quality_check:
    name: seq_quality_check
    recommended: true
  sop:
    name: sop
    recommended: true
  source_mat_id:
    name: source_mat_id
    recommended: true
  subspecf_gen_lin:
    name: subspecf_gen_lin
    recommended: true
  target_gene:
    name: target_gene
    required: true
  target_subfragment:
    name: target_subfragment
    recommended: true
  temp:
    name: temp
    recommended: true
  trophic_level:
    name: trophic_level
    recommended: true
attributes:
  samp_name:
    name: samp_name
    annotations:
      Preferred_unit:
        tag: Preferred_unit
        value: ''
    description: A local identifier or name that for the material sample used for
      extracting nucleic acids, and subsequent sequencing. It can refer either to
      the original material collected or to any derived sub-samples. It can have any
      format, but we suggest that you make it concise, unique and consistent within
      your lab, and as informative as possible. INSDC requires every sample name from
      a single Submitter to be unique. Use of a globally unique identifier for the
      field source_mat_id is recommended in addition to sample_name
    title: sample name
    examples:
    - value: ISDsoil1
    in_subset:
    - investigation
    from_schema: https://w3id.org/mixs
    keywords:
    - sample
    slot_uri: MIXS:0001107
    alias: samp_name
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - Air
    - BuiltEnvironment
    - FoodAnimalAndAnimalFeed
    - FoodFarmEnvironment
    - FoodFoodProductionFacility
    - FoodHumanFoods
    - HostAssociated
    - HumanAssociated
    - HumanGut
    - HumanOral
    - HumanSkin
    - HumanVaginal
    - HydrocarbonResourcesCores
    - HydrocarbonResourcesFluidsSwabs
    - MicrobialMatBiofilm
    - MiscellaneousNaturalOrArtificialEnvironment
    - PlantAssociated
    - Sediment
    - Soil
    - SymbiontAssociated
    - WastewaterSludge
    - Water
    range: string
    required: true
  pcr_primers:
    name: pcr_primers
    description: PCR primers that were used to amplify the sequence of the targeted
      gene, locus or subfragment. This field should contain all the primers used for
      a single PCR reaction if multiple forward or reverse primers are present in
      a single PCR reaction. The primer sequence should be reported in uppercase letters
    title: pcr primers
    examples:
    - value: FWD:GTGCCAGCMGCCGCGGTAA;REV:GGACTACHVGGGTWTCTAAT
    in_subset:
    - sequencing
    from_schema: https://w3id.org/mixs
    keywords:
    - pcr
    slot_uri: MIXS:0000046
    alias: pcr_primers
    owner: MimarksC
    domain_of:
    - MimarksC
    - MimarksS
    - Agriculture
    range: string
    recommended: true
    structured_pattern:
      syntax: FWD:{dna_bases};REV:{dna_bases}
      interpolated: true
  nucl_acid_amp:
    name: nucl_acid_amp
    description: A link to a literature reference, electronic resource or a standard
      operating procedure (SOP), that describes the enzymatic amplification (PCR,
      TMA, NASBA) of specific nucleic acids
    title: nucleic acid amplification
    examples:
    - value: https://phylogenomics.me/protocols/16s-pcr-protocol/
    in_subset:
    - sequencing
    from_schema: https://w3id.org/mixs
    slot_uri: MIXS:0000038
    alias: nucl_acid_amp
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - Agriculture
    range: string
    recommended: true
    structured_pattern:
      syntax: ^{PMID}|{DOI}|{URL}$
      interpolated: true
      partial_match: true
  target_subfragment:
    name: target_subfragment
    description: Name of subfragment of a gene or locus. Important to e.g. identify
      special regions on marker genes like V6 on 16S rRNA
    title: target subfragment
    examples:
    - value: V6, V9, ITS
    in_subset:
    - sequencing
    from_schema: https://w3id.org/mixs
    keywords:
    - target
    slot_uri: MIXS:0000045
    alias: target_subfragment
    owner: MimarksC
    domain_of:
    - MimarksC
    - MimarksS
    - Agriculture
    range: string
    recommended: true
  temp:
    name: temp
    annotations:
      Preferred_unit:
        tag: Preferred_unit
        value: degree Celsius
    description: Temperature of the sample at the time of sampling
    title: temperature
    examples:
    - value: 25 degree Celsius
    in_subset:
    - environment
    from_schema: https://w3id.org/mixs
    keywords:
    - temperature
    slot_uri: MIXS:0000113
    alias: temp
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - Agriculture
    - Air
    - FoodAnimalAndAnimalFeed
    - FoodFarmEnvironment
    - FoodHumanFoods
    - HostAssociated
    - HumanAssociated
    - HumanGut
    - HumanOral
    - HumanSkin
    - HumanVaginal
    - HydrocarbonResourcesCores
    - HydrocarbonResourcesFluidsSwabs
    - MicrobialMatBiofilm
    - MiscellaneousNaturalOrArtificialEnvironment
    - PlantAssociated
    - Sediment
    - Soil
    - SymbiontAssociated
    - WastewaterSludge
    - Water
    range: string
    recommended: true
    structured_pattern:
      syntax: ^{scientific_float}( *- *{scientific_float})? *{text}$
      interpolated: true
      partial_match: true
  pcr_cond:
    name: pcr_cond
    annotations:
      Expected_value:
        tag: Expected_value
        value: initial denaturation:degrees_minutes;annealing:degrees_minutes;elongation:degrees_minutes;final
          elongation:degrees_minutes;total cycles
    description: Description of reaction conditions and components of polymerase chain
      reaction performed during library preparation.
    title: pcr conditions
    examples:
    - value: initial denaturation:94_3;annealing:50_1;elongation:72_1.5;final elongation:72_10;35
    - value: initial denaturation:94degC_1.5min
    in_subset:
    - sequencing
    from_schema: https://w3id.org/mixs
    keywords:
    - condition
    - pcr
    string_serialization: initial denaturation:degrees_minutes;annealing:degrees_minutes;elongation:degrees_minutes;final
      elongation:degrees_minutes;total cycles
    slot_uri: MIXS:0000049
    alias: pcr_cond
    owner: MimarksC
    domain_of:
    - MimarksC
    - MimarksS
    - Agriculture
    range: string
    recommended: true
  nucl_acid_ext:
    name: nucl_acid_ext
    description: A link to a literature reference, electronic resource or a standard
      operating procedure (SOP), that describes the material separation to recover
      the nucleic acid fraction from a sample
    title: nucleic acid extraction
    examples:
    - value: https://mobio.com/media/wysiwyg/pdfs/protocols/12888.pdf
    in_subset:
    - sequencing
    from_schema: https://w3id.org/mixs
    slot_uri: MIXS:0000037
    alias: nucl_acid_ext
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - Agriculture
    - FoodAnimalAndAnimalFeed
    - FoodFarmEnvironment
    - FoodFoodProductionFacility
    - FoodHumanFoods
    range: string
    recommended: true
    structured_pattern:
      syntax: ^{PMID}|{DOI}|{URL}$
      interpolated: true
      partial_match: true
  samp_size:
    name: samp_size
    description: The total amount or size (volume (ml), mass (g) or area (m2) ) of
      sample collected
    title: amount or size of sample collected
    examples:
    - value: 5 liter
    in_subset:
    - nucleic acid sequence source
    from_schema: https://w3id.org/mixs
    keywords:
    - sample
    - size
    slot_uri: MIXS:0000001
    alias: samp_size
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - Agriculture
    - FoodAnimalAndAnimalFeed
    - FoodFarmEnvironment
    - FoodFoodProductionFacility
    - FoodHumanFoods
    range: string
    structured_pattern:
      syntax: ^{scientific_float}( *- *{scientific_float})? *{text}$
      interpolated: true
      partial_match: true
  isol_growth_condt:
    name: isol_growth_condt
    description: Publication reference in the form of pubmed ID (pmid), digital object
      identifier (doi) or url for isolation and growth condition specifications of
      the organism/material
    title: isolation and growth condition
    examples:
    - value: doi:10.1016/j.syapm.2018.01.009
    in_subset:
    - nucleic acid sequence source
    from_schema: https://w3id.org/mixs
    keywords:
    - condition
    - growth
    - isolation
    slot_uri: MIXS:0000003
    alias: isol_growth_condt
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - MimarksC
    - Agriculture
    range: string
    required: true
    structured_pattern:
      syntax: ^{PMID}|{DOI}|{URL}$
      interpolated: true
      partial_match: true
  alt:
    name: alt
    annotations:
      Preferred_unit:
        tag: Preferred_unit
        value: meter
    description: Heights of objects such as airplanes, space shuttles, rockets, atmospheric
      balloons and heights of places such as atmospheric layers and clouds. It is
      used to measure the height of an object which is above the earth's surface.
      In this context, the altitude measurement is the vertical distance between the
      earth's surface above sea level and the sampled position in the air
    title: altitude
    examples:
    - value: 100 meter
    in_subset:
    - environment
    from_schema: https://w3id.org/mixs
    slot_uri: MIXS:0000094
    alias: alt
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - Air
    - HostAssociated
    - MiscellaneousNaturalOrArtificialEnvironment
    - SymbiontAssociated
    range: string
    recommended: true
    structured_pattern:
      syntax: ^{scientific_float}( *- *{scientific_float})? *{text}$
      interpolated: true
      partial_match: true
  source_mat_id:
    name: source_mat_id
    annotations:
      Expected_value:
        tag: Expected_value
        value: 'for cultures of microorganisms: identifiers for two culture collections;
          for other material a unique arbitrary identifer'
    description: A unique identifier assigned to a material sample (as defined by
      http://rs.tdwg.org/dwc/terms/materialSampleID, and as opposed to a particular
      digital record of a material sample) used for extracting nucleic acids, and
      subsequent sequencing. The identifier can refer either to the original material
      collected or to any derived sub-samples. The INSDC qualifiers /specimen_voucher,
      /bio_material, or /culture_collection may or may not share the same value as
      the source_mat_id field. For instance, the /specimen_voucher qualifier and source_mat_id
      may both contain 'UAM:Herps:14' , referring to both the specimen voucher and
      sampled tissue with the same identifier. However, the /culture_collection qualifier
      may refer to a value from an initial culture (e.g. ATCC:11775) while source_mat_id
      would refer to an identifier from some derived culture from which the nucleic
      acids were extracted (e.g. xatc123 or ark:/2154/R2)
    title: source material identifiers
    examples:
    - value: MPI012345
    in_subset:
    - nucleic acid sequence source
    from_schema: https://w3id.org/mixs
    keywords:
    - identifier
    - material
    - source
    slot_uri: MIXS:0000026
    alias: source_mat_id
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - Agriculture
    - SymbiontAssociated
    range: string
    recommended: true
    multivalued: true
  extrachrom_elements:
    name: extrachrom_elements
    description: Do plasmids exist of significant phenotypic consequence (e.g. ones
      that determine virulence or antibiotic resistance). Megaplasmids? Other plasmids
      (borrelia has 15+ plasmids)
    title: extrachromosomal elements
    examples:
    - value: '5'
    in_subset:
    - nucleic acid sequence source
    from_schema: https://w3id.org/mixs
    slot_uri: MIXS:0000023
    alias: extrachrom_elements
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MimarksC
    range: integer
  samp_vol_we_dna_ext:
    name: samp_vol_we_dna_ext
    annotations:
      Preferred_unit:
        tag: Preferred_unit
        value: milliliter, gram, milligram, square centimeter
    description: 'Volume (ml) or mass (g) of total collected sample processed for
      DNA extraction. Note: total sample collected should be entered under the term
      Sample Size (MIXS:0000001)'
    title: sample volume or weight for DNA extraction
    examples:
    - value: 1500 milliliter
    in_subset:
    - nucleic acid sequence source
    from_schema: https://w3id.org/mixs
    keywords:
    - dna
    - sample
    - volume
    - weight
    slot_uri: MIXS:0000111
    alias: samp_vol_we_dna_ext
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - Agriculture
    - Air
    - FoodAnimalAndAnimalFeed
    - FoodFarmEnvironment
    - FoodFoodProductionFacility
    - FoodHumanFoods
    - HostAssociated
    - HumanAssociated
    - HumanGut
    - HumanOral
    - HumanSkin
    - HumanVaginal
    - HydrocarbonResourcesCores
    - HydrocarbonResourcesFluidsSwabs
    - MicrobialMatBiofilm
    - MiscellaneousNaturalOrArtificialEnvironment
    - PlantAssociated
    - Sediment
    - Soil
    - SymbiontAssociated
    - WastewaterSludge
    - Water
    range: string
    structured_pattern:
      syntax: ^{scientific_float}( *- *{scientific_float})? *{text}$
      interpolated: true
      partial_match: true
  rel_to_oxygen:
    name: rel_to_oxygen
    description: Is this organism an aerobe, anaerobe? Please note that aerobic and
      anaerobic are valid descriptors for microbial environments
    title: relationship to oxygen
    examples:
    - value: aerobe
    in_subset:
    - nucleic acid sequence source
    from_schema: https://w3id.org/mixs
    keywords:
    - oxygen
    - relationship
    slot_uri: MIXS:0000015
    alias: rel_to_oxygen
    owner: MimarksC
    domain_of:
    - MigsBa
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    range: RelToOxygenEnum
    recommended: true
  samp_collect_device:
    name: samp_collect_device
    annotations:
      Expected_value:
        tag: Expected_value
        value: device name
    description: The device used to collect an environmental sample. This field accepts
      terms listed under environmental sampling device (http://purl.obolibrary.org/obo/ENVO).
      This field also accepts terms listed under specimen collection device (http://purl.obolibrary.org/obo/GENEPIO_0002094)
    title: sample collection device
    examples:
    - value: swab, biopsy, niskin bottle, push core, drag swab [GENEPIO:0002713]
    in_subset:
    - nucleic acid sequence source
    from_schema: https://w3id.org/mixs
    keywords:
    - device
    - sample
    string_serialization: '{termLabel} [{termID}]|{text}'
    slot_uri: MIXS:0000002
    alias: samp_collect_device
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - Agriculture
    - FoodAnimalAndAnimalFeed
    - FoodFarmEnvironment
    - FoodFoodProductionFacility
    - FoodHumanFoods
    range: string
  biotic_relationship:
    name: biotic_relationship
    description: Description of relationship(s) between the subject organism and other
      organism(s) it is associated with. E.g., parasite on species X; mutualist with
      species Y. The target organism is the subject of the relationship, and the other
      organism(s) is the object
    title: observed biotic relationship
    examples:
    - value: free living
    in_subset:
    - nucleic acid sequence source
    from_schema: https://w3id.org/mixs
    keywords:
    - observed
    - relationship
    slot_uri: MIXS:0000028
    alias: biotic_relationship
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsVi
    - MimarksC
    - Miuvig
    - Agriculture
    range: BioticRelationshipEnum
    recommended: true
  seq_quality_check:
    name: seq_quality_check
    annotations:
      Expected_value:
        tag: Expected_value
        value: none or manually edited
    description: Indicate if the sequence has been called by automatic systems (none)
      or undergone a manual editing procedure (e.g. by inspecting the raw data or
      chromatograms). Applied only for sequences that are not submitted to SRA,ENA
      or DRA
    title: sequence quality check
    examples:
    - value: none
    in_subset:
    - sequencing
    from_schema: https://w3id.org/mixs
    keywords:
    - quality
    slot_uri: MIXS:0000051
    alias: seq_quality_check
    owner: MimarksC
    domain_of:
    - MimarksC
    - MimarksS
    - Agriculture
    range: SeqQualityCheckEnum
    recommended: true
  project_name:
    name: project_name
    description: Name of the project within which the sequencing was organized
    title: project name
    examples:
    - value: Forest soil metagenome
    in_subset:
    - investigation
    from_schema: https://w3id.org/mixs
    keywords:
    - project
    slot_uri: MIXS:0000092
    alias: project_name
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - Air
    - BuiltEnvironment
    - FoodAnimalAndAnimalFeed
    - FoodFarmEnvironment
    - FoodFoodProductionFacility
    - FoodHumanFoods
    - HostAssociated
    - HumanAssociated
    - HumanGut
    - HumanOral
    - HumanSkin
    - HumanVaginal
    - HydrocarbonResourcesCores
    - HydrocarbonResourcesFluidsSwabs
    - MicrobialMatBiofilm
    - MiscellaneousNaturalOrArtificialEnvironment
    - PlantAssociated
    - Sediment
    - Soil
    - SymbiontAssociated
    - WastewaterSludge
    - Water
    range: string
    required: true
  neg_cont_type:
    name: neg_cont_type
    annotations:
      Expected_value:
        tag: Expected_value
        value: enumeration or text
    description: The substance or equipment used as a negative control in an investigation
    title: negative control type
    in_subset:
    - investigation
    from_schema: https://w3id.org/mixs
    keywords:
    - type
    slot_uri: MIXS:0001321
    alias: neg_cont_type
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    range: NegContTypeEnum
    recommended: true
  chimera_check:
    name: chimera_check
    description: Tool(s) used for chimera checking, including version number and parameters,
      to discover and remove chimeric sequences. A chimeric sequence is comprised
      of two or more phylogenetically distinct parent sequences
    title: chimera check software
    examples:
    - value: uchime;v4.1;default parameters
    in_subset:
    - sequencing
    from_schema: https://w3id.org/mixs
    keywords:
    - software
    slot_uri: MIXS:0000052
    alias: chimera_check
    owner: MimarksC
    domain_of:
    - MimarksC
    - MimarksS
    - Agriculture
    range: string
    recommended: true
    structured_pattern:
      syntax: ^{software};{version};{parameters}$
      interpolated: true
      partial_match: true
  trophic_level:
    name: trophic_level
    description: Trophic levels are the feeding position in a food chain. Microbes
      can be a range of producers (e.g. chemolithotroph)
    title: trophic level
    examples:
    - value: heterotroph
    in_subset:
    - nucleic acid sequence source
    from_schema: https://w3id.org/mixs
    keywords:
    - level
    slot_uri: MIXS:0000032
    alias: trophic_level
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MimarksC
    - Agriculture
    range: TrophicLevelEnum
    recommended: true
  pos_cont_type:
    name: pos_cont_type
    description: The substance, mixture, product, or apparatus used to verify that
      a process which is part of an investigation delivers a true positive
    title: positive control type
    in_subset:
    - investigation
    from_schema: https://w3id.org/mixs
    keywords:
    - type
    string_serialization: '{term} or {text}'
    slot_uri: MIXS:0001322
    alias: pos_cont_type
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    range: string
    recommended: true
  subspecf_gen_lin:
    name: subspecf_gen_lin
    annotations:
      Expected_value:
        tag: Expected_value
        value: Genetic lineage below lowest rank of NCBI taxonomy, which is subspecies,
          e.g. serovar, biotype, ecotype, variety, cultivar
    description: Information about the genetic distinctness of the sequenced organism
      below the subspecies level, e.g., serovar, serotype, biotype, ecotype, or any
      relevant genetic typing schemes like Group I plasmid. Subspecies should not
      be recorded in this term, but in the NCBI taxonomy. Supply both the lineage
      name and the lineage rank separated by a colon, e.g., biovar:abc123
    title: subspecific genetic lineage
    examples:
    - value: serovar:Newport
    in_subset:
    - nucleic acid sequence source
    from_schema: https://w3id.org/mixs
    keywords:
    - lineage
    string_serialization: '{rank name}:{text}'
    slot_uri: MIXS:0000020
    alias: subspecf_gen_lin
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - MimarksC
    - FoodFoodProductionFacility
    range: string
    recommended: true
  env_local_scale:
    name: env_local_scale
    annotations:
      Expected_value:
        tag: Expected_value
        value: Environmental entities having causal influences upon the entity at
          time of sampling
    description: 'Report the entity or entities which are in the sample or specimen
      s local vicinity and which you believe have significant causal influences on
      your sample or specimen. We recommend using EnvO terms which are of smaller
      spatial grain than your entry for env_broad_scale. Terms, such as anatomical
      sites, from other OBO Library ontologies which interoperate with EnvO (e.g.
      UBERON) are accepted in this field. EnvO documentation about how to use the
      field: https://github.com/EnvironmentOntology/envo/wiki/Using-ENVO-with-MIxS'
    title: local environmental context
    examples:
    - value: hillside [ENVO:01000333]
    in_subset:
    - environment
    from_schema: https://w3id.org/mixs
    keywords:
    - context
    - environmental
    slot_uri: MIXS:0000013
    alias: env_local_scale
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    range: string
    required: true
    structured_pattern:
      syntax: ^{termLabel} \[{termID}\]$
      interpolated: true
      partial_match: true
  samp_mat_process:
    name: samp_mat_process
    description: A brief description of any processing applied to the sample during
      or after retrieving the sample from environment, or a link to the relevant protocol(s)
      performed
    title: sample material processing
    examples:
    - value: filtering of seawater, storing samples in ethanol
    in_subset:
    - nucleic acid sequence source
    from_schema: https://w3id.org/mixs
    keywords:
    - material
    - process
    - sample
    slot_uri: MIXS:0000016
    alias: samp_mat_process
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - Agriculture
    range: string
    recommended: true
  depth:
    name: depth
    annotations:
      Preferred_unit:
        tag: Preferred_unit
        value: meter
    description: The vertical distance below local surface. For sediment or soil samples
      depth is measured from sediment or soil surface, respectively. Depth can be
      reported as an interval for subsurface samples
    title: depth
    examples:
    - value: 10 meter
    in_subset:
    - environment
    from_schema: https://w3id.org/mixs
    keywords:
    - depth
    slot_uri: MIXS:0000018
    alias: depth
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - Agriculture
    - FoodFarmEnvironment
    - HostAssociated
    - MicrobialMatBiofilm
    - MiscellaneousNaturalOrArtificialEnvironment
    - PlantAssociated
    - Sediment
    - Soil
    - SymbiontAssociated
    - WastewaterSludge
    - Water
    range: string
    recommended: true
    structured_pattern:
      syntax: ^{scientific_float}( *- *{scientific_float})? *{text}$
      interpolated: true
      partial_match: true
  samp_collect_method:
    name: samp_collect_method
    description: The method employed for collecting the sample
    title: sample collection method
    examples:
    - value: swabbing
    in_subset:
    - nucleic acid sequence source
    from_schema: https://w3id.org/mixs
    keywords:
    - method
    - sample
    slot_uri: MIXS:0001225
    alias: samp_collect_method
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - Agriculture
    - FoodAnimalAndAnimalFeed
    - FoodFoodProductionFacility
    - FoodHumanFoods
    range: string
    structured_pattern:
      syntax: ^{PMID}|{DOI}|{URL}|{text}$
      interpolated: true
      partial_match: true
  env_medium:
    name: env_medium
    description: 'Report the environmental material(s) immediately surrounding the
      sample or specimen at the time of sampling. We recommend using subclasses of
      ''environmental material'' (http://purl.obolibrary.org/obo/ENVO_00010483). EnvO
      documentation about how to use the field: https://github.com/EnvironmentOntology/envo/wiki/Using-ENVO-with-MIxS
      . Terms from other OBO ontologies are permissible as long as they reference
      mass/volume nouns (e.g. air, water, blood) and not discrete, countable entities
      (e.g. a tree, a leaf, a table top)'
    title: environmental medium
    examples:
    - value: bluegrass field soil [ENVO:00005789]
    in_subset:
    - environment
    from_schema: https://w3id.org/mixs
    keywords:
    - environmental
    slot_uri: MIXS:0000014
    alias: env_medium
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    range: string
    required: true
    structured_pattern:
      syntax: ^{termLabel} \[{termID}\]$
      interpolated: true
      partial_match: true
  samp_taxon_id:
    name: samp_taxon_id
    description: NCBI taxon id of the sample.  Maybe be a single taxon or mixed taxa
      sample. Use 'synthetic metagenome  for mock community/positive controls, or
      'blank sample' for negative controls
    title: taxonomy ID of DNA sample
    examples:
    - value: Gut Metagenome [NCBITaxon:749906]
    in_subset:
    - investigation
    from_schema: https://w3id.org/mixs
    keywords:
    - dna
    - identifier
    - sample
    - taxon
    slot_uri: MIXS:0001320
    alias: samp_taxon_id
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    range: string
    required: true
    structured_pattern:
      syntax: ^{text} \[{NCBItaxon_id}\]$
      interpolated: true
      partial_match: true
  geo_loc_name:
    name: geo_loc_name
    description: The geographical origin of the sample as defined by the country or
      sea name followed by specific region name. Country or sea names should be chosen
      from the INSDC country list (http://insdc.org/country.html), or the GAZ ontology
      (http://purl.bioontology.org/ontology/GAZ)
    title: geographic location (country and/or sea,region)
    examples:
    - value: 'USA: Maryland, Bethesda'
    in_subset:
    - environment
    from_schema: https://w3id.org/mixs
    keywords:
    - geographic
    - location
    slot_uri: MIXS:0000010
    alias: geo_loc_name
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - FoodAnimalAndAnimalFeed
    - FoodFarmEnvironment
    - FoodFoodProductionFacility
    - FoodHumanFoods
    - SymbiontAssociated
    range: string
    required: true
    structured_pattern:
      syntax: '^{country}: {region}, {specific_location}$'
      interpolated: true
      partial_match: true
  collection_date:
    name: collection_date
    description: 'The time of sampling, either as an instance (single point in time)
      or interval. In case no exact time is available, the date/time can be right
      truncated i.e. all of these are valid times: 2008-01-23T19:23:10+00:00; 2008-01-23T19:23:10;
      2008-01-23; 2008-01; 2008; Except: 2008-01; 2008 all are ISO8601 compliant'
    title: collection date
    examples:
    - value: '2013-03-25T12:42:31+01:00'
    in_subset:
    - environment
    from_schema: https://w3id.org/mixs
    keywords:
    - date
    slot_uri: MIXS:0000011
    alias: collection_date
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - FoodAnimalAndAnimalFeed
    - FoodFarmEnvironment
    - FoodFoodProductionFacility
    - FoodHumanFoods
    - SymbiontAssociated
    range: datetime
    required: true
  seq_meth:
    name: seq_meth
    description: Sequencing machine used. Where possible the term should be taken
      from the OBI list of DNA sequencers (http://purl.obolibrary.org/obo/OBI_0400103)
    title: sequencing method
    examples:
    - value: 454 Genome Sequencer FLX [OBI:0000702]
    in_subset:
    - sequencing
    from_schema: https://w3id.org/mixs
    keywords:
    - method
    slot_uri: MIXS:0000050
    alias: seq_meth
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - Agriculture
    - FoodAnimalAndAnimalFeed
    - FoodFarmEnvironment
    - FoodFoodProductionFacility
    - FoodHumanFoods
    range: string
    required: true
    structured_pattern:
      syntax: ^{text}|({termLabel} \[{termID}\])$
      interpolated: true
      partial_match: true
  lat_lon:
    name: lat_lon
    description: The geographical origin of the sample as defined by latitude and
      longitude. The values should be reported in decimal degrees, limited to 8 decimal
      points, and in WGS84 system
    title: geographic location (latitude and longitude)
    examples:
    - value: 50.586825 6.408977
    in_subset:
    - environment
    from_schema: https://w3id.org/mixs
    keywords:
    - geographic
    - location
    slot_uri: MIXS:0000009
    alias: lat_lon
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - FoodAnimalAndAnimalFeed
    - FoodFarmEnvironment
    - FoodFoodProductionFacility
    - FoodHumanFoods
    - SymbiontAssociated
    range: string
    required: true
    structured_pattern:
      syntax: ^{lat} {lon}$
      interpolated: true
      partial_match: true
  elev:
    name: elev
    annotations:
      Preferred_unit:
        tag: Preferred_unit
        value: meter
    description: Elevation of the sampling site is its height above a fixed reference
      point, most commonly the mean sea level. Elevation is mainly used when referring
      to points on the earth's surface, while altitude is used for points above the
      surface, such as an aircraft in flight or a spacecraft in orbit
    title: elevation
    examples:
    - value: 100 meter
    in_subset:
    - environment
    from_schema: https://w3id.org/mixs
    keywords:
    - elevation
    slot_uri: MIXS:0000093
    alias: elev
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - Agriculture
    - Air
    - HostAssociated
    - HydrocarbonResourcesCores
    - MicrobialMatBiofilm
    - MiscellaneousNaturalOrArtificialEnvironment
    - PlantAssociated
    - Sediment
    - Soil
    - SymbiontAssociated
    - Water
    range: string
    recommended: true
    structured_pattern:
      syntax: ^{scientific_float}( *- *{scientific_float})? *{text}$
      interpolated: true
      partial_match: true
  env_broad_scale:
    name: env_broad_scale
    description: 'Report the major environmental system the sample or specimen came
      from. The system(s) identified should have a coarse spatial grain, to provide
      the general environmental context of where the sampling was done (e.g. in the
      desert or a rainforest). We recommend using subclasses of EnvO s biome class:  http://purl.obolibrary.org/obo/ENVO_00000428.
      EnvO documentation about how to use the field: https://github.com/EnvironmentOntology/envo/wiki/Using-ENVO-with-MIxS'
    title: broad-scale environmental context
    examples:
    - value: rangeland biome [ENVO:01000247]
    in_subset:
    - environment
    from_schema: https://w3id.org/mixs
    keywords:
    - context
    - environmental
    slot_uri: MIXS:0000012
    alias: env_broad_scale
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    range: string
    required: true
    structured_pattern:
      syntax: ^{termLabel} \[{termID}\]$
      interpolated: true
      partial_match: true
  experimental_factor:
    name: experimental_factor
    annotations:
      Expected_value:
        tag: Expected_value
        value: text or EFO and/or OBI
    description: Variable aspects of an experiment design that can be used to describe
      an experiment, or set of experiments, in an increasingly detailed manner. This
      field accepts ontology terms from Experimental Factor Ontology (EFO) and/or
      Ontology for Biomedical Investigations (OBI)
    title: experimental factor
    examples:
    - value: time series design [EFO:0001779]
    in_subset:
    - investigation
    from_schema: https://w3id.org/mixs
    keywords:
    - experimental
    - factor
    string_serialization: '{termLabel} [{termID}]|{text}'
    slot_uri: MIXS:0000008
    alias: experimental_factor
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - FoodAnimalAndAnimalFeed
    - FoodFoodProductionFacility
    - FoodHumanFoods
    range: string
    multivalued: true
    pattern: ^\S+.*\S+ \[[a-zA-Z]{2,}:\d+\]$
  target_gene:
    name: target_gene
    description: Targeted gene or locus name for marker gene studies
    title: target gene
    examples:
    - value: 16S rRNA, 18S rRNA, nif, amoA, rpo
    in_subset:
    - sequencing
    from_schema: https://w3id.org/mixs
    keywords:
    - target
    slot_uri: MIXS:0000044
    alias: target_gene
    owner: MimarksC
    domain_of:
    - MimarksC
    - MimarksS
    - Agriculture
    range: string
    required: true
  associated_resource:
    name: associated_resource
    annotations:
      Expected_value:
        tag: Expected_value
        value: reference to resource
    description: A related resource that is referenced, cited, or otherwise associated
      to the sequence
    title: relevant electronic resources
    examples:
    - value: http://www.earthmicrobiome.org/
    in_subset:
    - sequencing
    from_schema: https://w3id.org/mixs
    keywords:
    - resource
    slot_uri: MIXS:0000091
    alias: associated_resource
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - Agriculture
    range: string
    recommended: true
    multivalued: true
    structured_pattern:
      syntax: ^{PMID}|{DOI}|{URL}$
      interpolated: true
  sop:
    name: sop
    annotations:
      Expected_value:
        tag: Expected_value
        value: reference to SOP
    description: Standard operating procedures used in assembly and/or annotation
      of genomes, metagenomes or environmental sequences
    title: relevant standard operating procedures
    examples:
    - value: http://press.igsb.anl.gov/earthmicrobiome/protocols-and-standards/its/
    in_subset:
    - sequencing
    from_schema: https://w3id.org/mixs
    keywords:
    - procedures
    slot_uri: MIXS:0000090
    alias: sop
    owner: MimarksC
    domain_of:
    - MigsBa
    - MigsEu
    - MigsOrg
    - MigsPl
    - MigsVi
    - Mimag
    - MimarksC
    - MimarksS
    - Mims
    - Misag
    - Miuvig
    - Agriculture
    range: string
    recommended: true
    multivalued: true
    structured_pattern:
      syntax: ^{PMID}|{DOI}|{URL}$
      interpolated: true
class_uri: MIXS:0010009

</code></pre>

</details>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": ["content.tabs.link", "search.suggest"], "search": "../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.92b07e13.min.js"></script>
      
        <script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
      
        <script src="../javascripts/tablesort.js"></script>
      
    
  </body>
</html>