
# Class: human-associated


human-associated

URI: [mixs.vocab:Human-associated](https://w3id.org/mixs/vocab/Human-associated)


[![img](https://yuml.me/diagram/nofunky;dir:TB/class/[QuantityValue],[Human-associatedMIUVIG],[Human-associatedMISAG],[Human-associatedMIMS],[Human-associatedMIMARKSSurvey],[Human-associatedMIMARKSSpecimen],[Human-associatedMIMAG],[Human-associatedMIGSVirus],[Human-associatedMIGSPlant],[Human-associatedMIGSOrg],[Human-associatedMIGSEukaryote],[Human-associatedMIGSBacteria],[QuantityValue]<samp_vol_we_dna_ext%200..1-++[Human-associated&#124;lat_lon:string%20%3F;geo_loc_name:string%20%3F;collection_date:date%20%3F;env_broad_scale:string%20%3F;env_local_scale:string%20%3F;env_medium:string%20%3F;host_subject_id:string%20%3F;host_sex:host_sex_enum%20%3F;ihmc_medication_code:integer%20*;chem_administration:string%20*;host_body_site:string%20%3F;host_body_product:string%20%3F;host_diet:string%20*;host_last_meal:string%20*;host_family_relation:string%20*;host_genotype:string%20%3F;host_phenotype:string%20%3F;smoker:string%20%3F;host_hiv_stat:string%20%3F;drug_usage:drug_usage_enum%20*;diet_last_six_month:string%20%3F;weight_loss_3_month:string%20%3F;ethnicity:string%20*;host_occupation:integer%20%3F;pet_farm_animal:string%20*;travel_out_six_month:string%20*;twin_sibling:string%20%3F;medic_hist_perform:string%20%3F;study_complt_stat:study_complt_stat_enum%20%3F;pulmonary_disord:string%20*;nose_throat_disord:string%20*;blood_blood_disord:string%20*;gestation_state:string%20%3F;maternal_health_stat:string%20%3F;foetal_health_stat:string%20%3F;amniotic_fluid_color:string%20%3F;kidney_disord:string%20*;urogenit_tract_disor:string%20*;urine_collect_meth:string%20%3F;perturbation:string%20*;oxy_stat_samp:oxy_stat_samp_enum%20%3F;organism_count:organism_count_enum%20*;samp_store_dur:string%20%3F;host_symbiont:string%20*;samp_store_loc:string%20%3F;misc_param:string%20*;samp_name:string;project_name:string;host_disease_stat:string%20%3F],[QuantityValue]<samp_store_temp%200..1-++[Human-associated],[QuantityValue]<salinity%200..1-++[Human-associated],[QuantityValue]<samp_salinity%200..1-++[Human-associated],[QuantityValue]<host_pulse%200..1-++[Human-associated],[QuantityValue]<host_body_mass_index%200..1-++[Human-associated],[QuantityValue]<host_body_temp%200..1-++[Human-associated],[QuantityValue]<host_height%200..1-++[Human-associated],[QuantityValue]<host_tot_mass%200..1-++[Human-associated],[QuantityValue]<host_age%200..1-++[Human-associated],[QuantityValue]<temp%200..1-++[Human-associated],[QuantityValue]<elev%200..1-++[Human-associated],[QuantityValue]<alt%200..1-++[Human-associated],[QuantityValue]<depth%200..1-++[Human-associated],[Human-associated]^-[Human-associatedMIUVIG],[Human-associated]^-[Human-associatedMISAG],[Human-associated]^-[Human-associatedMIMS],[Human-associated]^-[Human-associatedMIMARKSSurvey],[Human-associated]^-[Human-associatedMIMARKSSpecimen],[Human-associated]^-[Human-associatedMIMAG],[Human-associated]^-[Human-associatedMIGSVirus],[Human-associated]^-[Human-associatedMIGSPlant],[Human-associated]^-[Human-associatedMIGSOrg],[Human-associated]^-[Human-associatedMIGSEukaryote],[Human-associated]^-[Human-associatedMIGSBacteria])](https://yuml.me/diagram/nofunky;dir:TB/class/[QuantityValue],[Human-associatedMIUVIG],[Human-associatedMISAG],[Human-associatedMIMS],[Human-associatedMIMARKSSurvey],[Human-associatedMIMARKSSpecimen],[Human-associatedMIMAG],[Human-associatedMIGSVirus],[Human-associatedMIGSPlant],[Human-associatedMIGSOrg],[Human-associatedMIGSEukaryote],[Human-associatedMIGSBacteria],[QuantityValue]<samp_vol_we_dna_ext%200..1-++[Human-associated&#124;lat_lon:string%20%3F;geo_loc_name:string%20%3F;collection_date:date%20%3F;env_broad_scale:string%20%3F;env_local_scale:string%20%3F;env_medium:string%20%3F;host_subject_id:string%20%3F;host_sex:host_sex_enum%20%3F;ihmc_medication_code:integer%20*;chem_administration:string%20*;host_body_site:string%20%3F;host_body_product:string%20%3F;host_diet:string%20*;host_last_meal:string%20*;host_family_relation:string%20*;host_genotype:string%20%3F;host_phenotype:string%20%3F;smoker:string%20%3F;host_hiv_stat:string%20%3F;drug_usage:drug_usage_enum%20*;diet_last_six_month:string%20%3F;weight_loss_3_month:string%20%3F;ethnicity:string%20*;host_occupation:integer%20%3F;pet_farm_animal:string%20*;travel_out_six_month:string%20*;twin_sibling:string%20%3F;medic_hist_perform:string%20%3F;study_complt_stat:study_complt_stat_enum%20%3F;pulmonary_disord:string%20*;nose_throat_disord:string%20*;blood_blood_disord:string%20*;gestation_state:string%20%3F;maternal_health_stat:string%20%3F;foetal_health_stat:string%20%3F;amniotic_fluid_color:string%20%3F;kidney_disord:string%20*;urogenit_tract_disor:string%20*;urine_collect_meth:string%20%3F;perturbation:string%20*;oxy_stat_samp:oxy_stat_samp_enum%20%3F;organism_count:organism_count_enum%20*;samp_store_dur:string%20%3F;host_symbiont:string%20*;samp_store_loc:string%20%3F;misc_param:string%20*;samp_name:string;project_name:string;host_disease_stat:string%20%3F],[QuantityValue]<samp_store_temp%200..1-++[Human-associated],[QuantityValue]<salinity%200..1-++[Human-associated],[QuantityValue]<samp_salinity%200..1-++[Human-associated],[QuantityValue]<host_pulse%200..1-++[Human-associated],[QuantityValue]<host_body_mass_index%200..1-++[Human-associated],[QuantityValue]<host_body_temp%200..1-++[Human-associated],[QuantityValue]<host_height%200..1-++[Human-associated],[QuantityValue]<host_tot_mass%200..1-++[Human-associated],[QuantityValue]<host_age%200..1-++[Human-associated],[QuantityValue]<temp%200..1-++[Human-associated],[QuantityValue]<elev%200..1-++[Human-associated],[QuantityValue]<alt%200..1-++[Human-associated],[QuantityValue]<depth%200..1-++[Human-associated],[Human-associated]^-[Human-associatedMIUVIG],[Human-associated]^-[Human-associatedMISAG],[Human-associated]^-[Human-associatedMIMS],[Human-associated]^-[Human-associatedMIMARKSSurvey],[Human-associated]^-[Human-associatedMIMARKSSpecimen],[Human-associated]^-[Human-associatedMIMAG],[Human-associated]^-[Human-associatedMIGSVirus],[Human-associated]^-[Human-associatedMIGSPlant],[Human-associated]^-[Human-associatedMIGSOrg],[Human-associated]^-[Human-associatedMIGSEukaryote],[Human-associated]^-[Human-associatedMIGSBacteria])

## Children

 * [Human-associatedMIGSBacteria](Human-associatedMIGSBacteria.md) - Combinatorial checklist Minimal Information about a Genome Sequence: cultured bacteria/archaea with environmental package human-associated
 * [Human-associatedMIGSEukaryote](Human-associatedMIGSEukaryote.md) - Combinatorial checklist Minimal Information about a Genome Sequence: eukaryote with environmental package human-associated
 * [Human-associatedMIGSOrg](Human-associatedMIGSOrg.md) - Combinatorial checklist Minimal Information about a Genome Sequence: org with environmental package human-associated
 * [Human-associatedMIGSPlant](Human-associatedMIGSPlant.md) - Combinatorial checklist Minimal Information about a Genome Sequence: plant with environmental package human-associated
 * [Human-associatedMIGSVirus](Human-associatedMIGSVirus.md) - Combinatorial checklist Minimal Information about a Genome Sequence: cultured bacteria/archaea with environmental package human-associated
 * [Human-associatedMIMAG](Human-associatedMIMAG.md) - Combinatorial checklist Minimum Information About a Metagenome-Assembled Genome with environmental package human-associated
 * [Human-associatedMIMARKSSpecimen](Human-associatedMIMARKSSpecimen.md) - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package human-associated
 * [Human-associatedMIMARKSSurvey](Human-associatedMIMARKSSurvey.md) - Combinatorial checklist Minimal Information about a Marker Specimen: survey with environmental package human-associated
 * [Human-associatedMIMS](Human-associatedMIMS.md) - Combinatorial checklist Metagenome or Environmental with environmental package human-associated
 * [Human-associatedMISAG](Human-associatedMISAG.md) - Combinatorial checklist Minimum Information About a Single Amplified Genome with environmental package human-associated
 * [Human-associatedMIUVIG](Human-associatedMIUVIG.md) - Combinatorial checklist Minimum Information About an Uncultivated Virus Genome with environmental package human-associated

## Referenced by Class


## Attributes


### Own

 * [lat_lon](lat_lon.md)  <sub>0..1</sub>
     * Description: The geographical origin of the sample as defined by latitude and longitude. The values should be reported in decimal degrees and in WGS84 system
     * Range: [String](types/String.md)
     * Example: 50.586825 6.408977 None
 * [depth](depth.md)  <sub>0..1</sub>
     * Description: The vertical distance below local surface, e.g. for sediment or soil samples depth is measured from sediment or soil surface, respectively. Depth can be reported as an interval for subsurface samples.
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 10 meter None
 * [alt](alt.md)  <sub>0..1</sub>
     * Description: Altitude is a term used to identify heights of objects such as airplanes, space shuttles, rockets, atmospheric balloons and heights of places such as atmospheric layers and clouds. It is used to measure the height of an object which is above the earth's surface. In this context, the altitude measurement is the vertical distance between the earth's surface above sea level and the sampled position in the air
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 100 meter None
 * [elev](elev.md)  <sub>0..1</sub>
     * Description: Elevation of the sampling site is its height above a fixed reference point, most commonly the mean sea level. Elevation is mainly used when referring to points on the earth's surface, while altitude is used for points above the surface, such as an aircraft in flight or a spacecraft in orbit.
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 100 meter None
 * [human-associated➞temp](human_associated_temp.md)  <sub>0..1</sub>
     * Description: Temperature of the sample at the time of sampling.
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 25 degree Celsius None
 * [geo_loc_name](geo_loc_name.md)  <sub>0..1</sub>
     * Description: The geographical origin of the sample as defined by the country or sea name followed by specific region name. Country or sea names should be chosen from the INSDC country list (http://insdc.org/country.html), or the GAZ ontology (http://purl.bioontology.org/ontology/GAZ)
     * Range: [String](types/String.md)
     * Example: USA: Maryland, Bethesda None
 * [collection_date](collection_date.md)  <sub>0..1</sub>
     * Description: The time of sampling, either as an instance (single point in time) or interval. In case no exact time is available, the date/time can be right truncated i.e. all of these are valid times: 2008-01-23T19:23:10+00:00; 2008-01-23T19:23:10; 2008-01-23; 2008-01; 2008; Except: 2008-01; 2008 all are ISO8601 compliant
     * Range: [Date](types/Date.md)
     * Example: 2018-05-11T10:00:00+01:00; 2018-05-11 None
 * [env_broad_scale](env_broad_scale.md)  <sub>0..1</sub>
     * Description: Report the major environmental system the sample or specimen came from. The system(s) identified should have a coarse spatial grain, to provide the general environmental context of where the sampling was done (e.g. in the desert or a rainforest). We recommend using subclasses of EnvO’s biome class:  http://purl.obolibrary.org/obo/ENVO_00000428. EnvO documentation about how to use the field: https://github.com/EnvironmentOntology/envo/wiki/Using-ENVO-with-MIxS
     * Range: [String](types/String.md)
     * Example: oceanic epipelagic zone biome [ENVO:01000033] for annotating a water sample from the photic zone in middle of the Atlantic Ocean None
 * [env_local_scale](env_local_scale.md)  <sub>0..1</sub>
     * Description: Report the entity or entities which are in the sample or specimen’s local vicinity and which you believe have significant causal influences on your sample or specimen. We recommend using EnvO terms which are of smaller spatial grain than your entry for env_broad_scale. Terms, such as anatomical sites, from other OBO Library ontologies which interoperate with EnvO (e.g. UBERON) are accepted in this field. EnvO documentation about how to use the field: https://github.com/EnvironmentOntology/envo/wiki/Using-ENVO-with-MIxS.
     * Range: [String](types/String.md)
     * Example: litter layer [ENVO:01000338]; Annotating a pooled sample taken from various vegetation layers in a forest consider: canopy [ENVO:00000047]|herb and fern layer [ENVO:01000337]|litter layer [ENVO:01000338]|understory [01000335]|shrub layer [ENVO:01000336]. None
 * [env_medium](env_medium.md)  <sub>0..1</sub>
     * Description: Report the environmental material(s) immediately surrounding the sample or specimen at the time of sampling. We recommend using subclasses of 'environmental material' (http://purl.obolibrary.org/obo/ENVO_00010483). EnvO documentation about how to use the field: https://github.com/EnvironmentOntology/envo/wiki/Using-ENVO-with-MIxS . Terms from other OBO ontologies are permissible as long as they reference mass/volume nouns (e.g. air, water, blood) and not discrete, countable entities (e.g. a tree, a leaf, a table top).
     * Range: [String](types/String.md)
     * Example: soil [ENVO:00001998]; Annotating a fish swimming in the upper 100 m of the Atlantic Ocean, consider: ocean water [ENVO:00002151]. Example: Annotating a duck on a pond consider: pond water [ENVO:00002228]|air [ENVO_00002005] None
 * [human-associated➞host_subject_id](human_associated_host_subject_id.md)  <sub>0..1</sub>
     * Description: A unique identifier by which each subject can be referred to, de-identified.
     * Range: [String](types/String.md)
     * Example: MPI123 None
 * [human-associated➞host_age](human_associated_host_age.md)  <sub>0..1</sub>
     * Description: Age of host at the time of sampling; relevant scale depends on species and study, e.g. Could be seconds for amoebae or centuries for trees
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 10 days None
 * [human-associated➞host_sex](human_associated_host_sex.md)  <sub>0..1</sub>
     * Description: Gender or physical sex of the host.
     * Range: [host_sex_enum](host_sex_enum.md)
     * Example: non-binary None
 * [human-associated➞ihmc_medication_code](human_associated_ihmc_medication_code.md)  <sub>0..\*</sub>
     * Description: Can include multiple medication codes
     * Range: [Integer](types/Integer.md)
     * Example: 810 None
 * [human-associated➞chem_administration](human_associated_chem_administration.md)  <sub>0..\*</sub>
     * Description: List of chemical compounds administered to the host or site where sampling occurred, and when (e.g. Antibiotics, n fertilizer, air filter); can include multiple compounds. For chemical entities of biological interest ontology (chebi) (v 163), http://purl.bioontology.org/ontology/chebi
     * Range: [String](types/String.md)
     * Example: agar [CHEBI:2509];2018-05-11T20:00Z None
 * [human-associated➞host_body_site](human_associated_host_body_site.md)  <sub>0..1</sub>
     * Description: Name of body site where the sample was obtained from, such as a specific organ or tissue (tongue, lung etc...). For foundational model of anatomy ontology (fma) (v 4.11.0) or Uber-anatomy ontology (UBERON) (v releases/2014-06-15) terms, please see http://purl.bioontology.org/ontology/FMA or http://purl.bioontology.org/ontology/UBERON
     * Range: [String](types/String.md)
     * Example: gill [UBERON:0002535] None
 * [human-associated➞host_body_product](human_associated_host_body_product.md)  <sub>0..1</sub>
     * Description: Substance produced by the body, e.g. Stool, mucus, where the sample was obtained from. For foundational model of anatomy ontology (fma) or Uber-anatomy ontology (UBERON) terms, please see https://www.ebi.ac.uk/ols/ontologies/fma or https://www.ebi.ac.uk/ols/ontologies/uberon
     * Range: [String](types/String.md)
     * Example: Portion of mucus [fma66938] None
 * [human-associated➞host_tot_mass](human_associated_host_tot_mass.md)  <sub>0..1</sub>
     * Description: Total mass of the host at collection, the unit depends on host
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 2500 gram None
 * [human-associated➞host_height](human_associated_host_height.md)  <sub>0..1</sub>
     * Description: The height of subject
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 0.1 meter None
 * [human-associated➞host_diet](human_associated_host_diet.md)  <sub>0..\*</sub>
     * Description: Type of diet depending on the host, for animals omnivore, herbivore etc., for humans high-fat, meditteranean etc.; can include multiple diet types
     * Range: [String](types/String.md)
     * Example: herbivore None
 * [human-associated➞host_last_meal](human_associated_host_last_meal.md)  <sub>0..\*</sub>
     * Description: Content of last meal and time since feeding; can include multiple values
     * Range: [String](types/String.md)
     * Example: corn feed;P2H None
 * [human-associated➞host_family_relation](human_associated_host_family_relation.md)  <sub>0..\*</sub>
     * Description: Familial relationships to other hosts in the same study; can include multiple relationships
     * Range: [String](types/String.md)
     * Example: offspring;Mussel25 None
 * [human-associated➞host_genotype](human_associated_host_genotype.md)  <sub>0..1</sub>
     * Description: Observed genotype
     * Range: [String](types/String.md)
     * Example: C57BL/6 None
 * [human-associated➞host_phenotype](human_associated_host_phenotype.md)  <sub>0..1</sub>
     * Description: Phenotype of human or other host. For phenotypic quality ontology (pato) (v 2018-03-27) terms, please see http://purl.bioontology.org/ontology/pato. For Human Phenotype Ontology (HP) (v 2018-06-13) please see http://purl.bioontology.org/ontology/HP
     * Range: [String](types/String.md)
     * Example: elongated [PATO:0001154] None
 * [human-associated➞host_body_temp](human_associated_host_body_temp.md)  <sub>0..1</sub>
     * Description: Core body temperature of the host when sample was collected
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 15 degree Celsius None
 * [human-associated➞smoker](human_associated_smoker.md)  <sub>0..1</sub>
     * Description: Specification of smoking status
     * Range: [String](types/String.md)
     * Example: yes None
 * [human-associated➞host_hiv_stat](human_associated_host_hiv_stat.md)  <sub>0..1</sub>
     * Description: HIV status of subject, if yes HAART initiation status should also be indicated as [YES or NO]
     * Range: [String](types/String.md)
     * Example: yes;yes None
 * [human-associated➞drug_usage](human_associated_drug_usage.md)  <sub>0..\*</sub>
     * Description: Any drug used by subject and the frequency of usage; can include multiple drugs used
     * Range: [drug_usage_enum](drug_usage_enum.md)
     * Example: Lipitor;2/day None
 * [human-associated➞host_body_mass_index](human_associated_host_body_mass_index.md)  <sub>0..1</sub>
     * Description: Body mass index, calculated as weight/(height)squared
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 22 kilogram per square meter None
 * [human-associated➞diet_last_six_month](human_associated_diet_last_six_month.md)  <sub>0..1</sub>
     * Description: Specification of major diet changes in the last six months, if yes the change should be specified
     * Range: [String](types/String.md)
     * Example: yes;vegetarian diet None
 * [human-associated➞weight_loss_3_month](human_associated_weight_loss_3_month.md)  <sub>0..1</sub>
     * Description: Specification of weight loss in the last three months, if yes should be further specified to include amount of weight loss
     * Range: [String](types/String.md)
     * Example: yes;5 kilogram None
 * [human-associated➞ethnicity](human_associated_ethnicity.md)  <sub>0..\*</sub>
     * Description: A category of people who identify with each other, usually on the basis of presumed similarities such as a common language, ancestry, history, society, culture, nation or social treatment within their residing area. https://en.wikipedia.org/wiki/List_of_contemporary_ethnic_groups
     * Range: [String](types/String.md)
     * Example: native american None
 * [human-associated➞host_occupation](human_associated_host_occupation.md)  <sub>0..1</sub>
     * Description: Most frequent job performed by subject
     * Range: [Integer](types/Integer.md)
     * Example: veterinary None
 * [human-associated➞pet_farm_animal](human_associated_pet_farm_animal.md)  <sub>0..\*</sub>
     * Description: Specification of presence of pets or farm animals in the environment of subject, if yes the animals should be specified; can include multiple animals present
     * Range: [String](types/String.md)
     * Example: yes; 5 cats None
 * [human-associated➞travel_out_six_month](human_associated_travel_out_six_month.md)  <sub>0..\*</sub>
     * Description: Specification of the countries travelled in the last six months; can include multiple travels
     * Range: [String](types/String.md)
     * Example:  None
 * [human-associated➞twin_sibling](human_associated_twin_sibling.md)  <sub>0..1</sub>
     * Description: Specification of twin sibling presence
     * Range: [String](types/String.md)
     * Example: yes None
 * [human-associated➞medic_hist_perform](human_associated_medic_hist_perform.md)  <sub>0..1</sub>
     * Description: Whether full medical history was collected
     * Range: [String](types/String.md)
     * Example: TRUE None
 * [human-associated➞study_complt_stat](human_associated_study_complt_stat.md)  <sub>0..1</sub>
     * Description: Specification of study completion status, if no the reason should be specified
     * Range: [study_complt_stat_enum](study_complt_stat_enum.md)
     * Example: no;non-compliance None
 * [human-associated➞pulmonary_disord](human_associated_pulmonary_disord.md)  <sub>0..\*</sub>
     * Description: History of pulmonary disorders; can include multiple disorders. The terms should be chosen from the DO (Human Disease Ontology) at http://www.disease-ontology.org, lung disease (https://disease-ontology.org/?id=DOID:850).
     * Range: [String](types/String.md)
     * Example:  None
 * [human-associated➞nose_throat_disord](human_associated_nose_throat_disord.md)  <sub>0..\*</sub>
     * Description: History of nose-throat disorders; can include multiple disorders,  The terms should be chosen from the DO (Human Disease Ontology) at http://www.disease-ontology.org, lung disease (https://disease-ontology.org/?id=DOID:850), upper respiratory tract disease (https://disease-ontology.org/?id=DOID:974).
     * Range: [String](types/String.md)
     * Example:  None
 * [human-associated➞blood_blood_disord](human_associated_blood_blood_disord.md)  <sub>0..\*</sub>
     * Description: History of blood disorders; can include multiple disorders.  The terms should be chosen from the DO (Human Disease Ontology) at http://www.disease-ontology.org, hematopoietic system disease (https://disease-ontology.org/?id=DOID:74).
     * Range: [String](types/String.md)
     * Example:  None
 * [human-associated➞host_pulse](human_associated_host_pulse.md)  <sub>0..1</sub>
     * Description: Resting pulse, measured as beats per minute
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 65 beats per minute None
 * [human-associated➞gestation_state](human_associated_gestation_state.md)  <sub>0..1</sub>
     * Description: Specification of the gestation state
     * Range: [String](types/String.md)
     * Example:  None
 * [human-associated➞maternal_health_stat](human_associated_maternal_health_stat.md)  <sub>0..1</sub>
     * Description: Specification of the maternal health status
     * Range: [String](types/String.md)
     * Example:  None
 * [human-associated➞foetal_health_stat](human_associated_foetal_health_stat.md)  <sub>0..1</sub>
     * Description: Specification of foetal health status, should also include abortion
     * Range: [String](types/String.md)
     * Example:  None
 * [human-associated➞amniotic_fluid_color](human_associated_amniotic_fluid_color.md)  <sub>0..1</sub>
     * Description: Specification of the color of the amniotic fluid sample
     * Range: [String](types/String.md)
     * Example:  None
 * [human-associated➞kidney_disord](human_associated_kidney_disord.md)  <sub>0..\*</sub>
     * Description: History of kidney disorders; can include multiple disorders. The terms should be chosen from the DO (Human Disease Ontology) at http://www.disease-ontology.org, kidney disease (https://disease-ontology.org/?id=DOID:557).
     * Range: [String](types/String.md)
     * Example:  None
 * [human-associated➞urogenit_tract_disor](human_associated_urogenit_tract_disor.md)  <sub>0..\*</sub>
     * Description: History of urogenital tract disorders; can include multiple disorders. The terms should be chosen from the DO (Human Disease Ontology) at http://www.disease-ontology.org, urinary system disease (https://disease-ontology.org/?id=DOID:18).
     * Range: [String](types/String.md)
     * Example:  None
 * [human-associated➞urine_collect_meth](human_associated_urine_collect_meth.md)  <sub>0..1</sub>
     * Description: Specification of urine collection method
     * Range: [String](types/String.md)
     * Example: catheter None
 * [human-associated➞perturbation](human_associated_perturbation.md)  <sub>0..\*</sub>
     * Description: Type of perturbation, e.g. chemical administration, physical disturbance, etc., coupled with perturbation regimen including how many times the perturbation was repeated, how long each perturbation lasted, and the start and end time of the entire perturbation period; can include multiple perturbation types
     * Range: [String](types/String.md)
     * Example: antibiotic addition;R2/2018-05-11T14:30Z/2018-05-11T19:30Z/P1H30M None
 * [human-associated➞samp_salinity](human_associated_samp_salinity.md)  <sub>0..1</sub>
     * Description: Salinity is the total concentration of all dissolved salts in a liquid or solid (in the form of an extract obtained by centrifugation) sample. While salinity can be measured by a complete chemical analysis, this method is difficult and time consuming. More often, it is instead derived from the conductivity measurement. This is known as practical salinity. These derivations compare the specific conductance of the sample to a salinity standard such as seawater
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 1 milligram per liter None
 * [human-associated➞salinity](human_associated_salinity.md)  <sub>0..1</sub>
     * Description: The total concentration of all dissolved salts in a liquid or solid sample. While salinity can be measured by a complete chemical analysis, this method is difficult and time consuming. More often, it is instead derived from the conductivity measurement. This is known as practical salinity. These derivations compare the specific conductance of the sample to a salinity standard such as seawater.
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 25 practical salinity unit None
 * [human-associated➞oxy_stat_samp](human_associated_oxy_stat_samp.md)  <sub>0..1</sub>
     * Description: Oxygenation status of sample
     * Range: [oxy_stat_samp_enum](oxy_stat_samp_enum.md)
     * Example: aerobic None
 * [human-associated➞organism_count](human_associated_organism_count.md)  <sub>0..\*</sub>
     * Description: Total cell count of any organism (or group of organisms) per gram, volume or area of sample, should include name of organism followed by count. The method that was used for the enumeration (e.g. qPCR, atp, mpn, etc.) Should also be provided. (example: total prokaryotes; 3.5e7 cells per ml; qpcr)
     * Range: [organism_count_enum](organism_count_enum.md)
     * Example: total prokaryotes;3.5e7 cells per milliliter;qPCR None
 * [human-associated➞samp_store_temp](human_associated_samp_store_temp.md)  <sub>0..1</sub>
     * Description: Temperature at which sample was stored, e.g. -80 degree Celsius
     * Range: [QuantityValue](QuantityValue.md)
     * Example: -80 degree Celsius None
 * [human-associated➞samp_store_dur](human_associated_samp_store_dur.md)  <sub>0..1</sub>
     * Description: Duration for which the sample was stored
     * Range: [String](types/String.md)
     * Example: P1Y6M None
 * [human-associated➞host_symbiont](human_associated_host_symbiont.md)  <sub>0..\*</sub>
     * Description: The taxonomic name of the organism(s) found living in mutualistic, commensalistic, or parasitic symbiosis with the specific host.
     * Range: [String](types/String.md)
     * Example: flukeworms None
 * [human-associated➞samp_store_loc](human_associated_samp_store_loc.md)  <sub>0..1</sub>
     * Description: Location at which sample was stored, usually name of a specific freezer/room
     * Range: [String](types/String.md)
     * Example: Freezer no:5 None
 * [human-associated➞misc_param](human_associated_misc_param.md)  <sub>0..\*</sub>
     * Description: Any other measurement performed or parameter collected, that is not listed here
     * Range: [String](types/String.md)
     * Example: Bicarbonate ion concentration;2075 micromole per kilogram None
 * [human-associated➞samp_name](human_associated_samp_name.md)  <sub>1..1</sub>
     * Description: A local identifier or name that for the material sample used for extracting nucleic acids, and subsequent sequencing. It can refer either to the original material collected or to any derived sub-samples. It can have any format, but we suggest that you make it concise, unique and consistent within your lab, and as informative as possible. INSDC requires every sample name from a single Submitter to be unique. Use of a globally unique identifier for the field source_mat_id is recommended in addition to sample_name.
     * Range: [String](types/String.md)
     * Example: ISDsoil1 None
 * [human-associated➞project_name](human_associated_project_name.md)  <sub>1..1</sub>
     * Description: Name of the project within which the sequencing was organized
     * Range: [String](types/String.md)
     * Example: Forest soil metagenome None
 * [human-associated➞host_disease_stat](human_associated_host_disease_stat.md)  <sub>0..1</sub>
     * Description: List of diseases with which the host has been diagnosed; can include multiple diagnoses. The value of the field depends on host; for humans the terms should be chosen from the DO (Human Disease Ontology) at https://www.disease-ontology.org, non-human host diseases are free text
     * Range: [String](types/String.md)
     * Example: rabies [DOID:11260] None
 * [human-associated➞samp_vol_we_dna_ext](human_associated_samp_vol_we_dna_ext.md)  <sub>0..1</sub>
     * Description: Volume (ml) or mass (g) of total collected sample processed for DNA extraction. Note: total sample collected should be entered under the term Sample Size (MIXS:0000001).
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 1500 milliliter None
