
# Class: hydrocarbon resources-cores MIUVIG


Combinatorial checklist Minimum Information About an Uncultivated Virus Genome with environmental package hydrocarbon resources-cores

URI: [mixs.vocab:HydrocarbonResources-coresMIUVIG](https://w3id.org/mixs/vocab/HydrocarbonResources-coresMIUVIG)


[![img](https://yuml.me/diagram/nofunky;dir:TB/class/[QuantityValue],[HydrocarbonResources-coresMIUVIG&#124;submitted_to_insdc:string;investigation_type:investigation_type_enum;samp_taxon_id:string;experimental_factor:string%20%3F;neg_cont_type:neg_cont_type_enum%20%3F;pos_cont_type:string%20%3F;env_package:env_package_enum%20%3F;estimated_size:string%20%3F;ref_biomaterial:string%20%3F;source_mat_id:string%20%3F;pathogenicity:string%20%3F;biotic_relationship:biotic_relationship_enum%20%3F;specific_host:string%20%3F;host_spec_range:integer%20%3F;host_disease_stat:string%20%3F;samp_collec_device:string%20%3F;samp_collec_method:string%20%3F;samp_mat_process:string%20%3F;size_frac:string%20%3F;source_uvig:source_uvig_enum;virus_enrich_appr:virus_enrich_appr_enum;nucl_acid_ext:string%20%3F;nucl_acid_amp:string%20%3F;lib_size:integer%20%3F;lib_reads_seqd:integer%20%3F;lib_layout:lib_layout_enum%20%3F;lib_vector:string%20%3F;lib_screen:string%20%3F;mid:string%20%3F;adapters:string%20%3F;seq_meth:string;tax_ident:tax_ident_enum%20%3F;assembly_qual:assembly_qual_enum;assembly_name:string%20%3F;assembly_software:string;annot:string%20%3F;number_contig:integer;feat_pred:string%20%3F;ref_db:string%20%3F;sim_search_meth:string%20%3F;tax_class:string%20%3F;trnas:integer%20%3F;trna_ext_software:string%20%3F;compl_score:compl_score_enum%20%3F;compl_software:string%20%3F;compl_appr:compl_appr_enum%20%3F;sort_tech:sort_tech_enum%20%3F;single_cell_lysis_appr:single_cell_lysis_appr_enum%20%3F;single_cell_lysis_prot:string%20%3F;wga_amp_appr:string%20%3F;wga_amp_kit:string%20%3F;bin_param:bin_param_enum%20%3F;bin_software:string%20%3F;reassembly_bin:string%20%3F;mag_cov_software:mag_cov_software_enum%20%3F;vir_ident_software:string;pred_genome_type:pred_genome_type_enum;pred_genome_struc:pred_genome_struc_enum;detec_type:string;otu_class_appr:string%20%3F;otu_seq_comp_appr:string%20%3F;otu_db:string%20%3F;host_pred_appr:host_pred_appr_enum%20%3F;host_pred_est_acc:string%20%3F;associated_resource:string%20%3F;sop:string%20%3F;lat_lon(i):string%20%3F;geo_loc_name(i):string%20%3F;collection_date(i):date%20%3F;env_broad_scale(i):string%20%3F;env_local_scale(i):string%20%3F;env_medium(i):string%20%3F;hcr(i):hcr_enum;hc_produced(i):hc_produced_enum;basin(i):string;field(i):string%20%3F;reservoir(i):string%20%3F;hcr_temp(i):string;hcr_pressure(i):string%20%3F;permeability(i):string%20%3F;porosity(i):string%20%3F;lithology(i):lithology_enum%20%3F;depos_env(i):depos_env_enum%20%3F;hcr_geol_age(i):hcr_geol_age_enum%20%3F;sr_kerog_type(i):sr_kerog_type_enum%20%3F;sr_lithology(i):sr_lithology_enum%20%3F;sr_dep_env(i):sr_dep_env_enum%20%3F;sr_geol_age(i):sr_geol_age_enum%20%3F;samp_well_name(i):string%20%3F;win(i):string%20%3F;samp_type(i):string;samp_subtype(i):samp_subtype_enum%20%3F;samp_tvdss(i):string%20%3F;samp_md(i):samp_md_enum%20%3F;oxy_stat_samp(i):oxy_stat_samp_enum%20%3F;samp_transport_cond(i):string%20%3F;samp_store_dur(i):string%20%3F;samp_store_loc(i):string%20%3F;organism_count(i):organism_count_enum%20*;org_count_qpcr_info(i):string%20%3F;ph(i):double%20%3F;alkalinity_method(i):string%20%3F;suspend_solids(i):string%20*;viscosity(i):string%20%3F;saturates_pc(i):string%20%3F;aromatics_pc(i):string%20%3F;resins_pc(i):string%20%3F;asphaltenes_pc(i):string%20%3F;misc_param(i):string%20*;additional_info(i):string%20%3F;samp_name(i):string;project_name(i):string]uses%20-.->[MIUVIG],[HydrocarbonResources-cores]^-[HydrocarbonResources-coresMIUVIG],[HydrocarbonResources-cores],[MIUVIG])](https://yuml.me/diagram/nofunky;dir:TB/class/[QuantityValue],[HydrocarbonResources-coresMIUVIG&#124;submitted_to_insdc:string;investigation_type:investigation_type_enum;samp_taxon_id:string;experimental_factor:string%20%3F;neg_cont_type:neg_cont_type_enum%20%3F;pos_cont_type:string%20%3F;env_package:env_package_enum%20%3F;estimated_size:string%20%3F;ref_biomaterial:string%20%3F;source_mat_id:string%20%3F;pathogenicity:string%20%3F;biotic_relationship:biotic_relationship_enum%20%3F;specific_host:string%20%3F;host_spec_range:integer%20%3F;host_disease_stat:string%20%3F;samp_collec_device:string%20%3F;samp_collec_method:string%20%3F;samp_mat_process:string%20%3F;size_frac:string%20%3F;source_uvig:source_uvig_enum;virus_enrich_appr:virus_enrich_appr_enum;nucl_acid_ext:string%20%3F;nucl_acid_amp:string%20%3F;lib_size:integer%20%3F;lib_reads_seqd:integer%20%3F;lib_layout:lib_layout_enum%20%3F;lib_vector:string%20%3F;lib_screen:string%20%3F;mid:string%20%3F;adapters:string%20%3F;seq_meth:string;tax_ident:tax_ident_enum%20%3F;assembly_qual:assembly_qual_enum;assembly_name:string%20%3F;assembly_software:string;annot:string%20%3F;number_contig:integer;feat_pred:string%20%3F;ref_db:string%20%3F;sim_search_meth:string%20%3F;tax_class:string%20%3F;trnas:integer%20%3F;trna_ext_software:string%20%3F;compl_score:compl_score_enum%20%3F;compl_software:string%20%3F;compl_appr:compl_appr_enum%20%3F;sort_tech:sort_tech_enum%20%3F;single_cell_lysis_appr:single_cell_lysis_appr_enum%20%3F;single_cell_lysis_prot:string%20%3F;wga_amp_appr:string%20%3F;wga_amp_kit:string%20%3F;bin_param:bin_param_enum%20%3F;bin_software:string%20%3F;reassembly_bin:string%20%3F;mag_cov_software:mag_cov_software_enum%20%3F;vir_ident_software:string;pred_genome_type:pred_genome_type_enum;pred_genome_struc:pred_genome_struc_enum;detec_type:string;otu_class_appr:string%20%3F;otu_seq_comp_appr:string%20%3F;otu_db:string%20%3F;host_pred_appr:host_pred_appr_enum%20%3F;host_pred_est_acc:string%20%3F;associated_resource:string%20%3F;sop:string%20%3F;lat_lon(i):string%20%3F;geo_loc_name(i):string%20%3F;collection_date(i):date%20%3F;env_broad_scale(i):string%20%3F;env_local_scale(i):string%20%3F;env_medium(i):string%20%3F;hcr(i):hcr_enum;hc_produced(i):hc_produced_enum;basin(i):string;field(i):string%20%3F;reservoir(i):string%20%3F;hcr_temp(i):string;hcr_pressure(i):string%20%3F;permeability(i):string%20%3F;porosity(i):string%20%3F;lithology(i):lithology_enum%20%3F;depos_env(i):depos_env_enum%20%3F;hcr_geol_age(i):hcr_geol_age_enum%20%3F;sr_kerog_type(i):sr_kerog_type_enum%20%3F;sr_lithology(i):sr_lithology_enum%20%3F;sr_dep_env(i):sr_dep_env_enum%20%3F;sr_geol_age(i):sr_geol_age_enum%20%3F;samp_well_name(i):string%20%3F;win(i):string%20%3F;samp_type(i):string;samp_subtype(i):samp_subtype_enum%20%3F;samp_tvdss(i):string%20%3F;samp_md(i):samp_md_enum%20%3F;oxy_stat_samp(i):oxy_stat_samp_enum%20%3F;samp_transport_cond(i):string%20%3F;samp_store_dur(i):string%20%3F;samp_store_loc(i):string%20%3F;organism_count(i):organism_count_enum%20*;org_count_qpcr_info(i):string%20%3F;ph(i):double%20%3F;alkalinity_method(i):string%20%3F;suspend_solids(i):string%20*;viscosity(i):string%20%3F;saturates_pc(i):string%20%3F;aromatics_pc(i):string%20%3F;resins_pc(i):string%20%3F;asphaltenes_pc(i):string%20%3F;misc_param(i):string%20*;additional_info(i):string%20%3F;samp_name(i):string;project_name(i):string]uses%20-.->[MIUVIG],[HydrocarbonResources-cores]^-[HydrocarbonResources-coresMIUVIG],[HydrocarbonResources-cores],[MIUVIG])

## Parents

 *  is_a: [HydrocarbonResources-cores](HydrocarbonResources-cores.md) - hydrocarbon resources-cores

## Uses Mixin

 *  mixin: [MIUVIG](MIUVIG.md) - Minimum Information About an Uncultivated Virus Genome

## Attributes


### Inherited from hydrocarbon resources-cores:

 * [lat_lon](lat_lon.md)  <sub>0..1</sub>
     * Description: The geographical origin of the sample as defined by latitude and longitude. The values should be reported in decimal degrees and in WGS84 system
     * Range: [String](types/String.md)
     * Example: 50.586825 6.408977 None
 * [depth](depth.md)  <sub>0..1</sub>
     * Description: The vertical distance below local surface, e.g. for sediment or soil samples depth is measured from sediment or soil surface, respectively. Depth can be reported as an interval for subsurface samples.
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 10 meter None
 * [alt](alt.md)  <sub>0..1</sub>
     * Description: Altitude is a term used to identify heights of objects such as airplanes, space shuttles, rockets, atmospheric balloons and heights of places such as atmospheric layers and clouds. It is used to measure the height of an object which is above the earth's surface. In this context, the altitude measurement is the vertical distance between the earth's surface above sea level and the sampled position in the air
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 100 meter None
 * [hydrocarbon resources-cores➞elev](hydrocarbon_resources_cores_elev.md)  <sub>0..1</sub>
     * Description: Elevation of the sampling site is its height above a fixed reference point, most commonly the mean sea level. Elevation is mainly used when referring to points on the earth's surface, while altitude is used for points above the surface, such as an aircraft in flight or a spacecraft in orbit.
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 100 meter None
 * [hydrocarbon resources-cores➞temp](hydrocarbon_resources_cores_temp.md)  <sub>1..1</sub>
     * Description: Temperature of the sample at the time of sampling.
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 25 degree Celsius None
 * [geo_loc_name](geo_loc_name.md)  <sub>0..1</sub>
     * Description: The geographical origin of the sample as defined by the country or sea name followed by specific region name. Country or sea names should be chosen from the INSDC country list (http://insdc.org/country.html), or the GAZ ontology (http://purl.bioontology.org/ontology/GAZ)
     * Range: [String](types/String.md)
     * Example: USA: Maryland, Bethesda None
 * [collection_date](collection_date.md)  <sub>0..1</sub>
     * Description: The time of sampling, either as an instance (single point in time) or interval. In case no exact time is available, the date/time can be right truncated i.e. all of these are valid times: 2008-01-23T19:23:10+00:00; 2008-01-23T19:23:10; 2008-01-23; 2008-01; 2008; Except: 2008-01; 2008 all are ISO8601 compliant
     * Range: [Date](types/Date.md)
     * Example: 2018-05-11T10:00:00+01:00; 2018-05-11 None
 * [env_broad_scale](env_broad_scale.md)  <sub>0..1</sub>
     * Description: Report the major environmental system the sample or specimen came from. The system(s) identified should have a coarse spatial grain, to provide the general environmental context of where the sampling was done (e.g. in the desert or a rainforest). We recommend using subclasses of EnvO’s biome class:  http://purl.obolibrary.org/obo/ENVO_00000428. EnvO documentation about how to use the field: https://github.com/EnvironmentOntology/envo/wiki/Using-ENVO-with-MIxS
     * Range: [String](types/String.md)
     * Example: oceanic epipelagic zone biome [ENVO:01000033] for annotating a water sample from the photic zone in middle of the Atlantic Ocean None
 * [env_local_scale](env_local_scale.md)  <sub>0..1</sub>
     * Description: Report the entity or entities which are in the sample or specimen’s local vicinity and which you believe have significant causal influences on your sample or specimen. We recommend using EnvO terms which are of smaller spatial grain than your entry for env_broad_scale. Terms, such as anatomical sites, from other OBO Library ontologies which interoperate with EnvO (e.g. UBERON) are accepted in this field. EnvO documentation about how to use the field: https://github.com/EnvironmentOntology/envo/wiki/Using-ENVO-with-MIxS.
     * Range: [String](types/String.md)
     * Example: litter layer [ENVO:01000338]; Annotating a pooled sample taken from various vegetation layers in a forest consider: canopy [ENVO:00000047]|herb and fern layer [ENVO:01000337]|litter layer [ENVO:01000338]|understory [01000335]|shrub layer [ENVO:01000336]. None
 * [env_medium](env_medium.md)  <sub>0..1</sub>
     * Description: Report the environmental material(s) immediately surrounding the sample or specimen at the time of sampling. We recommend using subclasses of 'environmental material' (http://purl.obolibrary.org/obo/ENVO_00010483). EnvO documentation about how to use the field: https://github.com/EnvironmentOntology/envo/wiki/Using-ENVO-with-MIxS . Terms from other OBO ontologies are permissible as long as they reference mass/volume nouns (e.g. air, water, blood) and not discrete, countable entities (e.g. a tree, a leaf, a table top).
     * Range: [String](types/String.md)
     * Example: soil [ENVO:00001998]; Annotating a fish swimming in the upper 100 m of the Atlantic Ocean, consider: ocean water [ENVO:00002151]. Example: Annotating a duck on a pond consider: pond water [ENVO:00002228]|air [ENVO_00002005] None
 * [hydrocarbon resources-cores➞hcr](hydrocarbon_resources_cores_hcr.md)  <sub>1..1</sub>
     * Description: Main Hydrocarbon Resource type. The term "Hydrocarbon Resource" HCR defined as a natural environmental feature containing large amounts of hydrocarbons at high concentrations potentially suitable for commercial exploitation. This term should not be confused with the Hydrocarbon Occurrence term which also includes hydrocarbon-rich environments with currently limited commercial interest such as seeps, outcrops, gas hydrates etc. If "other" is specified, please propose entry in "additional info" field
     * Range: [hcr_enum](hcr_enum.md)
     * Example: Oil Sand None
 * [hydrocarbon resources-cores➞hc_produced](hydrocarbon_resources_cores_hc_produced.md)  <sub>1..1</sub>
     * Description: Main hydrocarbon type produced from resource (i.e. Oil, gas, condensate, etc). If "other" is specified, please propose entry in "additional info" field
     * Range: [hc_produced_enum](hc_produced_enum.md)
     * Example: Gas None
 * [hydrocarbon resources-cores➞basin](hydrocarbon_resources_cores_basin.md)  <sub>1..1</sub>
     * Description: Name of the basin (e.g. Campos)
     * Range: [String](types/String.md)
     * Example:  None
 * [hydrocarbon resources-cores➞field](hydrocarbon_resources_cores_field.md)  <sub>0..1</sub>
     * Description: Name of the hydrocarbon field (e.g. Albacora)
     * Range: [String](types/String.md)
     * Example:  None
 * [hydrocarbon resources-cores➞reservoir](hydrocarbon_resources_cores_reservoir.md)  <sub>0..1</sub>
     * Description: Name of the reservoir (e.g. Carapebus)
     * Range: [String](types/String.md)
     * Example:  None
 * [hydrocarbon resources-cores➞hcr_temp](hydrocarbon_resources_cores_hcr_temp.md)  <sub>1..1</sub>
     * Description: Original temperature of the hydrocarbon resource
     * Range: [String](types/String.md)
     * Example: 150-295 degree Celsius None
 * [hydrocarbon resources-cores➞tvdss_of_hcr_temp](hydrocarbon_resources_cores_tvdss_of_hcr_temp.md)  <sub>0..1</sub>
     * Description: True vertical depth subsea (TVDSS) of the hydrocarbon resource where the original temperature was measured (e.g. 1345 m).
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞hcr_pressure](hydrocarbon_resources_cores_hcr_pressure.md)  <sub>0..1</sub>
     * Description: Original pressure of the hydrocarbon resource
     * Range: [String](types/String.md)
     * Example:  None
 * [hydrocarbon resources-cores➞tvdss_of_hcr_press](hydrocarbon_resources_cores_tvdss_of_hcr_press.md)  <sub>0..1</sub>
     * Description: True vertical depth subsea (TVDSS) of the hydrocarbon resource where the original pressure was measured (e.g. 1578 m).
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞permeability](hydrocarbon_resources_cores_permeability.md)  <sub>0..1</sub>
     * Description: Measure of the ability of a hydrocarbon resource to allow fluids to pass through it. (Additional information: https://en.wikipedia.org/wiki/Permeability_(earth_sciences))
     * Range: [String](types/String.md)
     * Example:  None
 * [hydrocarbon resources-cores➞porosity](hydrocarbon_resources_cores_porosity.md)  <sub>0..1</sub>
     * Description: Porosity of deposited sediment is volume of voids divided by the total volume of sample
     * Range: [String](types/String.md)
     * Example:  None
 * [hydrocarbon resources-cores➞lithology](hydrocarbon_resources_cores_lithology.md)  <sub>0..1</sub>
     * Description: Hydrocarbon resource main lithology (Additional information: http://petrowiki.org/Lithology_and_rock_type_determination). If "other" is specified, please propose entry in "additional info" field
     * Range: [lithology_enum](lithology_enum.md)
     * Example: Volcanic None
 * [hydrocarbon resources-cores➞depos_env](hydrocarbon_resources_cores_depos_env.md)  <sub>0..1</sub>
     * Description: Main depositional environment (https://en.wikipedia.org/wiki/Depositional_environment). If "other" is specified, please propose entry in "additional info" field
     * Range: [depos_env_enum](depos_env_enum.md)
     * Example: Continental - Alluvial None
 * [hydrocarbon resources-cores➞hcr_geol_age](hydrocarbon_resources_cores_hcr_geol_age.md)  <sub>0..1</sub>
     * Description: Geological age of hydrocarbon resource (Additional info: https://en.wikipedia.org/wiki/Period_(geology)). If "other" is specified, please propose entry in "additional info" field
     * Range: [hcr_geol_age_enum](hcr_geol_age_enum.md)
     * Example: Silurian None
 * [hydrocarbon resources-cores➞owc_tvdss](hydrocarbon_resources_cores_owc_tvdss.md)  <sub>0..1</sub>
     * Description: Depth of the original oil water contact (OWC) zone (average) (m TVDSS)
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞hcr_fw_salinity](hydrocarbon_resources_cores_hcr_fw_salinity.md)  <sub>0..1</sub>
     * Description: Original formation water salinity (prior to secondary recovery e.g. Waterflooding) expressed as TDS
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞sulfate_fw](hydrocarbon_resources_cores_sulfate_fw.md)  <sub>1..1</sub>
     * Description: Original sulfate concentration in the hydrocarbon resource
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞vfa_fw](hydrocarbon_resources_cores_vfa_fw.md)  <sub>1..1</sub>
     * Description: Original volatile fatty acid concentration in the hydrocarbon resource
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞sr_kerog_type](hydrocarbon_resources_cores_sr_kerog_type.md)  <sub>0..1</sub>
     * Description: Origin of kerogen. Type I: Algal (aquatic), Type II: planktonic and soft plant material (aquatic or terrestrial), Type III: terrestrial woody/ fibrous plant material (terrestrial), Type IV: oxidized recycled woody debris (terrestrial) (additional information: https://en.wikipedia.org/wiki/Kerogen). If "other" is specified, please propose entry in "additional info" field
     * Range: [sr_kerog_type_enum](sr_kerog_type_enum.md)
     * Example: Type IV None
 * [hydrocarbon resources-cores➞sr_lithology](hydrocarbon_resources_cores_sr_lithology.md)  <sub>0..1</sub>
     * Description: Lithology of source rock (https://en.wikipedia.org/wiki/Source_rock). If "other" is specified, please propose entry in "additional info" field
     * Range: [sr_lithology_enum](sr_lithology_enum.md)
     * Example: Coal None
 * [hydrocarbon resources-cores➞sr_dep_env](hydrocarbon_resources_cores_sr_dep_env.md)  <sub>0..1</sub>
     * Description: Source rock depositional environment (https://en.wikipedia.org/wiki/Source_rock). If "other" is specified, please propose entry in "additional info" field
     * Range: [sr_dep_env_enum](sr_dep_env_enum.md)
     * Example: Marine None
 * [hydrocarbon resources-cores➞sr_geol_age](hydrocarbon_resources_cores_sr_geol_age.md)  <sub>0..1</sub>
     * Description: Geological age of source rock (Additional info: https://en.wikipedia.org/wiki/Period_(geology)). If "other" is specified, please propose entry in "additional info" field
     * Range: [sr_geol_age_enum](sr_geol_age_enum.md)
     * Example: Silurian None
 * [hydrocarbon resources-cores➞samp_well_name](hydrocarbon_resources_cores_samp_well_name.md)  <sub>0..1</sub>
     * Description: Name of the well (e.g. BXA1123) where sample was taken
     * Range: [String](types/String.md)
     * Example:  None
 * [hydrocarbon resources-cores➞win](hydrocarbon_resources_cores_win.md)  <sub>0..1</sub>
     * Description: A unique identifier of a well or wellbore. This is part of the Global Framework for Well Identification initiative which is compiled by the Professional Petroleum Data Management Association (PPDM) in an effort to improve well identification systems. (Supporting information: https://ppdm.org/ and http://dl.ppdm.org/dl/690)
     * Range: [String](types/String.md)
     * Example:  None
 * [hydrocarbon resources-cores➞samp_type](hydrocarbon_resources_cores_samp_type.md)  <sub>1..1</sub>
     * Description: The type of material from which the sample was obtained. For the Hydrocarbon package, samples include types like core, rock trimmings, drill cuttings, piping section, coupon, pigging debris, solid deposit, produced fluid, produced water, injected water, swabs, etc. For the Food Package, samples are usually categorized as food, body products or tissues, or environmental material. This field accepts terms listed under environmental specimen (http://purl.obolibrary.org/obo/GENEPIO_0001246).
     * Range: [String](types/String.md)
     * Example: built environment sample [GENEPIO:0001248] None
 * [hydrocarbon resources-cores➞samp_subtype](hydrocarbon_resources_cores_samp_subtype.md)  <sub>0..1</sub>
     * Description: Name of sample sub-type. For example if "sample type" is "Produced Water" then subtype could be "Oil Phase" or "Water Phase". If "other" is specified, please propose entry in "additional info" field
     * Range: [samp_subtype_enum](samp_subtype_enum.md)
     * Example: biofilm None
 * [hydrocarbon resources-cores➞pressure](hydrocarbon_resources_cores_pressure.md)  <sub>0..1</sub>
     * Description: Pressure to which the sample is subject to, in atmospheres
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 50 atmosphere None
 * [hydrocarbon resources-cores➞samp_tvdss](hydrocarbon_resources_cores_samp_tvdss.md)  <sub>0..1</sub>
     * Description: Depth of the sample i.e. The vertical distance between the sea level and the sampled position in the subsurface. Depth can be reported as an interval for subsurface samples e.g. 1325.75-1362.25 m
     * Range: [String](types/String.md)
     * Example:  None
 * [hydrocarbon resources-cores➞samp_md](hydrocarbon_resources_cores_samp_md.md)  <sub>0..1</sub>
     * Description: In non deviated well, measured depth is equal to the true vertical depth, TVD (TVD=TVDSS plus the reference or datum it refers to). In deviated wells, the MD is the length of trajectory of the borehole measured from the same reference or datum. Common datums used are ground level (GL), drilling rig floor (DF), rotary table (RT), kelly bushing (KB) and mean sea level (MSL). If "other" is specified, please propose entry in "additional info" field
     * Range: [samp_md_enum](samp_md_enum.md)
     * Example: 1534 meter;MSL None
 * [hydrocarbon resources-cores➞oxy_stat_samp](hydrocarbon_resources_cores_oxy_stat_samp.md)  <sub>0..1</sub>
     * Description: Oxygenation status of sample
     * Range: [oxy_stat_samp_enum](oxy_stat_samp_enum.md)
     * Example: aerobic None
 * [hydrocarbon resources-cores➞samp_transport_cond](hydrocarbon_resources_cores_samp_transport_cond.md)  <sub>0..1</sub>
     * Description: Sample transport duration (in days or hrs) and temperature the sample was exposed to (e.g. 5.5 days; 20 ¬∞C)
     * Range: [String](types/String.md)
     * Example: 5 days;-20 degree Celsius None
 * [hydrocarbon resources-cores➞samp_store_temp](hydrocarbon_resources_cores_samp_store_temp.md)  <sub>0..1</sub>
     * Description: Temperature at which sample was stored, e.g. -80 degree Celsius
     * Range: [QuantityValue](QuantityValue.md)
     * Example: -80 degree Celsius None
 * [hydrocarbon resources-cores➞samp_store_dur](hydrocarbon_resources_cores_samp_store_dur.md)  <sub>0..1</sub>
     * Description: Duration for which the sample was stored
     * Range: [String](types/String.md)
     * Example: P1Y6M None
 * [hydrocarbon resources-cores➞samp_store_loc](hydrocarbon_resources_cores_samp_store_loc.md)  <sub>0..1</sub>
     * Description: Location at which sample was stored, usually name of a specific freezer/room
     * Range: [String](types/String.md)
     * Example: Freezer no:5 None
 * [hydrocarbon resources-cores➞organism_count](hydrocarbon_resources_cores_organism_count.md)  <sub>0..\*</sub>
     * Description: Total cell count of any organism (or group of organisms) per gram, volume or area of sample, should include name of organism followed by count. The method that was used for the enumeration (e.g. qPCR, atp, mpn, etc.) Should also be provided. (example: total prokaryotes; 3.5e7 cells per ml; qpcr)
     * Range: [organism_count_enum](organism_count_enum.md)
     * Example: total prokaryotes;3.5e7 cells per milliliter;qPCR None
 * [hydrocarbon resources-cores➞org_count_qpcr_info](hydrocarbon_resources_cores_org_count_qpcr_info.md)  <sub>0..1</sub>
     * Description: If qpcr was used for the cell count, the target gene name, the primer sequence and the cycling conditions should also be provided. (Example: 16S rrna; FWD:ACGTAGCTATGACGT REV:GTGCTAGTCGAGTAC; initial denaturation:90C_5min; denaturation:90C_2min; annealing:52C_30 sec; elongation:72C_30 sec; 90 C for 1 min; final elongation:72C_5min; 30 cycles)
     * Range: [String](types/String.md)
     * Example:  None
 * [hydrocarbon resources-cores➞ph](hydrocarbon_resources_cores_ph.md)  <sub>0..1</sub>
     * Description: Ph measurement of the sample, or liquid portion of sample, or aqueous phase of the fluid
     * Range: [Double](types/Double.md)
     * Example: 7.2 None
 * [hydrocarbon resources-cores➞samp_salinity](hydrocarbon_resources_cores_samp_salinity.md)  <sub>1..1</sub>
     * Description: Salinity is the total concentration of all dissolved salts in a liquid or solid (in the form of an extract obtained by centrifugation) sample. While salinity can be measured by a complete chemical analysis, this method is difficult and time consuming. More often, it is instead derived from the conductivity measurement. This is known as practical salinity. These derivations compare the specific conductance of the sample to a salinity standard such as seawater
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 1 milligram per liter None
 * [hydrocarbon resources-cores➞salinity](hydrocarbon_resources_cores_salinity.md)  <sub>0..1</sub>
     * Description: The total concentration of all dissolved salts in a liquid or solid sample. While salinity can be measured by a complete chemical analysis, this method is difficult and time consuming. More often, it is instead derived from the conductivity measurement. This is known as practical salinity. These derivations compare the specific conductance of the sample to a salinity standard such as seawater.
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 25 practical salinity unit None
 * [hydrocarbon resources-cores➞alkalinity](hydrocarbon_resources_cores_alkalinity.md)  <sub>0..1</sub>
     * Description: Alkalinity, the ability of a solution to neutralize acids to the equivalence point of carbonate or bicarbonate
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 50 milligram per liter None
 * [hydrocarbon resources-cores➞alkalinity_method](hydrocarbon_resources_cores_alkalinity_method.md)  <sub>0..1</sub>
     * Description: Method used for alkalinity measurement
     * Range: [String](types/String.md)
     * Example: titration None
 * [hydrocarbon resources-cores➞sulfate](hydrocarbon_resources_cores_sulfate.md)  <sub>0..1</sub>
     * Description: Concentration of sulfate in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 5 micromole per liter None
 * [hydrocarbon resources-cores➞sulfide](hydrocarbon_resources_cores_sulfide.md)  <sub>0..1</sub>
     * Description: Concentration of sulfide in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 2 micromole per liter None
 * [hydrocarbon resources-cores➞tot_sulfur](hydrocarbon_resources_cores_tot_sulfur.md)  <sub>0..1</sub>
     * Description: Concentration of total sulfur in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞nitrate](hydrocarbon_resources_cores_nitrate.md)  <sub>0..1</sub>
     * Description: Concentration of nitrate in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 65 micromole per liter None
 * [hydrocarbon resources-cores➞nitrite](hydrocarbon_resources_cores_nitrite.md)  <sub>0..1</sub>
     * Description: Concentration of nitrite in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 0.5 micromole per liter None
 * [hydrocarbon resources-cores➞ammonium](hydrocarbon_resources_cores_ammonium.md)  <sub>0..1</sub>
     * Description: Concentration of ammonium in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 1.5 milligram per liter None
 * [hydrocarbon resources-cores➞tot_nitro](hydrocarbon_resources_cores_tot_nitro.md)  <sub>0..1</sub>
     * Description: Total nitrogen concentration of water samples, calculated by: total nitrogen = total dissolved nitrogen + particulate nitrogen. Can also be measured without filtering, reported as nitrogen
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 50 micromole per liter None
 * [hydrocarbon resources-cores➞diss_iron](hydrocarbon_resources_cores_diss_iron.md)  <sub>0..1</sub>
     * Description: Concentration of dissolved iron in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞sodium](hydrocarbon_resources_cores_sodium.md)  <sub>0..1</sub>
     * Description: Sodium concentration in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 10.5 milligram per liter None
 * [hydrocarbon resources-cores➞chloride](hydrocarbon_resources_cores_chloride.md)  <sub>0..1</sub>
     * Description: Concentration of chloride in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 5000 milligram per liter None
 * [hydrocarbon resources-cores➞potassium](hydrocarbon_resources_cores_potassium.md)  <sub>0..1</sub>
     * Description: Concentration of potassium in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 463 milligram per liter None
 * [hydrocarbon resources-cores➞magnesium](hydrocarbon_resources_cores_magnesium.md)  <sub>0..1</sub>
     * Description: Concentration of magnesium in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 52.8 micromole per kilogram None
 * [hydrocarbon resources-cores➞calcium](hydrocarbon_resources_cores_calcium.md)  <sub>0..1</sub>
     * Description: Concentration of calcium in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 0.2 micromole per liter None
 * [hydrocarbon resources-cores➞tot_iron](hydrocarbon_resources_cores_tot_iron.md)  <sub>0..1</sub>
     * Description: Concentration of total iron in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞diss_org_carb](hydrocarbon_resources_cores_diss_org_carb.md)  <sub>0..1</sub>
     * Description: Concentration of dissolved organic carbon in the sample, liquid portion of the sample, or aqueous phase of the fluid
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 197 micromole per liter None
 * [hydrocarbon resources-cores➞diss_inorg_carb](hydrocarbon_resources_cores_diss_inorg_carb.md)  <sub>0..1</sub>
     * Description: Dissolved inorganic carbon concentration in the sample, typically measured after filtering the sample using a 0.45 micrometer filter
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 2059 micromole per kilogram None
 * [hydrocarbon resources-cores➞diss_inorg_phosp](hydrocarbon_resources_cores_diss_inorg_phosp.md)  <sub>0..1</sub>
     * Description: Concentration of dissolved inorganic phosphorus in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 56.5 micromole per liter None
 * [hydrocarbon resources-cores➞tot_phosp](hydrocarbon_resources_cores_tot_phosp.md)  <sub>0..1</sub>
     * Description: Total phosphorus concentration in the sample, calculated by: total phosphorus = total dissolved phosphorus + particulate phosphorus
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 0.03 milligram per liter None
 * [hydrocarbon resources-cores➞suspend_solids](hydrocarbon_resources_cores_suspend_solids.md)  <sub>0..\*</sub>
     * Description: Concentration of substances including a wide variety of material, such as silt, decaying plant and animal matter; can include multiple substances
     * Range: [String](types/String.md)
     * Example:  None
 * [hydrocarbon resources-cores➞density](hydrocarbon_resources_cores_density.md)  <sub>0..1</sub>
     * Description: Density of the sample, which is its mass per unit volume (aka volumetric mass density)
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 1000 kilogram per cubic meter None
 * [hydrocarbon resources-cores➞diss_carb_dioxide](hydrocarbon_resources_cores_diss_carb_dioxide.md)  <sub>0..1</sub>
     * Description: Concentration of dissolved carbon dioxide in the sample or liquid portion of the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 5 milligram per liter None
 * [hydrocarbon resources-cores➞diss_oxygen_fluid](hydrocarbon_resources_cores_diss_oxygen_fluid.md)  <sub>0..1</sub>
     * Description: Concentration of dissolved oxygen in the oil field produced fluids as it contributes to oxgen-corrosion and microbial activity (e.g. Mic).
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞vfa](hydrocarbon_resources_cores_vfa.md)  <sub>0..1</sub>
     * Description: Concentration of Volatile Fatty Acids in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞benzene](hydrocarbon_resources_cores_benzene.md)  <sub>0..1</sub>
     * Description: Concentration of benzene in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞toluene](hydrocarbon_resources_cores_toluene.md)  <sub>0..1</sub>
     * Description: Concentration of toluene in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞ethylbenzene](hydrocarbon_resources_cores_ethylbenzene.md)  <sub>0..1</sub>
     * Description: Concentration of ethylbenzene in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞xylene](hydrocarbon_resources_cores_xylene.md)  <sub>0..1</sub>
     * Description: Concentration of xylene in the sample
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞api](hydrocarbon_resources_cores_api.md)  <sub>1..1</sub>
     * Description: API gravity is a measure of how heavy or light a petroleum liquid is compared to water (source: https://en.wikipedia.org/wiki/API_gravity) (e.g. 31.1¬∞ API)
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞tan](hydrocarbon_resources_cores_tan.md)  <sub>0..1</sub>
     * Description: Total Acid Number¬†(TAN) is a measurement of acidity that is determined by the amount of¬†potassium hydroxide¬†in milligrams that is needed to neutralize the acids in one gram of oil.¬†It is an important quality measurement of¬†crude oil. (source: https://en.wikipedia.org/wiki/Total_acid_number)
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞viscosity](hydrocarbon_resources_cores_viscosity.md)  <sub>0..1</sub>
     * Description: A measure of oil's resistance¬†to gradual deformation by¬†shear stress¬†or¬†tensile stress (e.g. 3.5 cp; 100 ¬∞C)
     * Range: [String](types/String.md)
     * Example:  None
 * [hydrocarbon resources-cores➞pour_point](hydrocarbon_resources_cores_pour_point.md)  <sub>0..1</sub>
     * Description: Temperature at which a liquid becomes semi solid and loses its flow characteristics. In crude oil a high¬†pour point¬†is generally associated with a high paraffin content, typically found in crude deriving from a larger proportion of plant material. (soure: https://en.wikipedia.org/wiki/pour_point)
     * Range: [QuantityValue](QuantityValue.md)
     * Example:  None
 * [hydrocarbon resources-cores➞saturates_pc](hydrocarbon_resources_cores_saturates_pc.md)  <sub>0..1</sub>
     * Description: Saturate, Aromatic, Resin and Asphaltene¬†(SARA) is an analysis method that divides¬†crude oil¬†components according to their polarizability and polarity. There are three main methods to obtain SARA results. The most popular one is known as the Iatroscan TLC-FID and is referred to as IP-143 (source: https://en.wikipedia.org/wiki/Saturate,_aromatic,_resin_and_asphaltene)
     * Range: [String](types/String.md)
     * Example:  None
 * [hydrocarbon resources-cores➞aromatics_pc](hydrocarbon_resources_cores_aromatics_pc.md)  <sub>0..1</sub>
     * Description: Saturate, Aromatic, Resin and Asphaltene¬†(SARA) is an analysis method that divides¬†crude oil¬†components according to their polarizability and polarity. There are three main methods to obtain SARA results. The most popular one is known as the Iatroscan TLC-FID and is referred to as IP-143 (source: https://en.wikipedia.org/wiki/Saturate,_aromatic,_resin_and_asphaltene)
     * Range: [String](types/String.md)
     * Example:  None
 * [hydrocarbon resources-cores➞resins_pc](hydrocarbon_resources_cores_resins_pc.md)  <sub>0..1</sub>
     * Description: Saturate, Aromatic, Resin and Asphaltene¬†(SARA) is an analysis method that divides¬†crude oil¬†components according to their polarizability and polarity. There are three main methods to obtain SARA results. The most popular one is known as the Iatroscan TLC-FID and is referred to as IP-143 (source: https://en.wikipedia.org/wiki/Saturate,_aromatic,_resin_and_asphaltene)
     * Range: [String](types/String.md)
     * Example:  None
 * [hydrocarbon resources-cores➞asphaltenes_pc](hydrocarbon_resources_cores_asphaltenes_pc.md)  <sub>0..1</sub>
     * Description: Saturate, Aromatic, Resin and Asphaltene¬†(SARA) is an analysis method that divides¬†crude oil¬†components according to their polarizability and polarity. There are three main methods to obtain SARA results. The most popular one is known as the Iatroscan TLC-FID and is referred to as IP-143 (source: https://en.wikipedia.org/wiki/Saturate,_aromatic,_resin_and_asphaltene)
     * Range: [String](types/String.md)
     * Example:  None
 * [hydrocarbon resources-cores➞misc_param](hydrocarbon_resources_cores_misc_param.md)  <sub>0..\*</sub>
     * Description: Any other measurement performed or parameter collected, that is not listed here
     * Range: [String](types/String.md)
     * Example: Bicarbonate ion concentration;2075 micromole per kilogram None
 * [hydrocarbon resources-cores➞additional_info](hydrocarbon_resources_cores_additional_info.md)  <sub>0..1</sub>
     * Description: Information that doesn't fit anywhere else. Can also be used to propose new entries for fields with controlled vocabulary
     * Range: [String](types/String.md)
     * Example:  None
 * [hydrocarbon resources-cores➞samp_name](hydrocarbon_resources_cores_samp_name.md)  <sub>1..1</sub>
     * Description: A local identifier or name that for the material sample used for extracting nucleic acids, and subsequent sequencing. It can refer either to the original material collected or to any derived sub-samples. It can have any format, but we suggest that you make it concise, unique and consistent within your lab, and as informative as possible. INSDC requires every sample name from a single Submitter to be unique. Use of a globally unique identifier for the field source_mat_id is recommended in addition to sample_name.
     * Range: [String](types/String.md)
     * Example: ISDsoil1 None
 * [hydrocarbon resources-cores➞project_name](hydrocarbon_resources_cores_project_name.md)  <sub>1..1</sub>
     * Description: Name of the project within which the sequencing was organized
     * Range: [String](types/String.md)
     * Example: Forest soil metagenome None
 * [hydrocarbon resources-cores➞samp_vol_we_dna_ext](hydrocarbon_resources_cores_samp_vol_we_dna_ext.md)  <sub>0..1</sub>
     * Description: Volume (ml) or mass (g) of total collected sample processed for DNA extraction. Note: total sample collected should be entered under the term Sample Size (MIXS:0000001).
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 1500 milliliter None

### Mixed in from MIUVIG:

 * [MIUVIG➞submitted_to_insdc](MIUVIG_submitted_to_insdc.md)  <sub>1..1</sub>
     * Description: Depending on the study (large-scale e.g. done with next generation sequencing technology, or small-scale) sequences have to be submitted to SRA (Sequence Read Archive), DRA (DDBJ Read Archive) or via the classical Webin/Sequin systems to Genbank, ENA and DDBJ. Although this field is mandatory, it is meant as a self-test field, therefore it is not necessary to include this field in contextual data submitted to databases
     * Range: [String](types/String.md)
     * Example: yes None

### Mixed in from MIUVIG:

 * [MIUVIG➞investigation_type](MIUVIG_investigation_type.md)  <sub>1..1</sub>
     * Description: Nucleic Acid Sequence Report is the root element of all MIGS/MIMS compliant reports as standardized by Genomic Standards Consortium. This field is either eukaryote,bacteria,virus,plasmid,organelle, metagenome,mimarks-survey, mimarks-specimen, metatranscriptome, single amplified genome, metagenome-assembled genome, or uncultivated viral genome
     * Range: [investigation_type_enum](investigation_type_enum.md)
     * Example: metagenome None

### Mixed in from MIUVIG:

 * [MIUVIG➞samp_taxon_id](MIUVIG_samp_taxon_id.md)  <sub>1..1</sub>
     * Description: NCBI taxon id of the sample.  Maybe be a single taxon or mixed taxa sample. Use 'synthetic metagenome’ for mock community/positive controls, or 'blank sample' for negative controls.
     * Range: [String](types/String.md)
     * Example: Gut Metagenome [NCBI:txid749906] None

### Mixed in from MIUVIG:

 * [MIUVIG➞experimental_factor](MIUVIG_experimental_factor.md)  <sub>0..1</sub>
     * Description: Experimental factors are essentially the variable aspects of an experiment design which can be used to describe an experiment, or set of experiments, in an increasingly detailed manner. This field accepts ontology terms from Experimental Factor Ontology (EFO) and/or Ontology for Biomedical Investigations (OBI). For a browser of EFO (v 2.95) terms, please see http://purl.bioontology.org/ontology/EFO; for a browser of OBI (v 2018-02-12) terms please see http://purl.bioontology.org/ontology/OBI
     * Range: [String](types/String.md)
     * Example: time series design [EFO:EFO_0001779] None

### Mixed in from MIUVIG:

 * [MIUVIG➞neg_cont_type](MIUVIG_neg_cont_type.md)  <sub>0..1</sub>
     * Description: The substance or equipment used as a negative control in an investigation
     * Range: [neg_cont_type_enum](neg_cont_type_enum.md)
     * Example:  None

### Mixed in from MIUVIG:

 * [MIUVIG➞pos_cont_type](MIUVIG_pos_cont_type.md)  <sub>0..1</sub>
     * Description: The substance, mixture, product, or apparatus used to verify that a process which is part of an investigation delivers a true positive.
     * Range: [String](types/String.md)
     * Example:  None

### Mixed in from MIUVIG:

 * [MIUVIG➞env_package](MIUVIG_env_package.md)  <sub>0..1</sub>
     * Description: MIxS extension for reporting of measurements and observations obtained from one or more of the environments where the sample was obtained. All environmental packages listed here are further defined in separate subtables. By giving the name of the environmental package, a selection of fields can be made from the subtables and can be reported
     * Range: [env_package_enum](env_package_enum.md)
     * Example: soil None

### Mixed in from MIUVIG:

 * [MIUVIG➞estimated_size](MIUVIG_estimated_size.md)  <sub>0..1</sub>
     * Description: The estimated size of the genome prior to sequencing. Of particular importance in the sequencing of (eukaryotic) genome which could remain in draft form for a long or unspecified period.
     * Range: [String](types/String.md)
     * Example: 300000 bp None

### Mixed in from MIUVIG:

 * [MIUVIG➞ref_biomaterial](MIUVIG_ref_biomaterial.md)  <sub>0..1</sub>
     * Description: Primary publication if isolated before genome publication; otherwise, primary genome report.
     * Range: [String](types/String.md)
     * Example: doi:10.1016/j.syapm.2018.01.009 None

### Mixed in from MIUVIG:

 * [MIUVIG➞source_mat_id](MIUVIG_source_mat_id.md)  <sub>0..1</sub>
     * Description: A unique identifier assigned to a material sample (as defined by http://rs.tdwg.org/dwc/terms/materialSampleID, and as opposed to a particular digital record of a material sample) used for extracting nucleic acids, and subsequent sequencing. The identifier can refer either to the original material collected or to any derived sub-samples. The INSDC qualifiers /specimen_voucher, /bio_material, or /culture_collection may or may not share the same value as the source_mat_id field. For instance, the /specimen_voucher qualifier and source_mat_id may both contain 'UAM:Herps:14' , referring to both the specimen voucher and sampled tissue with the same identifier. However, the /culture_collection qualifier may refer to a value from an initial culture (e.g. ATCC:11775) while source_mat_id would refer to an identifier from some derived culture from which the nucleic acids were extracted (e.g. xatc123 or ark:/2154/R2).
     * Range: [String](types/String.md)
     * Example: MPI012345 None

### Mixed in from MIUVIG:

 * [MIUVIG➞pathogenicity](MIUVIG_pathogenicity.md)  <sub>0..1</sub>
     * Description: To what is the entity pathogenic
     * Range: [String](types/String.md)
     * Example: human, animal, plant, fungi, bacteria None

### Mixed in from MIUVIG:

 * [MIUVIG➞biotic_relationship](MIUVIG_biotic_relationship.md)  <sub>0..1</sub>
     * Description: Description of relationship(s) between the subject organism and other organism(s) it is associated with. E.g., parasite on species X; mutualist with species Y. The target organism is the subject of the relationship, and the other organism(s) is the object
     * Range: [biotic_relationship_enum](biotic_relationship_enum.md)
     * Example: free living None

### Mixed in from MIUVIG:

 * [MIUVIG➞specific_host](MIUVIG_specific_host.md)  <sub>0..1</sub>
     * Description: Report the host's taxonomic name and/or NCBI taxonomy ID.
     * Range: [String](types/String.md)
     * Example: Homo sapiens and/or 9606 None

### Mixed in from MIUVIG:

 * [MIUVIG➞host_spec_range](MIUVIG_host_spec_range.md)  <sub>0..1</sub>
     * Description: The range and diversity of host species that an organism is capable of infecting, defined by NCBI taxonomy identifier.
     * Range: [Integer](types/Integer.md)
     * Example: 9606 None

### Mixed in from MIUVIG:

 * [MIUVIG➞host_disease_stat](MIUVIG_host_disease_stat.md)  <sub>0..1</sub>
     * Description: List of diseases with which the host has been diagnosed; can include multiple diagnoses. The value of the field depends on host; for humans the terms should be chosen from the DO (Human Disease Ontology) at https://www.disease-ontology.org, non-human host diseases are free text
     * Range: [String](types/String.md)
     * Example: rabies [DOID:11260] None

### Mixed in from MIUVIG:

 * [MIUVIG➞samp_collec_device](MIUVIG_samp_collec_device.md)  <sub>0..1</sub>
     * Description: The device used to collect an environmental sample. This field accepts terms listed under environmental sampling device (http://purl.obolibrary.org/obo/ENVO). This field also accepts terms listed under specimen collection device (http://purl.obolibrary.org/obo/GENEPIO_0002094).
     * Range: [String](types/String.md)
     * Example: swab, biopsy, niskin bottle, push core, drag swab [GENEPIO:0002713] None

### Mixed in from MIUVIG:

 * [MIUVIG➞samp_collec_method](MIUVIG_samp_collec_method.md)  <sub>0..1</sub>
     * Description: The method employed for collecting the sample.
     * Range: [String](types/String.md)
     * Example: swabbing None

### Mixed in from MIUVIG:

 * [MIUVIG➞samp_mat_process](MIUVIG_samp_mat_process.md)  <sub>0..1</sub>
     * Description: A brief description of any processing applied to the sample during or after retrieving the sample from environment, or a link to the relevant protocol(s) performed.
     * Range: [String](types/String.md)
     * Example: filtering of seawater, storing samples in ethanol None

### Mixed in from MIUVIG:

 * [MIUVIG➞size_frac](MIUVIG_size_frac.md)  <sub>0..1</sub>
     * Description: Filtering pore size used in sample preparation
     * Range: [String](types/String.md)
     * Example: 0-0.22 micrometer None

### Mixed in from MIUVIG:

 * [MIUVIG➞samp_size](MIUVIG_samp_size.md)  <sub>0..1</sub>
     * Description: The total amount or size (volume (ml), mass (g) or area (m2) ) of sample collected.
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 5 liter None

### Mixed in from MIUVIG:

 * [MIUVIG➞source_uvig](MIUVIG_source_uvig.md)  <sub>1..1</sub>
     * Description: Type of dataset from which the UViG was obtained
     * Range: [source_uvig_enum](source_uvig_enum.md)
     * Example: viral fraction metagenome (virome) None

### Mixed in from MIUVIG:

 * [MIUVIG➞virus_enrich_appr](MIUVIG_virus_enrich_appr.md)  <sub>1..1</sub>
     * Description: List of approaches used to enrich the sample for viruses, if any
     * Range: [virus_enrich_appr_enum](virus_enrich_appr_enum.md)
     * Example: filtration + FeCl Precipitation + ultracentrifugation + DNAse None

### Mixed in from MIUVIG:

 * [MIUVIG➞nucl_acid_ext](MIUVIG_nucl_acid_ext.md)  <sub>0..1</sub>
     * Description: A link to a literature reference, electronic resource or a standard operating procedure (SOP), that describes the material separation to recover the nucleic acid fraction from a sample
     * Range: [String](types/String.md)
     * Example: https://mobio.com/media/wysiwyg/pdfs/protocols/12888.pdf None

### Mixed in from MIUVIG:

 * [MIUVIG➞nucl_acid_amp](MIUVIG_nucl_acid_amp.md)  <sub>0..1</sub>
     * Description: A link to a literature reference, electronic resource or a standard operating procedure (SOP), that describes the enzymatic amplification (PCR, TMA, NASBA) of specific nucleic acids
     * Range: [String](types/String.md)
     * Example: https://phylogenomics.me/protocols/16s-pcr-protocol/ None

### Mixed in from MIUVIG:

 * [MIUVIG➞lib_size](MIUVIG_lib_size.md)  <sub>0..1</sub>
     * Description: Total number of clones in the library prepared for the project
     * Range: [Integer](types/Integer.md)
     * Example: 50 None

### Mixed in from MIUVIG:

 * [MIUVIG➞lib_reads_seqd](MIUVIG_lib_reads_seqd.md)  <sub>0..1</sub>
     * Description: Total number of clones sequenced from the library
     * Range: [Integer](types/Integer.md)
     * Example: 20 None

### Mixed in from MIUVIG:

 * [MIUVIG➞lib_layout](MIUVIG_lib_layout.md)  <sub>0..1</sub>
     * Description: Specify whether to expect single, paired, or other configuration of reads
     * Range: [lib_layout_enum](lib_layout_enum.md)
     * Example: paired None

### Mixed in from MIUVIG:

 * [MIUVIG➞lib_vector](MIUVIG_lib_vector.md)  <sub>0..1</sub>
     * Description: Cloning vector type(s) used in construction of libraries
     * Range: [String](types/String.md)
     * Example: Bacteriophage P1 None

### Mixed in from MIUVIG:

 * [MIUVIG➞lib_screen](MIUVIG_lib_screen.md)  <sub>0..1</sub>
     * Description: Specific enrichment or screening methods applied before and/or after creating libraries
     * Range: [String](types/String.md)
     * Example: enriched, screened, normalized None

### Mixed in from MIUVIG:

 * [MIUVIG➞mid](MIUVIG_mid.md)  <sub>0..1</sub>
     * Description: Molecular barcodes, called Multiplex Identifiers (MIDs), that are used to specifically tag unique samples in a sequencing run. Sequence should be reported in uppercase letters
     * Range: [String](types/String.md)
     * Example: GTGAATAT None

### Mixed in from MIUVIG:

 * [MIUVIG➞adapters](MIUVIG_adapters.md)  <sub>0..1</sub>
     * Description: Adapters provide priming sequences for both amplification and sequencing of the sample-library fragments. Both adapters should be reported; in uppercase letters
     * Range: [String](types/String.md)
     * Example: AATGATACGGCGACCACCGAGATCTACACGCT;CAAGCAGAAGACGGCATACGAGAT None

### Mixed in from MIUVIG:

 * [MIUVIG➞seq_meth](MIUVIG_seq_meth.md)  <sub>1..1</sub>
     * Description: Sequencing machine used. Where possible the term should be taken from the OBI list of DNA sequencers (http://purl.obolibrary.org/obo/OBI_0400103).
     * Range: [String](types/String.md)
     * Example: 454 Genome Sequencer FLX [OBI:0000702] None

### Mixed in from MIUVIG:

 * [MIUVIG➞tax_ident](MIUVIG_tax_ident.md)  <sub>0..1</sub>
     * Description: The phylogenetic marker(s) used to assign an organism name to the SAG or MAG
     * Range: [tax_ident_enum](tax_ident_enum.md)
     * Example: other: rpoB gene None

### Mixed in from MIUVIG:

 * [MIUVIG➞assembly_qual](MIUVIG_assembly_qual.md)  <sub>1..1</sub>
     * Description: The assembly quality category is based on sets of criteria outlined for each assembly quality category. For MISAG/MIMAG; Finished: Single, validated, contiguous sequence per replicon without gaps or ambiguities with a consensus error rate equivalent to Q50 or better. High Quality Draft:Multiple fragments where gaps span repetitive regions. Presence of the 23S, 16S and 5S rRNA genes and at least 18 tRNAs. Medium Quality Draft:Many fragments with little to no review of assembly other than reporting of standard assembly statistics. Low Quality Draft:Many fragments with little to no review of assembly other than reporting of standard assembly statistics. Assembly statistics include, but are not limited to total assembly size, number of contigs, contig N50/L50, and maximum contig length. For MIUVIG; Finished: Single, validated, contiguous sequence per replicon without gaps or ambiguities, with extensive manual review and editing to annotate putative gene functions and transcriptional units. High-quality draft genome: One or multiple fragments, totaling ≥ 90% of the expected genome or replicon sequence or predicted complete. Genome fragment(s): One or multiple fragments, totalling < 90% of the expected genome or replicon sequence, or for which no genome size could be estimated
     * Range: [assembly_qual_enum](assembly_qual_enum.md)
     * Example: High-quality draft genome None

### Mixed in from MIUVIG:

 * [MIUVIG➞assembly_name](MIUVIG_assembly_name.md)  <sub>0..1</sub>
     * Description: Name/version of the assembly provided by the submitter that is used in the genome browsers and in the community
     * Range: [String](types/String.md)
     * Example: HuRef, JCVI_ISG_i3_1.0 None

### Mixed in from MIUVIG:

 * [MIUVIG➞assembly_software](MIUVIG_assembly_software.md)  <sub>1..1</sub>
     * Description: Tool(s) used for assembly, including version number and parameters
     * Range: [String](types/String.md)
     * Example: metaSPAdes;3.11.0;kmer set 21,33,55,77,99,121, default parameters otherwise None

### Mixed in from MIUVIG:

 * [MIUVIG➞annot](MIUVIG_annot.md)  <sub>0..1</sub>
     * Description: Tool used for annotation, or for cases where annotation was provided by a community jamboree or model organism database rather than by a specific submitter
     * Range: [String](types/String.md)
     * Example: prokka None

### Mixed in from MIUVIG:

 * [MIUVIG➞number_contig](MIUVIG_number_contig.md)  <sub>1..1</sub>
     * Description: Total number of contigs in the cleaned/submitted assembly that makes up a given genome, SAG, MAG, or UViG
     * Range: [Integer](types/Integer.md)
     * Example: 40 None

### Mixed in from MIUVIG:

 * [MIUVIG➞feat_pred](MIUVIG_feat_pred.md)  <sub>0..1</sub>
     * Description: Method used to predict UViGs features such as ORFs, integration site, etc.
     * Range: [String](types/String.md)
     * Example: Prodigal;2.6.3;default parameters None

### Mixed in from MIUVIG:

 * [MIUVIG➞ref_db](MIUVIG_ref_db.md)  <sub>0..1</sub>
     * Description: List of database(s) used for ORF annotation, along with version number and reference to website or publication
     * Range: [String](types/String.md)
     * Example: pVOGs;5;http://dmk-brain.ecn.uiowa.edu/pVOGs/ Grazziotin et al. 2017 doi:10.1093/nar/gkw975 None

### Mixed in from MIUVIG:

 * [MIUVIG➞sim_search_meth](MIUVIG_sim_search_meth.md)  <sub>0..1</sub>
     * Description: Tool used to compare ORFs with database, along with version and cutoffs used
     * Range: [String](types/String.md)
     * Example: HMMER3;3.1b2;hmmsearch, cutoff of 50 on score None

### Mixed in from MIUVIG:

 * [MIUVIG➞tax_class](MIUVIG_tax_class.md)  <sub>0..1</sub>
     * Description: Method used for taxonomic classification, along with reference database used, classification rank, and thresholds used to classify new genomes
     * Range: [String](types/String.md)
     * Example: vConTACT vContact2 (references from NCBI RefSeq v83, genus rank classification, default parameters) None

### Mixed in from MIUVIG:

 * [MIUVIG➞trnas](MIUVIG_trnas.md)  <sub>0..1</sub>
     * Description: The total number of tRNAs identified from the SAG or MAG
     * Range: [Integer](types/Integer.md)
     * Example: 18 None

### Mixed in from MIUVIG:

 * [MIUVIG➞trna_ext_software](MIUVIG_trna_ext_software.md)  <sub>0..1</sub>
     * Description: Tools used for tRNA identification
     * Range: [String](types/String.md)
     * Example: infernal;v2;default parameters None

### Mixed in from MIUVIG:

 * [MIUVIG➞compl_score](MIUVIG_compl_score.md)  <sub>0..1</sub>
     * Description: Completeness score is typically based on either the fraction of markers found as compared to a database or the percent of a genome found as compared to a closely related reference genome. High Quality Draft: >90%, Medium Quality Draft: >50%, and Low Quality Draft: < 50% should have the indicated completeness scores
     * Range: [compl_score_enum](compl_score_enum.md)
     * Example: med;60% None

### Mixed in from MIUVIG:

 * [MIUVIG➞compl_software](MIUVIG_compl_software.md)  <sub>0..1</sub>
     * Description: Tools used for completion estimate, i.e. checkm, anvi'o, busco
     * Range: [String](types/String.md)
     * Example: checkm None

### Mixed in from MIUVIG:

 * [MIUVIG➞compl_appr](MIUVIG_compl_appr.md)  <sub>0..1</sub>
     * Description: The approach used to determine the completeness of a given genomic assembly, which would typically make use of a set of conserved marker genes or a closely related reference genome. For UViG completeness, include reference genome or group used, and contig feature suggesting a complete genome
     * Range: [compl_appr_enum](compl_appr_enum.md)
     * Example: other: UViG length compared to the average length of reference genomes from the P22virus genus (NCBI RefSeq v83) None

### Mixed in from MIUVIG:

 * [MIUVIG➞sort_tech](MIUVIG_sort_tech.md)  <sub>0..1</sub>
     * Description: Method used to sort/isolate cells or particles of interest
     * Range: [sort_tech_enum](sort_tech_enum.md)
     * Example: optical manipulation None

### Mixed in from MIUVIG:

 * [MIUVIG➞single_cell_lysis_appr](MIUVIG_single_cell_lysis_appr.md)  <sub>0..1</sub>
     * Description: Method used to free DNA from interior of the cell(s) or particle(s)
     * Range: [single_cell_lysis_appr_enum](single_cell_lysis_appr_enum.md)
     * Example: enzymatic None

### Mixed in from MIUVIG:

 * [MIUVIG➞single_cell_lysis_prot](MIUVIG_single_cell_lysis_prot.md)  <sub>0..1</sub>
     * Description: Name of the kit or standard protocol used for cell(s) or particle(s) lysis
     * Range: [String](types/String.md)
     * Example: ambion single cell lysis kit None

### Mixed in from MIUVIG:

 * [MIUVIG➞wga_amp_appr](MIUVIG_wga_amp_appr.md)  <sub>0..1</sub>
     * Description: Method used to amplify genomic DNA in preparation for sequencing
     * Range: [String](types/String.md)
     * Example: mda based None

### Mixed in from MIUVIG:

 * [MIUVIG➞wga_amp_kit](MIUVIG_wga_amp_kit.md)  <sub>0..1</sub>
     * Description: Kit used to amplify genomic DNA in preparation for sequencing
     * Range: [String](types/String.md)
     * Example: qiagen repli-g None

### Mixed in from MIUVIG:

 * [MIUVIG➞bin_param](MIUVIG_bin_param.md)  <sub>0..1</sub>
     * Description: The parameters that have been applied during the extraction of genomes from metagenomic datasets
     * Range: [bin_param_enum](bin_param_enum.md)
     * Example: coverage and kmer None

### Mixed in from MIUVIG:

 * [MIUVIG➞bin_software](MIUVIG_bin_software.md)  <sub>0..1</sub>
     * Description: Tool(s) used for the extraction of genomes from metagenomic datasets, where possible include a product ID (PID) of the tool(s) used.
     * Range: [String](types/String.md)
     * Example: MetaCluster-TA (RRID:SCR_004599), MaxBin (biotools:maxbin) None

### Mixed in from MIUVIG:

 * [MIUVIG➞reassembly_bin](MIUVIG_reassembly_bin.md)  <sub>0..1</sub>
     * Description: Has an assembly been performed on a genome bin extracted from a metagenomic assembly?
     * Range: [String](types/String.md)
     * Example: no None

### Mixed in from MIUVIG:

 * [MIUVIG➞mag_cov_software](MIUVIG_mag_cov_software.md)  <sub>0..1</sub>
     * Description: Tool(s) used to determine the genome coverage if coverage is used as a binning parameter in the extraction of genomes from metagenomic datasets
     * Range: [mag_cov_software_enum](mag_cov_software_enum.md)
     * Example: bbmap None

### Mixed in from MIUVIG:

 * [MIUVIG➞vir_ident_software](MIUVIG_vir_ident_software.md)  <sub>1..1</sub>
     * Description: Tool(s) used for the identification of UViG as a viral genome, software or protocol name including version number, parameters, and cutoffs used
     * Range: [String](types/String.md)
     * Example: VirSorter; 1.0.4; Virome database, category 2 None

### Mixed in from MIUVIG:

 * [MIUVIG➞pred_genome_type](MIUVIG_pred_genome_type.md)  <sub>1..1</sub>
     * Description: Type of genome predicted for the UViG
     * Range: [pred_genome_type_enum](pred_genome_type_enum.md)
     * Example: dsDNA None

### Mixed in from MIUVIG:

 * [MIUVIG➞pred_genome_struc](MIUVIG_pred_genome_struc.md)  <sub>1..1</sub>
     * Description: Expected structure of the viral genome
     * Range: [pred_genome_struc_enum](pred_genome_struc_enum.md)
     * Example: non-segmented None

### Mixed in from MIUVIG:

 * [MIUVIG➞detec_type](MIUVIG_detec_type.md)  <sub>1..1</sub>
     * Description: Type of UViG detection
     * Range: [String](types/String.md)
     * Example: independent sequence (UViG) None

### Mixed in from MIUVIG:

 * [MIUVIG➞otu_class_appr](MIUVIG_otu_class_appr.md)  <sub>0..1</sub>
     * Description: Cutoffs and approach used when clustering “species-level” OTUs. Note that results from standard 95% ANI / 85% AF clustering should be provided alongside OTUS defined from another set of thresholds, even if the latter are the ones primarily used during the analysis
     * Range: [String](types/String.md)
     * Example: 95% ANI;85% AF; greedy incremental clustering None

### Mixed in from MIUVIG:

 * [MIUVIG➞otu_seq_comp_appr](MIUVIG_otu_seq_comp_appr.md)  <sub>0..1</sub>
     * Description: Tool and thresholds used to compare sequences when computing "species-level" OTUs
     * Range: [String](types/String.md)
     * Example: blastn;2.6.0+;e-value cutoff: 0.001 None

### Mixed in from MIUVIG:

 * [MIUVIG➞otu_db](MIUVIG_otu_db.md)  <sub>0..1</sub>
     * Description: Reference database (i.e. sequences not generated as part of the current study) used to cluster new genomes in "species-level" OTUs, if any
     * Range: [String](types/String.md)
     * Example: NCBI Viral RefSeq;83 None

### Mixed in from MIUVIG:

 * [MIUVIG➞host_pred_appr](MIUVIG_host_pred_appr.md)  <sub>0..1</sub>
     * Description: Tool or approach used for host prediction
     * Range: [host_pred_appr_enum](host_pred_appr_enum.md)
     * Example: CRISPR spacer match None

### Mixed in from MIUVIG:

 * [MIUVIG➞host_pred_est_acc](MIUVIG_host_pred_est_acc.md)  <sub>0..1</sub>
     * Description: For each tool or approach used for host prediction, estimated false discovery rates should be included, either computed de novo or from the literature
     * Range: [String](types/String.md)
     * Example: CRISPR spacer match: 0 or 1 mismatches, estimated 8% FDR at the host genus rank (Edwards et al. 2016 doi:10.1093/femsre/fuv048) None

### Mixed in from MIUVIG:

 * [MIUVIG➞associated resource](MIUVIG_associated_resource.md)  <sub>0..1</sub>
     * Description: A related resource that is referenced, cited, or otherwise associated to the sequence.
     * Range: [String](types/String.md)
     * Example: http://www.earthmicrobiome.org/ None

### Mixed in from MIUVIG:

 * [MIUVIG➞sop](MIUVIG_sop.md)  <sub>0..1</sub>
     * Description: Standard operating procedures used in assembly and/or annotation of genomes, metagenomes or environmental sequences
     * Range: [String](types/String.md)
     * Example: http://press.igsb.anl.gov/earthmicrobiome/protocols-and-standards/its/ None
