
# Class: MIMARKS specimen


Minimal Information about a Marker Specimen: specimen

URI: [mixs.vocab:MIMARKSSpecimen](https://w3id.org/mixs/vocab/MIMARKSSpecimen)


[![img](https://yuml.me/diagram/nofunky;dir:TB/class/[QuantityValue],[QuantityValue]<samp_vol_we_dna_ext%200..1-++[MIMARKSSpecimen&#124;submitted_to_insdc:string;investigation_type:investigation_type_enum;samp_name:string;samp_taxon_id:string;project_name:string;experimental_factor:string%20%3F;lat_lon:string;geo_loc_name:string;collection_date:date;neg_cont_type:neg_cont_type_enum%20%3F;pos_cont_type:string%20%3F;env_broad_scale:string;env_local_scale:string;env_medium:string;env_package:env_package_enum%20%3F;source_mat_id:string%20%3F;rel_to_oxygen:rel_to_oxygen_enum%20%3F;samp_collec_device:string%20%3F;samp_collec_method:string%20%3F;samp_mat_process:string%20%3F;size_frac:string%20%3F;nucl_acid_ext:string%20%3F;nucl_acid_amp:string%20%3F;lib_size:integer%20%3F;lib_reads_seqd:integer%20%3F;lib_layout:lib_layout_enum%20%3F;lib_vector:string%20%3F;lib_screen:string%20%3F;target_gene:string;target_subfragment:string%20%3F;pcr_primers:string%20%3F;mid:string%20%3F;adapters:string%20%3F;pcr_cond:string%20%3F;seq_meth:string;seq_quality_check:string%20%3F;chimera_check:string%20%3F;assembly_software:string%20%3F;associated_resource:string%20%3F;sop:string%20%3F],[QuantityValue]<samp_size%200..1-++[MIMARKSSpecimen],[WaterMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[WastewaterSludgeMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[SoilMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[SedimentMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[Plant-associatedMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[MiscellaneousNaturalOrArtificialEnvironmentMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[MicrobialMatBiofilmMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[HydrocarbonResources-fluidsSwabsMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[HydrocarbonResources-coresMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[Human-vaginalMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[Human-skinMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[Human-oralMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[Human-gutMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[Human-associatedMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[Host-associatedMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[BuiltEnvironmentMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[AirMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[WaterMIMARKSSpecimen],[WastewaterSludgeMIMARKSSpecimen],[SoilMIMARKSSpecimen],[SedimentMIMARKSSpecimen],[Plant-associatedMIMARKSSpecimen],[MiscellaneousNaturalOrArtificialEnvironmentMIMARKSSpecimen],[MicrobialMatBiofilmMIMARKSSpecimen],[HydrocarbonResources-fluidsSwabsMIMARKSSpecimen],[HydrocarbonResources-coresMIMARKSSpecimen],[Human-vaginalMIMARKSSpecimen],[Human-skinMIMARKSSpecimen],[Human-oralMIMARKSSpecimen],[Human-gutMIMARKSSpecimen],[Human-associatedMIMARKSSpecimen],[Host-associatedMIMARKSSpecimen],[BuiltEnvironmentMIMARKSSpecimen],[AirMIMARKSSpecimen])](https://yuml.me/diagram/nofunky;dir:TB/class/[QuantityValue],[QuantityValue]<samp_vol_we_dna_ext%200..1-++[MIMARKSSpecimen&#124;submitted_to_insdc:string;investigation_type:investigation_type_enum;samp_name:string;samp_taxon_id:string;project_name:string;experimental_factor:string%20%3F;lat_lon:string;geo_loc_name:string;collection_date:date;neg_cont_type:neg_cont_type_enum%20%3F;pos_cont_type:string%20%3F;env_broad_scale:string;env_local_scale:string;env_medium:string;env_package:env_package_enum%20%3F;source_mat_id:string%20%3F;rel_to_oxygen:rel_to_oxygen_enum%20%3F;samp_collec_device:string%20%3F;samp_collec_method:string%20%3F;samp_mat_process:string%20%3F;size_frac:string%20%3F;nucl_acid_ext:string%20%3F;nucl_acid_amp:string%20%3F;lib_size:integer%20%3F;lib_reads_seqd:integer%20%3F;lib_layout:lib_layout_enum%20%3F;lib_vector:string%20%3F;lib_screen:string%20%3F;target_gene:string;target_subfragment:string%20%3F;pcr_primers:string%20%3F;mid:string%20%3F;adapters:string%20%3F;pcr_cond:string%20%3F;seq_meth:string;seq_quality_check:string%20%3F;chimera_check:string%20%3F;assembly_software:string%20%3F;associated_resource:string%20%3F;sop:string%20%3F],[QuantityValue]<samp_size%200..1-++[MIMARKSSpecimen],[WaterMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[WastewaterSludgeMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[SoilMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[SedimentMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[Plant-associatedMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[MiscellaneousNaturalOrArtificialEnvironmentMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[MicrobialMatBiofilmMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[HydrocarbonResources-fluidsSwabsMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[HydrocarbonResources-coresMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[Human-vaginalMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[Human-skinMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[Human-oralMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[Human-gutMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[Human-associatedMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[Host-associatedMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[BuiltEnvironmentMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[AirMIMARKSSpecimen]uses%20-.->[MIMARKSSpecimen],[WaterMIMARKSSpecimen],[WastewaterSludgeMIMARKSSpecimen],[SoilMIMARKSSpecimen],[SedimentMIMARKSSpecimen],[Plant-associatedMIMARKSSpecimen],[MiscellaneousNaturalOrArtificialEnvironmentMIMARKSSpecimen],[MicrobialMatBiofilmMIMARKSSpecimen],[HydrocarbonResources-fluidsSwabsMIMARKSSpecimen],[HydrocarbonResources-coresMIMARKSSpecimen],[Human-vaginalMIMARKSSpecimen],[Human-skinMIMARKSSpecimen],[Human-oralMIMARKSSpecimen],[Human-gutMIMARKSSpecimen],[Human-associatedMIMARKSSpecimen],[Host-associatedMIMARKSSpecimen],[BuiltEnvironmentMIMARKSSpecimen],[AirMIMARKSSpecimen])

## Mixin for

 * [AirMIMARKSSpecimen](AirMIMARKSSpecimen.md) (mixin)  - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package air
 * [BuiltEnvironmentMIMARKSSpecimen](BuiltEnvironmentMIMARKSSpecimen.md) (mixin)  - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package built environment
 * [Host-associatedMIMARKSSpecimen](Host-associatedMIMARKSSpecimen.md) (mixin)  - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package host-associated
 * [Human-associatedMIMARKSSpecimen](Human-associatedMIMARKSSpecimen.md) (mixin)  - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package human-associated
 * [Human-gutMIMARKSSpecimen](Human-gutMIMARKSSpecimen.md) (mixin)  - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package human-gut
 * [Human-oralMIMARKSSpecimen](Human-oralMIMARKSSpecimen.md) (mixin)  - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package human-oral
 * [Human-skinMIMARKSSpecimen](Human-skinMIMARKSSpecimen.md) (mixin)  - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package human-skin
 * [Human-vaginalMIMARKSSpecimen](Human-vaginalMIMARKSSpecimen.md) (mixin)  - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package human-vaginal
 * [HydrocarbonResources-coresMIMARKSSpecimen](HydrocarbonResources-coresMIMARKSSpecimen.md) (mixin)  - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package hydrocarbon resources-cores
 * [HydrocarbonResources-fluidsSwabsMIMARKSSpecimen](HydrocarbonResources-fluidsSwabsMIMARKSSpecimen.md) (mixin)  - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package hydrocarbon resources-fluids_swabs
 * [MicrobialMatBiofilmMIMARKSSpecimen](MicrobialMatBiofilmMIMARKSSpecimen.md) (mixin)  - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package microbial mat_biofilm
 * [MiscellaneousNaturalOrArtificialEnvironmentMIMARKSSpecimen](MiscellaneousNaturalOrArtificialEnvironmentMIMARKSSpecimen.md) (mixin)  - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package miscellaneous natural or artificial environment
 * [Plant-associatedMIMARKSSpecimen](Plant-associatedMIMARKSSpecimen.md) (mixin)  - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package plant-associated
 * [SedimentMIMARKSSpecimen](SedimentMIMARKSSpecimen.md) (mixin)  - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package sediment
 * [SoilMIMARKSSpecimen](SoilMIMARKSSpecimen.md) (mixin)  - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package soil
 * [WastewaterSludgeMIMARKSSpecimen](WastewaterSludgeMIMARKSSpecimen.md) (mixin)  - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package wastewater_sludge
 * [WaterMIMARKSSpecimen](WaterMIMARKSSpecimen.md) (mixin)  - Combinatorial checklist Minimal Information about a Marker Specimen: specimen with environmental package water

## Referenced by Class


## Attributes


### Own

 * [MIMARKS specimenâžžsubmitted_to_insdc](MIMARKS_specimen_submitted_to_insdc.md)  <sub>1..1</sub>
     * Description: Depending on the study (large-scale e.g. done with next generation sequencing technology, or small-scale) sequences have to be submitted to SRA (Sequence Read Archive), DRA (DDBJ Read Archive) or via the classical Webin/Sequin systems to Genbank, ENA and DDBJ. Although this field is mandatory, it is meant as a self-test field, therefore it is not necessary to include this field in contextual data submitted to databases
     * Range: [String](types/String.md)
     * Example: yes None
 * [MIMARKS specimenâžžinvestigation_type](MIMARKS_specimen_investigation_type.md)  <sub>1..1</sub>
     * Description: Nucleic Acid Sequence Report is the root element of all MIGS/MIMS compliant reports as standardized by Genomic Standards Consortium. This field is either eukaryote,bacteria,virus,plasmid,organelle, metagenome,mimarks-survey, mimarks-specimen, metatranscriptome, single amplified genome, metagenome-assembled genome, or uncultivated viral genome
     * Range: [investigation_type_enum](investigation_type_enum.md)
     * Example: metagenome None
 * [MIMARKS specimenâžžsamp_name](MIMARKS_specimen_samp_name.md)  <sub>1..1</sub>
     * Description: A local identifier or name that for the material sample used for extracting nucleic acids, and subsequent sequencing. It can refer either to the original material collected or to any derived sub-samples. It can have any format, but we suggest that you make it concise, unique and consistent within your lab, and as informative as possible. INSDC requires every sample name from a single Submitter to be unique. Use of a globally unique identifier for the field source_mat_id is recommended in addition to sample_name.
     * Range: [String](types/String.md)
     * Example: ISDsoil1 None
 * [MIMARKS specimenâžžsamp_taxon_id](MIMARKS_specimen_samp_taxon_id.md)  <sub>1..1</sub>
     * Description: NCBI taxon id of the sample.  Maybe be a single taxon or mixed taxa sample. Use 'synthetic metagenomeâ€™ for mock community/positive controls, or 'blank sample' for negative controls.
     * Range: [String](types/String.md)
     * Example: Gut Metagenome [NCBI:txid749906] None
 * [MIMARKS specimenâžžproject_name](MIMARKS_specimen_project_name.md)  <sub>1..1</sub>
     * Description: Name of the project within which the sequencing was organized
     * Range: [String](types/String.md)
     * Example: Forest soil metagenome None
 * [MIMARKS specimenâžžexperimental_factor](MIMARKS_specimen_experimental_factor.md)  <sub>0..1</sub>
     * Description: Experimental factors are essentially the variable aspects of an experiment design which can be used to describe an experiment, or set of experiments, in an increasingly detailed manner. This field accepts ontology terms from Experimental Factor Ontology (EFO) and/or Ontology for Biomedical Investigations (OBI). For a browser of EFO (v 2.95) terms, please see http://purl.bioontology.org/ontology/EFO; for a browser of OBI (v 2018-02-12) terms please see http://purl.bioontology.org/ontology/OBI
     * Range: [String](types/String.md)
     * Example: time series design [EFO:EFO_0001779] None
 * [MIMARKS specimenâžžlat_lon](MIMARKS_specimen_lat_lon.md)  <sub>1..1</sub>
     * Description: The geographical origin of the sample as defined by latitude and longitude. The values should be reported in decimal degrees and in WGS84 system
     * Range: [String](types/String.md)
     * Example: 50.586825 6.408977 None
 * [MIMARKS specimenâžžgeo_loc_name](MIMARKS_specimen_geo_loc_name.md)  <sub>1..1</sub>
     * Description: The geographical origin of the sample as defined by the country or sea name followed by specific region name. Country or sea names should be chosen from the INSDC country list (http://insdc.org/country.html), or the GAZ ontology (http://purl.bioontology.org/ontology/GAZ)
     * Range: [String](types/String.md)
     * Example: USA: Maryland, Bethesda None
 * [MIMARKS specimenâžžcollection_date](MIMARKS_specimen_collection_date.md)  <sub>1..1</sub>
     * Description: The time of sampling, either as an instance (single point in time) or interval. In case no exact time is available, the date/time can be right truncated i.e. all of these are valid times: 2008-01-23T19:23:10+00:00; 2008-01-23T19:23:10; 2008-01-23; 2008-01; 2008; Except: 2008-01; 2008 all are ISO8601 compliant
     * Range: [Date](types/Date.md)
     * Example: 2018-05-11T10:00:00+01:00; 2018-05-11 None
 * [MIMARKS specimenâžžneg_cont_type](MIMARKS_specimen_neg_cont_type.md)  <sub>0..1</sub>
     * Description: The substance or equipment used as a negative control in an investigation
     * Range: [neg_cont_type_enum](neg_cont_type_enum.md)
     * Example:  None
 * [MIMARKS specimenâžžpos_cont_type](MIMARKS_specimen_pos_cont_type.md)  <sub>0..1</sub>
     * Description: The substance, mixture, product, or apparatus used to verify that a process which is part of an investigation delivers a true positive.
     * Range: [String](types/String.md)
     * Example:  None
 * [MIMARKS specimenâžženv_broad_scale](MIMARKS_specimen_env_broad_scale.md)  <sub>1..1</sub>
     * Description: Report the major environmental system the sample or specimen came from. The system(s) identified should have a coarse spatial grain, to provide the general environmental context of where the sampling was done (e.g. in the desert or a rainforest). We recommend using subclasses of EnvOâ€™s biome class:  http://purl.obolibrary.org/obo/ENVO_00000428. EnvO documentation about how to use the field: https://github.com/EnvironmentOntology/envo/wiki/Using-ENVO-with-MIxS
     * Range: [String](types/String.md)
     * Example: oceanic epipelagic zone biome [ENVO:01000033] for annotating a water sample from the photic zone in middle of the Atlantic Ocean None
 * [MIMARKS specimenâžženv_local_scale](MIMARKS_specimen_env_local_scale.md)  <sub>1..1</sub>
     * Description: Report the entity or entities which are in the sample or specimenâ€™s local vicinity and which you believe have significant causal influences on your sample or specimen. We recommend using EnvO terms which are of smaller spatial grain than your entry for env_broad_scale. Terms, such as anatomical sites, from other OBO Library ontologies which interoperate with EnvO (e.g. UBERON) are accepted in this field. EnvO documentation about how to use the field: https://github.com/EnvironmentOntology/envo/wiki/Using-ENVO-with-MIxS.
     * Range: [String](types/String.md)
     * Example: litter layer [ENVO:01000338]; Annotating a pooled sample taken from various vegetation layers in a forest consider: canopy [ENVO:00000047]|herb and fern layer [ENVO:01000337]|litter layer [ENVO:01000338]|understory [01000335]|shrub layer [ENVO:01000336]. None
 * [MIMARKS specimenâžženv_medium](MIMARKS_specimen_env_medium.md)  <sub>1..1</sub>
     * Description: Report the environmental material(s) immediately surrounding the sample or specimen at the time of sampling. We recommend using subclasses of 'environmental material' (http://purl.obolibrary.org/obo/ENVO_00010483). EnvO documentation about how to use the field: https://github.com/EnvironmentOntology/envo/wiki/Using-ENVO-with-MIxS . Terms from other OBO ontologies are permissible as long as they reference mass/volume nouns (e.g. air, water, blood) and not discrete, countable entities (e.g. a tree, a leaf, a table top).
     * Range: [String](types/String.md)
     * Example: soil [ENVO:00001998]; Annotating a fish swimming in the upper 100 m of the Atlantic Ocean, consider: ocean water [ENVO:00002151]. Example: Annotating a duck on a pond consider: pond water [ENVO:00002228]|air [ENVO_00002005] None
 * [MIMARKS specimenâžženv_package](MIMARKS_specimen_env_package.md)  <sub>0..1</sub>
     * Description: MIxS extension for reporting of measurements and observations obtained from one or more of the environments where the sample was obtained. All environmental packages listed here are further defined in separate subtables. By giving the name of the environmental package, a selection of fields can be made from the subtables and can be reported
     * Range: [env_package_enum](env_package_enum.md)
     * Example: soil None
 * [MIMARKS specimenâžžsource_mat_id](MIMARKS_specimen_source_mat_id.md)  <sub>0..1</sub>
     * Description: A unique identifier assigned to a material sample (as defined by http://rs.tdwg.org/dwc/terms/materialSampleID, and as opposed to a particular digital record of a material sample) used for extracting nucleic acids, and subsequent sequencing. The identifier can refer either to the original material collected or to any derived sub-samples. The INSDC qualifiers /specimen_voucher, /bio_material, or /culture_collection may or may not share the same value as the source_mat_id field. For instance, the /specimen_voucher qualifier and source_mat_id may both contain 'UAM:Herps:14' , referring to both the specimen voucher and sampled tissue with the same identifier. However, the /culture_collection qualifier may refer to a value from an initial culture (e.g. ATCC:11775) while source_mat_id would refer to an identifier from some derived culture from which the nucleic acids were extracted (e.g. xatc123 or ark:/2154/R2).
     * Range: [String](types/String.md)
     * Example: MPI012345 None
 * [MIMARKS specimenâžžrel_to_oxygen](MIMARKS_specimen_rel_to_oxygen.md)  <sub>0..1</sub>
     * Description: Is this organism an aerobe, anaerobe? Please note that aerobic and anaerobic are valid descriptors for microbial environments
     * Range: [rel_to_oxygen_enum](rel_to_oxygen_enum.md)
     * Example: aerobe None
 * [MIMARKS specimenâžžsamp_collec_device](MIMARKS_specimen_samp_collec_device.md)  <sub>0..1</sub>
     * Description: The device used to collect an environmental sample. This field accepts terms listed under environmental sampling device (http://purl.obolibrary.org/obo/ENVO). This field also accepts terms listed under specimen collection device (http://purl.obolibrary.org/obo/GENEPIO_0002094).
     * Range: [String](types/String.md)
     * Example: swab, biopsy, niskin bottle, push core, drag swab [GENEPIO:0002713] None
 * [MIMARKS specimenâžžsamp_collec_method](MIMARKS_specimen_samp_collec_method.md)  <sub>0..1</sub>
     * Description: The method employed for collecting the sample.
     * Range: [String](types/String.md)
     * Example: swabbing None
 * [MIMARKS specimenâžžsamp_mat_process](MIMARKS_specimen_samp_mat_process.md)  <sub>0..1</sub>
     * Description: A brief description of any processing applied to the sample during or after retrieving the sample from environment, or a link to the relevant protocol(s) performed.
     * Range: [String](types/String.md)
     * Example: filtering of seawater, storing samples in ethanol None
 * [MIMARKS specimenâžžsize_frac](MIMARKS_specimen_size_frac.md)  <sub>0..1</sub>
     * Description: Filtering pore size used in sample preparation
     * Range: [String](types/String.md)
     * Example: 0-0.22 micrometer None
 * [MIMARKS specimenâžžsamp_size](MIMARKS_specimen_samp_size.md)  <sub>0..1</sub>
     * Description: The total amount or size (volume (ml), mass (g) or area (m2) ) of sample collected.
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 5 liter None
 * [MIMARKS specimenâžžsamp_vol_we_dna_ext](MIMARKS_specimen_samp_vol_we_dna_ext.md)  <sub>0..1</sub>
     * Description: Volume (ml) or mass (g) of total collected sample processed for DNA extraction. Note: total sample collected should be entered under the term Sample Size (MIXS:0000001).
     * Range: [QuantityValue](QuantityValue.md)
     * Example: 1500 milliliter None
 * [MIMARKS specimenâžžnucl_acid_ext](MIMARKS_specimen_nucl_acid_ext.md)  <sub>0..1</sub>
     * Description: A link to a literature reference, electronic resource or a standard operating procedure (SOP), that describes the material separation to recover the nucleic acid fraction from a sample
     * Range: [String](types/String.md)
     * Example: https://mobio.com/media/wysiwyg/pdfs/protocols/12888.pdf None
 * [MIMARKS specimenâžžnucl_acid_amp](MIMARKS_specimen_nucl_acid_amp.md)  <sub>0..1</sub>
     * Description: A link to a literature reference, electronic resource or a standard operating procedure (SOP), that describes the enzymatic amplification (PCR, TMA, NASBA) of specific nucleic acids
     * Range: [String](types/String.md)
     * Example: https://phylogenomics.me/protocols/16s-pcr-protocol/ None
 * [MIMARKS specimenâžžlib_size](MIMARKS_specimen_lib_size.md)  <sub>0..1</sub>
     * Description: Total number of clones in the library prepared for the project
     * Range: [Integer](types/Integer.md)
     * Example: 50 None
 * [MIMARKS specimenâžžlib_reads_seqd](MIMARKS_specimen_lib_reads_seqd.md)  <sub>0..1</sub>
     * Description: Total number of clones sequenced from the library
     * Range: [Integer](types/Integer.md)
     * Example: 20 None
 * [MIMARKS specimenâžžlib_layout](MIMARKS_specimen_lib_layout.md)  <sub>0..1</sub>
     * Description: Specify whether to expect single, paired, or other configuration of reads
     * Range: [lib_layout_enum](lib_layout_enum.md)
     * Example: paired None
 * [MIMARKS specimenâžžlib_vector](MIMARKS_specimen_lib_vector.md)  <sub>0..1</sub>
     * Description: Cloning vector type(s) used in construction of libraries
     * Range: [String](types/String.md)
     * Example: Bacteriophage P1 None
 * [MIMARKS specimenâžžlib_screen](MIMARKS_specimen_lib_screen.md)  <sub>0..1</sub>
     * Description: Specific enrichment or screening methods applied before and/or after creating libraries
     * Range: [String](types/String.md)
     * Example: enriched, screened, normalized None
 * [MIMARKS specimenâžžtarget_gene](MIMARKS_specimen_target_gene.md)  <sub>1..1</sub>
     * Description: Targeted gene or locus name for marker gene studies
     * Range: [String](types/String.md)
     * Example: 16S rRNA, 18S rRNA, nif, amoA, rpo None
 * [MIMARKS specimenâžžtarget_subfragment](MIMARKS_specimen_target_subfragment.md)  <sub>0..1</sub>
     * Description: Name of subfragment of a gene or locus. Important to e.g. identify special regions on marker genes like V6 on 16S rRNA
     * Range: [String](types/String.md)
     * Example: V6, V9, ITS None
 * [MIMARKS specimenâžžpcr_primers](MIMARKS_specimen_pcr_primers.md)  <sub>0..1</sub>
     * Description: PCR primers that were used to amplify the sequence of the targeted gene, locus or subfragment. This field should contain all the primers used for a single PCR reaction if multiple forward or reverse primers are present in a single PCR reaction. The primer sequence should be reported in uppercase letters
     * Range: [String](types/String.md)
     * Example: FWD:GTGCCAGCMGCCGCGGTAA;REV:GGACTACHVGGGTWTCTAAT None
 * [MIMARKS specimenâžžmid](MIMARKS_specimen_mid.md)  <sub>0..1</sub>
     * Description: Molecular barcodes, called Multiplex Identifiers (MIDs), that are used to specifically tag unique samples in a sequencing run. Sequence should be reported in uppercase letters
     * Range: [String](types/String.md)
     * Example: GTGAATAT None
 * [MIMARKS specimenâžžadapters](MIMARKS_specimen_adapters.md)  <sub>0..1</sub>
     * Description: Adapters provide priming sequences for both amplification and sequencing of the sample-library fragments. Both adapters should be reported; in uppercase letters
     * Range: [String](types/String.md)
     * Example: AATGATACGGCGACCACCGAGATCTACACGCT;CAAGCAGAAGACGGCATACGAGAT None
 * [MIMARKS specimenâžžpcr_cond](MIMARKS_specimen_pcr_cond.md)  <sub>0..1</sub>
     * Description: Description of reaction conditions and components of PCR in the form of 'initial denaturation:94degC_1.5min; annealing=...'
     * Range: [String](types/String.md)
     * Example: initial denaturation:94_3;annealing:50_1;elongation:72_1.5;final elongation:72_10;35 None
 * [MIMARKS specimenâžžseq_meth](MIMARKS_specimen_seq_meth.md)  <sub>1..1</sub>
     * Description: Sequencing machine used. Where possible the term should be taken from the OBI list of DNA sequencers (http://purl.obolibrary.org/obo/OBI_0400103).
     * Range: [String](types/String.md)
     * Example: 454 Genome Sequencer FLX [OBI:0000702] None
 * [MIMARKS specimenâžžseq_quality_check](MIMARKS_specimen_seq_quality_check.md)  <sub>0..1</sub>
     * Description: Indicate if the sequence has been called by automatic systems (none) or undergone a manual editing procedure (e.g. by inspecting the raw data or chromatograms). Applied only for sequences that are not submitted to SRA,ENA or DRA
     * Range: [String](types/String.md)
     * Example: none None
 * [MIMARKS specimenâžžchimera_check](MIMARKS_specimen_chimera_check.md)  <sub>0..1</sub>
     * Description: Tool(s) used for chimera checking, including version number and parameters, to discover and remove chimeric sequences. A chimeric sequence is comprised of two or more phylogenetically distinct parent sequences.
     * Range: [String](types/String.md)
     * Example: uchime;v4.1;default parameters None
 * [MIMARKS specimenâžžassembly_software](MIMARKS_specimen_assembly_software.md)  <sub>0..1</sub>
     * Description: Tool(s) used for assembly, including version number and parameters
     * Range: [String](types/String.md)
     * Example: metaSPAdes;3.11.0;kmer set 21,33,55,77,99,121, default parameters otherwise None
 * [MIMARKS specimenâžžassociated resource](MIMARKS_specimen_associated_resource.md)  <sub>0..1</sub>
     * Description: A related resource that is referenced, cited, or otherwise associated to the sequence.
     * Range: [String](types/String.md)
     * Example: http://www.earthmicrobiome.org/ None
 * [MIMARKS specimenâžžsop](MIMARKS_specimen_sop.md)  <sub>0..1</sub>
     * Description: Standard operating procedures used in assembly and/or annotation of genomes, metagenomes or environmental sequences
     * Range: [String](types/String.md)
     * Example: http://press.igsb.anl.gov/earthmicrobiome/protocols-and-standards/its/ None

## Other properties

|  |  |  |
| --- | --- | --- |
| **Aliases:** | | MIMARKS.specimen |
| **See also:** | | https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3367316 |

