

CREATE TABLE agriculture (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	pres_animal_insect VARCHAR(15), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT NOT NULL, 
	pool_dna_extracts TEXT NOT NULL, 
	crop_rotation TEXT, 
	root_med_ph FLOAT, 
	"water_pH" FLOAT, 
	porosity TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	host_age TEXT NOT NULL, 
	host_dry_mass TEXT, 
	host_height TEXT NOT NULL, 
	host_length TEXT NOT NULL, 
	host_tot_mass TEXT NOT NULL, 
	humidity TEXT, 
	microbial_biomass TEXT NOT NULL, 
	photosynt_activ TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	solar_irradiance TEXT, 
	texture TEXT, 
	tot_car TEXT, 
	tot_nitro TEXT, 
	tot_org_carb TEXT, 
	tot_phos TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	crop_yield TEXT, 
	host_taxid INTEGER NOT NULL, 
	lot_number TEXT, 
	samp_store_dur TEXT NOT NULL, 
	texture_meth TEXT, 
	enrichment_protocol TEXT, 
	genetic_mod TEXT, 
	photosynt_activ_meth TEXT, 
	samp_collect_method TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	microbial_biomass_meth TEXT NOT NULL, 
	ph_meth TEXT, 
	previous_land_use_meth TEXT, 
	soil_type_meth TEXT NOT NULL, 
	tot_n_meth TEXT, 
	tot_org_c_meth TEXT, 
	host_subspecf_genlin TEXT, 
	assembly_quality TEXT, 
	fao_class VARCHAR(13), 
	host_phenotype TEXT NOT NULL, 
	anim_water_method TEXT, 
	farm_equip TEXT, 
	fertilizer_admin TEXT, 
	food_prod TEXT, 
	food_trav_mode TEXT, 
	plant_struc TEXT, 
	plant_water_method TEXT, 
	season TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	spikein_strain TEXT, 
	adjacent_environment TEXT, 
	ances_data TEXT, 
	cur_vegetation TEXT, 
	"Food_Product_type" TEXT, 
	"Food_source" TEXT NOT NULL, 
	host_common_name TEXT NOT NULL, 
	host_genotype TEXT NOT NULL, 
	host_life_stage TEXT NOT NULL, 
	host_symbiont TEXT, 
	library_prep_kit TEXT, 
	local_class TEXT, 
	plant_product TEXT, 
	samp_collect_device TEXT NOT NULL, 
	samp_store_loc TEXT NOT NULL, 
	sequencing_location TEXT, 
	soil_type TEXT NOT NULL, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	organism_count TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	herbicide_regm TEXT, 
	non_mineral_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	salt_regm TEXT, 
	agrochem_addition TEXT, 
	perturbation TEXT, 
	store_cond TEXT NOT NULL, 
	climate_environment TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	previous_land_use TEXT, 
	food_trav_vehic TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	url TEXT, 
	soil_horizon VARCHAR(10), 
	cult_isol_date DATE, 
	farm_equip_san TEXT, 
	farm_equip_shared TEXT, 
	food_harvest_proc TEXT, 
	rel_location TEXT, 
	samp_pooling TEXT, 
	soil_conductivity TEXT, 
	soil_cover TEXT, 
	"soil_pH" FLOAT, 
	trophic_level VARCHAR(22), 
	assembly_software TEXT, 
	biotic_relationship VARCHAR(11), 
	seq_quality_check TEXT, 
	lib_layout VARCHAR(6), 
	mid TEXT NOT NULL, 
	adapters TEXT NOT NULL, 
	samp_size TEXT NOT NULL, 
	samp_vol_we_dna_ext TEXT NOT NULL, 
	host_spec_range INTEGER NOT NULL, 
	lib_reads_seqd INTEGER NOT NULL, 
	lib_size INTEGER, 
	isol_growth_condt TEXT, 
	nucl_acid_amp TEXT NOT NULL, 
	nucl_acid_ext TEXT NOT NULL, 
	sop TEXT, 
	pathogenicity TEXT NOT NULL, 
	seq_meth TEXT NOT NULL, 
	annot TEXT, 
	lib_screen TEXT NOT NULL, 
	lib_vector TEXT NOT NULL, 
	source_mat_id TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT NOT NULL, 
	assembly_name TEXT NOT NULL, 
	chimera_check TEXT NOT NULL, 
	specific_host TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_mat_process TEXT NOT NULL, 
	pcr_cond TEXT NOT NULL, 
	pcr_primers TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, pres_animal_insect, cur_land_use, tillage, farm_water_source, water_source_shared, horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, root_med_ph, "water_pH", porosity, annual_precpt, annual_temp, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, microbial_biomass, photosynt_activ, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, slope_aspect, slope_gradient, solar_irradiance, texture, tot_car, tot_nitro, tot_org_carb, tot_phos, water_temp_regm, watering_regm, ph_regm, crop_yield, host_taxid, lot_number, samp_store_dur, texture_meth, enrichment_protocol, genetic_mod, photosynt_activ_meth, samp_collect_method, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_class_info, link_climate_info, local_class_meth, microbial_biomass_meth, ph_meth, previous_land_use_meth, soil_type_meth, tot_n_meth, tot_org_c_meth, host_subspecf_genlin, assembly_quality, fao_class, host_phenotype, anim_water_method, farm_equip, fertilizer_admin, food_prod, food_trav_mode, plant_struc, plant_water_method, season, plant_growth_med, chem_administration, spikein_strain, adjacent_environment, ances_data, cur_vegetation, "Food_Product_type", "Food_source", host_common_name, host_genotype, host_life_stage, host_symbiont, library_prep_kit, local_class, plant_product, samp_collect_device, samp_store_loc, sequencing_location, soil_type, misc_param, root_med_carbon, root_med_macronutr, organism_count, fungicide_regm, gaseous_environment, herbicide_regm, non_mineral_nutr_regm, pesticide_regm, salt_regm, agrochem_addition, perturbation, store_cond, climate_environment, season_environment, standing_water_regm, previous_land_use, food_trav_vehic, extreme_event, fire, flooding, url, soil_horizon, cult_isol_date, farm_equip_san, farm_equip_shared, food_harvest_proc, rel_location, samp_pooling, soil_conductivity, soil_cover, "soil_pH", trophic_level, assembly_software, biotic_relationship, seq_quality_check, lib_layout, mid, adapters, samp_size, samp_vol_we_dna_ext, host_spec_range, lib_reads_seqd, lib_size, isol_growth_condt, nucl_acid_amp, nucl_acid_ext, sop, pathogenicity, seq_meth, annot, lib_screen, lib_vector, source_mat_id, target_gene, target_subfragment, assembly_name, chimera_check, specific_host, host_disease_stat, samp_mat_process, pcr_cond, pcr_primers)
);

CREATE TABLE "agriculture_MIGS_bacteria" (
	lat_lon TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	pres_animal_insect VARCHAR(15), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT NOT NULL, 
	pool_dna_extracts TEXT NOT NULL, 
	crop_rotation TEXT, 
	root_med_ph FLOAT, 
	"water_pH" FLOAT, 
	porosity TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	host_age TEXT NOT NULL, 
	host_dry_mass TEXT, 
	host_height TEXT NOT NULL, 
	host_length TEXT NOT NULL, 
	host_tot_mass TEXT NOT NULL, 
	humidity TEXT, 
	microbial_biomass TEXT NOT NULL, 
	photosynt_activ TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	depth TEXT NOT NULL, 
	solar_irradiance TEXT, 
	texture TEXT, 
	tot_car TEXT, 
	tot_nitro TEXT, 
	tot_org_carb TEXT, 
	tot_phos TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	crop_yield TEXT, 
	"temp" TEXT, 
	host_taxid INTEGER NOT NULL, 
	lot_number TEXT, 
	samp_store_dur TEXT NOT NULL, 
	texture_meth TEXT, 
	enrichment_protocol TEXT, 
	genetic_mod TEXT, 
	photosynt_activ_meth TEXT, 
	samp_collect_method TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	microbial_biomass_meth TEXT NOT NULL, 
	ph_meth TEXT, 
	previous_land_use_meth TEXT, 
	soil_type_meth TEXT NOT NULL, 
	tot_n_meth TEXT, 
	tot_org_c_meth TEXT, 
	host_subspecf_genlin TEXT, 
	assembly_quality TEXT, 
	fao_class VARCHAR(13), 
	host_phenotype TEXT NOT NULL, 
	anim_water_method TEXT, 
	farm_equip TEXT, 
	fertilizer_admin TEXT, 
	food_prod TEXT, 
	food_trav_mode TEXT, 
	plant_struc TEXT, 
	plant_water_method TEXT, 
	season TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	spikein_strain TEXT, 
	adjacent_environment TEXT, 
	ances_data TEXT, 
	cur_vegetation TEXT, 
	"Food_Product_type" TEXT, 
	"Food_source" TEXT NOT NULL, 
	host_common_name TEXT NOT NULL, 
	host_genotype TEXT NOT NULL, 
	host_life_stage TEXT NOT NULL, 
	host_symbiont TEXT, 
	library_prep_kit TEXT, 
	local_class TEXT, 
	plant_product TEXT, 
	samp_collect_device TEXT NOT NULL, 
	samp_store_loc TEXT NOT NULL, 
	sequencing_location TEXT, 
	soil_type TEXT NOT NULL, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	organism_count TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	herbicide_regm TEXT, 
	non_mineral_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	salt_regm TEXT, 
	agrochem_addition TEXT, 
	perturbation TEXT, 
	store_cond TEXT NOT NULL, 
	climate_environment TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	previous_land_use TEXT, 
	food_trav_vehic TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	url TEXT, 
	soil_horizon VARCHAR(10), 
	cult_isol_date DATE, 
	farm_equip_san TEXT, 
	farm_equip_shared TEXT, 
	food_harvest_proc TEXT, 
	rel_location TEXT, 
	samp_pooling TEXT, 
	soil_conductivity TEXT, 
	soil_cover TEXT, 
	"soil_pH" FLOAT, 
	trophic_level VARCHAR(22), 
	assembly_software TEXT, 
	biotic_relationship VARCHAR(11), 
	seq_quality_check TEXT, 
	lib_layout VARCHAR(6), 
	mid TEXT NOT NULL, 
	adapters TEXT NOT NULL, 
	samp_size TEXT NOT NULL, 
	samp_vol_we_dna_ext TEXT NOT NULL, 
	host_spec_range INTEGER NOT NULL, 
	lib_reads_seqd INTEGER NOT NULL, 
	lib_size INTEGER, 
	isol_growth_condt TEXT, 
	nucl_acid_amp TEXT NOT NULL, 
	nucl_acid_ext TEXT NOT NULL, 
	sop TEXT, 
	pathogenicity TEXT NOT NULL, 
	seq_meth TEXT NOT NULL, 
	annot TEXT, 
	lib_screen TEXT NOT NULL, 
	lib_vector TEXT NOT NULL, 
	source_mat_id TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT NOT NULL, 
	assembly_name TEXT NOT NULL, 
	chimera_check TEXT NOT NULL, 
	specific_host TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_mat_process TEXT NOT NULL, 
	pcr_cond TEXT NOT NULL, 
	pcr_primers TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	PRIMARY KEY (lat_lon, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, pres_animal_insect, cur_land_use, tillage, farm_water_source, water_source_shared, horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, root_med_ph, "water_pH", porosity, annual_precpt, annual_temp, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, microbial_biomass, photosynt_activ, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, slope_aspect, slope_gradient, depth, solar_irradiance, texture, tot_car, tot_nitro, tot_org_carb, tot_phos, water_temp_regm, watering_regm, ph_regm, crop_yield, "temp", host_taxid, lot_number, samp_store_dur, texture_meth, enrichment_protocol, genetic_mod, photosynt_activ_meth, samp_collect_method, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_class_info, link_climate_info, local_class_meth, microbial_biomass_meth, ph_meth, previous_land_use_meth, soil_type_meth, tot_n_meth, tot_org_c_meth, host_subspecf_genlin, assembly_quality, fao_class, host_phenotype, anim_water_method, farm_equip, fertilizer_admin, food_prod, food_trav_mode, plant_struc, plant_water_method, season, plant_growth_med, chem_administration, spikein_strain, adjacent_environment, ances_data, cur_vegetation, "Food_Product_type", "Food_source", host_common_name, host_genotype, host_life_stage, host_symbiont, library_prep_kit, local_class, plant_product, samp_collect_device, samp_store_loc, sequencing_location, soil_type, misc_param, root_med_carbon, root_med_macronutr, organism_count, fungicide_regm, gaseous_environment, herbicide_regm, non_mineral_nutr_regm, pesticide_regm, salt_regm, agrochem_addition, perturbation, store_cond, climate_environment, season_environment, standing_water_regm, previous_land_use, food_trav_vehic, extreme_event, fire, flooding, url, soil_horizon, cult_isol_date, farm_equip_san, farm_equip_shared, food_harvest_proc, rel_location, samp_pooling, soil_conductivity, soil_cover, "soil_pH", trophic_level, assembly_software, biotic_relationship, seq_quality_check, lib_layout, mid, adapters, samp_size, samp_vol_we_dna_ext, host_spec_range, lib_reads_seqd, lib_size, isol_growth_condt, nucl_acid_amp, nucl_acid_ext, sop, pathogenicity, seq_meth, annot, lib_screen, lib_vector, source_mat_id, target_gene, target_subfragment, assembly_name, chimera_check, specific_host, host_disease_stat, samp_mat_process, pcr_cond, pcr_primers, samp_name, samp_taxon_id, project_name, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, propagation, encoded_traits, rel_to_oxygen, samp_collec_device, samp_collec_method, size_frac, source_uvig, virus_enrich_appr, tax_ident, assembly_qual, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource)
);

CREATE TABLE "agriculture_MIGS_eukaryote" (
	lat_lon TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	pres_animal_insect VARCHAR(15), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT NOT NULL, 
	pool_dna_extracts TEXT NOT NULL, 
	crop_rotation TEXT, 
	root_med_ph FLOAT, 
	"water_pH" FLOAT, 
	porosity TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	host_age TEXT NOT NULL, 
	host_dry_mass TEXT, 
	host_height TEXT NOT NULL, 
	host_length TEXT NOT NULL, 
	host_tot_mass TEXT NOT NULL, 
	humidity TEXT, 
	microbial_biomass TEXT NOT NULL, 
	photosynt_activ TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	depth TEXT NOT NULL, 
	solar_irradiance TEXT, 
	texture TEXT, 
	tot_car TEXT, 
	tot_nitro TEXT, 
	tot_org_carb TEXT, 
	tot_phos TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	crop_yield TEXT, 
	"temp" TEXT, 
	host_taxid INTEGER NOT NULL, 
	lot_number TEXT, 
	samp_store_dur TEXT NOT NULL, 
	texture_meth TEXT, 
	enrichment_protocol TEXT, 
	genetic_mod TEXT, 
	photosynt_activ_meth TEXT, 
	samp_collect_method TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	microbial_biomass_meth TEXT NOT NULL, 
	ph_meth TEXT, 
	previous_land_use_meth TEXT, 
	soil_type_meth TEXT NOT NULL, 
	tot_n_meth TEXT, 
	tot_org_c_meth TEXT, 
	host_subspecf_genlin TEXT, 
	assembly_quality TEXT, 
	fao_class VARCHAR(13), 
	host_phenotype TEXT NOT NULL, 
	anim_water_method TEXT, 
	farm_equip TEXT, 
	fertilizer_admin TEXT, 
	food_prod TEXT, 
	food_trav_mode TEXT, 
	plant_struc TEXT, 
	plant_water_method TEXT, 
	season TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	spikein_strain TEXT, 
	adjacent_environment TEXT, 
	ances_data TEXT, 
	cur_vegetation TEXT, 
	"Food_Product_type" TEXT, 
	"Food_source" TEXT NOT NULL, 
	host_common_name TEXT NOT NULL, 
	host_genotype TEXT NOT NULL, 
	host_life_stage TEXT NOT NULL, 
	host_symbiont TEXT, 
	library_prep_kit TEXT, 
	local_class TEXT, 
	plant_product TEXT, 
	samp_collect_device TEXT NOT NULL, 
	samp_store_loc TEXT NOT NULL, 
	sequencing_location TEXT, 
	soil_type TEXT NOT NULL, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	organism_count TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	herbicide_regm TEXT, 
	non_mineral_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	salt_regm TEXT, 
	agrochem_addition TEXT, 
	perturbation TEXT, 
	store_cond TEXT NOT NULL, 
	climate_environment TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	previous_land_use TEXT, 
	food_trav_vehic TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	url TEXT, 
	soil_horizon VARCHAR(10), 
	cult_isol_date DATE, 
	farm_equip_san TEXT, 
	farm_equip_shared TEXT, 
	food_harvest_proc TEXT, 
	rel_location TEXT, 
	samp_pooling TEXT, 
	soil_conductivity TEXT, 
	soil_cover TEXT, 
	"soil_pH" FLOAT, 
	trophic_level VARCHAR(22), 
	assembly_software TEXT, 
	biotic_relationship VARCHAR(11), 
	seq_quality_check TEXT, 
	lib_layout VARCHAR(6), 
	mid TEXT NOT NULL, 
	adapters TEXT NOT NULL, 
	samp_size TEXT NOT NULL, 
	samp_vol_we_dna_ext TEXT NOT NULL, 
	host_spec_range INTEGER NOT NULL, 
	lib_reads_seqd INTEGER NOT NULL, 
	lib_size INTEGER, 
	isol_growth_condt TEXT, 
	nucl_acid_amp TEXT NOT NULL, 
	nucl_acid_ext TEXT NOT NULL, 
	sop TEXT, 
	pathogenicity TEXT NOT NULL, 
	seq_meth TEXT NOT NULL, 
	annot TEXT, 
	lib_screen TEXT NOT NULL, 
	lib_vector TEXT NOT NULL, 
	source_mat_id TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT NOT NULL, 
	assembly_name TEXT NOT NULL, 
	chimera_check TEXT NOT NULL, 
	specific_host TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_mat_process TEXT NOT NULL, 
	pcr_cond TEXT NOT NULL, 
	pcr_primers TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	PRIMARY KEY (lat_lon, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, pres_animal_insect, cur_land_use, tillage, farm_water_source, water_source_shared, horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, root_med_ph, "water_pH", porosity, annual_precpt, annual_temp, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, microbial_biomass, photosynt_activ, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, slope_aspect, slope_gradient, depth, solar_irradiance, texture, tot_car, tot_nitro, tot_org_carb, tot_phos, water_temp_regm, watering_regm, ph_regm, crop_yield, "temp", host_taxid, lot_number, samp_store_dur, texture_meth, enrichment_protocol, genetic_mod, photosynt_activ_meth, samp_collect_method, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_class_info, link_climate_info, local_class_meth, microbial_biomass_meth, ph_meth, previous_land_use_meth, soil_type_meth, tot_n_meth, tot_org_c_meth, host_subspecf_genlin, assembly_quality, fao_class, host_phenotype, anim_water_method, farm_equip, fertilizer_admin, food_prod, food_trav_mode, plant_struc, plant_water_method, season, plant_growth_med, chem_administration, spikein_strain, adjacent_environment, ances_data, cur_vegetation, "Food_Product_type", "Food_source", host_common_name, host_genotype, host_life_stage, host_symbiont, library_prep_kit, local_class, plant_product, samp_collect_device, samp_store_loc, sequencing_location, soil_type, misc_param, root_med_carbon, root_med_macronutr, organism_count, fungicide_regm, gaseous_environment, herbicide_regm, non_mineral_nutr_regm, pesticide_regm, salt_regm, agrochem_addition, perturbation, store_cond, climate_environment, season_environment, standing_water_regm, previous_land_use, food_trav_vehic, extreme_event, fire, flooding, url, soil_horizon, cult_isol_date, farm_equip_san, farm_equip_shared, food_harvest_proc, rel_location, samp_pooling, soil_conductivity, soil_cover, "soil_pH", trophic_level, assembly_software, biotic_relationship, seq_quality_check, lib_layout, mid, adapters, samp_size, samp_vol_we_dna_ext, host_spec_range, lib_reads_seqd, lib_size, isol_growth_condt, nucl_acid_amp, nucl_acid_ext, sop, pathogenicity, seq_meth, annot, lib_screen, lib_vector, source_mat_id, target_gene, target_subfragment, assembly_name, chimera_check, specific_host, host_disease_stat, samp_mat_process, pcr_cond, pcr_primers, samp_name, samp_taxon_id, project_name, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, propagation, encoded_traits, rel_to_oxygen, samp_collec_device, samp_collec_method, size_frac, source_uvig, virus_enrich_appr, tax_ident, assembly_qual, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource)
);

CREATE TABLE "agriculture_MIGS_org" (
	lat_lon TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	pres_animal_insect VARCHAR(15), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT NOT NULL, 
	pool_dna_extracts TEXT NOT NULL, 
	crop_rotation TEXT, 
	root_med_ph FLOAT, 
	"water_pH" FLOAT, 
	porosity TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	host_age TEXT NOT NULL, 
	host_dry_mass TEXT, 
	host_height TEXT NOT NULL, 
	host_length TEXT NOT NULL, 
	host_tot_mass TEXT NOT NULL, 
	humidity TEXT, 
	microbial_biomass TEXT NOT NULL, 
	photosynt_activ TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	depth TEXT NOT NULL, 
	solar_irradiance TEXT, 
	texture TEXT, 
	tot_car TEXT, 
	tot_nitro TEXT, 
	tot_org_carb TEXT, 
	tot_phos TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	crop_yield TEXT, 
	"temp" TEXT, 
	host_taxid INTEGER NOT NULL, 
	lot_number TEXT, 
	samp_store_dur TEXT NOT NULL, 
	texture_meth TEXT, 
	enrichment_protocol TEXT, 
	genetic_mod TEXT, 
	photosynt_activ_meth TEXT, 
	samp_collect_method TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	microbial_biomass_meth TEXT NOT NULL, 
	ph_meth TEXT, 
	previous_land_use_meth TEXT, 
	soil_type_meth TEXT NOT NULL, 
	tot_n_meth TEXT, 
	tot_org_c_meth TEXT, 
	host_subspecf_genlin TEXT, 
	assembly_quality TEXT, 
	fao_class VARCHAR(13), 
	host_phenotype TEXT NOT NULL, 
	anim_water_method TEXT, 
	farm_equip TEXT, 
	fertilizer_admin TEXT, 
	food_prod TEXT, 
	food_trav_mode TEXT, 
	plant_struc TEXT, 
	plant_water_method TEXT, 
	season TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	spikein_strain TEXT, 
	adjacent_environment TEXT, 
	ances_data TEXT, 
	cur_vegetation TEXT, 
	"Food_Product_type" TEXT, 
	"Food_source" TEXT NOT NULL, 
	host_common_name TEXT NOT NULL, 
	host_genotype TEXT NOT NULL, 
	host_life_stage TEXT NOT NULL, 
	host_symbiont TEXT, 
	library_prep_kit TEXT, 
	local_class TEXT, 
	plant_product TEXT, 
	samp_collect_device TEXT NOT NULL, 
	samp_store_loc TEXT NOT NULL, 
	sequencing_location TEXT, 
	soil_type TEXT NOT NULL, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	organism_count TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	herbicide_regm TEXT, 
	non_mineral_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	salt_regm TEXT, 
	agrochem_addition TEXT, 
	perturbation TEXT, 
	store_cond TEXT NOT NULL, 
	climate_environment TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	previous_land_use TEXT, 
	food_trav_vehic TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	url TEXT, 
	soil_horizon VARCHAR(10), 
	cult_isol_date DATE, 
	farm_equip_san TEXT, 
	farm_equip_shared TEXT, 
	food_harvest_proc TEXT, 
	rel_location TEXT, 
	samp_pooling TEXT, 
	soil_conductivity TEXT, 
	soil_cover TEXT, 
	"soil_pH" FLOAT, 
	trophic_level VARCHAR(22), 
	assembly_software TEXT, 
	biotic_relationship VARCHAR(11), 
	seq_quality_check TEXT, 
	lib_layout VARCHAR(6), 
	mid TEXT NOT NULL, 
	adapters TEXT NOT NULL, 
	samp_size TEXT NOT NULL, 
	samp_vol_we_dna_ext TEXT NOT NULL, 
	host_spec_range INTEGER NOT NULL, 
	lib_reads_seqd INTEGER NOT NULL, 
	lib_size INTEGER, 
	isol_growth_condt TEXT, 
	nucl_acid_amp TEXT NOT NULL, 
	nucl_acid_ext TEXT NOT NULL, 
	sop TEXT, 
	pathogenicity TEXT NOT NULL, 
	seq_meth TEXT NOT NULL, 
	annot TEXT, 
	lib_screen TEXT NOT NULL, 
	lib_vector TEXT NOT NULL, 
	source_mat_id TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT NOT NULL, 
	assembly_name TEXT NOT NULL, 
	chimera_check TEXT NOT NULL, 
	specific_host TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_mat_process TEXT NOT NULL, 
	pcr_cond TEXT NOT NULL, 
	pcr_primers TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	PRIMARY KEY (lat_lon, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, pres_animal_insect, cur_land_use, tillage, farm_water_source, water_source_shared, horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, root_med_ph, "water_pH", porosity, annual_precpt, annual_temp, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, microbial_biomass, photosynt_activ, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, slope_aspect, slope_gradient, depth, solar_irradiance, texture, tot_car, tot_nitro, tot_org_carb, tot_phos, water_temp_regm, watering_regm, ph_regm, crop_yield, "temp", host_taxid, lot_number, samp_store_dur, texture_meth, enrichment_protocol, genetic_mod, photosynt_activ_meth, samp_collect_method, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_class_info, link_climate_info, local_class_meth, microbial_biomass_meth, ph_meth, previous_land_use_meth, soil_type_meth, tot_n_meth, tot_org_c_meth, host_subspecf_genlin, assembly_quality, fao_class, host_phenotype, anim_water_method, farm_equip, fertilizer_admin, food_prod, food_trav_mode, plant_struc, plant_water_method, season, plant_growth_med, chem_administration, spikein_strain, adjacent_environment, ances_data, cur_vegetation, "Food_Product_type", "Food_source", host_common_name, host_genotype, host_life_stage, host_symbiont, library_prep_kit, local_class, plant_product, samp_collect_device, samp_store_loc, sequencing_location, soil_type, misc_param, root_med_carbon, root_med_macronutr, organism_count, fungicide_regm, gaseous_environment, herbicide_regm, non_mineral_nutr_regm, pesticide_regm, salt_regm, agrochem_addition, perturbation, store_cond, climate_environment, season_environment, standing_water_regm, previous_land_use, food_trav_vehic, extreme_event, fire, flooding, url, soil_horizon, cult_isol_date, farm_equip_san, farm_equip_shared, food_harvest_proc, rel_location, samp_pooling, soil_conductivity, soil_cover, "soil_pH", trophic_level, assembly_software, biotic_relationship, seq_quality_check, lib_layout, mid, adapters, samp_size, samp_vol_we_dna_ext, host_spec_range, lib_reads_seqd, lib_size, isol_growth_condt, nucl_acid_amp, nucl_acid_ext, sop, pathogenicity, seq_meth, annot, lib_screen, lib_vector, source_mat_id, target_gene, target_subfragment, assembly_name, chimera_check, specific_host, host_disease_stat, samp_mat_process, pcr_cond, pcr_primers, samp_name, samp_taxon_id, project_name, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, propagation, encoded_traits, rel_to_oxygen, samp_collec_device, samp_collec_method, size_frac, source_uvig, virus_enrich_appr, tax_ident, assembly_qual, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource)
);

CREATE TABLE "agriculture_MIGS_plant" (
	lat_lon TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	pres_animal_insect VARCHAR(15), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT NOT NULL, 
	pool_dna_extracts TEXT NOT NULL, 
	crop_rotation TEXT, 
	root_med_ph FLOAT, 
	"water_pH" FLOAT, 
	porosity TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	host_age TEXT NOT NULL, 
	host_dry_mass TEXT, 
	host_height TEXT NOT NULL, 
	host_length TEXT NOT NULL, 
	host_tot_mass TEXT NOT NULL, 
	humidity TEXT, 
	microbial_biomass TEXT NOT NULL, 
	photosynt_activ TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	depth TEXT NOT NULL, 
	solar_irradiance TEXT, 
	texture TEXT, 
	tot_car TEXT, 
	tot_nitro TEXT, 
	tot_org_carb TEXT, 
	tot_phos TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	crop_yield TEXT, 
	"temp" TEXT, 
	host_taxid INTEGER NOT NULL, 
	lot_number TEXT, 
	samp_store_dur TEXT NOT NULL, 
	texture_meth TEXT, 
	enrichment_protocol TEXT, 
	genetic_mod TEXT, 
	photosynt_activ_meth TEXT, 
	samp_collect_method TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	microbial_biomass_meth TEXT NOT NULL, 
	ph_meth TEXT, 
	previous_land_use_meth TEXT, 
	soil_type_meth TEXT NOT NULL, 
	tot_n_meth TEXT, 
	tot_org_c_meth TEXT, 
	host_subspecf_genlin TEXT, 
	assembly_quality TEXT, 
	fao_class VARCHAR(13), 
	host_phenotype TEXT NOT NULL, 
	anim_water_method TEXT, 
	farm_equip TEXT, 
	fertilizer_admin TEXT, 
	food_prod TEXT, 
	food_trav_mode TEXT, 
	plant_struc TEXT, 
	plant_water_method TEXT, 
	season TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	spikein_strain TEXT, 
	adjacent_environment TEXT, 
	ances_data TEXT, 
	cur_vegetation TEXT, 
	"Food_Product_type" TEXT, 
	"Food_source" TEXT NOT NULL, 
	host_common_name TEXT NOT NULL, 
	host_genotype TEXT NOT NULL, 
	host_life_stage TEXT NOT NULL, 
	host_symbiont TEXT, 
	library_prep_kit TEXT, 
	local_class TEXT, 
	plant_product TEXT, 
	samp_collect_device TEXT NOT NULL, 
	samp_store_loc TEXT NOT NULL, 
	sequencing_location TEXT, 
	soil_type TEXT NOT NULL, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	organism_count TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	herbicide_regm TEXT, 
	non_mineral_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	salt_regm TEXT, 
	agrochem_addition TEXT, 
	perturbation TEXT, 
	store_cond TEXT NOT NULL, 
	climate_environment TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	previous_land_use TEXT, 
	food_trav_vehic TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	url TEXT, 
	soil_horizon VARCHAR(10), 
	cult_isol_date DATE, 
	farm_equip_san TEXT, 
	farm_equip_shared TEXT, 
	food_harvest_proc TEXT, 
	rel_location TEXT, 
	samp_pooling TEXT, 
	soil_conductivity TEXT, 
	soil_cover TEXT, 
	"soil_pH" FLOAT, 
	trophic_level VARCHAR(22), 
	assembly_software TEXT, 
	biotic_relationship VARCHAR(11), 
	seq_quality_check TEXT, 
	lib_layout VARCHAR(6), 
	mid TEXT NOT NULL, 
	adapters TEXT NOT NULL, 
	samp_size TEXT NOT NULL, 
	samp_vol_we_dna_ext TEXT NOT NULL, 
	host_spec_range INTEGER NOT NULL, 
	lib_reads_seqd INTEGER NOT NULL, 
	lib_size INTEGER, 
	isol_growth_condt TEXT, 
	nucl_acid_amp TEXT NOT NULL, 
	nucl_acid_ext TEXT NOT NULL, 
	sop TEXT, 
	pathogenicity TEXT NOT NULL, 
	seq_meth TEXT NOT NULL, 
	annot TEXT, 
	lib_screen TEXT NOT NULL, 
	lib_vector TEXT NOT NULL, 
	source_mat_id TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT NOT NULL, 
	assembly_name TEXT NOT NULL, 
	chimera_check TEXT NOT NULL, 
	specific_host TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_mat_process TEXT NOT NULL, 
	pcr_cond TEXT NOT NULL, 
	pcr_primers TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	PRIMARY KEY (lat_lon, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, pres_animal_insect, cur_land_use, tillage, farm_water_source, water_source_shared, horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, root_med_ph, "water_pH", porosity, annual_precpt, annual_temp, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, microbial_biomass, photosynt_activ, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, slope_aspect, slope_gradient, depth, solar_irradiance, texture, tot_car, tot_nitro, tot_org_carb, tot_phos, water_temp_regm, watering_regm, ph_regm, crop_yield, "temp", host_taxid, lot_number, samp_store_dur, texture_meth, enrichment_protocol, genetic_mod, photosynt_activ_meth, samp_collect_method, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_class_info, link_climate_info, local_class_meth, microbial_biomass_meth, ph_meth, previous_land_use_meth, soil_type_meth, tot_n_meth, tot_org_c_meth, host_subspecf_genlin, assembly_quality, fao_class, host_phenotype, anim_water_method, farm_equip, fertilizer_admin, food_prod, food_trav_mode, plant_struc, plant_water_method, season, plant_growth_med, chem_administration, spikein_strain, adjacent_environment, ances_data, cur_vegetation, "Food_Product_type", "Food_source", host_common_name, host_genotype, host_life_stage, host_symbiont, library_prep_kit, local_class, plant_product, samp_collect_device, samp_store_loc, sequencing_location, soil_type, misc_param, root_med_carbon, root_med_macronutr, organism_count, fungicide_regm, gaseous_environment, herbicide_regm, non_mineral_nutr_regm, pesticide_regm, salt_regm, agrochem_addition, perturbation, store_cond, climate_environment, season_environment, standing_water_regm, previous_land_use, food_trav_vehic, extreme_event, fire, flooding, url, soil_horizon, cult_isol_date, farm_equip_san, farm_equip_shared, food_harvest_proc, rel_location, samp_pooling, soil_conductivity, soil_cover, "soil_pH", trophic_level, assembly_software, biotic_relationship, seq_quality_check, lib_layout, mid, adapters, samp_size, samp_vol_we_dna_ext, host_spec_range, lib_reads_seqd, lib_size, isol_growth_condt, nucl_acid_amp, nucl_acid_ext, sop, pathogenicity, seq_meth, annot, lib_screen, lib_vector, source_mat_id, target_gene, target_subfragment, assembly_name, chimera_check, specific_host, host_disease_stat, samp_mat_process, pcr_cond, pcr_primers, samp_name, samp_taxon_id, project_name, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, propagation, encoded_traits, rel_to_oxygen, samp_collec_device, samp_collec_method, size_frac, source_uvig, virus_enrich_appr, tax_ident, assembly_qual, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource)
);

CREATE TABLE "agriculture_MIGS_virus" (
	lat_lon TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	pres_animal_insect VARCHAR(15), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT NOT NULL, 
	pool_dna_extracts TEXT NOT NULL, 
	crop_rotation TEXT, 
	root_med_ph FLOAT, 
	"water_pH" FLOAT, 
	porosity TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	host_age TEXT NOT NULL, 
	host_dry_mass TEXT, 
	host_height TEXT NOT NULL, 
	host_length TEXT NOT NULL, 
	host_tot_mass TEXT NOT NULL, 
	humidity TEXT, 
	microbial_biomass TEXT NOT NULL, 
	photosynt_activ TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	depth TEXT NOT NULL, 
	solar_irradiance TEXT, 
	texture TEXT, 
	tot_car TEXT, 
	tot_nitro TEXT, 
	tot_org_carb TEXT, 
	tot_phos TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	crop_yield TEXT, 
	"temp" TEXT, 
	host_taxid INTEGER NOT NULL, 
	lot_number TEXT, 
	samp_store_dur TEXT NOT NULL, 
	texture_meth TEXT, 
	enrichment_protocol TEXT, 
	genetic_mod TEXT, 
	photosynt_activ_meth TEXT, 
	samp_collect_method TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	microbial_biomass_meth TEXT NOT NULL, 
	ph_meth TEXT, 
	previous_land_use_meth TEXT, 
	soil_type_meth TEXT NOT NULL, 
	tot_n_meth TEXT, 
	tot_org_c_meth TEXT, 
	host_subspecf_genlin TEXT, 
	assembly_quality TEXT, 
	fao_class VARCHAR(13), 
	host_phenotype TEXT NOT NULL, 
	anim_water_method TEXT, 
	farm_equip TEXT, 
	fertilizer_admin TEXT, 
	food_prod TEXT, 
	food_trav_mode TEXT, 
	plant_struc TEXT, 
	plant_water_method TEXT, 
	season TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	spikein_strain TEXT, 
	adjacent_environment TEXT, 
	ances_data TEXT, 
	cur_vegetation TEXT, 
	"Food_Product_type" TEXT, 
	"Food_source" TEXT NOT NULL, 
	host_common_name TEXT NOT NULL, 
	host_genotype TEXT NOT NULL, 
	host_life_stage TEXT NOT NULL, 
	host_symbiont TEXT, 
	library_prep_kit TEXT, 
	local_class TEXT, 
	plant_product TEXT, 
	samp_collect_device TEXT NOT NULL, 
	samp_store_loc TEXT NOT NULL, 
	sequencing_location TEXT, 
	soil_type TEXT NOT NULL, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	organism_count TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	herbicide_regm TEXT, 
	non_mineral_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	salt_regm TEXT, 
	agrochem_addition TEXT, 
	perturbation TEXT, 
	store_cond TEXT NOT NULL, 
	climate_environment TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	previous_land_use TEXT, 
	food_trav_vehic TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	url TEXT, 
	soil_horizon VARCHAR(10), 
	cult_isol_date DATE, 
	farm_equip_san TEXT, 
	farm_equip_shared TEXT, 
	food_harvest_proc TEXT, 
	rel_location TEXT, 
	samp_pooling TEXT, 
	soil_conductivity TEXT, 
	soil_cover TEXT, 
	"soil_pH" FLOAT, 
	trophic_level VARCHAR(22), 
	assembly_software TEXT, 
	biotic_relationship VARCHAR(11), 
	seq_quality_check TEXT, 
	lib_layout VARCHAR(6), 
	mid TEXT NOT NULL, 
	adapters TEXT NOT NULL, 
	samp_size TEXT NOT NULL, 
	samp_vol_we_dna_ext TEXT NOT NULL, 
	host_spec_range INTEGER NOT NULL, 
	lib_reads_seqd INTEGER NOT NULL, 
	lib_size INTEGER, 
	isol_growth_condt TEXT, 
	nucl_acid_amp TEXT NOT NULL, 
	nucl_acid_ext TEXT NOT NULL, 
	sop TEXT, 
	pathogenicity TEXT NOT NULL, 
	seq_meth TEXT NOT NULL, 
	annot TEXT, 
	lib_screen TEXT NOT NULL, 
	lib_vector TEXT NOT NULL, 
	source_mat_id TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT NOT NULL, 
	assembly_name TEXT NOT NULL, 
	chimera_check TEXT NOT NULL, 
	specific_host TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_mat_process TEXT NOT NULL, 
	pcr_cond TEXT NOT NULL, 
	pcr_primers TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	PRIMARY KEY (lat_lon, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, pres_animal_insect, cur_land_use, tillage, farm_water_source, water_source_shared, horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, root_med_ph, "water_pH", porosity, annual_precpt, annual_temp, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, microbial_biomass, photosynt_activ, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, slope_aspect, slope_gradient, depth, solar_irradiance, texture, tot_car, tot_nitro, tot_org_carb, tot_phos, water_temp_regm, watering_regm, ph_regm, crop_yield, "temp", host_taxid, lot_number, samp_store_dur, texture_meth, enrichment_protocol, genetic_mod, photosynt_activ_meth, samp_collect_method, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_class_info, link_climate_info, local_class_meth, microbial_biomass_meth, ph_meth, previous_land_use_meth, soil_type_meth, tot_n_meth, tot_org_c_meth, host_subspecf_genlin, assembly_quality, fao_class, host_phenotype, anim_water_method, farm_equip, fertilizer_admin, food_prod, food_trav_mode, plant_struc, plant_water_method, season, plant_growth_med, chem_administration, spikein_strain, adjacent_environment, ances_data, cur_vegetation, "Food_Product_type", "Food_source", host_common_name, host_genotype, host_life_stage, host_symbiont, library_prep_kit, local_class, plant_product, samp_collect_device, samp_store_loc, sequencing_location, soil_type, misc_param, root_med_carbon, root_med_macronutr, organism_count, fungicide_regm, gaseous_environment, herbicide_regm, non_mineral_nutr_regm, pesticide_regm, salt_regm, agrochem_addition, perturbation, store_cond, climate_environment, season_environment, standing_water_regm, previous_land_use, food_trav_vehic, extreme_event, fire, flooding, url, soil_horizon, cult_isol_date, farm_equip_san, farm_equip_shared, food_harvest_proc, rel_location, samp_pooling, soil_conductivity, soil_cover, "soil_pH", trophic_level, assembly_software, biotic_relationship, seq_quality_check, lib_layout, mid, adapters, samp_size, samp_vol_we_dna_ext, host_spec_range, lib_reads_seqd, lib_size, isol_growth_condt, nucl_acid_amp, nucl_acid_ext, sop, pathogenicity, seq_meth, annot, lib_screen, lib_vector, source_mat_id, target_gene, target_subfragment, assembly_name, chimera_check, specific_host, host_disease_stat, samp_mat_process, pcr_cond, pcr_primers, samp_name, samp_taxon_id, project_name, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, propagation, encoded_traits, rel_to_oxygen, samp_collec_device, samp_collec_method, size_frac, source_uvig, virus_enrich_appr, tax_ident, assembly_qual, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource)
);

CREATE TABLE "agriculture_MIMAG" (
	lat_lon TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	pres_animal_insect VARCHAR(15), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT NOT NULL, 
	pool_dna_extracts TEXT NOT NULL, 
	crop_rotation TEXT, 
	root_med_ph FLOAT, 
	"water_pH" FLOAT, 
	porosity TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	host_age TEXT NOT NULL, 
	host_dry_mass TEXT, 
	host_height TEXT NOT NULL, 
	host_length TEXT NOT NULL, 
	host_tot_mass TEXT NOT NULL, 
	humidity TEXT, 
	microbial_biomass TEXT NOT NULL, 
	photosynt_activ TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	depth TEXT NOT NULL, 
	solar_irradiance TEXT, 
	texture TEXT, 
	tot_car TEXT, 
	tot_nitro TEXT, 
	tot_org_carb TEXT, 
	tot_phos TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	crop_yield TEXT, 
	"temp" TEXT, 
	host_taxid INTEGER NOT NULL, 
	lot_number TEXT, 
	samp_store_dur TEXT NOT NULL, 
	texture_meth TEXT, 
	enrichment_protocol TEXT, 
	genetic_mod TEXT, 
	photosynt_activ_meth TEXT, 
	samp_collect_method TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	microbial_biomass_meth TEXT NOT NULL, 
	ph_meth TEXT, 
	previous_land_use_meth TEXT, 
	soil_type_meth TEXT NOT NULL, 
	tot_n_meth TEXT, 
	tot_org_c_meth TEXT, 
	host_subspecf_genlin TEXT, 
	assembly_quality TEXT, 
	fao_class VARCHAR(13), 
	host_phenotype TEXT NOT NULL, 
	anim_water_method TEXT, 
	farm_equip TEXT, 
	fertilizer_admin TEXT, 
	food_prod TEXT, 
	food_trav_mode TEXT, 
	plant_struc TEXT, 
	plant_water_method TEXT, 
	season TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	spikein_strain TEXT, 
	adjacent_environment TEXT, 
	ances_data TEXT, 
	cur_vegetation TEXT, 
	"Food_Product_type" TEXT, 
	"Food_source" TEXT NOT NULL, 
	host_common_name TEXT NOT NULL, 
	host_genotype TEXT NOT NULL, 
	host_life_stage TEXT NOT NULL, 
	host_symbiont TEXT, 
	library_prep_kit TEXT, 
	local_class TEXT, 
	plant_product TEXT, 
	samp_collect_device TEXT NOT NULL, 
	samp_store_loc TEXT NOT NULL, 
	sequencing_location TEXT, 
	soil_type TEXT NOT NULL, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	organism_count TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	herbicide_regm TEXT, 
	non_mineral_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	salt_regm TEXT, 
	agrochem_addition TEXT, 
	perturbation TEXT, 
	store_cond TEXT NOT NULL, 
	climate_environment TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	previous_land_use TEXT, 
	food_trav_vehic TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	url TEXT, 
	soil_horizon VARCHAR(10), 
	cult_isol_date DATE, 
	farm_equip_san TEXT, 
	farm_equip_shared TEXT, 
	food_harvest_proc TEXT, 
	rel_location TEXT, 
	samp_pooling TEXT, 
	soil_conductivity TEXT, 
	soil_cover TEXT, 
	"soil_pH" FLOAT, 
	trophic_level VARCHAR(22), 
	assembly_software TEXT, 
	biotic_relationship VARCHAR(11), 
	seq_quality_check TEXT, 
	lib_layout VARCHAR(6), 
	mid TEXT NOT NULL, 
	adapters TEXT NOT NULL, 
	samp_size TEXT NOT NULL, 
	samp_vol_we_dna_ext TEXT NOT NULL, 
	host_spec_range INTEGER NOT NULL, 
	lib_reads_seqd INTEGER NOT NULL, 
	lib_size INTEGER, 
	isol_growth_condt TEXT, 
	nucl_acid_amp TEXT NOT NULL, 
	nucl_acid_ext TEXT NOT NULL, 
	sop TEXT, 
	pathogenicity TEXT NOT NULL, 
	seq_meth TEXT NOT NULL, 
	annot TEXT, 
	lib_screen TEXT NOT NULL, 
	lib_vector TEXT NOT NULL, 
	source_mat_id TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT NOT NULL, 
	assembly_name TEXT NOT NULL, 
	chimera_check TEXT NOT NULL, 
	specific_host TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_mat_process TEXT NOT NULL, 
	pcr_cond TEXT NOT NULL, 
	pcr_primers TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	PRIMARY KEY (lat_lon, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, pres_animal_insect, cur_land_use, tillage, farm_water_source, water_source_shared, horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, root_med_ph, "water_pH", porosity, annual_precpt, annual_temp, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, microbial_biomass, photosynt_activ, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, slope_aspect, slope_gradient, depth, solar_irradiance, texture, tot_car, tot_nitro, tot_org_carb, tot_phos, water_temp_regm, watering_regm, ph_regm, crop_yield, "temp", host_taxid, lot_number, samp_store_dur, texture_meth, enrichment_protocol, genetic_mod, photosynt_activ_meth, samp_collect_method, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_class_info, link_climate_info, local_class_meth, microbial_biomass_meth, ph_meth, previous_land_use_meth, soil_type_meth, tot_n_meth, tot_org_c_meth, host_subspecf_genlin, assembly_quality, fao_class, host_phenotype, anim_water_method, farm_equip, fertilizer_admin, food_prod, food_trav_mode, plant_struc, plant_water_method, season, plant_growth_med, chem_administration, spikein_strain, adjacent_environment, ances_data, cur_vegetation, "Food_Product_type", "Food_source", host_common_name, host_genotype, host_life_stage, host_symbiont, library_prep_kit, local_class, plant_product, samp_collect_device, samp_store_loc, sequencing_location, soil_type, misc_param, root_med_carbon, root_med_macronutr, organism_count, fungicide_regm, gaseous_environment, herbicide_regm, non_mineral_nutr_regm, pesticide_regm, salt_regm, agrochem_addition, perturbation, store_cond, climate_environment, season_environment, standing_water_regm, previous_land_use, food_trav_vehic, extreme_event, fire, flooding, url, soil_horizon, cult_isol_date, farm_equip_san, farm_equip_shared, food_harvest_proc, rel_location, samp_pooling, soil_conductivity, soil_cover, "soil_pH", trophic_level, assembly_software, biotic_relationship, seq_quality_check, lib_layout, mid, adapters, samp_size, samp_vol_we_dna_ext, host_spec_range, lib_reads_seqd, lib_size, isol_growth_condt, nucl_acid_amp, nucl_acid_ext, sop, pathogenicity, seq_meth, annot, lib_screen, lib_vector, source_mat_id, target_gene, target_subfragment, assembly_name, chimera_check, specific_host, host_disease_stat, samp_mat_process, pcr_cond, pcr_primers, samp_name, samp_taxon_id, project_name, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, propagation, encoded_traits, rel_to_oxygen, samp_collec_device, samp_collec_method, size_frac, source_uvig, virus_enrich_appr, tax_ident, assembly_qual, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource)
);

CREATE TABLE "agriculture_MIMARKS_specimen" (
	lat_lon TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	pres_animal_insect VARCHAR(15), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT NOT NULL, 
	pool_dna_extracts TEXT NOT NULL, 
	crop_rotation TEXT, 
	root_med_ph FLOAT, 
	"water_pH" FLOAT, 
	porosity TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	host_age TEXT NOT NULL, 
	host_dry_mass TEXT, 
	host_height TEXT NOT NULL, 
	host_length TEXT NOT NULL, 
	host_tot_mass TEXT NOT NULL, 
	humidity TEXT, 
	microbial_biomass TEXT NOT NULL, 
	photosynt_activ TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	depth TEXT NOT NULL, 
	solar_irradiance TEXT, 
	texture TEXT, 
	tot_car TEXT, 
	tot_nitro TEXT, 
	tot_org_carb TEXT, 
	tot_phos TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	crop_yield TEXT, 
	"temp" TEXT, 
	host_taxid INTEGER NOT NULL, 
	lot_number TEXT, 
	samp_store_dur TEXT NOT NULL, 
	texture_meth TEXT, 
	enrichment_protocol TEXT, 
	genetic_mod TEXT, 
	photosynt_activ_meth TEXT, 
	samp_collect_method TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	microbial_biomass_meth TEXT NOT NULL, 
	ph_meth TEXT, 
	previous_land_use_meth TEXT, 
	soil_type_meth TEXT NOT NULL, 
	tot_n_meth TEXT, 
	tot_org_c_meth TEXT, 
	host_subspecf_genlin TEXT, 
	assembly_quality TEXT, 
	fao_class VARCHAR(13), 
	host_phenotype TEXT NOT NULL, 
	anim_water_method TEXT, 
	farm_equip TEXT, 
	fertilizer_admin TEXT, 
	food_prod TEXT, 
	food_trav_mode TEXT, 
	plant_struc TEXT, 
	plant_water_method TEXT, 
	season TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	spikein_strain TEXT, 
	adjacent_environment TEXT, 
	ances_data TEXT, 
	cur_vegetation TEXT, 
	"Food_Product_type" TEXT, 
	"Food_source" TEXT NOT NULL, 
	host_common_name TEXT NOT NULL, 
	host_genotype TEXT NOT NULL, 
	host_life_stage TEXT NOT NULL, 
	host_symbiont TEXT, 
	library_prep_kit TEXT, 
	local_class TEXT, 
	plant_product TEXT, 
	samp_collect_device TEXT NOT NULL, 
	samp_store_loc TEXT NOT NULL, 
	sequencing_location TEXT, 
	soil_type TEXT NOT NULL, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	organism_count TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	herbicide_regm TEXT, 
	non_mineral_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	salt_regm TEXT, 
	agrochem_addition TEXT, 
	perturbation TEXT, 
	store_cond TEXT NOT NULL, 
	climate_environment TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	previous_land_use TEXT, 
	food_trav_vehic TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	url TEXT, 
	soil_horizon VARCHAR(10), 
	cult_isol_date DATE, 
	farm_equip_san TEXT, 
	farm_equip_shared TEXT, 
	food_harvest_proc TEXT, 
	rel_location TEXT, 
	samp_pooling TEXT, 
	soil_conductivity TEXT, 
	soil_cover TEXT, 
	"soil_pH" FLOAT, 
	trophic_level VARCHAR(22), 
	assembly_software TEXT, 
	biotic_relationship VARCHAR(11), 
	seq_quality_check TEXT, 
	lib_layout VARCHAR(6), 
	mid TEXT NOT NULL, 
	adapters TEXT NOT NULL, 
	samp_size TEXT NOT NULL, 
	samp_vol_we_dna_ext TEXT NOT NULL, 
	host_spec_range INTEGER NOT NULL, 
	lib_reads_seqd INTEGER NOT NULL, 
	lib_size INTEGER, 
	isol_growth_condt TEXT, 
	nucl_acid_amp TEXT NOT NULL, 
	nucl_acid_ext TEXT NOT NULL, 
	sop TEXT, 
	pathogenicity TEXT NOT NULL, 
	seq_meth TEXT NOT NULL, 
	annot TEXT, 
	lib_screen TEXT NOT NULL, 
	lib_vector TEXT NOT NULL, 
	source_mat_id TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT NOT NULL, 
	assembly_name TEXT NOT NULL, 
	chimera_check TEXT NOT NULL, 
	specific_host TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_mat_process TEXT NOT NULL, 
	pcr_cond TEXT NOT NULL, 
	pcr_primers TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	PRIMARY KEY (lat_lon, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, pres_animal_insect, cur_land_use, tillage, farm_water_source, water_source_shared, horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, root_med_ph, "water_pH", porosity, annual_precpt, annual_temp, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, microbial_biomass, photosynt_activ, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, slope_aspect, slope_gradient, depth, solar_irradiance, texture, tot_car, tot_nitro, tot_org_carb, tot_phos, water_temp_regm, watering_regm, ph_regm, crop_yield, "temp", host_taxid, lot_number, samp_store_dur, texture_meth, enrichment_protocol, genetic_mod, photosynt_activ_meth, samp_collect_method, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_class_info, link_climate_info, local_class_meth, microbial_biomass_meth, ph_meth, previous_land_use_meth, soil_type_meth, tot_n_meth, tot_org_c_meth, host_subspecf_genlin, assembly_quality, fao_class, host_phenotype, anim_water_method, farm_equip, fertilizer_admin, food_prod, food_trav_mode, plant_struc, plant_water_method, season, plant_growth_med, chem_administration, spikein_strain, adjacent_environment, ances_data, cur_vegetation, "Food_Product_type", "Food_source", host_common_name, host_genotype, host_life_stage, host_symbiont, library_prep_kit, local_class, plant_product, samp_collect_device, samp_store_loc, sequencing_location, soil_type, misc_param, root_med_carbon, root_med_macronutr, organism_count, fungicide_regm, gaseous_environment, herbicide_regm, non_mineral_nutr_regm, pesticide_regm, salt_regm, agrochem_addition, perturbation, store_cond, climate_environment, season_environment, standing_water_regm, previous_land_use, food_trav_vehic, extreme_event, fire, flooding, url, soil_horizon, cult_isol_date, farm_equip_san, farm_equip_shared, food_harvest_proc, rel_location, samp_pooling, soil_conductivity, soil_cover, "soil_pH", trophic_level, assembly_software, biotic_relationship, seq_quality_check, lib_layout, mid, adapters, samp_size, samp_vol_we_dna_ext, host_spec_range, lib_reads_seqd, lib_size, isol_growth_condt, nucl_acid_amp, nucl_acid_ext, sop, pathogenicity, seq_meth, annot, lib_screen, lib_vector, source_mat_id, target_gene, target_subfragment, assembly_name, chimera_check, specific_host, host_disease_stat, samp_mat_process, pcr_cond, pcr_primers, samp_name, samp_taxon_id, project_name, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, propagation, encoded_traits, rel_to_oxygen, samp_collec_device, samp_collec_method, size_frac, source_uvig, virus_enrich_appr, tax_ident, assembly_qual, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource)
);

CREATE TABLE "agriculture_MIMARKS_survey" (
	lat_lon TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	pres_animal_insect VARCHAR(15), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT NOT NULL, 
	pool_dna_extracts TEXT NOT NULL, 
	crop_rotation TEXT, 
	root_med_ph FLOAT, 
	"water_pH" FLOAT, 
	porosity TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	host_age TEXT NOT NULL, 
	host_dry_mass TEXT, 
	host_height TEXT NOT NULL, 
	host_length TEXT NOT NULL, 
	host_tot_mass TEXT NOT NULL, 
	humidity TEXT, 
	microbial_biomass TEXT NOT NULL, 
	photosynt_activ TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	depth TEXT NOT NULL, 
	solar_irradiance TEXT, 
	texture TEXT, 
	tot_car TEXT, 
	tot_nitro TEXT, 
	tot_org_carb TEXT, 
	tot_phos TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	crop_yield TEXT, 
	"temp" TEXT, 
	host_taxid INTEGER NOT NULL, 
	lot_number TEXT, 
	samp_store_dur TEXT NOT NULL, 
	texture_meth TEXT, 
	enrichment_protocol TEXT, 
	genetic_mod TEXT, 
	photosynt_activ_meth TEXT, 
	samp_collect_method TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	microbial_biomass_meth TEXT NOT NULL, 
	ph_meth TEXT, 
	previous_land_use_meth TEXT, 
	soil_type_meth TEXT NOT NULL, 
	tot_n_meth TEXT, 
	tot_org_c_meth TEXT, 
	host_subspecf_genlin TEXT, 
	assembly_quality TEXT, 
	fao_class VARCHAR(13), 
	host_phenotype TEXT NOT NULL, 
	anim_water_method TEXT, 
	farm_equip TEXT, 
	fertilizer_admin TEXT, 
	food_prod TEXT, 
	food_trav_mode TEXT, 
	plant_struc TEXT, 
	plant_water_method TEXT, 
	season TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	spikein_strain TEXT, 
	adjacent_environment TEXT, 
	ances_data TEXT, 
	cur_vegetation TEXT, 
	"Food_Product_type" TEXT, 
	"Food_source" TEXT NOT NULL, 
	host_common_name TEXT NOT NULL, 
	host_genotype TEXT NOT NULL, 
	host_life_stage TEXT NOT NULL, 
	host_symbiont TEXT, 
	library_prep_kit TEXT, 
	local_class TEXT, 
	plant_product TEXT, 
	samp_collect_device TEXT NOT NULL, 
	samp_store_loc TEXT NOT NULL, 
	sequencing_location TEXT, 
	soil_type TEXT NOT NULL, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	organism_count TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	herbicide_regm TEXT, 
	non_mineral_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	salt_regm TEXT, 
	agrochem_addition TEXT, 
	perturbation TEXT, 
	store_cond TEXT NOT NULL, 
	climate_environment TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	previous_land_use TEXT, 
	food_trav_vehic TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	url TEXT, 
	soil_horizon VARCHAR(10), 
	cult_isol_date DATE, 
	farm_equip_san TEXT, 
	farm_equip_shared TEXT, 
	food_harvest_proc TEXT, 
	rel_location TEXT, 
	samp_pooling TEXT, 
	soil_conductivity TEXT, 
	soil_cover TEXT, 
	"soil_pH" FLOAT, 
	trophic_level VARCHAR(22), 
	assembly_software TEXT, 
	biotic_relationship VARCHAR(11), 
	seq_quality_check TEXT, 
	lib_layout VARCHAR(6), 
	mid TEXT NOT NULL, 
	adapters TEXT NOT NULL, 
	samp_size TEXT NOT NULL, 
	samp_vol_we_dna_ext TEXT NOT NULL, 
	host_spec_range INTEGER NOT NULL, 
	lib_reads_seqd INTEGER NOT NULL, 
	lib_size INTEGER, 
	isol_growth_condt TEXT, 
	nucl_acid_amp TEXT NOT NULL, 
	nucl_acid_ext TEXT NOT NULL, 
	sop TEXT, 
	pathogenicity TEXT NOT NULL, 
	seq_meth TEXT NOT NULL, 
	annot TEXT, 
	lib_screen TEXT NOT NULL, 
	lib_vector TEXT NOT NULL, 
	source_mat_id TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT NOT NULL, 
	assembly_name TEXT NOT NULL, 
	chimera_check TEXT NOT NULL, 
	specific_host TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_mat_process TEXT NOT NULL, 
	pcr_cond TEXT NOT NULL, 
	pcr_primers TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	PRIMARY KEY (lat_lon, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, pres_animal_insect, cur_land_use, tillage, farm_water_source, water_source_shared, horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, root_med_ph, "water_pH", porosity, annual_precpt, annual_temp, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, microbial_biomass, photosynt_activ, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, slope_aspect, slope_gradient, depth, solar_irradiance, texture, tot_car, tot_nitro, tot_org_carb, tot_phos, water_temp_regm, watering_regm, ph_regm, crop_yield, "temp", host_taxid, lot_number, samp_store_dur, texture_meth, enrichment_protocol, genetic_mod, photosynt_activ_meth, samp_collect_method, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_class_info, link_climate_info, local_class_meth, microbial_biomass_meth, ph_meth, previous_land_use_meth, soil_type_meth, tot_n_meth, tot_org_c_meth, host_subspecf_genlin, assembly_quality, fao_class, host_phenotype, anim_water_method, farm_equip, fertilizer_admin, food_prod, food_trav_mode, plant_struc, plant_water_method, season, plant_growth_med, chem_administration, spikein_strain, adjacent_environment, ances_data, cur_vegetation, "Food_Product_type", "Food_source", host_common_name, host_genotype, host_life_stage, host_symbiont, library_prep_kit, local_class, plant_product, samp_collect_device, samp_store_loc, sequencing_location, soil_type, misc_param, root_med_carbon, root_med_macronutr, organism_count, fungicide_regm, gaseous_environment, herbicide_regm, non_mineral_nutr_regm, pesticide_regm, salt_regm, agrochem_addition, perturbation, store_cond, climate_environment, season_environment, standing_water_regm, previous_land_use, food_trav_vehic, extreme_event, fire, flooding, url, soil_horizon, cult_isol_date, farm_equip_san, farm_equip_shared, food_harvest_proc, rel_location, samp_pooling, soil_conductivity, soil_cover, "soil_pH", trophic_level, assembly_software, biotic_relationship, seq_quality_check, lib_layout, mid, adapters, samp_size, samp_vol_we_dna_ext, host_spec_range, lib_reads_seqd, lib_size, isol_growth_condt, nucl_acid_amp, nucl_acid_ext, sop, pathogenicity, seq_meth, annot, lib_screen, lib_vector, source_mat_id, target_gene, target_subfragment, assembly_name, chimera_check, specific_host, host_disease_stat, samp_mat_process, pcr_cond, pcr_primers, samp_name, samp_taxon_id, project_name, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, propagation, encoded_traits, rel_to_oxygen, samp_collec_device, samp_collec_method, size_frac, source_uvig, virus_enrich_appr, tax_ident, assembly_qual, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource)
);

CREATE TABLE "agriculture_MIMS" (
	lat_lon TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	pres_animal_insect VARCHAR(15), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT NOT NULL, 
	pool_dna_extracts TEXT NOT NULL, 
	crop_rotation TEXT, 
	root_med_ph FLOAT, 
	"water_pH" FLOAT, 
	porosity TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	host_age TEXT NOT NULL, 
	host_dry_mass TEXT, 
	host_height TEXT NOT NULL, 
	host_length TEXT NOT NULL, 
	host_tot_mass TEXT NOT NULL, 
	humidity TEXT, 
	microbial_biomass TEXT NOT NULL, 
	photosynt_activ TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	depth TEXT NOT NULL, 
	solar_irradiance TEXT, 
	texture TEXT, 
	tot_car TEXT, 
	tot_nitro TEXT, 
	tot_org_carb TEXT, 
	tot_phos TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	crop_yield TEXT, 
	"temp" TEXT, 
	host_taxid INTEGER NOT NULL, 
	lot_number TEXT, 
	samp_store_dur TEXT NOT NULL, 
	texture_meth TEXT, 
	enrichment_protocol TEXT, 
	genetic_mod TEXT, 
	photosynt_activ_meth TEXT, 
	samp_collect_method TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	microbial_biomass_meth TEXT NOT NULL, 
	ph_meth TEXT, 
	previous_land_use_meth TEXT, 
	soil_type_meth TEXT NOT NULL, 
	tot_n_meth TEXT, 
	tot_org_c_meth TEXT, 
	host_subspecf_genlin TEXT, 
	assembly_quality TEXT, 
	fao_class VARCHAR(13), 
	host_phenotype TEXT NOT NULL, 
	anim_water_method TEXT, 
	farm_equip TEXT, 
	fertilizer_admin TEXT, 
	food_prod TEXT, 
	food_trav_mode TEXT, 
	plant_struc TEXT, 
	plant_water_method TEXT, 
	season TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	spikein_strain TEXT, 
	adjacent_environment TEXT, 
	ances_data TEXT, 
	cur_vegetation TEXT, 
	"Food_Product_type" TEXT, 
	"Food_source" TEXT NOT NULL, 
	host_common_name TEXT NOT NULL, 
	host_genotype TEXT NOT NULL, 
	host_life_stage TEXT NOT NULL, 
	host_symbiont TEXT, 
	library_prep_kit TEXT, 
	local_class TEXT, 
	plant_product TEXT, 
	samp_collect_device TEXT NOT NULL, 
	samp_store_loc TEXT NOT NULL, 
	sequencing_location TEXT, 
	soil_type TEXT NOT NULL, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	organism_count TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	herbicide_regm TEXT, 
	non_mineral_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	salt_regm TEXT, 
	agrochem_addition TEXT, 
	perturbation TEXT, 
	store_cond TEXT NOT NULL, 
	climate_environment TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	previous_land_use TEXT, 
	food_trav_vehic TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	url TEXT, 
	soil_horizon VARCHAR(10), 
	cult_isol_date DATE, 
	farm_equip_san TEXT, 
	farm_equip_shared TEXT, 
	food_harvest_proc TEXT, 
	rel_location TEXT, 
	samp_pooling TEXT, 
	soil_conductivity TEXT, 
	soil_cover TEXT, 
	"soil_pH" FLOAT, 
	trophic_level VARCHAR(22), 
	assembly_software TEXT, 
	biotic_relationship VARCHAR(11), 
	seq_quality_check TEXT, 
	lib_layout VARCHAR(6), 
	mid TEXT NOT NULL, 
	adapters TEXT NOT NULL, 
	samp_size TEXT NOT NULL, 
	samp_vol_we_dna_ext TEXT NOT NULL, 
	host_spec_range INTEGER NOT NULL, 
	lib_reads_seqd INTEGER NOT NULL, 
	lib_size INTEGER, 
	isol_growth_condt TEXT, 
	nucl_acid_amp TEXT NOT NULL, 
	nucl_acid_ext TEXT NOT NULL, 
	sop TEXT, 
	pathogenicity TEXT NOT NULL, 
	seq_meth TEXT NOT NULL, 
	annot TEXT, 
	lib_screen TEXT NOT NULL, 
	lib_vector TEXT NOT NULL, 
	source_mat_id TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT NOT NULL, 
	assembly_name TEXT NOT NULL, 
	chimera_check TEXT NOT NULL, 
	specific_host TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_mat_process TEXT NOT NULL, 
	pcr_cond TEXT NOT NULL, 
	pcr_primers TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	PRIMARY KEY (lat_lon, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, pres_animal_insect, cur_land_use, tillage, farm_water_source, water_source_shared, horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, root_med_ph, "water_pH", porosity, annual_precpt, annual_temp, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, microbial_biomass, photosynt_activ, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, slope_aspect, slope_gradient, depth, solar_irradiance, texture, tot_car, tot_nitro, tot_org_carb, tot_phos, water_temp_regm, watering_regm, ph_regm, crop_yield, "temp", host_taxid, lot_number, samp_store_dur, texture_meth, enrichment_protocol, genetic_mod, photosynt_activ_meth, samp_collect_method, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_class_info, link_climate_info, local_class_meth, microbial_biomass_meth, ph_meth, previous_land_use_meth, soil_type_meth, tot_n_meth, tot_org_c_meth, host_subspecf_genlin, assembly_quality, fao_class, host_phenotype, anim_water_method, farm_equip, fertilizer_admin, food_prod, food_trav_mode, plant_struc, plant_water_method, season, plant_growth_med, chem_administration, spikein_strain, adjacent_environment, ances_data, cur_vegetation, "Food_Product_type", "Food_source", host_common_name, host_genotype, host_life_stage, host_symbiont, library_prep_kit, local_class, plant_product, samp_collect_device, samp_store_loc, sequencing_location, soil_type, misc_param, root_med_carbon, root_med_macronutr, organism_count, fungicide_regm, gaseous_environment, herbicide_regm, non_mineral_nutr_regm, pesticide_regm, salt_regm, agrochem_addition, perturbation, store_cond, climate_environment, season_environment, standing_water_regm, previous_land_use, food_trav_vehic, extreme_event, fire, flooding, url, soil_horizon, cult_isol_date, farm_equip_san, farm_equip_shared, food_harvest_proc, rel_location, samp_pooling, soil_conductivity, soil_cover, "soil_pH", trophic_level, assembly_software, biotic_relationship, seq_quality_check, lib_layout, mid, adapters, samp_size, samp_vol_we_dna_ext, host_spec_range, lib_reads_seqd, lib_size, isol_growth_condt, nucl_acid_amp, nucl_acid_ext, sop, pathogenicity, seq_meth, annot, lib_screen, lib_vector, source_mat_id, target_gene, target_subfragment, assembly_name, chimera_check, specific_host, host_disease_stat, samp_mat_process, pcr_cond, pcr_primers, samp_name, samp_taxon_id, project_name, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, propagation, encoded_traits, rel_to_oxygen, samp_collec_device, samp_collec_method, size_frac, source_uvig, virus_enrich_appr, tax_ident, assembly_qual, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource)
);

CREATE TABLE "agriculture_MISAG" (
	lat_lon TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	pres_animal_insect VARCHAR(15), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT NOT NULL, 
	pool_dna_extracts TEXT NOT NULL, 
	crop_rotation TEXT, 
	root_med_ph FLOAT, 
	"water_pH" FLOAT, 
	porosity TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	host_age TEXT NOT NULL, 
	host_dry_mass TEXT, 
	host_height TEXT NOT NULL, 
	host_length TEXT NOT NULL, 
	host_tot_mass TEXT NOT NULL, 
	humidity TEXT, 
	microbial_biomass TEXT NOT NULL, 
	photosynt_activ TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	depth TEXT NOT NULL, 
	solar_irradiance TEXT, 
	texture TEXT, 
	tot_car TEXT, 
	tot_nitro TEXT, 
	tot_org_carb TEXT, 
	tot_phos TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	crop_yield TEXT, 
	"temp" TEXT, 
	host_taxid INTEGER NOT NULL, 
	lot_number TEXT, 
	samp_store_dur TEXT NOT NULL, 
	texture_meth TEXT, 
	enrichment_protocol TEXT, 
	genetic_mod TEXT, 
	photosynt_activ_meth TEXT, 
	samp_collect_method TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	microbial_biomass_meth TEXT NOT NULL, 
	ph_meth TEXT, 
	previous_land_use_meth TEXT, 
	soil_type_meth TEXT NOT NULL, 
	tot_n_meth TEXT, 
	tot_org_c_meth TEXT, 
	host_subspecf_genlin TEXT, 
	assembly_quality TEXT, 
	fao_class VARCHAR(13), 
	host_phenotype TEXT NOT NULL, 
	anim_water_method TEXT, 
	farm_equip TEXT, 
	fertilizer_admin TEXT, 
	food_prod TEXT, 
	food_trav_mode TEXT, 
	plant_struc TEXT, 
	plant_water_method TEXT, 
	season TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	spikein_strain TEXT, 
	adjacent_environment TEXT, 
	ances_data TEXT, 
	cur_vegetation TEXT, 
	"Food_Product_type" TEXT, 
	"Food_source" TEXT NOT NULL, 
	host_common_name TEXT NOT NULL, 
	host_genotype TEXT NOT NULL, 
	host_life_stage TEXT NOT NULL, 
	host_symbiont TEXT, 
	library_prep_kit TEXT, 
	local_class TEXT, 
	plant_product TEXT, 
	samp_collect_device TEXT NOT NULL, 
	samp_store_loc TEXT NOT NULL, 
	sequencing_location TEXT, 
	soil_type TEXT NOT NULL, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	organism_count TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	herbicide_regm TEXT, 
	non_mineral_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	salt_regm TEXT, 
	agrochem_addition TEXT, 
	perturbation TEXT, 
	store_cond TEXT NOT NULL, 
	climate_environment TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	previous_land_use TEXT, 
	food_trav_vehic TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	url TEXT, 
	soil_horizon VARCHAR(10), 
	cult_isol_date DATE, 
	farm_equip_san TEXT, 
	farm_equip_shared TEXT, 
	food_harvest_proc TEXT, 
	rel_location TEXT, 
	samp_pooling TEXT, 
	soil_conductivity TEXT, 
	soil_cover TEXT, 
	"soil_pH" FLOAT, 
	trophic_level VARCHAR(22), 
	assembly_software TEXT, 
	biotic_relationship VARCHAR(11), 
	seq_quality_check TEXT, 
	lib_layout VARCHAR(6), 
	mid TEXT NOT NULL, 
	adapters TEXT NOT NULL, 
	samp_size TEXT NOT NULL, 
	samp_vol_we_dna_ext TEXT NOT NULL, 
	host_spec_range INTEGER NOT NULL, 
	lib_reads_seqd INTEGER NOT NULL, 
	lib_size INTEGER, 
	isol_growth_condt TEXT, 
	nucl_acid_amp TEXT NOT NULL, 
	nucl_acid_ext TEXT NOT NULL, 
	sop TEXT, 
	pathogenicity TEXT NOT NULL, 
	seq_meth TEXT NOT NULL, 
	annot TEXT, 
	lib_screen TEXT NOT NULL, 
	lib_vector TEXT NOT NULL, 
	source_mat_id TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT NOT NULL, 
	assembly_name TEXT NOT NULL, 
	chimera_check TEXT NOT NULL, 
	specific_host TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_mat_process TEXT NOT NULL, 
	pcr_cond TEXT NOT NULL, 
	pcr_primers TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	PRIMARY KEY (lat_lon, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, pres_animal_insect, cur_land_use, tillage, farm_water_source, water_source_shared, horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, root_med_ph, "water_pH", porosity, annual_precpt, annual_temp, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, microbial_biomass, photosynt_activ, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, slope_aspect, slope_gradient, depth, solar_irradiance, texture, tot_car, tot_nitro, tot_org_carb, tot_phos, water_temp_regm, watering_regm, ph_regm, crop_yield, "temp", host_taxid, lot_number, samp_store_dur, texture_meth, enrichment_protocol, genetic_mod, photosynt_activ_meth, samp_collect_method, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_class_info, link_climate_info, local_class_meth, microbial_biomass_meth, ph_meth, previous_land_use_meth, soil_type_meth, tot_n_meth, tot_org_c_meth, host_subspecf_genlin, assembly_quality, fao_class, host_phenotype, anim_water_method, farm_equip, fertilizer_admin, food_prod, food_trav_mode, plant_struc, plant_water_method, season, plant_growth_med, chem_administration, spikein_strain, adjacent_environment, ances_data, cur_vegetation, "Food_Product_type", "Food_source", host_common_name, host_genotype, host_life_stage, host_symbiont, library_prep_kit, local_class, plant_product, samp_collect_device, samp_store_loc, sequencing_location, soil_type, misc_param, root_med_carbon, root_med_macronutr, organism_count, fungicide_regm, gaseous_environment, herbicide_regm, non_mineral_nutr_regm, pesticide_regm, salt_regm, agrochem_addition, perturbation, store_cond, climate_environment, season_environment, standing_water_regm, previous_land_use, food_trav_vehic, extreme_event, fire, flooding, url, soil_horizon, cult_isol_date, farm_equip_san, farm_equip_shared, food_harvest_proc, rel_location, samp_pooling, soil_conductivity, soil_cover, "soil_pH", trophic_level, assembly_software, biotic_relationship, seq_quality_check, lib_layout, mid, adapters, samp_size, samp_vol_we_dna_ext, host_spec_range, lib_reads_seqd, lib_size, isol_growth_condt, nucl_acid_amp, nucl_acid_ext, sop, pathogenicity, seq_meth, annot, lib_screen, lib_vector, source_mat_id, target_gene, target_subfragment, assembly_name, chimera_check, specific_host, host_disease_stat, samp_mat_process, pcr_cond, pcr_primers, samp_name, samp_taxon_id, project_name, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, propagation, encoded_traits, rel_to_oxygen, samp_collec_device, samp_collec_method, size_frac, source_uvig, virus_enrich_appr, tax_ident, assembly_qual, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource)
);

CREATE TABLE "agriculture_MIUVIG" (
	lat_lon TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	pres_animal_insect VARCHAR(15), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT NOT NULL, 
	pool_dna_extracts TEXT NOT NULL, 
	crop_rotation TEXT, 
	root_med_ph FLOAT, 
	"water_pH" FLOAT, 
	porosity TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	host_age TEXT NOT NULL, 
	host_dry_mass TEXT, 
	host_height TEXT NOT NULL, 
	host_length TEXT NOT NULL, 
	host_tot_mass TEXT NOT NULL, 
	humidity TEXT, 
	microbial_biomass TEXT NOT NULL, 
	photosynt_activ TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	depth TEXT NOT NULL, 
	solar_irradiance TEXT, 
	texture TEXT, 
	tot_car TEXT, 
	tot_nitro TEXT, 
	tot_org_carb TEXT, 
	tot_phos TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	crop_yield TEXT, 
	"temp" TEXT, 
	host_taxid INTEGER NOT NULL, 
	lot_number TEXT, 
	samp_store_dur TEXT NOT NULL, 
	texture_meth TEXT, 
	enrichment_protocol TEXT, 
	genetic_mod TEXT, 
	photosynt_activ_meth TEXT, 
	samp_collect_method TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	microbial_biomass_meth TEXT NOT NULL, 
	ph_meth TEXT, 
	previous_land_use_meth TEXT, 
	soil_type_meth TEXT NOT NULL, 
	tot_n_meth TEXT, 
	tot_org_c_meth TEXT, 
	host_subspecf_genlin TEXT, 
	assembly_quality TEXT, 
	fao_class VARCHAR(13), 
	host_phenotype TEXT NOT NULL, 
	anim_water_method TEXT, 
	farm_equip TEXT, 
	fertilizer_admin TEXT, 
	food_prod TEXT, 
	food_trav_mode TEXT, 
	plant_struc TEXT, 
	plant_water_method TEXT, 
	season TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	spikein_strain TEXT, 
	adjacent_environment TEXT, 
	ances_data TEXT, 
	cur_vegetation TEXT, 
	"Food_Product_type" TEXT, 
	"Food_source" TEXT NOT NULL, 
	host_common_name TEXT NOT NULL, 
	host_genotype TEXT NOT NULL, 
	host_life_stage TEXT NOT NULL, 
	host_symbiont TEXT, 
	library_prep_kit TEXT, 
	local_class TEXT, 
	plant_product TEXT, 
	samp_collect_device TEXT NOT NULL, 
	samp_store_loc TEXT NOT NULL, 
	sequencing_location TEXT, 
	soil_type TEXT NOT NULL, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	organism_count TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	herbicide_regm TEXT, 
	non_mineral_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	salt_regm TEXT, 
	agrochem_addition TEXT, 
	perturbation TEXT, 
	store_cond TEXT NOT NULL, 
	climate_environment TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	previous_land_use TEXT, 
	food_trav_vehic TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	url TEXT, 
	soil_horizon VARCHAR(10), 
	cult_isol_date DATE, 
	farm_equip_san TEXT, 
	farm_equip_shared TEXT, 
	food_harvest_proc TEXT, 
	rel_location TEXT, 
	samp_pooling TEXT, 
	soil_conductivity TEXT, 
	soil_cover TEXT, 
	"soil_pH" FLOAT, 
	trophic_level VARCHAR(22), 
	assembly_software TEXT, 
	biotic_relationship VARCHAR(11), 
	seq_quality_check TEXT, 
	lib_layout VARCHAR(6), 
	mid TEXT NOT NULL, 
	adapters TEXT NOT NULL, 
	samp_size TEXT NOT NULL, 
	samp_vol_we_dna_ext TEXT NOT NULL, 
	host_spec_range INTEGER NOT NULL, 
	lib_reads_seqd INTEGER NOT NULL, 
	lib_size INTEGER, 
	isol_growth_condt TEXT, 
	nucl_acid_amp TEXT NOT NULL, 
	nucl_acid_ext TEXT NOT NULL, 
	sop TEXT, 
	pathogenicity TEXT NOT NULL, 
	seq_meth TEXT NOT NULL, 
	annot TEXT, 
	lib_screen TEXT NOT NULL, 
	lib_vector TEXT NOT NULL, 
	source_mat_id TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT NOT NULL, 
	assembly_name TEXT NOT NULL, 
	chimera_check TEXT NOT NULL, 
	specific_host TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_mat_process TEXT NOT NULL, 
	pcr_cond TEXT NOT NULL, 
	pcr_primers TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	PRIMARY KEY (lat_lon, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, pres_animal_insect, cur_land_use, tillage, farm_water_source, water_source_shared, horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, root_med_ph, "water_pH", porosity, annual_precpt, annual_temp, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, microbial_biomass, photosynt_activ, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, slope_aspect, slope_gradient, depth, solar_irradiance, texture, tot_car, tot_nitro, tot_org_carb, tot_phos, water_temp_regm, watering_regm, ph_regm, crop_yield, "temp", host_taxid, lot_number, samp_store_dur, texture_meth, enrichment_protocol, genetic_mod, photosynt_activ_meth, samp_collect_method, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_class_info, link_climate_info, local_class_meth, microbial_biomass_meth, ph_meth, previous_land_use_meth, soil_type_meth, tot_n_meth, tot_org_c_meth, host_subspecf_genlin, assembly_quality, fao_class, host_phenotype, anim_water_method, farm_equip, fertilizer_admin, food_prod, food_trav_mode, plant_struc, plant_water_method, season, plant_growth_med, chem_administration, spikein_strain, adjacent_environment, ances_data, cur_vegetation, "Food_Product_type", "Food_source", host_common_name, host_genotype, host_life_stage, host_symbiont, library_prep_kit, local_class, plant_product, samp_collect_device, samp_store_loc, sequencing_location, soil_type, misc_param, root_med_carbon, root_med_macronutr, organism_count, fungicide_regm, gaseous_environment, herbicide_regm, non_mineral_nutr_regm, pesticide_regm, salt_regm, agrochem_addition, perturbation, store_cond, climate_environment, season_environment, standing_water_regm, previous_land_use, food_trav_vehic, extreme_event, fire, flooding, url, soil_horizon, cult_isol_date, farm_equip_san, farm_equip_shared, food_harvest_proc, rel_location, samp_pooling, soil_conductivity, soil_cover, "soil_pH", trophic_level, assembly_software, biotic_relationship, seq_quality_check, lib_layout, mid, adapters, samp_size, samp_vol_we_dna_ext, host_spec_range, lib_reads_seqd, lib_size, isol_growth_condt, nucl_acid_amp, nucl_acid_ext, sop, pathogenicity, seq_meth, annot, lib_screen, lib_vector, source_mat_id, target_gene, target_subfragment, assembly_name, chimera_check, specific_host, host_disease_stat, samp_mat_process, pcr_cond, pcr_primers, samp_name, samp_taxon_id, project_name, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, propagation, encoded_traits, rel_to_oxygen, samp_collec_device, samp_collec_method, size_frac, source_uvig, virus_enrich_appr, tax_ident, assembly_qual, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource)
);

CREATE TABLE air (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT NOT NULL, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	barometric_press TEXT, 
	carb_dioxide TEXT, 
	carb_monoxide TEXT, 
	humidity TEXT, 
	methane TEXT, 
	oxygen TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	solar_irradiance TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	ventilation_type TEXT, 
	wind_direction TEXT, 
	air_particulate_matter_concentration TEXT, 
	misc_param TEXT, 
	pollutants TEXT, 
	volatile_org_comp TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, barometric_press, carb_dioxide, carb_monoxide, humidity, methane, oxygen, salinity, samp_store_temp, solar_irradiance, ventilation_rate, wind_speed, chem_administration, samp_store_loc, ventilation_type, wind_direction, air_particulate_matter_concentration, misc_param, pollutants, volatile_org_comp, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name)
);

CREATE TABLE "air_MIGS_bacteria" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT NOT NULL, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	barometric_press TEXT, 
	carb_dioxide TEXT, 
	carb_monoxide TEXT, 
	humidity TEXT, 
	methane TEXT, 
	oxygen TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	solar_irradiance TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	ventilation_type TEXT, 
	wind_direction TEXT, 
	air_particulate_matter_concentration TEXT, 
	misc_param TEXT, 
	pollutants TEXT, 
	volatile_org_comp TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, barometric_press, carb_dioxide, carb_monoxide, humidity, methane, oxygen, salinity, samp_store_temp, solar_irradiance, ventilation_rate, wind_speed, chem_administration, samp_store_loc, ventilation_type, wind_direction, air_particulate_matter_concentration, misc_param, pollutants, volatile_org_comp, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "air_MIGS_eukaryote" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT NOT NULL, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	barometric_press TEXT, 
	carb_dioxide TEXT, 
	carb_monoxide TEXT, 
	humidity TEXT, 
	methane TEXT, 
	oxygen TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	solar_irradiance TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	ventilation_type TEXT, 
	wind_direction TEXT, 
	air_particulate_matter_concentration TEXT, 
	misc_param TEXT, 
	pollutants TEXT, 
	volatile_org_comp TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, barometric_press, carb_dioxide, carb_monoxide, humidity, methane, oxygen, salinity, samp_store_temp, solar_irradiance, ventilation_rate, wind_speed, chem_administration, samp_store_loc, ventilation_type, wind_direction, air_particulate_matter_concentration, misc_param, pollutants, volatile_org_comp, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "air_MIGS_org" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT NOT NULL, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	barometric_press TEXT, 
	carb_dioxide TEXT, 
	carb_monoxide TEXT, 
	humidity TEXT, 
	methane TEXT, 
	oxygen TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	solar_irradiance TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	ventilation_type TEXT, 
	wind_direction TEXT, 
	air_particulate_matter_concentration TEXT, 
	misc_param TEXT, 
	pollutants TEXT, 
	volatile_org_comp TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, barometric_press, carb_dioxide, carb_monoxide, humidity, methane, oxygen, salinity, samp_store_temp, solar_irradiance, ventilation_rate, wind_speed, chem_administration, samp_store_loc, ventilation_type, wind_direction, air_particulate_matter_concentration, misc_param, pollutants, volatile_org_comp, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "air_MIGS_plant" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT NOT NULL, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	barometric_press TEXT, 
	carb_dioxide TEXT, 
	carb_monoxide TEXT, 
	humidity TEXT, 
	methane TEXT, 
	oxygen TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	solar_irradiance TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	ventilation_type TEXT, 
	wind_direction TEXT, 
	air_particulate_matter_concentration TEXT, 
	misc_param TEXT, 
	pollutants TEXT, 
	volatile_org_comp TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, barometric_press, carb_dioxide, carb_monoxide, humidity, methane, oxygen, salinity, samp_store_temp, solar_irradiance, ventilation_rate, wind_speed, chem_administration, samp_store_loc, ventilation_type, wind_direction, air_particulate_matter_concentration, misc_param, pollutants, volatile_org_comp, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "air_MIGS_virus" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT NOT NULL, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	barometric_press TEXT, 
	carb_dioxide TEXT, 
	carb_monoxide TEXT, 
	humidity TEXT, 
	methane TEXT, 
	oxygen TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	solar_irradiance TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	ventilation_type TEXT, 
	wind_direction TEXT, 
	air_particulate_matter_concentration TEXT, 
	misc_param TEXT, 
	pollutants TEXT, 
	volatile_org_comp TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, barometric_press, carb_dioxide, carb_monoxide, humidity, methane, oxygen, salinity, samp_store_temp, solar_irradiance, ventilation_rate, wind_speed, chem_administration, samp_store_loc, ventilation_type, wind_direction, air_particulate_matter_concentration, misc_param, pollutants, volatile_org_comp, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "air_MIMAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT NOT NULL, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	barometric_press TEXT, 
	carb_dioxide TEXT, 
	carb_monoxide TEXT, 
	humidity TEXT, 
	methane TEXT, 
	oxygen TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	solar_irradiance TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	ventilation_type TEXT, 
	wind_direction TEXT, 
	air_particulate_matter_concentration TEXT, 
	misc_param TEXT, 
	pollutants TEXT, 
	volatile_org_comp TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, barometric_press, carb_dioxide, carb_monoxide, humidity, methane, oxygen, salinity, samp_store_temp, solar_irradiance, ventilation_rate, wind_speed, chem_administration, samp_store_loc, ventilation_type, wind_direction, air_particulate_matter_concentration, misc_param, pollutants, volatile_org_comp, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "air_MIMARKS_specimen" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT NOT NULL, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	barometric_press TEXT, 
	carb_dioxide TEXT, 
	carb_monoxide TEXT, 
	humidity TEXT, 
	methane TEXT, 
	oxygen TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	solar_irradiance TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	ventilation_type TEXT, 
	wind_direction TEXT, 
	air_particulate_matter_concentration TEXT, 
	misc_param TEXT, 
	pollutants TEXT, 
	volatile_org_comp TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, barometric_press, carb_dioxide, carb_monoxide, humidity, methane, oxygen, salinity, samp_store_temp, solar_irradiance, ventilation_rate, wind_speed, chem_administration, samp_store_loc, ventilation_type, wind_direction, air_particulate_matter_concentration, misc_param, pollutants, volatile_org_comp, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "air_MIMARKS_survey" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT NOT NULL, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	barometric_press TEXT, 
	carb_dioxide TEXT, 
	carb_monoxide TEXT, 
	humidity TEXT, 
	methane TEXT, 
	oxygen TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	solar_irradiance TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	ventilation_type TEXT, 
	wind_direction TEXT, 
	air_particulate_matter_concentration TEXT, 
	misc_param TEXT, 
	pollutants TEXT, 
	volatile_org_comp TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, barometric_press, carb_dioxide, carb_monoxide, humidity, methane, oxygen, salinity, samp_store_temp, solar_irradiance, ventilation_rate, wind_speed, chem_administration, samp_store_loc, ventilation_type, wind_direction, air_particulate_matter_concentration, misc_param, pollutants, volatile_org_comp, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "air_MIMS" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT NOT NULL, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	barometric_press TEXT, 
	carb_dioxide TEXT, 
	carb_monoxide TEXT, 
	humidity TEXT, 
	methane TEXT, 
	oxygen TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	solar_irradiance TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	ventilation_type TEXT, 
	wind_direction TEXT, 
	air_particulate_matter_concentration TEXT, 
	misc_param TEXT, 
	pollutants TEXT, 
	volatile_org_comp TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, barometric_press, carb_dioxide, carb_monoxide, humidity, methane, oxygen, salinity, samp_store_temp, solar_irradiance, ventilation_rate, wind_speed, chem_administration, samp_store_loc, ventilation_type, wind_direction, air_particulate_matter_concentration, misc_param, pollutants, volatile_org_comp, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "air_MISAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT NOT NULL, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	barometric_press TEXT, 
	carb_dioxide TEXT, 
	carb_monoxide TEXT, 
	humidity TEXT, 
	methane TEXT, 
	oxygen TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	solar_irradiance TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	ventilation_type TEXT, 
	wind_direction TEXT, 
	air_particulate_matter_concentration TEXT, 
	misc_param TEXT, 
	pollutants TEXT, 
	volatile_org_comp TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, barometric_press, carb_dioxide, carb_monoxide, humidity, methane, oxygen, salinity, samp_store_temp, solar_irradiance, ventilation_rate, wind_speed, chem_administration, samp_store_loc, ventilation_type, wind_direction, air_particulate_matter_concentration, misc_param, pollutants, volatile_org_comp, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "air_MIUVIG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT NOT NULL, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	barometric_press TEXT, 
	carb_dioxide TEXT, 
	carb_monoxide TEXT, 
	humidity TEXT, 
	methane TEXT, 
	oxygen TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	solar_irradiance TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	ventilation_type TEXT, 
	wind_direction TEXT, 
	air_particulate_matter_concentration TEXT, 
	misc_param TEXT, 
	pollutants TEXT, 
	volatile_org_comp TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, barometric_press, carb_dioxide, carb_monoxide, humidity, methane, oxygen, salinity, samp_store_temp, solar_irradiance, ventilation_rate, wind_speed, chem_administration, samp_store_loc, ventilation_type, wind_direction, air_particulate_matter_concentration, misc_param, pollutants, volatile_org_comp, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE built_environment (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	freq_clean VARCHAR(9), 
	samp_floor VARCHAR(9), 
	surf_material VARCHAR(15), 
	gender_restroom VARCHAR(15), 
	door_mat VARCHAR(22), 
	handidness VARCHAR(16), 
	room_connected VARCHAR(15), 
	room_type VARCHAR(29), 
	occup_document VARCHAR(15), 
	shading_device_type VARCHAR(21), 
	floor_struc VARCHAR(14), 
	indoor_space VARCHAR(11) NOT NULL, 
	door_type_wood VARCHAR(24), 
	window_cover VARCHAR(8), 
	window_vert_pos VARCHAR(6), 
	build_docs VARCHAR(28), 
	arch_struc VARCHAR(8), 
	indoor_surf VARCHAR(11), 
	furniture VARCHAR(7), 
	pres_animal_insect VARCHAR(15), 
	ceil_type VARCHAR(13), 
	window_mat VARCHAR(10), 
	samp_weather VARCHAR(9), 
	window_status TEXT, 
	door_type_metal VARCHAR(16), 
	door_move VARCHAR(15), 
	door_type VARCHAR(9), 
	heat_sys_deliv_meth TEXT, 
	room_loc VARCHAR(13), 
	substructure_type VARCHAR(13), 
	wall_texture VARCHAR(16), 
	ceil_texture VARCHAR(16), 
	door_cond VARCHAR(12), 
	shading_device_cond VARCHAR(12), 
	window_cond VARCHAR(12), 
	ceil_finish_mat VARCHAR(29), 
	surf_air_cont VARCHAR(26), 
	rel_samp_loc VARCHAR(13), 
	train_stop_loc VARCHAR(8), 
	shading_device_loc TEXT, 
	water_feat_type VARCHAR(16), 
	wall_const_type VARCHAR(23), 
	fireplace_type TEXT, 
	door_direct VARCHAR(8), 
	window_horiz_pos VARCHAR(6), 
	door_comp_type VARCHAR(13), 
	floor_water_mold VARCHAR(26), 
	weekday VARCHAR(9), 
	light_type VARCHAR(18) NOT NULL, 
	ceil_cond VARCHAR(12), 
	floor_cond VARCHAR(12), 
	int_wall_cond VARCHAR(12), 
	room_condt VARCHAR(28), 
	room_samp_pos VARCHAR(16), 
	quad_pos VARCHAR(10), 
	door_loc VARCHAR(5), 
	heat_deliv_loc VARCHAR(5), 
	wall_loc VARCHAR(5), 
	window_loc VARCHAR(5), 
	ext_wall_orient VARCHAR(9), 
	ext_window_orient VARCHAR(9), 
	build_occup_type VARCHAR(14) NOT NULL, 
	drawings VARCHAR(23), 
	specific VARCHAR(12), 
	wall_surf_treatment VARCHAR(12), 
	filter_type VARCHAR(39) NOT NULL, 
	vis_media VARCHAR(63), 
	aero_struc TEXT, 
	wall_finish_mat VARCHAR(20), 
	ceil_water_mold TEXT, 
	door_water_mold TEXT, 
	shad_dev_water_mold TEXT, 
	wall_water_mold TEXT, 
	window_water_mold TEXT, 
	heat_cool_type VARCHAR(17) NOT NULL, 
	train_line VARCHAR(6), 
	window_type VARCHAR(23), 
	train_stat_loc VARCHAR(26), 
	season_use VARCHAR(6), 
	mech_struc VARCHAR(9), 
	floor_finish_mat VARCHAR(23), 
	space_typ_state TEXT NOT NULL, 
	built_struc_set TEXT, 
	building_setting VARCHAR(8) NOT NULL, 
	ceil_struc TEXT, 
	avg_occup FLOAT, 
	occup_density_samp FLOAT NOT NULL, 
	samp_time_out FLOAT, 
	surf_moisture_ph FLOAT, 
	typ_occup_density FLOAT NOT NULL, 
	abs_air_humidity TEXT NOT NULL, 
	amount_light TEXT, 
	avg_dew_point TEXT, 
	avg_temp TEXT, 
	built_struc_age TEXT, 
	carb_dioxide TEXT NOT NULL, 
	ceil_area TEXT, 
	ceil_thermal_mass TEXT, 
	dew_point TEXT, 
	door_size TEXT, 
	exp_duct TEXT, 
	floor_age TEXT, 
	floor_area TEXT, 
	floor_thermal_mass TEXT, 
	height_carper_fiber TEXT, 
	inside_lux TEXT, 
	rel_air_humidity TEXT NOT NULL, 
	rel_humidity_out TEXT, 
	room_air_exch_rate TEXT, 
	specific_humidity TEXT, 
	surf_humidity TEXT, 
	surf_moisture TEXT, 
	surf_temp TEXT, 
	temp_out TEXT, 
	wall_area TEXT, 
	wall_height TEXT, 
	wall_thermal_mass TEXT, 
	water_feat_size TEXT, 
	window_size TEXT, 
	bathroom_count INTEGER, 
	bedroom_count INTEGER, 
	cool_syst_id INTEGER, 
	elevator INTEGER, 
	escalator INTEGER, 
	exp_pipe INTEGER, 
	ext_door INTEGER, 
	floor_count INTEGER, 
	freq_cook INTEGER, 
	hall_count INTEGER, 
	heat_system_id INTEGER, 
	max_occup INTEGER, 
	number_pets INTEGER, 
	number_plants INTEGER, 
	number_resident INTEGER, 
	occup_samp INTEGER NOT NULL, 
	room_count INTEGER, 
	room_moist_dam_hist INTEGER, 
	room_occup INTEGER, 
	room_window_count INTEGER, 
	samp_room_id INTEGER, 
	window_open_freq INTEGER, 
	room_door_dist TEXT, 
	room_net_area TEXT, 
	room_vol TEXT, 
	room_dim TEXT, 
	address TEXT, 
	season TEXT, 
	built_struc_type TEXT, 
	room_architec_elem TEXT, 
	samp_sort_meth TEXT, 
	shading_device_mat TEXT, 
	ventilation_type TEXT NOT NULL, 
	organism_count TEXT NOT NULL, 
	adj_room TEXT, 
	room_door_share TEXT, 
	room_hallway TEXT, 
	room_wall_share TEXT, 
	date_last_rain DATE, 
	last_clean DATE, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, freq_clean, samp_floor, surf_material, gender_restroom, door_mat, handidness, room_connected, room_type, occup_document, shading_device_type, floor_struc, indoor_space, door_type_wood, window_cover, window_vert_pos, build_docs, arch_struc, indoor_surf, furniture, pres_animal_insect, ceil_type, window_mat, samp_weather, window_status, door_type_metal, door_move, door_type, heat_sys_deliv_meth, room_loc, substructure_type, wall_texture, ceil_texture, door_cond, shading_device_cond, window_cond, ceil_finish_mat, surf_air_cont, rel_samp_loc, train_stop_loc, shading_device_loc, water_feat_type, wall_const_type, fireplace_type, door_direct, window_horiz_pos, door_comp_type, floor_water_mold, weekday, light_type, ceil_cond, floor_cond, int_wall_cond, room_condt, room_samp_pos, quad_pos, door_loc, heat_deliv_loc, wall_loc, window_loc, ext_wall_orient, ext_window_orient, build_occup_type, drawings, specific, wall_surf_treatment, filter_type, vis_media, aero_struc, wall_finish_mat, ceil_water_mold, door_water_mold, shad_dev_water_mold, wall_water_mold, window_water_mold, heat_cool_type, train_line, window_type, train_stat_loc, season_use, mech_struc, floor_finish_mat, space_typ_state, built_struc_set, building_setting, ceil_struc, avg_occup, occup_density_samp, samp_time_out, surf_moisture_ph, typ_occup_density, abs_air_humidity, amount_light, avg_dew_point, avg_temp, built_struc_age, carb_dioxide, ceil_area, ceil_thermal_mass, dew_point, door_size, exp_duct, floor_age, floor_area, floor_thermal_mass, height_carper_fiber, inside_lux, rel_air_humidity, rel_humidity_out, room_air_exch_rate, specific_humidity, surf_humidity, surf_moisture, surf_temp, temp_out, wall_area, wall_height, wall_thermal_mass, water_feat_size, window_size, bathroom_count, bedroom_count, cool_syst_id, elevator, escalator, exp_pipe, ext_door, floor_count, freq_cook, hall_count, heat_system_id, max_occup, number_pets, number_plants, number_resident, occup_samp, room_count, room_moist_dam_hist, room_occup, room_window_count, samp_room_id, window_open_freq, room_door_dist, room_net_area, room_vol, room_dim, address, season, built_struc_type, room_architec_elem, samp_sort_meth, shading_device_mat, ventilation_type, organism_count, adj_room, room_door_share, room_hallway, room_wall_share, date_last_rain, last_clean, project_name, samp_name)
);

CREATE TABLE "built_environment_MIGS_bacteria" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	freq_clean VARCHAR(9), 
	samp_floor VARCHAR(9), 
	surf_material VARCHAR(15), 
	gender_restroom VARCHAR(15), 
	door_mat VARCHAR(22), 
	handidness VARCHAR(16), 
	room_connected VARCHAR(15), 
	room_type VARCHAR(29), 
	occup_document VARCHAR(15), 
	shading_device_type VARCHAR(21), 
	floor_struc VARCHAR(14), 
	indoor_space VARCHAR(11) NOT NULL, 
	door_type_wood VARCHAR(24), 
	window_cover VARCHAR(8), 
	window_vert_pos VARCHAR(6), 
	build_docs VARCHAR(28), 
	arch_struc VARCHAR(8), 
	indoor_surf VARCHAR(11), 
	furniture VARCHAR(7), 
	pres_animal_insect VARCHAR(15), 
	ceil_type VARCHAR(13), 
	window_mat VARCHAR(10), 
	samp_weather VARCHAR(9), 
	window_status TEXT, 
	door_type_metal VARCHAR(16), 
	door_move VARCHAR(15), 
	door_type VARCHAR(9), 
	heat_sys_deliv_meth TEXT, 
	room_loc VARCHAR(13), 
	substructure_type VARCHAR(13), 
	wall_texture VARCHAR(16), 
	ceil_texture VARCHAR(16), 
	door_cond VARCHAR(12), 
	shading_device_cond VARCHAR(12), 
	window_cond VARCHAR(12), 
	ceil_finish_mat VARCHAR(29), 
	surf_air_cont VARCHAR(26), 
	rel_samp_loc VARCHAR(13), 
	train_stop_loc VARCHAR(8), 
	shading_device_loc TEXT, 
	water_feat_type VARCHAR(16), 
	wall_const_type VARCHAR(23), 
	fireplace_type TEXT, 
	door_direct VARCHAR(8), 
	window_horiz_pos VARCHAR(6), 
	door_comp_type VARCHAR(13), 
	floor_water_mold VARCHAR(26), 
	weekday VARCHAR(9), 
	light_type VARCHAR(18) NOT NULL, 
	ceil_cond VARCHAR(12), 
	floor_cond VARCHAR(12), 
	int_wall_cond VARCHAR(12), 
	room_condt VARCHAR(28), 
	room_samp_pos VARCHAR(16), 
	quad_pos VARCHAR(10), 
	door_loc VARCHAR(5), 
	heat_deliv_loc VARCHAR(5), 
	wall_loc VARCHAR(5), 
	window_loc VARCHAR(5), 
	ext_wall_orient VARCHAR(9), 
	ext_window_orient VARCHAR(9), 
	build_occup_type VARCHAR(14) NOT NULL, 
	drawings VARCHAR(23), 
	specific VARCHAR(12), 
	wall_surf_treatment VARCHAR(12), 
	filter_type VARCHAR(39) NOT NULL, 
	vis_media VARCHAR(63), 
	aero_struc TEXT, 
	wall_finish_mat VARCHAR(20), 
	ceil_water_mold TEXT, 
	door_water_mold TEXT, 
	shad_dev_water_mold TEXT, 
	wall_water_mold TEXT, 
	window_water_mold TEXT, 
	heat_cool_type VARCHAR(17) NOT NULL, 
	train_line VARCHAR(6), 
	window_type VARCHAR(23), 
	train_stat_loc VARCHAR(26), 
	season_use VARCHAR(6), 
	mech_struc VARCHAR(9), 
	floor_finish_mat VARCHAR(23), 
	space_typ_state TEXT NOT NULL, 
	built_struc_set TEXT, 
	building_setting VARCHAR(8) NOT NULL, 
	ceil_struc TEXT, 
	avg_occup FLOAT, 
	occup_density_samp FLOAT NOT NULL, 
	samp_time_out FLOAT, 
	surf_moisture_ph FLOAT, 
	typ_occup_density FLOAT NOT NULL, 
	abs_air_humidity TEXT NOT NULL, 
	air_temp TEXT NOT NULL, 
	amount_light TEXT, 
	avg_dew_point TEXT, 
	avg_temp TEXT, 
	built_struc_age TEXT, 
	carb_dioxide TEXT NOT NULL, 
	ceil_area TEXT, 
	ceil_thermal_mass TEXT, 
	dew_point TEXT, 
	door_size TEXT, 
	exp_duct TEXT, 
	floor_age TEXT, 
	floor_area TEXT, 
	floor_thermal_mass TEXT, 
	height_carper_fiber TEXT, 
	inside_lux TEXT, 
	rel_air_humidity TEXT NOT NULL, 
	rel_humidity_out TEXT, 
	room_air_exch_rate TEXT, 
	specific_humidity TEXT, 
	surf_humidity TEXT, 
	surf_moisture TEXT, 
	surf_temp TEXT, 
	temp_out TEXT, 
	wall_area TEXT, 
	wall_height TEXT, 
	wall_thermal_mass TEXT, 
	water_feat_size TEXT, 
	window_size TEXT, 
	bathroom_count INTEGER, 
	bedroom_count INTEGER, 
	cool_syst_id INTEGER, 
	elevator INTEGER, 
	escalator INTEGER, 
	exp_pipe INTEGER, 
	ext_door INTEGER, 
	floor_count INTEGER, 
	freq_cook INTEGER, 
	hall_count INTEGER, 
	heat_system_id INTEGER, 
	max_occup INTEGER, 
	number_pets INTEGER, 
	number_plants INTEGER, 
	number_resident INTEGER, 
	occup_samp INTEGER NOT NULL, 
	room_count INTEGER, 
	room_moist_dam_hist INTEGER, 
	room_occup INTEGER, 
	room_window_count INTEGER, 
	samp_room_id INTEGER, 
	window_open_freq INTEGER, 
	room_door_dist TEXT, 
	room_net_area TEXT, 
	room_vol TEXT, 
	room_dim TEXT, 
	address TEXT, 
	season TEXT, 
	built_struc_type TEXT, 
	room_architec_elem TEXT, 
	samp_sort_meth TEXT, 
	shading_device_mat TEXT, 
	ventilation_type TEXT NOT NULL, 
	organism_count TEXT NOT NULL, 
	adj_room TEXT, 
	room_door_share TEXT, 
	room_hallway TEXT, 
	room_wall_share TEXT, 
	date_last_rain DATE, 
	last_clean DATE, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, freq_clean, samp_floor, surf_material, gender_restroom, door_mat, handidness, room_connected, room_type, occup_document, shading_device_type, floor_struc, indoor_space, door_type_wood, window_cover, window_vert_pos, build_docs, arch_struc, indoor_surf, furniture, pres_animal_insect, ceil_type, window_mat, samp_weather, window_status, door_type_metal, door_move, door_type, heat_sys_deliv_meth, room_loc, substructure_type, wall_texture, ceil_texture, door_cond, shading_device_cond, window_cond, ceil_finish_mat, surf_air_cont, rel_samp_loc, train_stop_loc, shading_device_loc, water_feat_type, wall_const_type, fireplace_type, door_direct, window_horiz_pos, door_comp_type, floor_water_mold, weekday, light_type, ceil_cond, floor_cond, int_wall_cond, room_condt, room_samp_pos, quad_pos, door_loc, heat_deliv_loc, wall_loc, window_loc, ext_wall_orient, ext_window_orient, build_occup_type, drawings, specific, wall_surf_treatment, filter_type, vis_media, aero_struc, wall_finish_mat, ceil_water_mold, door_water_mold, shad_dev_water_mold, wall_water_mold, window_water_mold, heat_cool_type, train_line, window_type, train_stat_loc, season_use, mech_struc, floor_finish_mat, space_typ_state, built_struc_set, building_setting, ceil_struc, avg_occup, occup_density_samp, samp_time_out, surf_moisture_ph, typ_occup_density, abs_air_humidity, air_temp, amount_light, avg_dew_point, avg_temp, built_struc_age, carb_dioxide, ceil_area, ceil_thermal_mass, dew_point, door_size, exp_duct, floor_age, floor_area, floor_thermal_mass, height_carper_fiber, inside_lux, rel_air_humidity, rel_humidity_out, room_air_exch_rate, specific_humidity, surf_humidity, surf_moisture, surf_temp, temp_out, wall_area, wall_height, wall_thermal_mass, water_feat_size, window_size, bathroom_count, bedroom_count, cool_syst_id, elevator, escalator, exp_pipe, ext_door, floor_count, freq_cook, hall_count, heat_system_id, max_occup, number_pets, number_plants, number_resident, occup_samp, room_count, room_moist_dam_hist, room_occup, room_window_count, samp_room_id, window_open_freq, room_door_dist, room_net_area, room_vol, room_dim, address, season, built_struc_type, room_architec_elem, samp_sort_meth, shading_device_mat, ventilation_type, organism_count, adj_room, room_door_share, room_hallway, room_wall_share, date_last_rain, last_clean, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, samp_vol_we_dna_ext, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "built_environment_MIGS_eukaryote" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	freq_clean VARCHAR(9), 
	samp_floor VARCHAR(9), 
	surf_material VARCHAR(15), 
	gender_restroom VARCHAR(15), 
	door_mat VARCHAR(22), 
	handidness VARCHAR(16), 
	room_connected VARCHAR(15), 
	room_type VARCHAR(29), 
	occup_document VARCHAR(15), 
	shading_device_type VARCHAR(21), 
	floor_struc VARCHAR(14), 
	indoor_space VARCHAR(11) NOT NULL, 
	door_type_wood VARCHAR(24), 
	window_cover VARCHAR(8), 
	window_vert_pos VARCHAR(6), 
	build_docs VARCHAR(28), 
	arch_struc VARCHAR(8), 
	indoor_surf VARCHAR(11), 
	furniture VARCHAR(7), 
	pres_animal_insect VARCHAR(15), 
	ceil_type VARCHAR(13), 
	window_mat VARCHAR(10), 
	samp_weather VARCHAR(9), 
	window_status TEXT, 
	door_type_metal VARCHAR(16), 
	door_move VARCHAR(15), 
	door_type VARCHAR(9), 
	heat_sys_deliv_meth TEXT, 
	room_loc VARCHAR(13), 
	substructure_type VARCHAR(13), 
	wall_texture VARCHAR(16), 
	ceil_texture VARCHAR(16), 
	door_cond VARCHAR(12), 
	shading_device_cond VARCHAR(12), 
	window_cond VARCHAR(12), 
	ceil_finish_mat VARCHAR(29), 
	surf_air_cont VARCHAR(26), 
	rel_samp_loc VARCHAR(13), 
	train_stop_loc VARCHAR(8), 
	shading_device_loc TEXT, 
	water_feat_type VARCHAR(16), 
	wall_const_type VARCHAR(23), 
	fireplace_type TEXT, 
	door_direct VARCHAR(8), 
	window_horiz_pos VARCHAR(6), 
	door_comp_type VARCHAR(13), 
	floor_water_mold VARCHAR(26), 
	weekday VARCHAR(9), 
	light_type VARCHAR(18) NOT NULL, 
	ceil_cond VARCHAR(12), 
	floor_cond VARCHAR(12), 
	int_wall_cond VARCHAR(12), 
	room_condt VARCHAR(28), 
	room_samp_pos VARCHAR(16), 
	quad_pos VARCHAR(10), 
	door_loc VARCHAR(5), 
	heat_deliv_loc VARCHAR(5), 
	wall_loc VARCHAR(5), 
	window_loc VARCHAR(5), 
	ext_wall_orient VARCHAR(9), 
	ext_window_orient VARCHAR(9), 
	build_occup_type VARCHAR(14) NOT NULL, 
	drawings VARCHAR(23), 
	specific VARCHAR(12), 
	wall_surf_treatment VARCHAR(12), 
	filter_type VARCHAR(39) NOT NULL, 
	vis_media VARCHAR(63), 
	aero_struc TEXT, 
	wall_finish_mat VARCHAR(20), 
	ceil_water_mold TEXT, 
	door_water_mold TEXT, 
	shad_dev_water_mold TEXT, 
	wall_water_mold TEXT, 
	window_water_mold TEXT, 
	heat_cool_type VARCHAR(17) NOT NULL, 
	train_line VARCHAR(6), 
	window_type VARCHAR(23), 
	train_stat_loc VARCHAR(26), 
	season_use VARCHAR(6), 
	mech_struc VARCHAR(9), 
	floor_finish_mat VARCHAR(23), 
	space_typ_state TEXT NOT NULL, 
	built_struc_set TEXT, 
	building_setting VARCHAR(8) NOT NULL, 
	ceil_struc TEXT, 
	avg_occup FLOAT, 
	occup_density_samp FLOAT NOT NULL, 
	samp_time_out FLOAT, 
	surf_moisture_ph FLOAT, 
	typ_occup_density FLOAT NOT NULL, 
	abs_air_humidity TEXT NOT NULL, 
	air_temp TEXT NOT NULL, 
	amount_light TEXT, 
	avg_dew_point TEXT, 
	avg_temp TEXT, 
	built_struc_age TEXT, 
	carb_dioxide TEXT NOT NULL, 
	ceil_area TEXT, 
	ceil_thermal_mass TEXT, 
	dew_point TEXT, 
	door_size TEXT, 
	exp_duct TEXT, 
	floor_age TEXT, 
	floor_area TEXT, 
	floor_thermal_mass TEXT, 
	height_carper_fiber TEXT, 
	inside_lux TEXT, 
	rel_air_humidity TEXT NOT NULL, 
	rel_humidity_out TEXT, 
	room_air_exch_rate TEXT, 
	specific_humidity TEXT, 
	surf_humidity TEXT, 
	surf_moisture TEXT, 
	surf_temp TEXT, 
	temp_out TEXT, 
	wall_area TEXT, 
	wall_height TEXT, 
	wall_thermal_mass TEXT, 
	water_feat_size TEXT, 
	window_size TEXT, 
	bathroom_count INTEGER, 
	bedroom_count INTEGER, 
	cool_syst_id INTEGER, 
	elevator INTEGER, 
	escalator INTEGER, 
	exp_pipe INTEGER, 
	ext_door INTEGER, 
	floor_count INTEGER, 
	freq_cook INTEGER, 
	hall_count INTEGER, 
	heat_system_id INTEGER, 
	max_occup INTEGER, 
	number_pets INTEGER, 
	number_plants INTEGER, 
	number_resident INTEGER, 
	occup_samp INTEGER NOT NULL, 
	room_count INTEGER, 
	room_moist_dam_hist INTEGER, 
	room_occup INTEGER, 
	room_window_count INTEGER, 
	samp_room_id INTEGER, 
	window_open_freq INTEGER, 
	room_door_dist TEXT, 
	room_net_area TEXT, 
	room_vol TEXT, 
	room_dim TEXT, 
	address TEXT, 
	season TEXT, 
	built_struc_type TEXT, 
	room_architec_elem TEXT, 
	samp_sort_meth TEXT, 
	shading_device_mat TEXT, 
	ventilation_type TEXT NOT NULL, 
	organism_count TEXT NOT NULL, 
	adj_room TEXT, 
	room_door_share TEXT, 
	room_hallway TEXT, 
	room_wall_share TEXT, 
	date_last_rain DATE, 
	last_clean DATE, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, freq_clean, samp_floor, surf_material, gender_restroom, door_mat, handidness, room_connected, room_type, occup_document, shading_device_type, floor_struc, indoor_space, door_type_wood, window_cover, window_vert_pos, build_docs, arch_struc, indoor_surf, furniture, pres_animal_insect, ceil_type, window_mat, samp_weather, window_status, door_type_metal, door_move, door_type, heat_sys_deliv_meth, room_loc, substructure_type, wall_texture, ceil_texture, door_cond, shading_device_cond, window_cond, ceil_finish_mat, surf_air_cont, rel_samp_loc, train_stop_loc, shading_device_loc, water_feat_type, wall_const_type, fireplace_type, door_direct, window_horiz_pos, door_comp_type, floor_water_mold, weekday, light_type, ceil_cond, floor_cond, int_wall_cond, room_condt, room_samp_pos, quad_pos, door_loc, heat_deliv_loc, wall_loc, window_loc, ext_wall_orient, ext_window_orient, build_occup_type, drawings, specific, wall_surf_treatment, filter_type, vis_media, aero_struc, wall_finish_mat, ceil_water_mold, door_water_mold, shad_dev_water_mold, wall_water_mold, window_water_mold, heat_cool_type, train_line, window_type, train_stat_loc, season_use, mech_struc, floor_finish_mat, space_typ_state, built_struc_set, building_setting, ceil_struc, avg_occup, occup_density_samp, samp_time_out, surf_moisture_ph, typ_occup_density, abs_air_humidity, air_temp, amount_light, avg_dew_point, avg_temp, built_struc_age, carb_dioxide, ceil_area, ceil_thermal_mass, dew_point, door_size, exp_duct, floor_age, floor_area, floor_thermal_mass, height_carper_fiber, inside_lux, rel_air_humidity, rel_humidity_out, room_air_exch_rate, specific_humidity, surf_humidity, surf_moisture, surf_temp, temp_out, wall_area, wall_height, wall_thermal_mass, water_feat_size, window_size, bathroom_count, bedroom_count, cool_syst_id, elevator, escalator, exp_pipe, ext_door, floor_count, freq_cook, hall_count, heat_system_id, max_occup, number_pets, number_plants, number_resident, occup_samp, room_count, room_moist_dam_hist, room_occup, room_window_count, samp_room_id, window_open_freq, room_door_dist, room_net_area, room_vol, room_dim, address, season, built_struc_type, room_architec_elem, samp_sort_meth, shading_device_mat, ventilation_type, organism_count, adj_room, room_door_share, room_hallway, room_wall_share, date_last_rain, last_clean, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, samp_vol_we_dna_ext, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "built_environment_MIGS_org" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	freq_clean VARCHAR(9), 
	samp_floor VARCHAR(9), 
	surf_material VARCHAR(15), 
	gender_restroom VARCHAR(15), 
	door_mat VARCHAR(22), 
	handidness VARCHAR(16), 
	room_connected VARCHAR(15), 
	room_type VARCHAR(29), 
	occup_document VARCHAR(15), 
	shading_device_type VARCHAR(21), 
	floor_struc VARCHAR(14), 
	indoor_space VARCHAR(11) NOT NULL, 
	door_type_wood VARCHAR(24), 
	window_cover VARCHAR(8), 
	window_vert_pos VARCHAR(6), 
	build_docs VARCHAR(28), 
	arch_struc VARCHAR(8), 
	indoor_surf VARCHAR(11), 
	furniture VARCHAR(7), 
	pres_animal_insect VARCHAR(15), 
	ceil_type VARCHAR(13), 
	window_mat VARCHAR(10), 
	samp_weather VARCHAR(9), 
	window_status TEXT, 
	door_type_metal VARCHAR(16), 
	door_move VARCHAR(15), 
	door_type VARCHAR(9), 
	heat_sys_deliv_meth TEXT, 
	room_loc VARCHAR(13), 
	substructure_type VARCHAR(13), 
	wall_texture VARCHAR(16), 
	ceil_texture VARCHAR(16), 
	door_cond VARCHAR(12), 
	shading_device_cond VARCHAR(12), 
	window_cond VARCHAR(12), 
	ceil_finish_mat VARCHAR(29), 
	surf_air_cont VARCHAR(26), 
	rel_samp_loc VARCHAR(13), 
	train_stop_loc VARCHAR(8), 
	shading_device_loc TEXT, 
	water_feat_type VARCHAR(16), 
	wall_const_type VARCHAR(23), 
	fireplace_type TEXT, 
	door_direct VARCHAR(8), 
	window_horiz_pos VARCHAR(6), 
	door_comp_type VARCHAR(13), 
	floor_water_mold VARCHAR(26), 
	weekday VARCHAR(9), 
	light_type VARCHAR(18) NOT NULL, 
	ceil_cond VARCHAR(12), 
	floor_cond VARCHAR(12), 
	int_wall_cond VARCHAR(12), 
	room_condt VARCHAR(28), 
	room_samp_pos VARCHAR(16), 
	quad_pos VARCHAR(10), 
	door_loc VARCHAR(5), 
	heat_deliv_loc VARCHAR(5), 
	wall_loc VARCHAR(5), 
	window_loc VARCHAR(5), 
	ext_wall_orient VARCHAR(9), 
	ext_window_orient VARCHAR(9), 
	build_occup_type VARCHAR(14) NOT NULL, 
	drawings VARCHAR(23), 
	specific VARCHAR(12), 
	wall_surf_treatment VARCHAR(12), 
	filter_type VARCHAR(39) NOT NULL, 
	vis_media VARCHAR(63), 
	aero_struc TEXT, 
	wall_finish_mat VARCHAR(20), 
	ceil_water_mold TEXT, 
	door_water_mold TEXT, 
	shad_dev_water_mold TEXT, 
	wall_water_mold TEXT, 
	window_water_mold TEXT, 
	heat_cool_type VARCHAR(17) NOT NULL, 
	train_line VARCHAR(6), 
	window_type VARCHAR(23), 
	train_stat_loc VARCHAR(26), 
	season_use VARCHAR(6), 
	mech_struc VARCHAR(9), 
	floor_finish_mat VARCHAR(23), 
	space_typ_state TEXT NOT NULL, 
	built_struc_set TEXT, 
	building_setting VARCHAR(8) NOT NULL, 
	ceil_struc TEXT, 
	avg_occup FLOAT, 
	occup_density_samp FLOAT NOT NULL, 
	samp_time_out FLOAT, 
	surf_moisture_ph FLOAT, 
	typ_occup_density FLOAT NOT NULL, 
	abs_air_humidity TEXT NOT NULL, 
	air_temp TEXT NOT NULL, 
	amount_light TEXT, 
	avg_dew_point TEXT, 
	avg_temp TEXT, 
	built_struc_age TEXT, 
	carb_dioxide TEXT NOT NULL, 
	ceil_area TEXT, 
	ceil_thermal_mass TEXT, 
	dew_point TEXT, 
	door_size TEXT, 
	exp_duct TEXT, 
	floor_age TEXT, 
	floor_area TEXT, 
	floor_thermal_mass TEXT, 
	height_carper_fiber TEXT, 
	inside_lux TEXT, 
	rel_air_humidity TEXT NOT NULL, 
	rel_humidity_out TEXT, 
	room_air_exch_rate TEXT, 
	specific_humidity TEXT, 
	surf_humidity TEXT, 
	surf_moisture TEXT, 
	surf_temp TEXT, 
	temp_out TEXT, 
	wall_area TEXT, 
	wall_height TEXT, 
	wall_thermal_mass TEXT, 
	water_feat_size TEXT, 
	window_size TEXT, 
	bathroom_count INTEGER, 
	bedroom_count INTEGER, 
	cool_syst_id INTEGER, 
	elevator INTEGER, 
	escalator INTEGER, 
	exp_pipe INTEGER, 
	ext_door INTEGER, 
	floor_count INTEGER, 
	freq_cook INTEGER, 
	hall_count INTEGER, 
	heat_system_id INTEGER, 
	max_occup INTEGER, 
	number_pets INTEGER, 
	number_plants INTEGER, 
	number_resident INTEGER, 
	occup_samp INTEGER NOT NULL, 
	room_count INTEGER, 
	room_moist_dam_hist INTEGER, 
	room_occup INTEGER, 
	room_window_count INTEGER, 
	samp_room_id INTEGER, 
	window_open_freq INTEGER, 
	room_door_dist TEXT, 
	room_net_area TEXT, 
	room_vol TEXT, 
	room_dim TEXT, 
	address TEXT, 
	season TEXT, 
	built_struc_type TEXT, 
	room_architec_elem TEXT, 
	samp_sort_meth TEXT, 
	shading_device_mat TEXT, 
	ventilation_type TEXT NOT NULL, 
	organism_count TEXT NOT NULL, 
	adj_room TEXT, 
	room_door_share TEXT, 
	room_hallway TEXT, 
	room_wall_share TEXT, 
	date_last_rain DATE, 
	last_clean DATE, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, freq_clean, samp_floor, surf_material, gender_restroom, door_mat, handidness, room_connected, room_type, occup_document, shading_device_type, floor_struc, indoor_space, door_type_wood, window_cover, window_vert_pos, build_docs, arch_struc, indoor_surf, furniture, pres_animal_insect, ceil_type, window_mat, samp_weather, window_status, door_type_metal, door_move, door_type, heat_sys_deliv_meth, room_loc, substructure_type, wall_texture, ceil_texture, door_cond, shading_device_cond, window_cond, ceil_finish_mat, surf_air_cont, rel_samp_loc, train_stop_loc, shading_device_loc, water_feat_type, wall_const_type, fireplace_type, door_direct, window_horiz_pos, door_comp_type, floor_water_mold, weekday, light_type, ceil_cond, floor_cond, int_wall_cond, room_condt, room_samp_pos, quad_pos, door_loc, heat_deliv_loc, wall_loc, window_loc, ext_wall_orient, ext_window_orient, build_occup_type, drawings, specific, wall_surf_treatment, filter_type, vis_media, aero_struc, wall_finish_mat, ceil_water_mold, door_water_mold, shad_dev_water_mold, wall_water_mold, window_water_mold, heat_cool_type, train_line, window_type, train_stat_loc, season_use, mech_struc, floor_finish_mat, space_typ_state, built_struc_set, building_setting, ceil_struc, avg_occup, occup_density_samp, samp_time_out, surf_moisture_ph, typ_occup_density, abs_air_humidity, air_temp, amount_light, avg_dew_point, avg_temp, built_struc_age, carb_dioxide, ceil_area, ceil_thermal_mass, dew_point, door_size, exp_duct, floor_age, floor_area, floor_thermal_mass, height_carper_fiber, inside_lux, rel_air_humidity, rel_humidity_out, room_air_exch_rate, specific_humidity, surf_humidity, surf_moisture, surf_temp, temp_out, wall_area, wall_height, wall_thermal_mass, water_feat_size, window_size, bathroom_count, bedroom_count, cool_syst_id, elevator, escalator, exp_pipe, ext_door, floor_count, freq_cook, hall_count, heat_system_id, max_occup, number_pets, number_plants, number_resident, occup_samp, room_count, room_moist_dam_hist, room_occup, room_window_count, samp_room_id, window_open_freq, room_door_dist, room_net_area, room_vol, room_dim, address, season, built_struc_type, room_architec_elem, samp_sort_meth, shading_device_mat, ventilation_type, organism_count, adj_room, room_door_share, room_hallway, room_wall_share, date_last_rain, last_clean, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, samp_vol_we_dna_ext, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "built_environment_MIGS_plant" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	freq_clean VARCHAR(9), 
	samp_floor VARCHAR(9), 
	surf_material VARCHAR(15), 
	gender_restroom VARCHAR(15), 
	door_mat VARCHAR(22), 
	handidness VARCHAR(16), 
	room_connected VARCHAR(15), 
	room_type VARCHAR(29), 
	occup_document VARCHAR(15), 
	shading_device_type VARCHAR(21), 
	floor_struc VARCHAR(14), 
	indoor_space VARCHAR(11) NOT NULL, 
	door_type_wood VARCHAR(24), 
	window_cover VARCHAR(8), 
	window_vert_pos VARCHAR(6), 
	build_docs VARCHAR(28), 
	arch_struc VARCHAR(8), 
	indoor_surf VARCHAR(11), 
	furniture VARCHAR(7), 
	pres_animal_insect VARCHAR(15), 
	ceil_type VARCHAR(13), 
	window_mat VARCHAR(10), 
	samp_weather VARCHAR(9), 
	window_status TEXT, 
	door_type_metal VARCHAR(16), 
	door_move VARCHAR(15), 
	door_type VARCHAR(9), 
	heat_sys_deliv_meth TEXT, 
	room_loc VARCHAR(13), 
	substructure_type VARCHAR(13), 
	wall_texture VARCHAR(16), 
	ceil_texture VARCHAR(16), 
	door_cond VARCHAR(12), 
	shading_device_cond VARCHAR(12), 
	window_cond VARCHAR(12), 
	ceil_finish_mat VARCHAR(29), 
	surf_air_cont VARCHAR(26), 
	rel_samp_loc VARCHAR(13), 
	train_stop_loc VARCHAR(8), 
	shading_device_loc TEXT, 
	water_feat_type VARCHAR(16), 
	wall_const_type VARCHAR(23), 
	fireplace_type TEXT, 
	door_direct VARCHAR(8), 
	window_horiz_pos VARCHAR(6), 
	door_comp_type VARCHAR(13), 
	floor_water_mold VARCHAR(26), 
	weekday VARCHAR(9), 
	light_type VARCHAR(18) NOT NULL, 
	ceil_cond VARCHAR(12), 
	floor_cond VARCHAR(12), 
	int_wall_cond VARCHAR(12), 
	room_condt VARCHAR(28), 
	room_samp_pos VARCHAR(16), 
	quad_pos VARCHAR(10), 
	door_loc VARCHAR(5), 
	heat_deliv_loc VARCHAR(5), 
	wall_loc VARCHAR(5), 
	window_loc VARCHAR(5), 
	ext_wall_orient VARCHAR(9), 
	ext_window_orient VARCHAR(9), 
	build_occup_type VARCHAR(14) NOT NULL, 
	drawings VARCHAR(23), 
	specific VARCHAR(12), 
	wall_surf_treatment VARCHAR(12), 
	filter_type VARCHAR(39) NOT NULL, 
	vis_media VARCHAR(63), 
	aero_struc TEXT, 
	wall_finish_mat VARCHAR(20), 
	ceil_water_mold TEXT, 
	door_water_mold TEXT, 
	shad_dev_water_mold TEXT, 
	wall_water_mold TEXT, 
	window_water_mold TEXT, 
	heat_cool_type VARCHAR(17) NOT NULL, 
	train_line VARCHAR(6), 
	window_type VARCHAR(23), 
	train_stat_loc VARCHAR(26), 
	season_use VARCHAR(6), 
	mech_struc VARCHAR(9), 
	floor_finish_mat VARCHAR(23), 
	space_typ_state TEXT NOT NULL, 
	built_struc_set TEXT, 
	building_setting VARCHAR(8) NOT NULL, 
	ceil_struc TEXT, 
	avg_occup FLOAT, 
	occup_density_samp FLOAT NOT NULL, 
	samp_time_out FLOAT, 
	surf_moisture_ph FLOAT, 
	typ_occup_density FLOAT NOT NULL, 
	abs_air_humidity TEXT NOT NULL, 
	air_temp TEXT NOT NULL, 
	amount_light TEXT, 
	avg_dew_point TEXT, 
	avg_temp TEXT, 
	built_struc_age TEXT, 
	carb_dioxide TEXT NOT NULL, 
	ceil_area TEXT, 
	ceil_thermal_mass TEXT, 
	dew_point TEXT, 
	door_size TEXT, 
	exp_duct TEXT, 
	floor_age TEXT, 
	floor_area TEXT, 
	floor_thermal_mass TEXT, 
	height_carper_fiber TEXT, 
	inside_lux TEXT, 
	rel_air_humidity TEXT NOT NULL, 
	rel_humidity_out TEXT, 
	room_air_exch_rate TEXT, 
	specific_humidity TEXT, 
	surf_humidity TEXT, 
	surf_moisture TEXT, 
	surf_temp TEXT, 
	temp_out TEXT, 
	wall_area TEXT, 
	wall_height TEXT, 
	wall_thermal_mass TEXT, 
	water_feat_size TEXT, 
	window_size TEXT, 
	bathroom_count INTEGER, 
	bedroom_count INTEGER, 
	cool_syst_id INTEGER, 
	elevator INTEGER, 
	escalator INTEGER, 
	exp_pipe INTEGER, 
	ext_door INTEGER, 
	floor_count INTEGER, 
	freq_cook INTEGER, 
	hall_count INTEGER, 
	heat_system_id INTEGER, 
	max_occup INTEGER, 
	number_pets INTEGER, 
	number_plants INTEGER, 
	number_resident INTEGER, 
	occup_samp INTEGER NOT NULL, 
	room_count INTEGER, 
	room_moist_dam_hist INTEGER, 
	room_occup INTEGER, 
	room_window_count INTEGER, 
	samp_room_id INTEGER, 
	window_open_freq INTEGER, 
	room_door_dist TEXT, 
	room_net_area TEXT, 
	room_vol TEXT, 
	room_dim TEXT, 
	address TEXT, 
	season TEXT, 
	built_struc_type TEXT, 
	room_architec_elem TEXT, 
	samp_sort_meth TEXT, 
	shading_device_mat TEXT, 
	ventilation_type TEXT NOT NULL, 
	organism_count TEXT NOT NULL, 
	adj_room TEXT, 
	room_door_share TEXT, 
	room_hallway TEXT, 
	room_wall_share TEXT, 
	date_last_rain DATE, 
	last_clean DATE, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, freq_clean, samp_floor, surf_material, gender_restroom, door_mat, handidness, room_connected, room_type, occup_document, shading_device_type, floor_struc, indoor_space, door_type_wood, window_cover, window_vert_pos, build_docs, arch_struc, indoor_surf, furniture, pres_animal_insect, ceil_type, window_mat, samp_weather, window_status, door_type_metal, door_move, door_type, heat_sys_deliv_meth, room_loc, substructure_type, wall_texture, ceil_texture, door_cond, shading_device_cond, window_cond, ceil_finish_mat, surf_air_cont, rel_samp_loc, train_stop_loc, shading_device_loc, water_feat_type, wall_const_type, fireplace_type, door_direct, window_horiz_pos, door_comp_type, floor_water_mold, weekday, light_type, ceil_cond, floor_cond, int_wall_cond, room_condt, room_samp_pos, quad_pos, door_loc, heat_deliv_loc, wall_loc, window_loc, ext_wall_orient, ext_window_orient, build_occup_type, drawings, specific, wall_surf_treatment, filter_type, vis_media, aero_struc, wall_finish_mat, ceil_water_mold, door_water_mold, shad_dev_water_mold, wall_water_mold, window_water_mold, heat_cool_type, train_line, window_type, train_stat_loc, season_use, mech_struc, floor_finish_mat, space_typ_state, built_struc_set, building_setting, ceil_struc, avg_occup, occup_density_samp, samp_time_out, surf_moisture_ph, typ_occup_density, abs_air_humidity, air_temp, amount_light, avg_dew_point, avg_temp, built_struc_age, carb_dioxide, ceil_area, ceil_thermal_mass, dew_point, door_size, exp_duct, floor_age, floor_area, floor_thermal_mass, height_carper_fiber, inside_lux, rel_air_humidity, rel_humidity_out, room_air_exch_rate, specific_humidity, surf_humidity, surf_moisture, surf_temp, temp_out, wall_area, wall_height, wall_thermal_mass, water_feat_size, window_size, bathroom_count, bedroom_count, cool_syst_id, elevator, escalator, exp_pipe, ext_door, floor_count, freq_cook, hall_count, heat_system_id, max_occup, number_pets, number_plants, number_resident, occup_samp, room_count, room_moist_dam_hist, room_occup, room_window_count, samp_room_id, window_open_freq, room_door_dist, room_net_area, room_vol, room_dim, address, season, built_struc_type, room_architec_elem, samp_sort_meth, shading_device_mat, ventilation_type, organism_count, adj_room, room_door_share, room_hallway, room_wall_share, date_last_rain, last_clean, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, samp_vol_we_dna_ext, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "built_environment_MIGS_virus" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	freq_clean VARCHAR(9), 
	samp_floor VARCHAR(9), 
	surf_material VARCHAR(15), 
	gender_restroom VARCHAR(15), 
	door_mat VARCHAR(22), 
	handidness VARCHAR(16), 
	room_connected VARCHAR(15), 
	room_type VARCHAR(29), 
	occup_document VARCHAR(15), 
	shading_device_type VARCHAR(21), 
	floor_struc VARCHAR(14), 
	indoor_space VARCHAR(11) NOT NULL, 
	door_type_wood VARCHAR(24), 
	window_cover VARCHAR(8), 
	window_vert_pos VARCHAR(6), 
	build_docs VARCHAR(28), 
	arch_struc VARCHAR(8), 
	indoor_surf VARCHAR(11), 
	furniture VARCHAR(7), 
	pres_animal_insect VARCHAR(15), 
	ceil_type VARCHAR(13), 
	window_mat VARCHAR(10), 
	samp_weather VARCHAR(9), 
	window_status TEXT, 
	door_type_metal VARCHAR(16), 
	door_move VARCHAR(15), 
	door_type VARCHAR(9), 
	heat_sys_deliv_meth TEXT, 
	room_loc VARCHAR(13), 
	substructure_type VARCHAR(13), 
	wall_texture VARCHAR(16), 
	ceil_texture VARCHAR(16), 
	door_cond VARCHAR(12), 
	shading_device_cond VARCHAR(12), 
	window_cond VARCHAR(12), 
	ceil_finish_mat VARCHAR(29), 
	surf_air_cont VARCHAR(26), 
	rel_samp_loc VARCHAR(13), 
	train_stop_loc VARCHAR(8), 
	shading_device_loc TEXT, 
	water_feat_type VARCHAR(16), 
	wall_const_type VARCHAR(23), 
	fireplace_type TEXT, 
	door_direct VARCHAR(8), 
	window_horiz_pos VARCHAR(6), 
	door_comp_type VARCHAR(13), 
	floor_water_mold VARCHAR(26), 
	weekday VARCHAR(9), 
	light_type VARCHAR(18) NOT NULL, 
	ceil_cond VARCHAR(12), 
	floor_cond VARCHAR(12), 
	int_wall_cond VARCHAR(12), 
	room_condt VARCHAR(28), 
	room_samp_pos VARCHAR(16), 
	quad_pos VARCHAR(10), 
	door_loc VARCHAR(5), 
	heat_deliv_loc VARCHAR(5), 
	wall_loc VARCHAR(5), 
	window_loc VARCHAR(5), 
	ext_wall_orient VARCHAR(9), 
	ext_window_orient VARCHAR(9), 
	build_occup_type VARCHAR(14) NOT NULL, 
	drawings VARCHAR(23), 
	specific VARCHAR(12), 
	wall_surf_treatment VARCHAR(12), 
	filter_type VARCHAR(39) NOT NULL, 
	vis_media VARCHAR(63), 
	aero_struc TEXT, 
	wall_finish_mat VARCHAR(20), 
	ceil_water_mold TEXT, 
	door_water_mold TEXT, 
	shad_dev_water_mold TEXT, 
	wall_water_mold TEXT, 
	window_water_mold TEXT, 
	heat_cool_type VARCHAR(17) NOT NULL, 
	train_line VARCHAR(6), 
	window_type VARCHAR(23), 
	train_stat_loc VARCHAR(26), 
	season_use VARCHAR(6), 
	mech_struc VARCHAR(9), 
	floor_finish_mat VARCHAR(23), 
	space_typ_state TEXT NOT NULL, 
	built_struc_set TEXT, 
	building_setting VARCHAR(8) NOT NULL, 
	ceil_struc TEXT, 
	avg_occup FLOAT, 
	occup_density_samp FLOAT NOT NULL, 
	samp_time_out FLOAT, 
	surf_moisture_ph FLOAT, 
	typ_occup_density FLOAT NOT NULL, 
	abs_air_humidity TEXT NOT NULL, 
	air_temp TEXT NOT NULL, 
	amount_light TEXT, 
	avg_dew_point TEXT, 
	avg_temp TEXT, 
	built_struc_age TEXT, 
	carb_dioxide TEXT NOT NULL, 
	ceil_area TEXT, 
	ceil_thermal_mass TEXT, 
	dew_point TEXT, 
	door_size TEXT, 
	exp_duct TEXT, 
	floor_age TEXT, 
	floor_area TEXT, 
	floor_thermal_mass TEXT, 
	height_carper_fiber TEXT, 
	inside_lux TEXT, 
	rel_air_humidity TEXT NOT NULL, 
	rel_humidity_out TEXT, 
	room_air_exch_rate TEXT, 
	specific_humidity TEXT, 
	surf_humidity TEXT, 
	surf_moisture TEXT, 
	surf_temp TEXT, 
	temp_out TEXT, 
	wall_area TEXT, 
	wall_height TEXT, 
	wall_thermal_mass TEXT, 
	water_feat_size TEXT, 
	window_size TEXT, 
	bathroom_count INTEGER, 
	bedroom_count INTEGER, 
	cool_syst_id INTEGER, 
	elevator INTEGER, 
	escalator INTEGER, 
	exp_pipe INTEGER, 
	ext_door INTEGER, 
	floor_count INTEGER, 
	freq_cook INTEGER, 
	hall_count INTEGER, 
	heat_system_id INTEGER, 
	max_occup INTEGER, 
	number_pets INTEGER, 
	number_plants INTEGER, 
	number_resident INTEGER, 
	occup_samp INTEGER NOT NULL, 
	room_count INTEGER, 
	room_moist_dam_hist INTEGER, 
	room_occup INTEGER, 
	room_window_count INTEGER, 
	samp_room_id INTEGER, 
	window_open_freq INTEGER, 
	room_door_dist TEXT, 
	room_net_area TEXT, 
	room_vol TEXT, 
	room_dim TEXT, 
	address TEXT, 
	season TEXT, 
	built_struc_type TEXT, 
	room_architec_elem TEXT, 
	samp_sort_meth TEXT, 
	shading_device_mat TEXT, 
	ventilation_type TEXT NOT NULL, 
	organism_count TEXT NOT NULL, 
	adj_room TEXT, 
	room_door_share TEXT, 
	room_hallway TEXT, 
	room_wall_share TEXT, 
	date_last_rain DATE, 
	last_clean DATE, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, freq_clean, samp_floor, surf_material, gender_restroom, door_mat, handidness, room_connected, room_type, occup_document, shading_device_type, floor_struc, indoor_space, door_type_wood, window_cover, window_vert_pos, build_docs, arch_struc, indoor_surf, furniture, pres_animal_insect, ceil_type, window_mat, samp_weather, window_status, door_type_metal, door_move, door_type, heat_sys_deliv_meth, room_loc, substructure_type, wall_texture, ceil_texture, door_cond, shading_device_cond, window_cond, ceil_finish_mat, surf_air_cont, rel_samp_loc, train_stop_loc, shading_device_loc, water_feat_type, wall_const_type, fireplace_type, door_direct, window_horiz_pos, door_comp_type, floor_water_mold, weekday, light_type, ceil_cond, floor_cond, int_wall_cond, room_condt, room_samp_pos, quad_pos, door_loc, heat_deliv_loc, wall_loc, window_loc, ext_wall_orient, ext_window_orient, build_occup_type, drawings, specific, wall_surf_treatment, filter_type, vis_media, aero_struc, wall_finish_mat, ceil_water_mold, door_water_mold, shad_dev_water_mold, wall_water_mold, window_water_mold, heat_cool_type, train_line, window_type, train_stat_loc, season_use, mech_struc, floor_finish_mat, space_typ_state, built_struc_set, building_setting, ceil_struc, avg_occup, occup_density_samp, samp_time_out, surf_moisture_ph, typ_occup_density, abs_air_humidity, air_temp, amount_light, avg_dew_point, avg_temp, built_struc_age, carb_dioxide, ceil_area, ceil_thermal_mass, dew_point, door_size, exp_duct, floor_age, floor_area, floor_thermal_mass, height_carper_fiber, inside_lux, rel_air_humidity, rel_humidity_out, room_air_exch_rate, specific_humidity, surf_humidity, surf_moisture, surf_temp, temp_out, wall_area, wall_height, wall_thermal_mass, water_feat_size, window_size, bathroom_count, bedroom_count, cool_syst_id, elevator, escalator, exp_pipe, ext_door, floor_count, freq_cook, hall_count, heat_system_id, max_occup, number_pets, number_plants, number_resident, occup_samp, room_count, room_moist_dam_hist, room_occup, room_window_count, samp_room_id, window_open_freq, room_door_dist, room_net_area, room_vol, room_dim, address, season, built_struc_type, room_architec_elem, samp_sort_meth, shading_device_mat, ventilation_type, organism_count, adj_room, room_door_share, room_hallway, room_wall_share, date_last_rain, last_clean, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, samp_vol_we_dna_ext, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "built_environment_MIMAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	freq_clean VARCHAR(9), 
	samp_floor VARCHAR(9), 
	surf_material VARCHAR(15), 
	gender_restroom VARCHAR(15), 
	door_mat VARCHAR(22), 
	handidness VARCHAR(16), 
	room_connected VARCHAR(15), 
	room_type VARCHAR(29), 
	occup_document VARCHAR(15), 
	shading_device_type VARCHAR(21), 
	floor_struc VARCHAR(14), 
	indoor_space VARCHAR(11) NOT NULL, 
	door_type_wood VARCHAR(24), 
	window_cover VARCHAR(8), 
	window_vert_pos VARCHAR(6), 
	build_docs VARCHAR(28), 
	arch_struc VARCHAR(8), 
	indoor_surf VARCHAR(11), 
	furniture VARCHAR(7), 
	pres_animal_insect VARCHAR(15), 
	ceil_type VARCHAR(13), 
	window_mat VARCHAR(10), 
	samp_weather VARCHAR(9), 
	window_status TEXT, 
	door_type_metal VARCHAR(16), 
	door_move VARCHAR(15), 
	door_type VARCHAR(9), 
	heat_sys_deliv_meth TEXT, 
	room_loc VARCHAR(13), 
	substructure_type VARCHAR(13), 
	wall_texture VARCHAR(16), 
	ceil_texture VARCHAR(16), 
	door_cond VARCHAR(12), 
	shading_device_cond VARCHAR(12), 
	window_cond VARCHAR(12), 
	ceil_finish_mat VARCHAR(29), 
	surf_air_cont VARCHAR(26), 
	rel_samp_loc VARCHAR(13), 
	train_stop_loc VARCHAR(8), 
	shading_device_loc TEXT, 
	water_feat_type VARCHAR(16), 
	wall_const_type VARCHAR(23), 
	fireplace_type TEXT, 
	door_direct VARCHAR(8), 
	window_horiz_pos VARCHAR(6), 
	door_comp_type VARCHAR(13), 
	floor_water_mold VARCHAR(26), 
	weekday VARCHAR(9), 
	light_type VARCHAR(18) NOT NULL, 
	ceil_cond VARCHAR(12), 
	floor_cond VARCHAR(12), 
	int_wall_cond VARCHAR(12), 
	room_condt VARCHAR(28), 
	room_samp_pos VARCHAR(16), 
	quad_pos VARCHAR(10), 
	door_loc VARCHAR(5), 
	heat_deliv_loc VARCHAR(5), 
	wall_loc VARCHAR(5), 
	window_loc VARCHAR(5), 
	ext_wall_orient VARCHAR(9), 
	ext_window_orient VARCHAR(9), 
	build_occup_type VARCHAR(14) NOT NULL, 
	drawings VARCHAR(23), 
	specific VARCHAR(12), 
	wall_surf_treatment VARCHAR(12), 
	filter_type VARCHAR(39) NOT NULL, 
	vis_media VARCHAR(63), 
	aero_struc TEXT, 
	wall_finish_mat VARCHAR(20), 
	ceil_water_mold TEXT, 
	door_water_mold TEXT, 
	shad_dev_water_mold TEXT, 
	wall_water_mold TEXT, 
	window_water_mold TEXT, 
	heat_cool_type VARCHAR(17) NOT NULL, 
	train_line VARCHAR(6), 
	window_type VARCHAR(23), 
	train_stat_loc VARCHAR(26), 
	season_use VARCHAR(6), 
	mech_struc VARCHAR(9), 
	floor_finish_mat VARCHAR(23), 
	space_typ_state TEXT NOT NULL, 
	built_struc_set TEXT, 
	building_setting VARCHAR(8) NOT NULL, 
	ceil_struc TEXT, 
	avg_occup FLOAT, 
	occup_density_samp FLOAT NOT NULL, 
	samp_time_out FLOAT, 
	surf_moisture_ph FLOAT, 
	typ_occup_density FLOAT NOT NULL, 
	abs_air_humidity TEXT NOT NULL, 
	air_temp TEXT NOT NULL, 
	amount_light TEXT, 
	avg_dew_point TEXT, 
	avg_temp TEXT, 
	built_struc_age TEXT, 
	carb_dioxide TEXT NOT NULL, 
	ceil_area TEXT, 
	ceil_thermal_mass TEXT, 
	dew_point TEXT, 
	door_size TEXT, 
	exp_duct TEXT, 
	floor_age TEXT, 
	floor_area TEXT, 
	floor_thermal_mass TEXT, 
	height_carper_fiber TEXT, 
	inside_lux TEXT, 
	rel_air_humidity TEXT NOT NULL, 
	rel_humidity_out TEXT, 
	room_air_exch_rate TEXT, 
	specific_humidity TEXT, 
	surf_humidity TEXT, 
	surf_moisture TEXT, 
	surf_temp TEXT, 
	temp_out TEXT, 
	wall_area TEXT, 
	wall_height TEXT, 
	wall_thermal_mass TEXT, 
	water_feat_size TEXT, 
	window_size TEXT, 
	bathroom_count INTEGER, 
	bedroom_count INTEGER, 
	cool_syst_id INTEGER, 
	elevator INTEGER, 
	escalator INTEGER, 
	exp_pipe INTEGER, 
	ext_door INTEGER, 
	floor_count INTEGER, 
	freq_cook INTEGER, 
	hall_count INTEGER, 
	heat_system_id INTEGER, 
	max_occup INTEGER, 
	number_pets INTEGER, 
	number_plants INTEGER, 
	number_resident INTEGER, 
	occup_samp INTEGER NOT NULL, 
	room_count INTEGER, 
	room_moist_dam_hist INTEGER, 
	room_occup INTEGER, 
	room_window_count INTEGER, 
	samp_room_id INTEGER, 
	window_open_freq INTEGER, 
	room_door_dist TEXT, 
	room_net_area TEXT, 
	room_vol TEXT, 
	room_dim TEXT, 
	address TEXT, 
	season TEXT, 
	built_struc_type TEXT, 
	room_architec_elem TEXT, 
	samp_sort_meth TEXT, 
	shading_device_mat TEXT, 
	ventilation_type TEXT NOT NULL, 
	organism_count TEXT NOT NULL, 
	adj_room TEXT, 
	room_door_share TEXT, 
	room_hallway TEXT, 
	room_wall_share TEXT, 
	date_last_rain DATE, 
	last_clean DATE, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, freq_clean, samp_floor, surf_material, gender_restroom, door_mat, handidness, room_connected, room_type, occup_document, shading_device_type, floor_struc, indoor_space, door_type_wood, window_cover, window_vert_pos, build_docs, arch_struc, indoor_surf, furniture, pres_animal_insect, ceil_type, window_mat, samp_weather, window_status, door_type_metal, door_move, door_type, heat_sys_deliv_meth, room_loc, substructure_type, wall_texture, ceil_texture, door_cond, shading_device_cond, window_cond, ceil_finish_mat, surf_air_cont, rel_samp_loc, train_stop_loc, shading_device_loc, water_feat_type, wall_const_type, fireplace_type, door_direct, window_horiz_pos, door_comp_type, floor_water_mold, weekday, light_type, ceil_cond, floor_cond, int_wall_cond, room_condt, room_samp_pos, quad_pos, door_loc, heat_deliv_loc, wall_loc, window_loc, ext_wall_orient, ext_window_orient, build_occup_type, drawings, specific, wall_surf_treatment, filter_type, vis_media, aero_struc, wall_finish_mat, ceil_water_mold, door_water_mold, shad_dev_water_mold, wall_water_mold, window_water_mold, heat_cool_type, train_line, window_type, train_stat_loc, season_use, mech_struc, floor_finish_mat, space_typ_state, built_struc_set, building_setting, ceil_struc, avg_occup, occup_density_samp, samp_time_out, surf_moisture_ph, typ_occup_density, abs_air_humidity, air_temp, amount_light, avg_dew_point, avg_temp, built_struc_age, carb_dioxide, ceil_area, ceil_thermal_mass, dew_point, door_size, exp_duct, floor_age, floor_area, floor_thermal_mass, height_carper_fiber, inside_lux, rel_air_humidity, rel_humidity_out, room_air_exch_rate, specific_humidity, surf_humidity, surf_moisture, surf_temp, temp_out, wall_area, wall_height, wall_thermal_mass, water_feat_size, window_size, bathroom_count, bedroom_count, cool_syst_id, elevator, escalator, exp_pipe, ext_door, floor_count, freq_cook, hall_count, heat_system_id, max_occup, number_pets, number_plants, number_resident, occup_samp, room_count, room_moist_dam_hist, room_occup, room_window_count, samp_room_id, window_open_freq, room_door_dist, room_net_area, room_vol, room_dim, address, season, built_struc_type, room_architec_elem, samp_sort_meth, shading_device_mat, ventilation_type, organism_count, adj_room, room_door_share, room_hallway, room_wall_share, date_last_rain, last_clean, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, samp_vol_we_dna_ext, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "built_environment_MIMARKS_specimen" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	freq_clean VARCHAR(9), 
	samp_floor VARCHAR(9), 
	surf_material VARCHAR(15), 
	gender_restroom VARCHAR(15), 
	door_mat VARCHAR(22), 
	handidness VARCHAR(16), 
	room_connected VARCHAR(15), 
	room_type VARCHAR(29), 
	occup_document VARCHAR(15), 
	shading_device_type VARCHAR(21), 
	floor_struc VARCHAR(14), 
	indoor_space VARCHAR(11) NOT NULL, 
	door_type_wood VARCHAR(24), 
	window_cover VARCHAR(8), 
	window_vert_pos VARCHAR(6), 
	build_docs VARCHAR(28), 
	arch_struc VARCHAR(8), 
	indoor_surf VARCHAR(11), 
	furniture VARCHAR(7), 
	pres_animal_insect VARCHAR(15), 
	ceil_type VARCHAR(13), 
	window_mat VARCHAR(10), 
	samp_weather VARCHAR(9), 
	window_status TEXT, 
	door_type_metal VARCHAR(16), 
	door_move VARCHAR(15), 
	door_type VARCHAR(9), 
	heat_sys_deliv_meth TEXT, 
	room_loc VARCHAR(13), 
	substructure_type VARCHAR(13), 
	wall_texture VARCHAR(16), 
	ceil_texture VARCHAR(16), 
	door_cond VARCHAR(12), 
	shading_device_cond VARCHAR(12), 
	window_cond VARCHAR(12), 
	ceil_finish_mat VARCHAR(29), 
	surf_air_cont VARCHAR(26), 
	rel_samp_loc VARCHAR(13), 
	train_stop_loc VARCHAR(8), 
	shading_device_loc TEXT, 
	water_feat_type VARCHAR(16), 
	wall_const_type VARCHAR(23), 
	fireplace_type TEXT, 
	door_direct VARCHAR(8), 
	window_horiz_pos VARCHAR(6), 
	door_comp_type VARCHAR(13), 
	floor_water_mold VARCHAR(26), 
	weekday VARCHAR(9), 
	light_type VARCHAR(18) NOT NULL, 
	ceil_cond VARCHAR(12), 
	floor_cond VARCHAR(12), 
	int_wall_cond VARCHAR(12), 
	room_condt VARCHAR(28), 
	room_samp_pos VARCHAR(16), 
	quad_pos VARCHAR(10), 
	door_loc VARCHAR(5), 
	heat_deliv_loc VARCHAR(5), 
	wall_loc VARCHAR(5), 
	window_loc VARCHAR(5), 
	ext_wall_orient VARCHAR(9), 
	ext_window_orient VARCHAR(9), 
	build_occup_type VARCHAR(14) NOT NULL, 
	drawings VARCHAR(23), 
	specific VARCHAR(12), 
	wall_surf_treatment VARCHAR(12), 
	filter_type VARCHAR(39) NOT NULL, 
	vis_media VARCHAR(63), 
	aero_struc TEXT, 
	wall_finish_mat VARCHAR(20), 
	ceil_water_mold TEXT, 
	door_water_mold TEXT, 
	shad_dev_water_mold TEXT, 
	wall_water_mold TEXT, 
	window_water_mold TEXT, 
	heat_cool_type VARCHAR(17) NOT NULL, 
	train_line VARCHAR(6), 
	window_type VARCHAR(23), 
	train_stat_loc VARCHAR(26), 
	season_use VARCHAR(6), 
	mech_struc VARCHAR(9), 
	floor_finish_mat VARCHAR(23), 
	space_typ_state TEXT NOT NULL, 
	built_struc_set TEXT, 
	building_setting VARCHAR(8) NOT NULL, 
	ceil_struc TEXT, 
	avg_occup FLOAT, 
	occup_density_samp FLOAT NOT NULL, 
	samp_time_out FLOAT, 
	surf_moisture_ph FLOAT, 
	typ_occup_density FLOAT NOT NULL, 
	abs_air_humidity TEXT NOT NULL, 
	air_temp TEXT NOT NULL, 
	amount_light TEXT, 
	avg_dew_point TEXT, 
	avg_temp TEXT, 
	built_struc_age TEXT, 
	carb_dioxide TEXT NOT NULL, 
	ceil_area TEXT, 
	ceil_thermal_mass TEXT, 
	dew_point TEXT, 
	door_size TEXT, 
	exp_duct TEXT, 
	floor_age TEXT, 
	floor_area TEXT, 
	floor_thermal_mass TEXT, 
	height_carper_fiber TEXT, 
	inside_lux TEXT, 
	rel_air_humidity TEXT NOT NULL, 
	rel_humidity_out TEXT, 
	room_air_exch_rate TEXT, 
	specific_humidity TEXT, 
	surf_humidity TEXT, 
	surf_moisture TEXT, 
	surf_temp TEXT, 
	temp_out TEXT, 
	wall_area TEXT, 
	wall_height TEXT, 
	wall_thermal_mass TEXT, 
	water_feat_size TEXT, 
	window_size TEXT, 
	bathroom_count INTEGER, 
	bedroom_count INTEGER, 
	cool_syst_id INTEGER, 
	elevator INTEGER, 
	escalator INTEGER, 
	exp_pipe INTEGER, 
	ext_door INTEGER, 
	floor_count INTEGER, 
	freq_cook INTEGER, 
	hall_count INTEGER, 
	heat_system_id INTEGER, 
	max_occup INTEGER, 
	number_pets INTEGER, 
	number_plants INTEGER, 
	number_resident INTEGER, 
	occup_samp INTEGER NOT NULL, 
	room_count INTEGER, 
	room_moist_dam_hist INTEGER, 
	room_occup INTEGER, 
	room_window_count INTEGER, 
	samp_room_id INTEGER, 
	window_open_freq INTEGER, 
	room_door_dist TEXT, 
	room_net_area TEXT, 
	room_vol TEXT, 
	room_dim TEXT, 
	address TEXT, 
	season TEXT, 
	built_struc_type TEXT, 
	room_architec_elem TEXT, 
	samp_sort_meth TEXT, 
	shading_device_mat TEXT, 
	ventilation_type TEXT NOT NULL, 
	organism_count TEXT NOT NULL, 
	adj_room TEXT, 
	room_door_share TEXT, 
	room_hallway TEXT, 
	room_wall_share TEXT, 
	date_last_rain DATE, 
	last_clean DATE, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, freq_clean, samp_floor, surf_material, gender_restroom, door_mat, handidness, room_connected, room_type, occup_document, shading_device_type, floor_struc, indoor_space, door_type_wood, window_cover, window_vert_pos, build_docs, arch_struc, indoor_surf, furniture, pres_animal_insect, ceil_type, window_mat, samp_weather, window_status, door_type_metal, door_move, door_type, heat_sys_deliv_meth, room_loc, substructure_type, wall_texture, ceil_texture, door_cond, shading_device_cond, window_cond, ceil_finish_mat, surf_air_cont, rel_samp_loc, train_stop_loc, shading_device_loc, water_feat_type, wall_const_type, fireplace_type, door_direct, window_horiz_pos, door_comp_type, floor_water_mold, weekday, light_type, ceil_cond, floor_cond, int_wall_cond, room_condt, room_samp_pos, quad_pos, door_loc, heat_deliv_loc, wall_loc, window_loc, ext_wall_orient, ext_window_orient, build_occup_type, drawings, specific, wall_surf_treatment, filter_type, vis_media, aero_struc, wall_finish_mat, ceil_water_mold, door_water_mold, shad_dev_water_mold, wall_water_mold, window_water_mold, heat_cool_type, train_line, window_type, train_stat_loc, season_use, mech_struc, floor_finish_mat, space_typ_state, built_struc_set, building_setting, ceil_struc, avg_occup, occup_density_samp, samp_time_out, surf_moisture_ph, typ_occup_density, abs_air_humidity, air_temp, amount_light, avg_dew_point, avg_temp, built_struc_age, carb_dioxide, ceil_area, ceil_thermal_mass, dew_point, door_size, exp_duct, floor_age, floor_area, floor_thermal_mass, height_carper_fiber, inside_lux, rel_air_humidity, rel_humidity_out, room_air_exch_rate, specific_humidity, surf_humidity, surf_moisture, surf_temp, temp_out, wall_area, wall_height, wall_thermal_mass, water_feat_size, window_size, bathroom_count, bedroom_count, cool_syst_id, elevator, escalator, exp_pipe, ext_door, floor_count, freq_cook, hall_count, heat_system_id, max_occup, number_pets, number_plants, number_resident, occup_samp, room_count, room_moist_dam_hist, room_occup, room_window_count, samp_room_id, window_open_freq, room_door_dist, room_net_area, room_vol, room_dim, address, season, built_struc_type, room_architec_elem, samp_sort_meth, shading_device_mat, ventilation_type, organism_count, adj_room, room_door_share, room_hallway, room_wall_share, date_last_rain, last_clean, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, samp_vol_we_dna_ext, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "built_environment_MIMARKS_survey" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	freq_clean VARCHAR(9), 
	samp_floor VARCHAR(9), 
	surf_material VARCHAR(15), 
	gender_restroom VARCHAR(15), 
	door_mat VARCHAR(22), 
	handidness VARCHAR(16), 
	room_connected VARCHAR(15), 
	room_type VARCHAR(29), 
	occup_document VARCHAR(15), 
	shading_device_type VARCHAR(21), 
	floor_struc VARCHAR(14), 
	indoor_space VARCHAR(11) NOT NULL, 
	door_type_wood VARCHAR(24), 
	window_cover VARCHAR(8), 
	window_vert_pos VARCHAR(6), 
	build_docs VARCHAR(28), 
	arch_struc VARCHAR(8), 
	indoor_surf VARCHAR(11), 
	furniture VARCHAR(7), 
	pres_animal_insect VARCHAR(15), 
	ceil_type VARCHAR(13), 
	window_mat VARCHAR(10), 
	samp_weather VARCHAR(9), 
	window_status TEXT, 
	door_type_metal VARCHAR(16), 
	door_move VARCHAR(15), 
	door_type VARCHAR(9), 
	heat_sys_deliv_meth TEXT, 
	room_loc VARCHAR(13), 
	substructure_type VARCHAR(13), 
	wall_texture VARCHAR(16), 
	ceil_texture VARCHAR(16), 
	door_cond VARCHAR(12), 
	shading_device_cond VARCHAR(12), 
	window_cond VARCHAR(12), 
	ceil_finish_mat VARCHAR(29), 
	surf_air_cont VARCHAR(26), 
	rel_samp_loc VARCHAR(13), 
	train_stop_loc VARCHAR(8), 
	shading_device_loc TEXT, 
	water_feat_type VARCHAR(16), 
	wall_const_type VARCHAR(23), 
	fireplace_type TEXT, 
	door_direct VARCHAR(8), 
	window_horiz_pos VARCHAR(6), 
	door_comp_type VARCHAR(13), 
	floor_water_mold VARCHAR(26), 
	weekday VARCHAR(9), 
	light_type VARCHAR(18) NOT NULL, 
	ceil_cond VARCHAR(12), 
	floor_cond VARCHAR(12), 
	int_wall_cond VARCHAR(12), 
	room_condt VARCHAR(28), 
	room_samp_pos VARCHAR(16), 
	quad_pos VARCHAR(10), 
	door_loc VARCHAR(5), 
	heat_deliv_loc VARCHAR(5), 
	wall_loc VARCHAR(5), 
	window_loc VARCHAR(5), 
	ext_wall_orient VARCHAR(9), 
	ext_window_orient VARCHAR(9), 
	build_occup_type VARCHAR(14) NOT NULL, 
	drawings VARCHAR(23), 
	specific VARCHAR(12), 
	wall_surf_treatment VARCHAR(12), 
	filter_type VARCHAR(39) NOT NULL, 
	vis_media VARCHAR(63), 
	aero_struc TEXT, 
	wall_finish_mat VARCHAR(20), 
	ceil_water_mold TEXT, 
	door_water_mold TEXT, 
	shad_dev_water_mold TEXT, 
	wall_water_mold TEXT, 
	window_water_mold TEXT, 
	heat_cool_type VARCHAR(17) NOT NULL, 
	train_line VARCHAR(6), 
	window_type VARCHAR(23), 
	train_stat_loc VARCHAR(26), 
	season_use VARCHAR(6), 
	mech_struc VARCHAR(9), 
	floor_finish_mat VARCHAR(23), 
	space_typ_state TEXT NOT NULL, 
	built_struc_set TEXT, 
	building_setting VARCHAR(8) NOT NULL, 
	ceil_struc TEXT, 
	avg_occup FLOAT, 
	occup_density_samp FLOAT NOT NULL, 
	samp_time_out FLOAT, 
	surf_moisture_ph FLOAT, 
	typ_occup_density FLOAT NOT NULL, 
	abs_air_humidity TEXT NOT NULL, 
	air_temp TEXT NOT NULL, 
	amount_light TEXT, 
	avg_dew_point TEXT, 
	avg_temp TEXT, 
	built_struc_age TEXT, 
	carb_dioxide TEXT NOT NULL, 
	ceil_area TEXT, 
	ceil_thermal_mass TEXT, 
	dew_point TEXT, 
	door_size TEXT, 
	exp_duct TEXT, 
	floor_age TEXT, 
	floor_area TEXT, 
	floor_thermal_mass TEXT, 
	height_carper_fiber TEXT, 
	inside_lux TEXT, 
	rel_air_humidity TEXT NOT NULL, 
	rel_humidity_out TEXT, 
	room_air_exch_rate TEXT, 
	specific_humidity TEXT, 
	surf_humidity TEXT, 
	surf_moisture TEXT, 
	surf_temp TEXT, 
	temp_out TEXT, 
	wall_area TEXT, 
	wall_height TEXT, 
	wall_thermal_mass TEXT, 
	water_feat_size TEXT, 
	window_size TEXT, 
	bathroom_count INTEGER, 
	bedroom_count INTEGER, 
	cool_syst_id INTEGER, 
	elevator INTEGER, 
	escalator INTEGER, 
	exp_pipe INTEGER, 
	ext_door INTEGER, 
	floor_count INTEGER, 
	freq_cook INTEGER, 
	hall_count INTEGER, 
	heat_system_id INTEGER, 
	max_occup INTEGER, 
	number_pets INTEGER, 
	number_plants INTEGER, 
	number_resident INTEGER, 
	occup_samp INTEGER NOT NULL, 
	room_count INTEGER, 
	room_moist_dam_hist INTEGER, 
	room_occup INTEGER, 
	room_window_count INTEGER, 
	samp_room_id INTEGER, 
	window_open_freq INTEGER, 
	room_door_dist TEXT, 
	room_net_area TEXT, 
	room_vol TEXT, 
	room_dim TEXT, 
	address TEXT, 
	season TEXT, 
	built_struc_type TEXT, 
	room_architec_elem TEXT, 
	samp_sort_meth TEXT, 
	shading_device_mat TEXT, 
	ventilation_type TEXT NOT NULL, 
	organism_count TEXT NOT NULL, 
	adj_room TEXT, 
	room_door_share TEXT, 
	room_hallway TEXT, 
	room_wall_share TEXT, 
	date_last_rain DATE, 
	last_clean DATE, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, freq_clean, samp_floor, surf_material, gender_restroom, door_mat, handidness, room_connected, room_type, occup_document, shading_device_type, floor_struc, indoor_space, door_type_wood, window_cover, window_vert_pos, build_docs, arch_struc, indoor_surf, furniture, pres_animal_insect, ceil_type, window_mat, samp_weather, window_status, door_type_metal, door_move, door_type, heat_sys_deliv_meth, room_loc, substructure_type, wall_texture, ceil_texture, door_cond, shading_device_cond, window_cond, ceil_finish_mat, surf_air_cont, rel_samp_loc, train_stop_loc, shading_device_loc, water_feat_type, wall_const_type, fireplace_type, door_direct, window_horiz_pos, door_comp_type, floor_water_mold, weekday, light_type, ceil_cond, floor_cond, int_wall_cond, room_condt, room_samp_pos, quad_pos, door_loc, heat_deliv_loc, wall_loc, window_loc, ext_wall_orient, ext_window_orient, build_occup_type, drawings, specific, wall_surf_treatment, filter_type, vis_media, aero_struc, wall_finish_mat, ceil_water_mold, door_water_mold, shad_dev_water_mold, wall_water_mold, window_water_mold, heat_cool_type, train_line, window_type, train_stat_loc, season_use, mech_struc, floor_finish_mat, space_typ_state, built_struc_set, building_setting, ceil_struc, avg_occup, occup_density_samp, samp_time_out, surf_moisture_ph, typ_occup_density, abs_air_humidity, air_temp, amount_light, avg_dew_point, avg_temp, built_struc_age, carb_dioxide, ceil_area, ceil_thermal_mass, dew_point, door_size, exp_duct, floor_age, floor_area, floor_thermal_mass, height_carper_fiber, inside_lux, rel_air_humidity, rel_humidity_out, room_air_exch_rate, specific_humidity, surf_humidity, surf_moisture, surf_temp, temp_out, wall_area, wall_height, wall_thermal_mass, water_feat_size, window_size, bathroom_count, bedroom_count, cool_syst_id, elevator, escalator, exp_pipe, ext_door, floor_count, freq_cook, hall_count, heat_system_id, max_occup, number_pets, number_plants, number_resident, occup_samp, room_count, room_moist_dam_hist, room_occup, room_window_count, samp_room_id, window_open_freq, room_door_dist, room_net_area, room_vol, room_dim, address, season, built_struc_type, room_architec_elem, samp_sort_meth, shading_device_mat, ventilation_type, organism_count, adj_room, room_door_share, room_hallway, room_wall_share, date_last_rain, last_clean, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, samp_vol_we_dna_ext, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "built_environment_MIMS" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	freq_clean VARCHAR(9), 
	samp_floor VARCHAR(9), 
	surf_material VARCHAR(15), 
	gender_restroom VARCHAR(15), 
	door_mat VARCHAR(22), 
	handidness VARCHAR(16), 
	room_connected VARCHAR(15), 
	room_type VARCHAR(29), 
	occup_document VARCHAR(15), 
	shading_device_type VARCHAR(21), 
	floor_struc VARCHAR(14), 
	indoor_space VARCHAR(11) NOT NULL, 
	door_type_wood VARCHAR(24), 
	window_cover VARCHAR(8), 
	window_vert_pos VARCHAR(6), 
	build_docs VARCHAR(28), 
	arch_struc VARCHAR(8), 
	indoor_surf VARCHAR(11), 
	furniture VARCHAR(7), 
	pres_animal_insect VARCHAR(15), 
	ceil_type VARCHAR(13), 
	window_mat VARCHAR(10), 
	samp_weather VARCHAR(9), 
	window_status TEXT, 
	door_type_metal VARCHAR(16), 
	door_move VARCHAR(15), 
	door_type VARCHAR(9), 
	heat_sys_deliv_meth TEXT, 
	room_loc VARCHAR(13), 
	substructure_type VARCHAR(13), 
	wall_texture VARCHAR(16), 
	ceil_texture VARCHAR(16), 
	door_cond VARCHAR(12), 
	shading_device_cond VARCHAR(12), 
	window_cond VARCHAR(12), 
	ceil_finish_mat VARCHAR(29), 
	surf_air_cont VARCHAR(26), 
	rel_samp_loc VARCHAR(13), 
	train_stop_loc VARCHAR(8), 
	shading_device_loc TEXT, 
	water_feat_type VARCHAR(16), 
	wall_const_type VARCHAR(23), 
	fireplace_type TEXT, 
	door_direct VARCHAR(8), 
	window_horiz_pos VARCHAR(6), 
	door_comp_type VARCHAR(13), 
	floor_water_mold VARCHAR(26), 
	weekday VARCHAR(9), 
	light_type VARCHAR(18) NOT NULL, 
	ceil_cond VARCHAR(12), 
	floor_cond VARCHAR(12), 
	int_wall_cond VARCHAR(12), 
	room_condt VARCHAR(28), 
	room_samp_pos VARCHAR(16), 
	quad_pos VARCHAR(10), 
	door_loc VARCHAR(5), 
	heat_deliv_loc VARCHAR(5), 
	wall_loc VARCHAR(5), 
	window_loc VARCHAR(5), 
	ext_wall_orient VARCHAR(9), 
	ext_window_orient VARCHAR(9), 
	build_occup_type VARCHAR(14) NOT NULL, 
	drawings VARCHAR(23), 
	specific VARCHAR(12), 
	wall_surf_treatment VARCHAR(12), 
	filter_type VARCHAR(39) NOT NULL, 
	vis_media VARCHAR(63), 
	aero_struc TEXT, 
	wall_finish_mat VARCHAR(20), 
	ceil_water_mold TEXT, 
	door_water_mold TEXT, 
	shad_dev_water_mold TEXT, 
	wall_water_mold TEXT, 
	window_water_mold TEXT, 
	heat_cool_type VARCHAR(17) NOT NULL, 
	train_line VARCHAR(6), 
	window_type VARCHAR(23), 
	train_stat_loc VARCHAR(26), 
	season_use VARCHAR(6), 
	mech_struc VARCHAR(9), 
	floor_finish_mat VARCHAR(23), 
	space_typ_state TEXT NOT NULL, 
	built_struc_set TEXT, 
	building_setting VARCHAR(8) NOT NULL, 
	ceil_struc TEXT, 
	avg_occup FLOAT, 
	occup_density_samp FLOAT NOT NULL, 
	samp_time_out FLOAT, 
	surf_moisture_ph FLOAT, 
	typ_occup_density FLOAT NOT NULL, 
	abs_air_humidity TEXT NOT NULL, 
	air_temp TEXT NOT NULL, 
	amount_light TEXT, 
	avg_dew_point TEXT, 
	avg_temp TEXT, 
	built_struc_age TEXT, 
	carb_dioxide TEXT NOT NULL, 
	ceil_area TEXT, 
	ceil_thermal_mass TEXT, 
	dew_point TEXT, 
	door_size TEXT, 
	exp_duct TEXT, 
	floor_age TEXT, 
	floor_area TEXT, 
	floor_thermal_mass TEXT, 
	height_carper_fiber TEXT, 
	inside_lux TEXT, 
	rel_air_humidity TEXT NOT NULL, 
	rel_humidity_out TEXT, 
	room_air_exch_rate TEXT, 
	specific_humidity TEXT, 
	surf_humidity TEXT, 
	surf_moisture TEXT, 
	surf_temp TEXT, 
	temp_out TEXT, 
	wall_area TEXT, 
	wall_height TEXT, 
	wall_thermal_mass TEXT, 
	water_feat_size TEXT, 
	window_size TEXT, 
	bathroom_count INTEGER, 
	bedroom_count INTEGER, 
	cool_syst_id INTEGER, 
	elevator INTEGER, 
	escalator INTEGER, 
	exp_pipe INTEGER, 
	ext_door INTEGER, 
	floor_count INTEGER, 
	freq_cook INTEGER, 
	hall_count INTEGER, 
	heat_system_id INTEGER, 
	max_occup INTEGER, 
	number_pets INTEGER, 
	number_plants INTEGER, 
	number_resident INTEGER, 
	occup_samp INTEGER NOT NULL, 
	room_count INTEGER, 
	room_moist_dam_hist INTEGER, 
	room_occup INTEGER, 
	room_window_count INTEGER, 
	samp_room_id INTEGER, 
	window_open_freq INTEGER, 
	room_door_dist TEXT, 
	room_net_area TEXT, 
	room_vol TEXT, 
	room_dim TEXT, 
	address TEXT, 
	season TEXT, 
	built_struc_type TEXT, 
	room_architec_elem TEXT, 
	samp_sort_meth TEXT, 
	shading_device_mat TEXT, 
	ventilation_type TEXT NOT NULL, 
	organism_count TEXT NOT NULL, 
	adj_room TEXT, 
	room_door_share TEXT, 
	room_hallway TEXT, 
	room_wall_share TEXT, 
	date_last_rain DATE, 
	last_clean DATE, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, freq_clean, samp_floor, surf_material, gender_restroom, door_mat, handidness, room_connected, room_type, occup_document, shading_device_type, floor_struc, indoor_space, door_type_wood, window_cover, window_vert_pos, build_docs, arch_struc, indoor_surf, furniture, pres_animal_insect, ceil_type, window_mat, samp_weather, window_status, door_type_metal, door_move, door_type, heat_sys_deliv_meth, room_loc, substructure_type, wall_texture, ceil_texture, door_cond, shading_device_cond, window_cond, ceil_finish_mat, surf_air_cont, rel_samp_loc, train_stop_loc, shading_device_loc, water_feat_type, wall_const_type, fireplace_type, door_direct, window_horiz_pos, door_comp_type, floor_water_mold, weekday, light_type, ceil_cond, floor_cond, int_wall_cond, room_condt, room_samp_pos, quad_pos, door_loc, heat_deliv_loc, wall_loc, window_loc, ext_wall_orient, ext_window_orient, build_occup_type, drawings, specific, wall_surf_treatment, filter_type, vis_media, aero_struc, wall_finish_mat, ceil_water_mold, door_water_mold, shad_dev_water_mold, wall_water_mold, window_water_mold, heat_cool_type, train_line, window_type, train_stat_loc, season_use, mech_struc, floor_finish_mat, space_typ_state, built_struc_set, building_setting, ceil_struc, avg_occup, occup_density_samp, samp_time_out, surf_moisture_ph, typ_occup_density, abs_air_humidity, air_temp, amount_light, avg_dew_point, avg_temp, built_struc_age, carb_dioxide, ceil_area, ceil_thermal_mass, dew_point, door_size, exp_duct, floor_age, floor_area, floor_thermal_mass, height_carper_fiber, inside_lux, rel_air_humidity, rel_humidity_out, room_air_exch_rate, specific_humidity, surf_humidity, surf_moisture, surf_temp, temp_out, wall_area, wall_height, wall_thermal_mass, water_feat_size, window_size, bathroom_count, bedroom_count, cool_syst_id, elevator, escalator, exp_pipe, ext_door, floor_count, freq_cook, hall_count, heat_system_id, max_occup, number_pets, number_plants, number_resident, occup_samp, room_count, room_moist_dam_hist, room_occup, room_window_count, samp_room_id, window_open_freq, room_door_dist, room_net_area, room_vol, room_dim, address, season, built_struc_type, room_architec_elem, samp_sort_meth, shading_device_mat, ventilation_type, organism_count, adj_room, room_door_share, room_hallway, room_wall_share, date_last_rain, last_clean, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, samp_vol_we_dna_ext, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "built_environment_MISAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	freq_clean VARCHAR(9), 
	samp_floor VARCHAR(9), 
	surf_material VARCHAR(15), 
	gender_restroom VARCHAR(15), 
	door_mat VARCHAR(22), 
	handidness VARCHAR(16), 
	room_connected VARCHAR(15), 
	room_type VARCHAR(29), 
	occup_document VARCHAR(15), 
	shading_device_type VARCHAR(21), 
	floor_struc VARCHAR(14), 
	indoor_space VARCHAR(11) NOT NULL, 
	door_type_wood VARCHAR(24), 
	window_cover VARCHAR(8), 
	window_vert_pos VARCHAR(6), 
	build_docs VARCHAR(28), 
	arch_struc VARCHAR(8), 
	indoor_surf VARCHAR(11), 
	furniture VARCHAR(7), 
	pres_animal_insect VARCHAR(15), 
	ceil_type VARCHAR(13), 
	window_mat VARCHAR(10), 
	samp_weather VARCHAR(9), 
	window_status TEXT, 
	door_type_metal VARCHAR(16), 
	door_move VARCHAR(15), 
	door_type VARCHAR(9), 
	heat_sys_deliv_meth TEXT, 
	room_loc VARCHAR(13), 
	substructure_type VARCHAR(13), 
	wall_texture VARCHAR(16), 
	ceil_texture VARCHAR(16), 
	door_cond VARCHAR(12), 
	shading_device_cond VARCHAR(12), 
	window_cond VARCHAR(12), 
	ceil_finish_mat VARCHAR(29), 
	surf_air_cont VARCHAR(26), 
	rel_samp_loc VARCHAR(13), 
	train_stop_loc VARCHAR(8), 
	shading_device_loc TEXT, 
	water_feat_type VARCHAR(16), 
	wall_const_type VARCHAR(23), 
	fireplace_type TEXT, 
	door_direct VARCHAR(8), 
	window_horiz_pos VARCHAR(6), 
	door_comp_type VARCHAR(13), 
	floor_water_mold VARCHAR(26), 
	weekday VARCHAR(9), 
	light_type VARCHAR(18) NOT NULL, 
	ceil_cond VARCHAR(12), 
	floor_cond VARCHAR(12), 
	int_wall_cond VARCHAR(12), 
	room_condt VARCHAR(28), 
	room_samp_pos VARCHAR(16), 
	quad_pos VARCHAR(10), 
	door_loc VARCHAR(5), 
	heat_deliv_loc VARCHAR(5), 
	wall_loc VARCHAR(5), 
	window_loc VARCHAR(5), 
	ext_wall_orient VARCHAR(9), 
	ext_window_orient VARCHAR(9), 
	build_occup_type VARCHAR(14) NOT NULL, 
	drawings VARCHAR(23), 
	specific VARCHAR(12), 
	wall_surf_treatment VARCHAR(12), 
	filter_type VARCHAR(39) NOT NULL, 
	vis_media VARCHAR(63), 
	aero_struc TEXT, 
	wall_finish_mat VARCHAR(20), 
	ceil_water_mold TEXT, 
	door_water_mold TEXT, 
	shad_dev_water_mold TEXT, 
	wall_water_mold TEXT, 
	window_water_mold TEXT, 
	heat_cool_type VARCHAR(17) NOT NULL, 
	train_line VARCHAR(6), 
	window_type VARCHAR(23), 
	train_stat_loc VARCHAR(26), 
	season_use VARCHAR(6), 
	mech_struc VARCHAR(9), 
	floor_finish_mat VARCHAR(23), 
	space_typ_state TEXT NOT NULL, 
	built_struc_set TEXT, 
	building_setting VARCHAR(8) NOT NULL, 
	ceil_struc TEXT, 
	avg_occup FLOAT, 
	occup_density_samp FLOAT NOT NULL, 
	samp_time_out FLOAT, 
	surf_moisture_ph FLOAT, 
	typ_occup_density FLOAT NOT NULL, 
	abs_air_humidity TEXT NOT NULL, 
	air_temp TEXT NOT NULL, 
	amount_light TEXT, 
	avg_dew_point TEXT, 
	avg_temp TEXT, 
	built_struc_age TEXT, 
	carb_dioxide TEXT NOT NULL, 
	ceil_area TEXT, 
	ceil_thermal_mass TEXT, 
	dew_point TEXT, 
	door_size TEXT, 
	exp_duct TEXT, 
	floor_age TEXT, 
	floor_area TEXT, 
	floor_thermal_mass TEXT, 
	height_carper_fiber TEXT, 
	inside_lux TEXT, 
	rel_air_humidity TEXT NOT NULL, 
	rel_humidity_out TEXT, 
	room_air_exch_rate TEXT, 
	specific_humidity TEXT, 
	surf_humidity TEXT, 
	surf_moisture TEXT, 
	surf_temp TEXT, 
	temp_out TEXT, 
	wall_area TEXT, 
	wall_height TEXT, 
	wall_thermal_mass TEXT, 
	water_feat_size TEXT, 
	window_size TEXT, 
	bathroom_count INTEGER, 
	bedroom_count INTEGER, 
	cool_syst_id INTEGER, 
	elevator INTEGER, 
	escalator INTEGER, 
	exp_pipe INTEGER, 
	ext_door INTEGER, 
	floor_count INTEGER, 
	freq_cook INTEGER, 
	hall_count INTEGER, 
	heat_system_id INTEGER, 
	max_occup INTEGER, 
	number_pets INTEGER, 
	number_plants INTEGER, 
	number_resident INTEGER, 
	occup_samp INTEGER NOT NULL, 
	room_count INTEGER, 
	room_moist_dam_hist INTEGER, 
	room_occup INTEGER, 
	room_window_count INTEGER, 
	samp_room_id INTEGER, 
	window_open_freq INTEGER, 
	room_door_dist TEXT, 
	room_net_area TEXT, 
	room_vol TEXT, 
	room_dim TEXT, 
	address TEXT, 
	season TEXT, 
	built_struc_type TEXT, 
	room_architec_elem TEXT, 
	samp_sort_meth TEXT, 
	shading_device_mat TEXT, 
	ventilation_type TEXT NOT NULL, 
	organism_count TEXT NOT NULL, 
	adj_room TEXT, 
	room_door_share TEXT, 
	room_hallway TEXT, 
	room_wall_share TEXT, 
	date_last_rain DATE, 
	last_clean DATE, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, freq_clean, samp_floor, surf_material, gender_restroom, door_mat, handidness, room_connected, room_type, occup_document, shading_device_type, floor_struc, indoor_space, door_type_wood, window_cover, window_vert_pos, build_docs, arch_struc, indoor_surf, furniture, pres_animal_insect, ceil_type, window_mat, samp_weather, window_status, door_type_metal, door_move, door_type, heat_sys_deliv_meth, room_loc, substructure_type, wall_texture, ceil_texture, door_cond, shading_device_cond, window_cond, ceil_finish_mat, surf_air_cont, rel_samp_loc, train_stop_loc, shading_device_loc, water_feat_type, wall_const_type, fireplace_type, door_direct, window_horiz_pos, door_comp_type, floor_water_mold, weekday, light_type, ceil_cond, floor_cond, int_wall_cond, room_condt, room_samp_pos, quad_pos, door_loc, heat_deliv_loc, wall_loc, window_loc, ext_wall_orient, ext_window_orient, build_occup_type, drawings, specific, wall_surf_treatment, filter_type, vis_media, aero_struc, wall_finish_mat, ceil_water_mold, door_water_mold, shad_dev_water_mold, wall_water_mold, window_water_mold, heat_cool_type, train_line, window_type, train_stat_loc, season_use, mech_struc, floor_finish_mat, space_typ_state, built_struc_set, building_setting, ceil_struc, avg_occup, occup_density_samp, samp_time_out, surf_moisture_ph, typ_occup_density, abs_air_humidity, air_temp, amount_light, avg_dew_point, avg_temp, built_struc_age, carb_dioxide, ceil_area, ceil_thermal_mass, dew_point, door_size, exp_duct, floor_age, floor_area, floor_thermal_mass, height_carper_fiber, inside_lux, rel_air_humidity, rel_humidity_out, room_air_exch_rate, specific_humidity, surf_humidity, surf_moisture, surf_temp, temp_out, wall_area, wall_height, wall_thermal_mass, water_feat_size, window_size, bathroom_count, bedroom_count, cool_syst_id, elevator, escalator, exp_pipe, ext_door, floor_count, freq_cook, hall_count, heat_system_id, max_occup, number_pets, number_plants, number_resident, occup_samp, room_count, room_moist_dam_hist, room_occup, room_window_count, samp_room_id, window_open_freq, room_door_dist, room_net_area, room_vol, room_dim, address, season, built_struc_type, room_architec_elem, samp_sort_meth, shading_device_mat, ventilation_type, organism_count, adj_room, room_door_share, room_hallway, room_wall_share, date_last_rain, last_clean, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, samp_vol_we_dna_ext, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "built_environment_MIUVIG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	freq_clean VARCHAR(9), 
	samp_floor VARCHAR(9), 
	surf_material VARCHAR(15), 
	gender_restroom VARCHAR(15), 
	door_mat VARCHAR(22), 
	handidness VARCHAR(16), 
	room_connected VARCHAR(15), 
	room_type VARCHAR(29), 
	occup_document VARCHAR(15), 
	shading_device_type VARCHAR(21), 
	floor_struc VARCHAR(14), 
	indoor_space VARCHAR(11) NOT NULL, 
	door_type_wood VARCHAR(24), 
	window_cover VARCHAR(8), 
	window_vert_pos VARCHAR(6), 
	build_docs VARCHAR(28), 
	arch_struc VARCHAR(8), 
	indoor_surf VARCHAR(11), 
	furniture VARCHAR(7), 
	pres_animal_insect VARCHAR(15), 
	ceil_type VARCHAR(13), 
	window_mat VARCHAR(10), 
	samp_weather VARCHAR(9), 
	window_status TEXT, 
	door_type_metal VARCHAR(16), 
	door_move VARCHAR(15), 
	door_type VARCHAR(9), 
	heat_sys_deliv_meth TEXT, 
	room_loc VARCHAR(13), 
	substructure_type VARCHAR(13), 
	wall_texture VARCHAR(16), 
	ceil_texture VARCHAR(16), 
	door_cond VARCHAR(12), 
	shading_device_cond VARCHAR(12), 
	window_cond VARCHAR(12), 
	ceil_finish_mat VARCHAR(29), 
	surf_air_cont VARCHAR(26), 
	rel_samp_loc VARCHAR(13), 
	train_stop_loc VARCHAR(8), 
	shading_device_loc TEXT, 
	water_feat_type VARCHAR(16), 
	wall_const_type VARCHAR(23), 
	fireplace_type TEXT, 
	door_direct VARCHAR(8), 
	window_horiz_pos VARCHAR(6), 
	door_comp_type VARCHAR(13), 
	floor_water_mold VARCHAR(26), 
	weekday VARCHAR(9), 
	light_type VARCHAR(18) NOT NULL, 
	ceil_cond VARCHAR(12), 
	floor_cond VARCHAR(12), 
	int_wall_cond VARCHAR(12), 
	room_condt VARCHAR(28), 
	room_samp_pos VARCHAR(16), 
	quad_pos VARCHAR(10), 
	door_loc VARCHAR(5), 
	heat_deliv_loc VARCHAR(5), 
	wall_loc VARCHAR(5), 
	window_loc VARCHAR(5), 
	ext_wall_orient VARCHAR(9), 
	ext_window_orient VARCHAR(9), 
	build_occup_type VARCHAR(14) NOT NULL, 
	drawings VARCHAR(23), 
	specific VARCHAR(12), 
	wall_surf_treatment VARCHAR(12), 
	filter_type VARCHAR(39) NOT NULL, 
	vis_media VARCHAR(63), 
	aero_struc TEXT, 
	wall_finish_mat VARCHAR(20), 
	ceil_water_mold TEXT, 
	door_water_mold TEXT, 
	shad_dev_water_mold TEXT, 
	wall_water_mold TEXT, 
	window_water_mold TEXT, 
	heat_cool_type VARCHAR(17) NOT NULL, 
	train_line VARCHAR(6), 
	window_type VARCHAR(23), 
	train_stat_loc VARCHAR(26), 
	season_use VARCHAR(6), 
	mech_struc VARCHAR(9), 
	floor_finish_mat VARCHAR(23), 
	space_typ_state TEXT NOT NULL, 
	built_struc_set TEXT, 
	building_setting VARCHAR(8) NOT NULL, 
	ceil_struc TEXT, 
	avg_occup FLOAT, 
	occup_density_samp FLOAT NOT NULL, 
	samp_time_out FLOAT, 
	surf_moisture_ph FLOAT, 
	typ_occup_density FLOAT NOT NULL, 
	abs_air_humidity TEXT NOT NULL, 
	air_temp TEXT NOT NULL, 
	amount_light TEXT, 
	avg_dew_point TEXT, 
	avg_temp TEXT, 
	built_struc_age TEXT, 
	carb_dioxide TEXT NOT NULL, 
	ceil_area TEXT, 
	ceil_thermal_mass TEXT, 
	dew_point TEXT, 
	door_size TEXT, 
	exp_duct TEXT, 
	floor_age TEXT, 
	floor_area TEXT, 
	floor_thermal_mass TEXT, 
	height_carper_fiber TEXT, 
	inside_lux TEXT, 
	rel_air_humidity TEXT NOT NULL, 
	rel_humidity_out TEXT, 
	room_air_exch_rate TEXT, 
	specific_humidity TEXT, 
	surf_humidity TEXT, 
	surf_moisture TEXT, 
	surf_temp TEXT, 
	temp_out TEXT, 
	wall_area TEXT, 
	wall_height TEXT, 
	wall_thermal_mass TEXT, 
	water_feat_size TEXT, 
	window_size TEXT, 
	bathroom_count INTEGER, 
	bedroom_count INTEGER, 
	cool_syst_id INTEGER, 
	elevator INTEGER, 
	escalator INTEGER, 
	exp_pipe INTEGER, 
	ext_door INTEGER, 
	floor_count INTEGER, 
	freq_cook INTEGER, 
	hall_count INTEGER, 
	heat_system_id INTEGER, 
	max_occup INTEGER, 
	number_pets INTEGER, 
	number_plants INTEGER, 
	number_resident INTEGER, 
	occup_samp INTEGER NOT NULL, 
	room_count INTEGER, 
	room_moist_dam_hist INTEGER, 
	room_occup INTEGER, 
	room_window_count INTEGER, 
	samp_room_id INTEGER, 
	window_open_freq INTEGER, 
	room_door_dist TEXT, 
	room_net_area TEXT, 
	room_vol TEXT, 
	room_dim TEXT, 
	address TEXT, 
	season TEXT, 
	built_struc_type TEXT, 
	room_architec_elem TEXT, 
	samp_sort_meth TEXT, 
	shading_device_mat TEXT, 
	ventilation_type TEXT NOT NULL, 
	organism_count TEXT NOT NULL, 
	adj_room TEXT, 
	room_door_share TEXT, 
	room_hallway TEXT, 
	room_wall_share TEXT, 
	date_last_rain DATE, 
	last_clean DATE, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, freq_clean, samp_floor, surf_material, gender_restroom, door_mat, handidness, room_connected, room_type, occup_document, shading_device_type, floor_struc, indoor_space, door_type_wood, window_cover, window_vert_pos, build_docs, arch_struc, indoor_surf, furniture, pres_animal_insect, ceil_type, window_mat, samp_weather, window_status, door_type_metal, door_move, door_type, heat_sys_deliv_meth, room_loc, substructure_type, wall_texture, ceil_texture, door_cond, shading_device_cond, window_cond, ceil_finish_mat, surf_air_cont, rel_samp_loc, train_stop_loc, shading_device_loc, water_feat_type, wall_const_type, fireplace_type, door_direct, window_horiz_pos, door_comp_type, floor_water_mold, weekday, light_type, ceil_cond, floor_cond, int_wall_cond, room_condt, room_samp_pos, quad_pos, door_loc, heat_deliv_loc, wall_loc, window_loc, ext_wall_orient, ext_window_orient, build_occup_type, drawings, specific, wall_surf_treatment, filter_type, vis_media, aero_struc, wall_finish_mat, ceil_water_mold, door_water_mold, shad_dev_water_mold, wall_water_mold, window_water_mold, heat_cool_type, train_line, window_type, train_stat_loc, season_use, mech_struc, floor_finish_mat, space_typ_state, built_struc_set, building_setting, ceil_struc, avg_occup, occup_density_samp, samp_time_out, surf_moisture_ph, typ_occup_density, abs_air_humidity, air_temp, amount_light, avg_dew_point, avg_temp, built_struc_age, carb_dioxide, ceil_area, ceil_thermal_mass, dew_point, door_size, exp_duct, floor_age, floor_area, floor_thermal_mass, height_carper_fiber, inside_lux, rel_air_humidity, rel_humidity_out, room_air_exch_rate, specific_humidity, surf_humidity, surf_moisture, surf_temp, temp_out, wall_area, wall_height, wall_thermal_mass, water_feat_size, window_size, bathroom_count, bedroom_count, cool_syst_id, elevator, escalator, exp_pipe, ext_door, floor_count, freq_cook, hall_count, heat_system_id, max_occup, number_pets, number_plants, number_resident, occup_samp, room_count, room_moist_dam_hist, room_occup, room_window_count, samp_room_id, window_open_freq, room_door_dist, room_net_area, room_vol, room_dim, address, season, built_struc_type, room_architec_elem, samp_sort_meth, shading_device_mat, ventilation_type, organism_count, adj_room, room_door_share, room_hallway, room_wall_share, date_last_rain, last_clean, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, samp_vol_we_dna_ext, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE core (
	samp_name TEXT, 
	samp_taxon_id TEXT, 
	project_name TEXT, 
	experimental_factor TEXT, 
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (samp_name, samp_taxon_id, project_name, experimental_factor, lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, neg_cont_type, pos_cont_type, env_broad_scale, env_local_scale, env_medium, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, samp_vol_we_dna_ext, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE food_animal_and_animal_feed (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48) NOT NULL, 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	animal_sex VARCHAR(16), 
	food_trace_list VARCHAR(45), 
	animal_body_cond VARCHAR(17), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	samp_stor_dur TEXT, 
	timepoint FLOAT, 
	animal_am_dur TEXT, 
	animal_am_freq TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	food_source_age TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	animal_group_size INTEGER, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT NOT NULL, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT NOT NULL, 
	animal_feed_equip TEXT, 
	animal_housing TEXT, 
	food_prod TEXT NOT NULL, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	animal_am TEXT, 
	animal_am_route TEXT, 
	animal_am_use TEXT, 
	animal_diet TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	food_additive TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, animal_sex, food_trace_list, animal_body_cond, pool_dna_extracts, cons_food_stor_dur, samp_stor_dur, timepoint, animal_am_dur, animal_am_freq, bacterial_density, food_pack_capacity, food_source_age, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, animal_group_size, cons_qty_purchased, lot_number, intended_consumer, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, animal_feed_equip, animal_housing, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, animal_am, animal_am_route, animal_am_use, animal_diet, food_prod_synonym, "IFSAC_category", library_prep_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, perturbation, coll_site_geo_feat, food_additive, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name)
);

CREATE TABLE "food_animal_and_animal_feed_MIGS_bacteria" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48) NOT NULL, 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	animal_sex VARCHAR(16), 
	food_trace_list VARCHAR(45), 
	animal_body_cond VARCHAR(17), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	samp_stor_dur TEXT, 
	timepoint FLOAT, 
	animal_am_dur TEXT, 
	animal_am_freq TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	food_source_age TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	animal_group_size INTEGER, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT NOT NULL, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT NOT NULL, 
	animal_feed_equip TEXT, 
	animal_housing TEXT, 
	food_prod TEXT NOT NULL, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	animal_am TEXT, 
	animal_am_route TEXT, 
	animal_am_use TEXT, 
	animal_diet TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	food_additive TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, animal_sex, food_trace_list, animal_body_cond, pool_dna_extracts, cons_food_stor_dur, samp_stor_dur, timepoint, animal_am_dur, animal_am_freq, bacterial_density, food_pack_capacity, food_source_age, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, animal_group_size, cons_qty_purchased, lot_number, intended_consumer, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, animal_feed_equip, animal_housing, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, animal_am, animal_am_route, animal_am_use, animal_diet, food_prod_synonym, "IFSAC_category", library_prep_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, perturbation, coll_site_geo_feat, food_additive, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_animal_and_animal_feed_MIGS_eukaryote" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48) NOT NULL, 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	animal_sex VARCHAR(16), 
	food_trace_list VARCHAR(45), 
	animal_body_cond VARCHAR(17), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	samp_stor_dur TEXT, 
	timepoint FLOAT, 
	animal_am_dur TEXT, 
	animal_am_freq TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	food_source_age TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	animal_group_size INTEGER, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT NOT NULL, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT NOT NULL, 
	animal_feed_equip TEXT, 
	animal_housing TEXT, 
	food_prod TEXT NOT NULL, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	animal_am TEXT, 
	animal_am_route TEXT, 
	animal_am_use TEXT, 
	animal_diet TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	food_additive TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, animal_sex, food_trace_list, animal_body_cond, pool_dna_extracts, cons_food_stor_dur, samp_stor_dur, timepoint, animal_am_dur, animal_am_freq, bacterial_density, food_pack_capacity, food_source_age, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, animal_group_size, cons_qty_purchased, lot_number, intended_consumer, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, animal_feed_equip, animal_housing, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, animal_am, animal_am_route, animal_am_use, animal_diet, food_prod_synonym, "IFSAC_category", library_prep_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, perturbation, coll_site_geo_feat, food_additive, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_animal_and_animal_feed_MIGS_org" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48) NOT NULL, 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	animal_sex VARCHAR(16), 
	food_trace_list VARCHAR(45), 
	animal_body_cond VARCHAR(17), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	samp_stor_dur TEXT, 
	timepoint FLOAT, 
	animal_am_dur TEXT, 
	animal_am_freq TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	food_source_age TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	animal_group_size INTEGER, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT NOT NULL, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT NOT NULL, 
	animal_feed_equip TEXT, 
	animal_housing TEXT, 
	food_prod TEXT NOT NULL, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	animal_am TEXT, 
	animal_am_route TEXT, 
	animal_am_use TEXT, 
	animal_diet TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	food_additive TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, animal_sex, food_trace_list, animal_body_cond, pool_dna_extracts, cons_food_stor_dur, samp_stor_dur, timepoint, animal_am_dur, animal_am_freq, bacterial_density, food_pack_capacity, food_source_age, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, animal_group_size, cons_qty_purchased, lot_number, intended_consumer, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, animal_feed_equip, animal_housing, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, animal_am, animal_am_route, animal_am_use, animal_diet, food_prod_synonym, "IFSAC_category", library_prep_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, perturbation, coll_site_geo_feat, food_additive, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_animal_and_animal_feed_MIGS_plant" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48) NOT NULL, 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	animal_sex VARCHAR(16), 
	food_trace_list VARCHAR(45), 
	animal_body_cond VARCHAR(17), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	samp_stor_dur TEXT, 
	timepoint FLOAT, 
	animal_am_dur TEXT, 
	animal_am_freq TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	food_source_age TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	animal_group_size INTEGER, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT NOT NULL, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT NOT NULL, 
	animal_feed_equip TEXT, 
	animal_housing TEXT, 
	food_prod TEXT NOT NULL, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	animal_am TEXT, 
	animal_am_route TEXT, 
	animal_am_use TEXT, 
	animal_diet TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	food_additive TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, animal_sex, food_trace_list, animal_body_cond, pool_dna_extracts, cons_food_stor_dur, samp_stor_dur, timepoint, animal_am_dur, animal_am_freq, bacterial_density, food_pack_capacity, food_source_age, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, animal_group_size, cons_qty_purchased, lot_number, intended_consumer, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, animal_feed_equip, animal_housing, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, animal_am, animal_am_route, animal_am_use, animal_diet, food_prod_synonym, "IFSAC_category", library_prep_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, perturbation, coll_site_geo_feat, food_additive, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_animal_and_animal_feed_MIGS_virus" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48) NOT NULL, 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	animal_sex VARCHAR(16), 
	food_trace_list VARCHAR(45), 
	animal_body_cond VARCHAR(17), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	samp_stor_dur TEXT, 
	timepoint FLOAT, 
	animal_am_dur TEXT, 
	animal_am_freq TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	food_source_age TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	animal_group_size INTEGER, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT NOT NULL, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT NOT NULL, 
	animal_feed_equip TEXT, 
	animal_housing TEXT, 
	food_prod TEXT NOT NULL, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	animal_am TEXT, 
	animal_am_route TEXT, 
	animal_am_use TEXT, 
	animal_diet TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	food_additive TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, animal_sex, food_trace_list, animal_body_cond, pool_dna_extracts, cons_food_stor_dur, samp_stor_dur, timepoint, animal_am_dur, animal_am_freq, bacterial_density, food_pack_capacity, food_source_age, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, animal_group_size, cons_qty_purchased, lot_number, intended_consumer, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, animal_feed_equip, animal_housing, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, animal_am, animal_am_route, animal_am_use, animal_diet, food_prod_synonym, "IFSAC_category", library_prep_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, perturbation, coll_site_geo_feat, food_additive, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_animal_and_animal_feed_MIMAG" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48) NOT NULL, 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	animal_sex VARCHAR(16), 
	food_trace_list VARCHAR(45), 
	animal_body_cond VARCHAR(17), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	samp_stor_dur TEXT, 
	timepoint FLOAT, 
	animal_am_dur TEXT, 
	animal_am_freq TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	food_source_age TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	animal_group_size INTEGER, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT NOT NULL, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT NOT NULL, 
	animal_feed_equip TEXT, 
	animal_housing TEXT, 
	food_prod TEXT NOT NULL, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	animal_am TEXT, 
	animal_am_route TEXT, 
	animal_am_use TEXT, 
	animal_diet TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	food_additive TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, animal_sex, food_trace_list, animal_body_cond, pool_dna_extracts, cons_food_stor_dur, samp_stor_dur, timepoint, animal_am_dur, animal_am_freq, bacterial_density, food_pack_capacity, food_source_age, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, animal_group_size, cons_qty_purchased, lot_number, intended_consumer, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, animal_feed_equip, animal_housing, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, animal_am, animal_am_route, animal_am_use, animal_diet, food_prod_synonym, "IFSAC_category", library_prep_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, perturbation, coll_site_geo_feat, food_additive, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_animal_and_animal_feed_MIMARKS_specimen" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48) NOT NULL, 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	animal_sex VARCHAR(16), 
	food_trace_list VARCHAR(45), 
	animal_body_cond VARCHAR(17), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	samp_stor_dur TEXT, 
	timepoint FLOAT, 
	animal_am_dur TEXT, 
	animal_am_freq TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	food_source_age TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	animal_group_size INTEGER, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT NOT NULL, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT NOT NULL, 
	animal_feed_equip TEXT, 
	animal_housing TEXT, 
	food_prod TEXT NOT NULL, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	animal_am TEXT, 
	animal_am_route TEXT, 
	animal_am_use TEXT, 
	animal_diet TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	food_additive TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, animal_sex, food_trace_list, animal_body_cond, pool_dna_extracts, cons_food_stor_dur, samp_stor_dur, timepoint, animal_am_dur, animal_am_freq, bacterial_density, food_pack_capacity, food_source_age, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, animal_group_size, cons_qty_purchased, lot_number, intended_consumer, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, animal_feed_equip, animal_housing, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, animal_am, animal_am_route, animal_am_use, animal_diet, food_prod_synonym, "IFSAC_category", library_prep_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, perturbation, coll_site_geo_feat, food_additive, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_animal_and_animal_feed_MIMARKS_survey" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48) NOT NULL, 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	animal_sex VARCHAR(16), 
	food_trace_list VARCHAR(45), 
	animal_body_cond VARCHAR(17), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	samp_stor_dur TEXT, 
	timepoint FLOAT, 
	animal_am_dur TEXT, 
	animal_am_freq TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	food_source_age TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	animal_group_size INTEGER, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT NOT NULL, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT NOT NULL, 
	animal_feed_equip TEXT, 
	animal_housing TEXT, 
	food_prod TEXT NOT NULL, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	animal_am TEXT, 
	animal_am_route TEXT, 
	animal_am_use TEXT, 
	animal_diet TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	food_additive TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, animal_sex, food_trace_list, animal_body_cond, pool_dna_extracts, cons_food_stor_dur, samp_stor_dur, timepoint, animal_am_dur, animal_am_freq, bacterial_density, food_pack_capacity, food_source_age, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, animal_group_size, cons_qty_purchased, lot_number, intended_consumer, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, animal_feed_equip, animal_housing, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, animal_am, animal_am_route, animal_am_use, animal_diet, food_prod_synonym, "IFSAC_category", library_prep_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, perturbation, coll_site_geo_feat, food_additive, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_animal_and_animal_feed_MIMS" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48) NOT NULL, 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	animal_sex VARCHAR(16), 
	food_trace_list VARCHAR(45), 
	animal_body_cond VARCHAR(17), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	samp_stor_dur TEXT, 
	timepoint FLOAT, 
	animal_am_dur TEXT, 
	animal_am_freq TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	food_source_age TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	animal_group_size INTEGER, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT NOT NULL, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT NOT NULL, 
	animal_feed_equip TEXT, 
	animal_housing TEXT, 
	food_prod TEXT NOT NULL, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	animal_am TEXT, 
	animal_am_route TEXT, 
	animal_am_use TEXT, 
	animal_diet TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	food_additive TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, animal_sex, food_trace_list, animal_body_cond, pool_dna_extracts, cons_food_stor_dur, samp_stor_dur, timepoint, animal_am_dur, animal_am_freq, bacterial_density, food_pack_capacity, food_source_age, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, animal_group_size, cons_qty_purchased, lot_number, intended_consumer, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, animal_feed_equip, animal_housing, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, animal_am, animal_am_route, animal_am_use, animal_diet, food_prod_synonym, "IFSAC_category", library_prep_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, perturbation, coll_site_geo_feat, food_additive, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_animal_and_animal_feed_MISAG" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48) NOT NULL, 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	animal_sex VARCHAR(16), 
	food_trace_list VARCHAR(45), 
	animal_body_cond VARCHAR(17), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	samp_stor_dur TEXT, 
	timepoint FLOAT, 
	animal_am_dur TEXT, 
	animal_am_freq TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	food_source_age TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	animal_group_size INTEGER, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT NOT NULL, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT NOT NULL, 
	animal_feed_equip TEXT, 
	animal_housing TEXT, 
	food_prod TEXT NOT NULL, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	animal_am TEXT, 
	animal_am_route TEXT, 
	animal_am_use TEXT, 
	animal_diet TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	food_additive TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, animal_sex, food_trace_list, animal_body_cond, pool_dna_extracts, cons_food_stor_dur, samp_stor_dur, timepoint, animal_am_dur, animal_am_freq, bacterial_density, food_pack_capacity, food_source_age, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, animal_group_size, cons_qty_purchased, lot_number, intended_consumer, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, animal_feed_equip, animal_housing, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, animal_am, animal_am_route, animal_am_use, animal_diet, food_prod_synonym, "IFSAC_category", library_prep_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, perturbation, coll_site_geo_feat, food_additive, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_animal_and_animal_feed_MIUVIG" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48) NOT NULL, 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	animal_sex VARCHAR(16), 
	food_trace_list VARCHAR(45), 
	animal_body_cond VARCHAR(17), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	samp_stor_dur TEXT, 
	timepoint FLOAT, 
	animal_am_dur TEXT, 
	animal_am_freq TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	food_source_age TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	animal_group_size INTEGER, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT NOT NULL, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT NOT NULL, 
	animal_feed_equip TEXT, 
	animal_housing TEXT, 
	food_prod TEXT NOT NULL, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	animal_am TEXT, 
	animal_am_route TEXT, 
	animal_am_use TEXT, 
	animal_diet TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	food_additive TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, animal_sex, food_trace_list, animal_body_cond, pool_dna_extracts, cons_food_stor_dur, samp_stor_dur, timepoint, animal_am_dur, animal_am_freq, bacterial_density, food_pack_capacity, food_source_age, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, animal_group_size, cons_qty_purchased, lot_number, intended_consumer, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, animal_feed_equip, animal_housing, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, animal_am, animal_am_route, animal_am_use, animal_diet, food_prod_synonym, "IFSAC_category", library_prep_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, perturbation, coll_site_geo_feat, food_additive, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE food_farm_environment (
	lat_lon TEXT NOT NULL, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	soil_texture_class VARCHAR(15), 
	tillage VARCHAR(13), 
	extr_weather_event VARCHAR(28), 
	food_clean_proc VARCHAR(30), 
	growth_habit VARCHAR(10), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	air_flow_impede TEXT, 
	plant_reprod_crop VARCHAR(18), 
	crop_rotation TEXT, 
	samp_stor_dur TEXT, 
	ph FLOAT, 
	root_med_ph FLOAT, 
	"soil_pH" FLOAT, 
	timepoint FLOAT, 
	"water_pH" FLOAT, 
	conduc TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	humidity TEXT, 
	salinity TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	soil_conductivity TEXT, 
	solar_irradiance TEXT, 
	tot_nitro_content TEXT, 
	turbidity TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	crop_yield TEXT, 
	water_frequency TEXT, 
	animal_group_size INTEGER, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	host_taxid INTEGER, 
	soil_porosity TEXT, 
	tot_org_carb TEXT, 
	genetic_mod TEXT, 
	ph_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	enrichment_protocol TEXT, 
	root_cond TEXT, 
	host_subspecf_genlin TEXT, 
	plant_part_maturity TEXT, 
	anim_water_method TEXT, 
	animal_housing TEXT, 
	farm_equip TEXT, 
	farm_equip_shared TEXT, 
	fertilizer_admin TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	plant_water_method TEXT, 
	samp_source_mat_cat TEXT, 
	samp_type TEXT NOT NULL, 
	season TEXT, 
	soil_cover TEXT, 
	soil_type TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	animal_intrusion TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	adjacent_environment TEXT, 
	animal_feed_equip TEXT, 
	ances_data TEXT, 
	animal_diet TEXT, 
	biotic_regm TEXT NOT NULL, 
	farm_equip_san_freq TEXT, 
	food_harvest_proc TEXT, 
	food_prod_char TEXT, 
	growth_medium TEXT, 
	host_genotype TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	prod_label_claims TEXT, 
	rel_location TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	ventilation_type TEXT, 
	water_source_adjac TEXT, 
	wind_direction TEXT, 
	farm_equip_san TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	"air_PM_concen" TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	previous_land_use TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	part_plant_animal TEXT, 
	spikein_growth_med TEXT, 
	cult_isol_date DATE, 
	date_extr_weath DATE, 
	fertilizer_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, soil_texture_class, tillage, extr_weather_event, food_clean_proc, growth_habit, farm_water_source, water_source_shared, air_flow_impede, plant_reprod_crop, crop_rotation, samp_stor_dur, ph, root_med_ph, "soil_pH", timepoint, "water_pH", conduc, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, salinity, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, soil_conductivity, solar_irradiance, tot_nitro_content, turbidity, ventilation_rate, wind_speed, crop_yield, water_frequency, animal_group_size, lot_number, intended_consumer, host_taxid, soil_porosity, tot_org_carb, genetic_mod, ph_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, enrichment_protocol, root_cond, host_subspecf_genlin, plant_part_maturity, anim_water_method, animal_housing, farm_equip, farm_equip_shared, fertilizer_admin, food_source, food_trav_mode, plant_water_method, samp_source_mat_cat, samp_type, season, soil_cover, soil_type, plant_growth_med, chem_administration, animal_intrusion, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, adjacent_environment, animal_feed_equip, ances_data, animal_diet, biotic_regm, farm_equip_san_freq, food_harvest_proc, food_prod_char, growth_medium, host_genotype, "IFSAC_category", library_prep_kit, prod_label_claims, rel_location, repository_name, samp_pooling, sequencing_kit, sequencing_location, ventilation_type, water_source_adjac, wind_direction, farm_equip_san, spikein_antibiotic, spikein_metal, "air_PM_concen", misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, organism_count, spikein_count, perturbation, host_phenotype, mechanical_damage, previous_land_use, food_contact_surf, food_contain_wrap, food_pack_medium, food_preserv_proc, food_product_type, food_trav_vehic, food_treat_proc, part_plant_animal, spikein_growth_med, cult_isol_date, date_extr_weath, fertilizer_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, seq_meth, project_name, samp_name, host_disease_stat)
);

CREATE TABLE "food_farm_environment_MIGS_bacteria" (
	lat_lon TEXT NOT NULL, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	soil_texture_class VARCHAR(15), 
	tillage VARCHAR(13), 
	extr_weather_event VARCHAR(28), 
	food_clean_proc VARCHAR(30), 
	growth_habit VARCHAR(10), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	air_flow_impede TEXT, 
	plant_reprod_crop VARCHAR(18), 
	crop_rotation TEXT, 
	samp_stor_dur TEXT, 
	ph FLOAT, 
	root_med_ph FLOAT, 
	"soil_pH" FLOAT, 
	timepoint FLOAT, 
	"water_pH" FLOAT, 
	conduc TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	humidity TEXT, 
	salinity TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	soil_conductivity TEXT, 
	solar_irradiance TEXT, 
	tot_nitro_content TEXT, 
	turbidity TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	crop_yield TEXT, 
	soil_temp TEXT, 
	water_frequency TEXT, 
	animal_group_size INTEGER, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	host_taxid INTEGER, 
	soil_porosity TEXT, 
	tot_org_carb TEXT, 
	genetic_mod TEXT, 
	ph_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	enrichment_protocol TEXT, 
	root_cond TEXT, 
	host_subspecf_genlin TEXT, 
	plant_part_maturity TEXT, 
	anim_water_method TEXT, 
	animal_housing TEXT, 
	farm_equip TEXT, 
	farm_equip_shared TEXT, 
	fertilizer_admin TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	plant_water_method TEXT, 
	samp_source_mat_cat TEXT, 
	samp_type TEXT NOT NULL, 
	season TEXT, 
	soil_cover TEXT, 
	soil_type TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	animal_intrusion TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	adjacent_environment TEXT, 
	animal_feed_equip TEXT, 
	ances_data TEXT, 
	animal_diet TEXT, 
	biotic_regm TEXT NOT NULL, 
	farm_equip_san_freq TEXT, 
	food_harvest_proc TEXT, 
	food_prod_char TEXT, 
	growth_medium TEXT, 
	host_genotype TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	prod_label_claims TEXT, 
	rel_location TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	ventilation_type TEXT, 
	water_source_adjac TEXT, 
	wind_direction TEXT, 
	farm_equip_san TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	"air_PM_concen" TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	previous_land_use TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	part_plant_animal TEXT, 
	spikein_growth_med TEXT, 
	cult_isol_date DATE, 
	date_extr_weath DATE, 
	fertilizer_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, soil_texture_class, tillage, extr_weather_event, food_clean_proc, growth_habit, farm_water_source, water_source_shared, air_flow_impede, plant_reprod_crop, crop_rotation, samp_stor_dur, ph, root_med_ph, "soil_pH", timepoint, "water_pH", conduc, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, salinity, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, soil_conductivity, solar_irradiance, tot_nitro_content, turbidity, ventilation_rate, wind_speed, crop_yield, soil_temp, water_frequency, animal_group_size, lot_number, intended_consumer, host_taxid, soil_porosity, tot_org_carb, genetic_mod, ph_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, enrichment_protocol, root_cond, host_subspecf_genlin, plant_part_maturity, anim_water_method, animal_housing, farm_equip, farm_equip_shared, fertilizer_admin, food_source, food_trav_mode, plant_water_method, samp_source_mat_cat, samp_type, season, soil_cover, soil_type, plant_growth_med, chem_administration, animal_intrusion, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, adjacent_environment, animal_feed_equip, ances_data, animal_diet, biotic_regm, farm_equip_san_freq, food_harvest_proc, food_prod_char, growth_medium, host_genotype, "IFSAC_category", library_prep_kit, prod_label_claims, rel_location, repository_name, samp_pooling, sequencing_kit, sequencing_location, ventilation_type, water_source_adjac, wind_direction, farm_equip_san, spikein_antibiotic, spikein_metal, "air_PM_concen", misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, organism_count, spikein_count, perturbation, host_phenotype, mechanical_damage, previous_land_use, food_contact_surf, food_contain_wrap, food_pack_medium, food_preserv_proc, food_product_type, food_trav_vehic, food_treat_proc, part_plant_animal, spikein_growth_med, cult_isol_date, date_extr_weath, fertilizer_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, seq_meth, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_farm_environment_MIGS_eukaryote" (
	lat_lon TEXT NOT NULL, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	soil_texture_class VARCHAR(15), 
	tillage VARCHAR(13), 
	extr_weather_event VARCHAR(28), 
	food_clean_proc VARCHAR(30), 
	growth_habit VARCHAR(10), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	air_flow_impede TEXT, 
	plant_reprod_crop VARCHAR(18), 
	crop_rotation TEXT, 
	samp_stor_dur TEXT, 
	ph FLOAT, 
	root_med_ph FLOAT, 
	"soil_pH" FLOAT, 
	timepoint FLOAT, 
	"water_pH" FLOAT, 
	conduc TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	humidity TEXT, 
	salinity TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	soil_conductivity TEXT, 
	solar_irradiance TEXT, 
	tot_nitro_content TEXT, 
	turbidity TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	crop_yield TEXT, 
	soil_temp TEXT, 
	water_frequency TEXT, 
	animal_group_size INTEGER, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	host_taxid INTEGER, 
	soil_porosity TEXT, 
	tot_org_carb TEXT, 
	genetic_mod TEXT, 
	ph_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	enrichment_protocol TEXT, 
	root_cond TEXT, 
	host_subspecf_genlin TEXT, 
	plant_part_maturity TEXT, 
	anim_water_method TEXT, 
	animal_housing TEXT, 
	farm_equip TEXT, 
	farm_equip_shared TEXT, 
	fertilizer_admin TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	plant_water_method TEXT, 
	samp_source_mat_cat TEXT, 
	samp_type TEXT NOT NULL, 
	season TEXT, 
	soil_cover TEXT, 
	soil_type TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	animal_intrusion TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	adjacent_environment TEXT, 
	animal_feed_equip TEXT, 
	ances_data TEXT, 
	animal_diet TEXT, 
	biotic_regm TEXT NOT NULL, 
	farm_equip_san_freq TEXT, 
	food_harvest_proc TEXT, 
	food_prod_char TEXT, 
	growth_medium TEXT, 
	host_genotype TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	prod_label_claims TEXT, 
	rel_location TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	ventilation_type TEXT, 
	water_source_adjac TEXT, 
	wind_direction TEXT, 
	farm_equip_san TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	"air_PM_concen" TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	previous_land_use TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	part_plant_animal TEXT, 
	spikein_growth_med TEXT, 
	cult_isol_date DATE, 
	date_extr_weath DATE, 
	fertilizer_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, soil_texture_class, tillage, extr_weather_event, food_clean_proc, growth_habit, farm_water_source, water_source_shared, air_flow_impede, plant_reprod_crop, crop_rotation, samp_stor_dur, ph, root_med_ph, "soil_pH", timepoint, "water_pH", conduc, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, salinity, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, soil_conductivity, solar_irradiance, tot_nitro_content, turbidity, ventilation_rate, wind_speed, crop_yield, soil_temp, water_frequency, animal_group_size, lot_number, intended_consumer, host_taxid, soil_porosity, tot_org_carb, genetic_mod, ph_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, enrichment_protocol, root_cond, host_subspecf_genlin, plant_part_maturity, anim_water_method, animal_housing, farm_equip, farm_equip_shared, fertilizer_admin, food_source, food_trav_mode, plant_water_method, samp_source_mat_cat, samp_type, season, soil_cover, soil_type, plant_growth_med, chem_administration, animal_intrusion, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, adjacent_environment, animal_feed_equip, ances_data, animal_diet, biotic_regm, farm_equip_san_freq, food_harvest_proc, food_prod_char, growth_medium, host_genotype, "IFSAC_category", library_prep_kit, prod_label_claims, rel_location, repository_name, samp_pooling, sequencing_kit, sequencing_location, ventilation_type, water_source_adjac, wind_direction, farm_equip_san, spikein_antibiotic, spikein_metal, "air_PM_concen", misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, organism_count, spikein_count, perturbation, host_phenotype, mechanical_damage, previous_land_use, food_contact_surf, food_contain_wrap, food_pack_medium, food_preserv_proc, food_product_type, food_trav_vehic, food_treat_proc, part_plant_animal, spikein_growth_med, cult_isol_date, date_extr_weath, fertilizer_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, seq_meth, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_farm_environment_MIGS_org" (
	lat_lon TEXT NOT NULL, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	soil_texture_class VARCHAR(15), 
	tillage VARCHAR(13), 
	extr_weather_event VARCHAR(28), 
	food_clean_proc VARCHAR(30), 
	growth_habit VARCHAR(10), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	air_flow_impede TEXT, 
	plant_reprod_crop VARCHAR(18), 
	crop_rotation TEXT, 
	samp_stor_dur TEXT, 
	ph FLOAT, 
	root_med_ph FLOAT, 
	"soil_pH" FLOAT, 
	timepoint FLOAT, 
	"water_pH" FLOAT, 
	conduc TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	humidity TEXT, 
	salinity TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	soil_conductivity TEXT, 
	solar_irradiance TEXT, 
	tot_nitro_content TEXT, 
	turbidity TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	crop_yield TEXT, 
	soil_temp TEXT, 
	water_frequency TEXT, 
	animal_group_size INTEGER, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	host_taxid INTEGER, 
	soil_porosity TEXT, 
	tot_org_carb TEXT, 
	genetic_mod TEXT, 
	ph_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	enrichment_protocol TEXT, 
	root_cond TEXT, 
	host_subspecf_genlin TEXT, 
	plant_part_maturity TEXT, 
	anim_water_method TEXT, 
	animal_housing TEXT, 
	farm_equip TEXT, 
	farm_equip_shared TEXT, 
	fertilizer_admin TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	plant_water_method TEXT, 
	samp_source_mat_cat TEXT, 
	samp_type TEXT NOT NULL, 
	season TEXT, 
	soil_cover TEXT, 
	soil_type TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	animal_intrusion TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	adjacent_environment TEXT, 
	animal_feed_equip TEXT, 
	ances_data TEXT, 
	animal_diet TEXT, 
	biotic_regm TEXT NOT NULL, 
	farm_equip_san_freq TEXT, 
	food_harvest_proc TEXT, 
	food_prod_char TEXT, 
	growth_medium TEXT, 
	host_genotype TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	prod_label_claims TEXT, 
	rel_location TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	ventilation_type TEXT, 
	water_source_adjac TEXT, 
	wind_direction TEXT, 
	farm_equip_san TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	"air_PM_concen" TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	previous_land_use TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	part_plant_animal TEXT, 
	spikein_growth_med TEXT, 
	cult_isol_date DATE, 
	date_extr_weath DATE, 
	fertilizer_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, soil_texture_class, tillage, extr_weather_event, food_clean_proc, growth_habit, farm_water_source, water_source_shared, air_flow_impede, plant_reprod_crop, crop_rotation, samp_stor_dur, ph, root_med_ph, "soil_pH", timepoint, "water_pH", conduc, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, salinity, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, soil_conductivity, solar_irradiance, tot_nitro_content, turbidity, ventilation_rate, wind_speed, crop_yield, soil_temp, water_frequency, animal_group_size, lot_number, intended_consumer, host_taxid, soil_porosity, tot_org_carb, genetic_mod, ph_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, enrichment_protocol, root_cond, host_subspecf_genlin, plant_part_maturity, anim_water_method, animal_housing, farm_equip, farm_equip_shared, fertilizer_admin, food_source, food_trav_mode, plant_water_method, samp_source_mat_cat, samp_type, season, soil_cover, soil_type, plant_growth_med, chem_administration, animal_intrusion, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, adjacent_environment, animal_feed_equip, ances_data, animal_diet, biotic_regm, farm_equip_san_freq, food_harvest_proc, food_prod_char, growth_medium, host_genotype, "IFSAC_category", library_prep_kit, prod_label_claims, rel_location, repository_name, samp_pooling, sequencing_kit, sequencing_location, ventilation_type, water_source_adjac, wind_direction, farm_equip_san, spikein_antibiotic, spikein_metal, "air_PM_concen", misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, organism_count, spikein_count, perturbation, host_phenotype, mechanical_damage, previous_land_use, food_contact_surf, food_contain_wrap, food_pack_medium, food_preserv_proc, food_product_type, food_trav_vehic, food_treat_proc, part_plant_animal, spikein_growth_med, cult_isol_date, date_extr_weath, fertilizer_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, seq_meth, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_farm_environment_MIGS_plant" (
	lat_lon TEXT NOT NULL, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	soil_texture_class VARCHAR(15), 
	tillage VARCHAR(13), 
	extr_weather_event VARCHAR(28), 
	food_clean_proc VARCHAR(30), 
	growth_habit VARCHAR(10), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	air_flow_impede TEXT, 
	plant_reprod_crop VARCHAR(18), 
	crop_rotation TEXT, 
	samp_stor_dur TEXT, 
	ph FLOAT, 
	root_med_ph FLOAT, 
	"soil_pH" FLOAT, 
	timepoint FLOAT, 
	"water_pH" FLOAT, 
	conduc TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	humidity TEXT, 
	salinity TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	soil_conductivity TEXT, 
	solar_irradiance TEXT, 
	tot_nitro_content TEXT, 
	turbidity TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	crop_yield TEXT, 
	soil_temp TEXT, 
	water_frequency TEXT, 
	animal_group_size INTEGER, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	host_taxid INTEGER, 
	soil_porosity TEXT, 
	tot_org_carb TEXT, 
	genetic_mod TEXT, 
	ph_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	enrichment_protocol TEXT, 
	root_cond TEXT, 
	host_subspecf_genlin TEXT, 
	plant_part_maturity TEXT, 
	anim_water_method TEXT, 
	animal_housing TEXT, 
	farm_equip TEXT, 
	farm_equip_shared TEXT, 
	fertilizer_admin TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	plant_water_method TEXT, 
	samp_source_mat_cat TEXT, 
	samp_type TEXT NOT NULL, 
	season TEXT, 
	soil_cover TEXT, 
	soil_type TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	animal_intrusion TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	adjacent_environment TEXT, 
	animal_feed_equip TEXT, 
	ances_data TEXT, 
	animal_diet TEXT, 
	biotic_regm TEXT NOT NULL, 
	farm_equip_san_freq TEXT, 
	food_harvest_proc TEXT, 
	food_prod_char TEXT, 
	growth_medium TEXT, 
	host_genotype TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	prod_label_claims TEXT, 
	rel_location TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	ventilation_type TEXT, 
	water_source_adjac TEXT, 
	wind_direction TEXT, 
	farm_equip_san TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	"air_PM_concen" TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	previous_land_use TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	part_plant_animal TEXT, 
	spikein_growth_med TEXT, 
	cult_isol_date DATE, 
	date_extr_weath DATE, 
	fertilizer_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, soil_texture_class, tillage, extr_weather_event, food_clean_proc, growth_habit, farm_water_source, water_source_shared, air_flow_impede, plant_reprod_crop, crop_rotation, samp_stor_dur, ph, root_med_ph, "soil_pH", timepoint, "water_pH", conduc, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, salinity, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, soil_conductivity, solar_irradiance, tot_nitro_content, turbidity, ventilation_rate, wind_speed, crop_yield, soil_temp, water_frequency, animal_group_size, lot_number, intended_consumer, host_taxid, soil_porosity, tot_org_carb, genetic_mod, ph_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, enrichment_protocol, root_cond, host_subspecf_genlin, plant_part_maturity, anim_water_method, animal_housing, farm_equip, farm_equip_shared, fertilizer_admin, food_source, food_trav_mode, plant_water_method, samp_source_mat_cat, samp_type, season, soil_cover, soil_type, plant_growth_med, chem_administration, animal_intrusion, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, adjacent_environment, animal_feed_equip, ances_data, animal_diet, biotic_regm, farm_equip_san_freq, food_harvest_proc, food_prod_char, growth_medium, host_genotype, "IFSAC_category", library_prep_kit, prod_label_claims, rel_location, repository_name, samp_pooling, sequencing_kit, sequencing_location, ventilation_type, water_source_adjac, wind_direction, farm_equip_san, spikein_antibiotic, spikein_metal, "air_PM_concen", misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, organism_count, spikein_count, perturbation, host_phenotype, mechanical_damage, previous_land_use, food_contact_surf, food_contain_wrap, food_pack_medium, food_preserv_proc, food_product_type, food_trav_vehic, food_treat_proc, part_plant_animal, spikein_growth_med, cult_isol_date, date_extr_weath, fertilizer_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, seq_meth, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_farm_environment_MIGS_virus" (
	lat_lon TEXT NOT NULL, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	soil_texture_class VARCHAR(15), 
	tillage VARCHAR(13), 
	extr_weather_event VARCHAR(28), 
	food_clean_proc VARCHAR(30), 
	growth_habit VARCHAR(10), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	air_flow_impede TEXT, 
	plant_reprod_crop VARCHAR(18), 
	crop_rotation TEXT, 
	samp_stor_dur TEXT, 
	ph FLOAT, 
	root_med_ph FLOAT, 
	"soil_pH" FLOAT, 
	timepoint FLOAT, 
	"water_pH" FLOAT, 
	conduc TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	humidity TEXT, 
	salinity TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	soil_conductivity TEXT, 
	solar_irradiance TEXT, 
	tot_nitro_content TEXT, 
	turbidity TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	crop_yield TEXT, 
	soil_temp TEXT, 
	water_frequency TEXT, 
	animal_group_size INTEGER, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	host_taxid INTEGER, 
	soil_porosity TEXT, 
	tot_org_carb TEXT, 
	genetic_mod TEXT, 
	ph_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	enrichment_protocol TEXT, 
	root_cond TEXT, 
	host_subspecf_genlin TEXT, 
	plant_part_maturity TEXT, 
	anim_water_method TEXT, 
	animal_housing TEXT, 
	farm_equip TEXT, 
	farm_equip_shared TEXT, 
	fertilizer_admin TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	plant_water_method TEXT, 
	samp_source_mat_cat TEXT, 
	samp_type TEXT NOT NULL, 
	season TEXT, 
	soil_cover TEXT, 
	soil_type TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	animal_intrusion TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	adjacent_environment TEXT, 
	animal_feed_equip TEXT, 
	ances_data TEXT, 
	animal_diet TEXT, 
	biotic_regm TEXT NOT NULL, 
	farm_equip_san_freq TEXT, 
	food_harvest_proc TEXT, 
	food_prod_char TEXT, 
	growth_medium TEXT, 
	host_genotype TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	prod_label_claims TEXT, 
	rel_location TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	ventilation_type TEXT, 
	water_source_adjac TEXT, 
	wind_direction TEXT, 
	farm_equip_san TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	"air_PM_concen" TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	previous_land_use TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	part_plant_animal TEXT, 
	spikein_growth_med TEXT, 
	cult_isol_date DATE, 
	date_extr_weath DATE, 
	fertilizer_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, soil_texture_class, tillage, extr_weather_event, food_clean_proc, growth_habit, farm_water_source, water_source_shared, air_flow_impede, plant_reprod_crop, crop_rotation, samp_stor_dur, ph, root_med_ph, "soil_pH", timepoint, "water_pH", conduc, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, salinity, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, soil_conductivity, solar_irradiance, tot_nitro_content, turbidity, ventilation_rate, wind_speed, crop_yield, soil_temp, water_frequency, animal_group_size, lot_number, intended_consumer, host_taxid, soil_porosity, tot_org_carb, genetic_mod, ph_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, enrichment_protocol, root_cond, host_subspecf_genlin, plant_part_maturity, anim_water_method, animal_housing, farm_equip, farm_equip_shared, fertilizer_admin, food_source, food_trav_mode, plant_water_method, samp_source_mat_cat, samp_type, season, soil_cover, soil_type, plant_growth_med, chem_administration, animal_intrusion, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, adjacent_environment, animal_feed_equip, ances_data, animal_diet, biotic_regm, farm_equip_san_freq, food_harvest_proc, food_prod_char, growth_medium, host_genotype, "IFSAC_category", library_prep_kit, prod_label_claims, rel_location, repository_name, samp_pooling, sequencing_kit, sequencing_location, ventilation_type, water_source_adjac, wind_direction, farm_equip_san, spikein_antibiotic, spikein_metal, "air_PM_concen", misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, organism_count, spikein_count, perturbation, host_phenotype, mechanical_damage, previous_land_use, food_contact_surf, food_contain_wrap, food_pack_medium, food_preserv_proc, food_product_type, food_trav_vehic, food_treat_proc, part_plant_animal, spikein_growth_med, cult_isol_date, date_extr_weath, fertilizer_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, seq_meth, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_farm_environment_MIMAG" (
	lat_lon TEXT NOT NULL, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	soil_texture_class VARCHAR(15), 
	tillage VARCHAR(13), 
	extr_weather_event VARCHAR(28), 
	food_clean_proc VARCHAR(30), 
	growth_habit VARCHAR(10), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	air_flow_impede TEXT, 
	plant_reprod_crop VARCHAR(18), 
	crop_rotation TEXT, 
	samp_stor_dur TEXT, 
	ph FLOAT, 
	root_med_ph FLOAT, 
	"soil_pH" FLOAT, 
	timepoint FLOAT, 
	"water_pH" FLOAT, 
	conduc TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	humidity TEXT, 
	salinity TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	soil_conductivity TEXT, 
	solar_irradiance TEXT, 
	tot_nitro_content TEXT, 
	turbidity TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	crop_yield TEXT, 
	soil_temp TEXT, 
	water_frequency TEXT, 
	animal_group_size INTEGER, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	host_taxid INTEGER, 
	soil_porosity TEXT, 
	tot_org_carb TEXT, 
	genetic_mod TEXT, 
	ph_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	enrichment_protocol TEXT, 
	root_cond TEXT, 
	host_subspecf_genlin TEXT, 
	plant_part_maturity TEXT, 
	anim_water_method TEXT, 
	animal_housing TEXT, 
	farm_equip TEXT, 
	farm_equip_shared TEXT, 
	fertilizer_admin TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	plant_water_method TEXT, 
	samp_source_mat_cat TEXT, 
	samp_type TEXT NOT NULL, 
	season TEXT, 
	soil_cover TEXT, 
	soil_type TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	animal_intrusion TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	adjacent_environment TEXT, 
	animal_feed_equip TEXT, 
	ances_data TEXT, 
	animal_diet TEXT, 
	biotic_regm TEXT NOT NULL, 
	farm_equip_san_freq TEXT, 
	food_harvest_proc TEXT, 
	food_prod_char TEXT, 
	growth_medium TEXT, 
	host_genotype TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	prod_label_claims TEXT, 
	rel_location TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	ventilation_type TEXT, 
	water_source_adjac TEXT, 
	wind_direction TEXT, 
	farm_equip_san TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	"air_PM_concen" TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	previous_land_use TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	part_plant_animal TEXT, 
	spikein_growth_med TEXT, 
	cult_isol_date DATE, 
	date_extr_weath DATE, 
	fertilizer_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, soil_texture_class, tillage, extr_weather_event, food_clean_proc, growth_habit, farm_water_source, water_source_shared, air_flow_impede, plant_reprod_crop, crop_rotation, samp_stor_dur, ph, root_med_ph, "soil_pH", timepoint, "water_pH", conduc, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, salinity, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, soil_conductivity, solar_irradiance, tot_nitro_content, turbidity, ventilation_rate, wind_speed, crop_yield, soil_temp, water_frequency, animal_group_size, lot_number, intended_consumer, host_taxid, soil_porosity, tot_org_carb, genetic_mod, ph_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, enrichment_protocol, root_cond, host_subspecf_genlin, plant_part_maturity, anim_water_method, animal_housing, farm_equip, farm_equip_shared, fertilizer_admin, food_source, food_trav_mode, plant_water_method, samp_source_mat_cat, samp_type, season, soil_cover, soil_type, plant_growth_med, chem_administration, animal_intrusion, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, adjacent_environment, animal_feed_equip, ances_data, animal_diet, biotic_regm, farm_equip_san_freq, food_harvest_proc, food_prod_char, growth_medium, host_genotype, "IFSAC_category", library_prep_kit, prod_label_claims, rel_location, repository_name, samp_pooling, sequencing_kit, sequencing_location, ventilation_type, water_source_adjac, wind_direction, farm_equip_san, spikein_antibiotic, spikein_metal, "air_PM_concen", misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, organism_count, spikein_count, perturbation, host_phenotype, mechanical_damage, previous_land_use, food_contact_surf, food_contain_wrap, food_pack_medium, food_preserv_proc, food_product_type, food_trav_vehic, food_treat_proc, part_plant_animal, spikein_growth_med, cult_isol_date, date_extr_weath, fertilizer_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, seq_meth, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_farm_environment_MIMARKS_specimen" (
	lat_lon TEXT NOT NULL, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	soil_texture_class VARCHAR(15), 
	tillage VARCHAR(13), 
	extr_weather_event VARCHAR(28), 
	food_clean_proc VARCHAR(30), 
	growth_habit VARCHAR(10), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	air_flow_impede TEXT, 
	plant_reprod_crop VARCHAR(18), 
	crop_rotation TEXT, 
	samp_stor_dur TEXT, 
	ph FLOAT, 
	root_med_ph FLOAT, 
	"soil_pH" FLOAT, 
	timepoint FLOAT, 
	"water_pH" FLOAT, 
	conduc TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	humidity TEXT, 
	salinity TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	soil_conductivity TEXT, 
	solar_irradiance TEXT, 
	tot_nitro_content TEXT, 
	turbidity TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	crop_yield TEXT, 
	soil_temp TEXT, 
	water_frequency TEXT, 
	animal_group_size INTEGER, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	host_taxid INTEGER, 
	soil_porosity TEXT, 
	tot_org_carb TEXT, 
	genetic_mod TEXT, 
	ph_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	enrichment_protocol TEXT, 
	root_cond TEXT, 
	host_subspecf_genlin TEXT, 
	plant_part_maturity TEXT, 
	anim_water_method TEXT, 
	animal_housing TEXT, 
	farm_equip TEXT, 
	farm_equip_shared TEXT, 
	fertilizer_admin TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	plant_water_method TEXT, 
	samp_source_mat_cat TEXT, 
	samp_type TEXT NOT NULL, 
	season TEXT, 
	soil_cover TEXT, 
	soil_type TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	animal_intrusion TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	adjacent_environment TEXT, 
	animal_feed_equip TEXT, 
	ances_data TEXT, 
	animal_diet TEXT, 
	biotic_regm TEXT NOT NULL, 
	farm_equip_san_freq TEXT, 
	food_harvest_proc TEXT, 
	food_prod_char TEXT, 
	growth_medium TEXT, 
	host_genotype TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	prod_label_claims TEXT, 
	rel_location TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	ventilation_type TEXT, 
	water_source_adjac TEXT, 
	wind_direction TEXT, 
	farm_equip_san TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	"air_PM_concen" TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	previous_land_use TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	part_plant_animal TEXT, 
	spikein_growth_med TEXT, 
	cult_isol_date DATE, 
	date_extr_weath DATE, 
	fertilizer_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, soil_texture_class, tillage, extr_weather_event, food_clean_proc, growth_habit, farm_water_source, water_source_shared, air_flow_impede, plant_reprod_crop, crop_rotation, samp_stor_dur, ph, root_med_ph, "soil_pH", timepoint, "water_pH", conduc, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, salinity, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, soil_conductivity, solar_irradiance, tot_nitro_content, turbidity, ventilation_rate, wind_speed, crop_yield, soil_temp, water_frequency, animal_group_size, lot_number, intended_consumer, host_taxid, soil_porosity, tot_org_carb, genetic_mod, ph_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, enrichment_protocol, root_cond, host_subspecf_genlin, plant_part_maturity, anim_water_method, animal_housing, farm_equip, farm_equip_shared, fertilizer_admin, food_source, food_trav_mode, plant_water_method, samp_source_mat_cat, samp_type, season, soil_cover, soil_type, plant_growth_med, chem_administration, animal_intrusion, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, adjacent_environment, animal_feed_equip, ances_data, animal_diet, biotic_regm, farm_equip_san_freq, food_harvest_proc, food_prod_char, growth_medium, host_genotype, "IFSAC_category", library_prep_kit, prod_label_claims, rel_location, repository_name, samp_pooling, sequencing_kit, sequencing_location, ventilation_type, water_source_adjac, wind_direction, farm_equip_san, spikein_antibiotic, spikein_metal, "air_PM_concen", misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, organism_count, spikein_count, perturbation, host_phenotype, mechanical_damage, previous_land_use, food_contact_surf, food_contain_wrap, food_pack_medium, food_preserv_proc, food_product_type, food_trav_vehic, food_treat_proc, part_plant_animal, spikein_growth_med, cult_isol_date, date_extr_weath, fertilizer_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, seq_meth, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_farm_environment_MIMARKS_survey" (
	lat_lon TEXT NOT NULL, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	soil_texture_class VARCHAR(15), 
	tillage VARCHAR(13), 
	extr_weather_event VARCHAR(28), 
	food_clean_proc VARCHAR(30), 
	growth_habit VARCHAR(10), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	air_flow_impede TEXT, 
	plant_reprod_crop VARCHAR(18), 
	crop_rotation TEXT, 
	samp_stor_dur TEXT, 
	ph FLOAT, 
	root_med_ph FLOAT, 
	"soil_pH" FLOAT, 
	timepoint FLOAT, 
	"water_pH" FLOAT, 
	conduc TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	humidity TEXT, 
	salinity TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	soil_conductivity TEXT, 
	solar_irradiance TEXT, 
	tot_nitro_content TEXT, 
	turbidity TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	crop_yield TEXT, 
	soil_temp TEXT, 
	water_frequency TEXT, 
	animal_group_size INTEGER, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	host_taxid INTEGER, 
	soil_porosity TEXT, 
	tot_org_carb TEXT, 
	genetic_mod TEXT, 
	ph_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	enrichment_protocol TEXT, 
	root_cond TEXT, 
	host_subspecf_genlin TEXT, 
	plant_part_maturity TEXT, 
	anim_water_method TEXT, 
	animal_housing TEXT, 
	farm_equip TEXT, 
	farm_equip_shared TEXT, 
	fertilizer_admin TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	plant_water_method TEXT, 
	samp_source_mat_cat TEXT, 
	samp_type TEXT NOT NULL, 
	season TEXT, 
	soil_cover TEXT, 
	soil_type TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	animal_intrusion TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	adjacent_environment TEXT, 
	animal_feed_equip TEXT, 
	ances_data TEXT, 
	animal_diet TEXT, 
	biotic_regm TEXT NOT NULL, 
	farm_equip_san_freq TEXT, 
	food_harvest_proc TEXT, 
	food_prod_char TEXT, 
	growth_medium TEXT, 
	host_genotype TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	prod_label_claims TEXT, 
	rel_location TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	ventilation_type TEXT, 
	water_source_adjac TEXT, 
	wind_direction TEXT, 
	farm_equip_san TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	"air_PM_concen" TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	previous_land_use TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	part_plant_animal TEXT, 
	spikein_growth_med TEXT, 
	cult_isol_date DATE, 
	date_extr_weath DATE, 
	fertilizer_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, soil_texture_class, tillage, extr_weather_event, food_clean_proc, growth_habit, farm_water_source, water_source_shared, air_flow_impede, plant_reprod_crop, crop_rotation, samp_stor_dur, ph, root_med_ph, "soil_pH", timepoint, "water_pH", conduc, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, salinity, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, soil_conductivity, solar_irradiance, tot_nitro_content, turbidity, ventilation_rate, wind_speed, crop_yield, soil_temp, water_frequency, animal_group_size, lot_number, intended_consumer, host_taxid, soil_porosity, tot_org_carb, genetic_mod, ph_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, enrichment_protocol, root_cond, host_subspecf_genlin, plant_part_maturity, anim_water_method, animal_housing, farm_equip, farm_equip_shared, fertilizer_admin, food_source, food_trav_mode, plant_water_method, samp_source_mat_cat, samp_type, season, soil_cover, soil_type, plant_growth_med, chem_administration, animal_intrusion, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, adjacent_environment, animal_feed_equip, ances_data, animal_diet, biotic_regm, farm_equip_san_freq, food_harvest_proc, food_prod_char, growth_medium, host_genotype, "IFSAC_category", library_prep_kit, prod_label_claims, rel_location, repository_name, samp_pooling, sequencing_kit, sequencing_location, ventilation_type, water_source_adjac, wind_direction, farm_equip_san, spikein_antibiotic, spikein_metal, "air_PM_concen", misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, organism_count, spikein_count, perturbation, host_phenotype, mechanical_damage, previous_land_use, food_contact_surf, food_contain_wrap, food_pack_medium, food_preserv_proc, food_product_type, food_trav_vehic, food_treat_proc, part_plant_animal, spikein_growth_med, cult_isol_date, date_extr_weath, fertilizer_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, seq_meth, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_farm_environment_MIMS" (
	lat_lon TEXT NOT NULL, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	soil_texture_class VARCHAR(15), 
	tillage VARCHAR(13), 
	extr_weather_event VARCHAR(28), 
	food_clean_proc VARCHAR(30), 
	growth_habit VARCHAR(10), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	air_flow_impede TEXT, 
	plant_reprod_crop VARCHAR(18), 
	crop_rotation TEXT, 
	samp_stor_dur TEXT, 
	ph FLOAT, 
	root_med_ph FLOAT, 
	"soil_pH" FLOAT, 
	timepoint FLOAT, 
	"water_pH" FLOAT, 
	conduc TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	humidity TEXT, 
	salinity TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	soil_conductivity TEXT, 
	solar_irradiance TEXT, 
	tot_nitro_content TEXT, 
	turbidity TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	crop_yield TEXT, 
	soil_temp TEXT, 
	water_frequency TEXT, 
	animal_group_size INTEGER, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	host_taxid INTEGER, 
	soil_porosity TEXT, 
	tot_org_carb TEXT, 
	genetic_mod TEXT, 
	ph_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	enrichment_protocol TEXT, 
	root_cond TEXT, 
	host_subspecf_genlin TEXT, 
	plant_part_maturity TEXT, 
	anim_water_method TEXT, 
	animal_housing TEXT, 
	farm_equip TEXT, 
	farm_equip_shared TEXT, 
	fertilizer_admin TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	plant_water_method TEXT, 
	samp_source_mat_cat TEXT, 
	samp_type TEXT NOT NULL, 
	season TEXT, 
	soil_cover TEXT, 
	soil_type TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	animal_intrusion TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	adjacent_environment TEXT, 
	animal_feed_equip TEXT, 
	ances_data TEXT, 
	animal_diet TEXT, 
	biotic_regm TEXT NOT NULL, 
	farm_equip_san_freq TEXT, 
	food_harvest_proc TEXT, 
	food_prod_char TEXT, 
	growth_medium TEXT, 
	host_genotype TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	prod_label_claims TEXT, 
	rel_location TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	ventilation_type TEXT, 
	water_source_adjac TEXT, 
	wind_direction TEXT, 
	farm_equip_san TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	"air_PM_concen" TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	previous_land_use TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	part_plant_animal TEXT, 
	spikein_growth_med TEXT, 
	cult_isol_date DATE, 
	date_extr_weath DATE, 
	fertilizer_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, soil_texture_class, tillage, extr_weather_event, food_clean_proc, growth_habit, farm_water_source, water_source_shared, air_flow_impede, plant_reprod_crop, crop_rotation, samp_stor_dur, ph, root_med_ph, "soil_pH", timepoint, "water_pH", conduc, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, salinity, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, soil_conductivity, solar_irradiance, tot_nitro_content, turbidity, ventilation_rate, wind_speed, crop_yield, soil_temp, water_frequency, animal_group_size, lot_number, intended_consumer, host_taxid, soil_porosity, tot_org_carb, genetic_mod, ph_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, enrichment_protocol, root_cond, host_subspecf_genlin, plant_part_maturity, anim_water_method, animal_housing, farm_equip, farm_equip_shared, fertilizer_admin, food_source, food_trav_mode, plant_water_method, samp_source_mat_cat, samp_type, season, soil_cover, soil_type, plant_growth_med, chem_administration, animal_intrusion, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, adjacent_environment, animal_feed_equip, ances_data, animal_diet, biotic_regm, farm_equip_san_freq, food_harvest_proc, food_prod_char, growth_medium, host_genotype, "IFSAC_category", library_prep_kit, prod_label_claims, rel_location, repository_name, samp_pooling, sequencing_kit, sequencing_location, ventilation_type, water_source_adjac, wind_direction, farm_equip_san, spikein_antibiotic, spikein_metal, "air_PM_concen", misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, organism_count, spikein_count, perturbation, host_phenotype, mechanical_damage, previous_land_use, food_contact_surf, food_contain_wrap, food_pack_medium, food_preserv_proc, food_product_type, food_trav_vehic, food_treat_proc, part_plant_animal, spikein_growth_med, cult_isol_date, date_extr_weath, fertilizer_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, seq_meth, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_farm_environment_MISAG" (
	lat_lon TEXT NOT NULL, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	soil_texture_class VARCHAR(15), 
	tillage VARCHAR(13), 
	extr_weather_event VARCHAR(28), 
	food_clean_proc VARCHAR(30), 
	growth_habit VARCHAR(10), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	air_flow_impede TEXT, 
	plant_reprod_crop VARCHAR(18), 
	crop_rotation TEXT, 
	samp_stor_dur TEXT, 
	ph FLOAT, 
	root_med_ph FLOAT, 
	"soil_pH" FLOAT, 
	timepoint FLOAT, 
	"water_pH" FLOAT, 
	conduc TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	humidity TEXT, 
	salinity TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	soil_conductivity TEXT, 
	solar_irradiance TEXT, 
	tot_nitro_content TEXT, 
	turbidity TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	crop_yield TEXT, 
	soil_temp TEXT, 
	water_frequency TEXT, 
	animal_group_size INTEGER, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	host_taxid INTEGER, 
	soil_porosity TEXT, 
	tot_org_carb TEXT, 
	genetic_mod TEXT, 
	ph_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	enrichment_protocol TEXT, 
	root_cond TEXT, 
	host_subspecf_genlin TEXT, 
	plant_part_maturity TEXT, 
	anim_water_method TEXT, 
	animal_housing TEXT, 
	farm_equip TEXT, 
	farm_equip_shared TEXT, 
	fertilizer_admin TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	plant_water_method TEXT, 
	samp_source_mat_cat TEXT, 
	samp_type TEXT NOT NULL, 
	season TEXT, 
	soil_cover TEXT, 
	soil_type TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	animal_intrusion TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	adjacent_environment TEXT, 
	animal_feed_equip TEXT, 
	ances_data TEXT, 
	animal_diet TEXT, 
	biotic_regm TEXT NOT NULL, 
	farm_equip_san_freq TEXT, 
	food_harvest_proc TEXT, 
	food_prod_char TEXT, 
	growth_medium TEXT, 
	host_genotype TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	prod_label_claims TEXT, 
	rel_location TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	ventilation_type TEXT, 
	water_source_adjac TEXT, 
	wind_direction TEXT, 
	farm_equip_san TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	"air_PM_concen" TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	previous_land_use TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	part_plant_animal TEXT, 
	spikein_growth_med TEXT, 
	cult_isol_date DATE, 
	date_extr_weath DATE, 
	fertilizer_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, soil_texture_class, tillage, extr_weather_event, food_clean_proc, growth_habit, farm_water_source, water_source_shared, air_flow_impede, plant_reprod_crop, crop_rotation, samp_stor_dur, ph, root_med_ph, "soil_pH", timepoint, "water_pH", conduc, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, salinity, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, soil_conductivity, solar_irradiance, tot_nitro_content, turbidity, ventilation_rate, wind_speed, crop_yield, soil_temp, water_frequency, animal_group_size, lot_number, intended_consumer, host_taxid, soil_porosity, tot_org_carb, genetic_mod, ph_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, enrichment_protocol, root_cond, host_subspecf_genlin, plant_part_maturity, anim_water_method, animal_housing, farm_equip, farm_equip_shared, fertilizer_admin, food_source, food_trav_mode, plant_water_method, samp_source_mat_cat, samp_type, season, soil_cover, soil_type, plant_growth_med, chem_administration, animal_intrusion, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, adjacent_environment, animal_feed_equip, ances_data, animal_diet, biotic_regm, farm_equip_san_freq, food_harvest_proc, food_prod_char, growth_medium, host_genotype, "IFSAC_category", library_prep_kit, prod_label_claims, rel_location, repository_name, samp_pooling, sequencing_kit, sequencing_location, ventilation_type, water_source_adjac, wind_direction, farm_equip_san, spikein_antibiotic, spikein_metal, "air_PM_concen", misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, organism_count, spikein_count, perturbation, host_phenotype, mechanical_damage, previous_land_use, food_contact_surf, food_contain_wrap, food_pack_medium, food_preserv_proc, food_product_type, food_trav_vehic, food_treat_proc, part_plant_animal, spikein_growth_med, cult_isol_date, date_extr_weath, fertilizer_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, seq_meth, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_farm_environment_MIUVIG" (
	lat_lon TEXT NOT NULL, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	soil_texture_class VARCHAR(15), 
	tillage VARCHAR(13), 
	extr_weather_event VARCHAR(28), 
	food_clean_proc VARCHAR(30), 
	growth_habit VARCHAR(10), 
	farm_water_source VARCHAR(19), 
	water_source_shared VARCHAR(27), 
	air_flow_impede TEXT, 
	plant_reprod_crop VARCHAR(18), 
	crop_rotation TEXT, 
	samp_stor_dur TEXT, 
	ph FLOAT, 
	root_med_ph FLOAT, 
	"soil_pH" FLOAT, 
	timepoint FLOAT, 
	"water_pH" FLOAT, 
	conduc TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	humidity TEXT, 
	salinity TEXT, 
	samp_stor_temp TEXT, 
	season_humidity TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	soil_conductivity TEXT, 
	solar_irradiance TEXT, 
	tot_nitro_content TEXT, 
	turbidity TEXT, 
	ventilation_rate TEXT, 
	wind_speed TEXT, 
	crop_yield TEXT, 
	soil_temp TEXT, 
	water_frequency TEXT, 
	animal_group_size INTEGER, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	host_taxid INTEGER, 
	soil_porosity TEXT, 
	tot_org_carb TEXT, 
	genetic_mod TEXT, 
	ph_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	enrichment_protocol TEXT, 
	root_cond TEXT, 
	host_subspecf_genlin TEXT, 
	plant_part_maturity TEXT, 
	anim_water_method TEXT, 
	animal_housing TEXT, 
	farm_equip TEXT, 
	farm_equip_shared TEXT, 
	fertilizer_admin TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	plant_water_method TEXT, 
	samp_source_mat_cat TEXT, 
	samp_type TEXT NOT NULL, 
	season TEXT, 
	soil_cover TEXT, 
	soil_type TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT NOT NULL, 
	animal_intrusion TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	adjacent_environment TEXT, 
	animal_feed_equip TEXT, 
	ances_data TEXT, 
	animal_diet TEXT, 
	biotic_regm TEXT NOT NULL, 
	farm_equip_san_freq TEXT, 
	food_harvest_proc TEXT, 
	food_prod_char TEXT, 
	growth_medium TEXT, 
	host_genotype TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	prod_label_claims TEXT, 
	rel_location TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	ventilation_type TEXT, 
	water_source_adjac TEXT, 
	wind_direction TEXT, 
	farm_equip_san TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	"air_PM_concen" TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	perturbation TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	previous_land_use TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	part_plant_animal TEXT, 
	spikein_growth_med TEXT, 
	cult_isol_date DATE, 
	date_extr_weath DATE, 
	fertilizer_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, soil_texture_class, tillage, extr_weather_event, food_clean_proc, growth_habit, farm_water_source, water_source_shared, air_flow_impede, plant_reprod_crop, crop_rotation, samp_stor_dur, ph, root_med_ph, "soil_pH", timepoint, "water_pH", conduc, host_age, host_dry_mass, host_height, host_length, host_tot_mass, humidity, salinity, samp_stor_temp, season_humidity, season_precpt, season_temp, size_frac_low, size_frac_up, soil_conductivity, solar_irradiance, tot_nitro_content, turbidity, ventilation_rate, wind_speed, crop_yield, soil_temp, water_frequency, animal_group_size, lot_number, intended_consumer, host_taxid, soil_porosity, tot_org_carb, genetic_mod, ph_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, enrichment_protocol, root_cond, host_subspecf_genlin, plant_part_maturity, anim_water_method, animal_housing, farm_equip, farm_equip_shared, fertilizer_admin, food_source, food_trav_mode, plant_water_method, samp_source_mat_cat, samp_type, season, soil_cover, soil_type, plant_growth_med, chem_administration, animal_intrusion, cult_result_org, cult_target, serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, adjacent_environment, animal_feed_equip, ances_data, animal_diet, biotic_regm, farm_equip_san_freq, food_harvest_proc, food_prod_char, growth_medium, host_genotype, "IFSAC_category", library_prep_kit, prod_label_claims, rel_location, repository_name, samp_pooling, sequencing_kit, sequencing_location, ventilation_type, water_source_adjac, wind_direction, farm_equip_san, spikein_antibiotic, spikein_metal, "air_PM_concen", misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, organism_count, spikein_count, perturbation, host_phenotype, mechanical_damage, previous_land_use, food_contact_surf, food_contain_wrap, food_pack_medium, food_preserv_proc, food_product_type, food_trav_vehic, food_treat_proc, part_plant_animal, spikein_growth_med, cult_isol_date, date_extr_weath, fertilizer_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, seq_meth, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE food_food_production_facility (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	samp_floor VARCHAR(9), 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	surf_material VARCHAR(15), 
	facility_type VARCHAR(24), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	indoor_surf VARCHAR(11), 
	food_trace_list VARCHAR(45), 
	samp_loc_condition VARCHAR(28), 
	samp_surf_moisture VARCHAR(21), 
	pool_dna_extracts TEXT, 
	samp_stor_dur TEXT, 
	avg_occup FLOAT, 
	timepoint FLOAT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	samp_room_id INTEGER, 
	area_samp_size TEXT, 
	room_dim TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	env_monitoring_zone TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT NOT NULL, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	hygienic_area TEXT, 
	samp_source_mat_cat TEXT NOT NULL, 
	samp_type TEXT NOT NULL, 
	spec_intended_cons TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	biocide_used TEXT, 
	food_prod_char TEXT, 
	food_prod_synonym TEXT, 
	freq_clean VARCHAR(9), 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	nucl_acid_ext_kit TEXT, 
	prod_label_claims TEXT, 
	repository_name TEXT, 
	samp_collect_device TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT NOT NULL, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT NOT NULL, 
	samp_stor_media TEXT NOT NULL, 
	ster_meth_samp_room TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	subspecf_gen_lin TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_floor, cult_result, samp_purpose, surf_material, facility_type, food_quality_date, samp_transport_cont, indoor_surf, food_trace_list, samp_loc_condition, samp_surf_moisture, pool_dna_extracts, samp_stor_dur, avg_occup, timepoint, bacterial_density, food_pack_capacity, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, samp_room_id, area_samp_size, room_dim, lot_number, intended_consumer, genetic_mod, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, env_monitoring_zone, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", hygienic_area, samp_source_mat_cat, samp_type, spec_intended_cons, cult_result_org, cult_target, biocide_used, food_prod_char, food_prod_synonym, freq_clean, "IFSAC_category", library_prep_kit, nucl_acid_ext_kit, prod_label_claims, repository_name, samp_collect_device, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, misc_param, organism_count, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, ster_meth_samp_room, study_design, study_tmnt, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, subspecf_gen_lin, experimental_factor, seq_meth, project_name, samp_name)
);

CREATE TABLE "food_food_production_facility_MIGS_bacteria" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	samp_floor VARCHAR(9), 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	surf_material VARCHAR(15), 
	facility_type VARCHAR(24), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	indoor_surf VARCHAR(11), 
	food_trace_list VARCHAR(45), 
	samp_loc_condition VARCHAR(28), 
	samp_surf_moisture VARCHAR(21), 
	pool_dna_extracts TEXT, 
	samp_stor_dur TEXT, 
	avg_occup FLOAT, 
	timepoint FLOAT, 
	air_temp TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	samp_room_id INTEGER, 
	area_samp_size TEXT, 
	room_dim TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	env_monitoring_zone TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT NOT NULL, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	hygienic_area TEXT, 
	samp_source_mat_cat TEXT NOT NULL, 
	samp_type TEXT NOT NULL, 
	spec_intended_cons TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	biocide_used TEXT, 
	food_prod_char TEXT, 
	food_prod_synonym TEXT, 
	freq_clean VARCHAR(9), 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	nucl_acid_ext_kit TEXT, 
	prod_label_claims TEXT, 
	repository_name TEXT, 
	samp_collect_device TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT NOT NULL, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT NOT NULL, 
	samp_stor_media TEXT NOT NULL, 
	ster_meth_samp_room TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	subspecf_gen_lin TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_floor, cult_result, samp_purpose, surf_material, facility_type, food_quality_date, samp_transport_cont, indoor_surf, food_trace_list, samp_loc_condition, samp_surf_moisture, pool_dna_extracts, samp_stor_dur, avg_occup, timepoint, air_temp, bacterial_density, food_pack_capacity, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, samp_room_id, area_samp_size, room_dim, lot_number, intended_consumer, genetic_mod, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, env_monitoring_zone, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", hygienic_area, samp_source_mat_cat, samp_type, spec_intended_cons, cult_result_org, cult_target, biocide_used, food_prod_char, food_prod_synonym, freq_clean, "IFSAC_category", library_prep_kit, nucl_acid_ext_kit, prod_label_claims, repository_name, samp_collect_device, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, misc_param, organism_count, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, ster_meth_samp_room, study_design, study_tmnt, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, subspecf_gen_lin, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_food_production_facility_MIGS_eukaryote" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	samp_floor VARCHAR(9), 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	surf_material VARCHAR(15), 
	facility_type VARCHAR(24), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	indoor_surf VARCHAR(11), 
	food_trace_list VARCHAR(45), 
	samp_loc_condition VARCHAR(28), 
	samp_surf_moisture VARCHAR(21), 
	pool_dna_extracts TEXT, 
	samp_stor_dur TEXT, 
	avg_occup FLOAT, 
	timepoint FLOAT, 
	air_temp TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	samp_room_id INTEGER, 
	area_samp_size TEXT, 
	room_dim TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	env_monitoring_zone TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT NOT NULL, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	hygienic_area TEXT, 
	samp_source_mat_cat TEXT NOT NULL, 
	samp_type TEXT NOT NULL, 
	spec_intended_cons TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	biocide_used TEXT, 
	food_prod_char TEXT, 
	food_prod_synonym TEXT, 
	freq_clean VARCHAR(9), 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	nucl_acid_ext_kit TEXT, 
	prod_label_claims TEXT, 
	repository_name TEXT, 
	samp_collect_device TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT NOT NULL, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT NOT NULL, 
	samp_stor_media TEXT NOT NULL, 
	ster_meth_samp_room TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	subspecf_gen_lin TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_floor, cult_result, samp_purpose, surf_material, facility_type, food_quality_date, samp_transport_cont, indoor_surf, food_trace_list, samp_loc_condition, samp_surf_moisture, pool_dna_extracts, samp_stor_dur, avg_occup, timepoint, air_temp, bacterial_density, food_pack_capacity, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, samp_room_id, area_samp_size, room_dim, lot_number, intended_consumer, genetic_mod, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, env_monitoring_zone, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", hygienic_area, samp_source_mat_cat, samp_type, spec_intended_cons, cult_result_org, cult_target, biocide_used, food_prod_char, food_prod_synonym, freq_clean, "IFSAC_category", library_prep_kit, nucl_acid_ext_kit, prod_label_claims, repository_name, samp_collect_device, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, misc_param, organism_count, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, ster_meth_samp_room, study_design, study_tmnt, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, subspecf_gen_lin, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_food_production_facility_MIGS_org" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	samp_floor VARCHAR(9), 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	surf_material VARCHAR(15), 
	facility_type VARCHAR(24), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	indoor_surf VARCHAR(11), 
	food_trace_list VARCHAR(45), 
	samp_loc_condition VARCHAR(28), 
	samp_surf_moisture VARCHAR(21), 
	pool_dna_extracts TEXT, 
	samp_stor_dur TEXT, 
	avg_occup FLOAT, 
	timepoint FLOAT, 
	air_temp TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	samp_room_id INTEGER, 
	area_samp_size TEXT, 
	room_dim TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	env_monitoring_zone TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT NOT NULL, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	hygienic_area TEXT, 
	samp_source_mat_cat TEXT NOT NULL, 
	samp_type TEXT NOT NULL, 
	spec_intended_cons TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	biocide_used TEXT, 
	food_prod_char TEXT, 
	food_prod_synonym TEXT, 
	freq_clean VARCHAR(9), 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	nucl_acid_ext_kit TEXT, 
	prod_label_claims TEXT, 
	repository_name TEXT, 
	samp_collect_device TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT NOT NULL, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT NOT NULL, 
	samp_stor_media TEXT NOT NULL, 
	ster_meth_samp_room TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	subspecf_gen_lin TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_floor, cult_result, samp_purpose, surf_material, facility_type, food_quality_date, samp_transport_cont, indoor_surf, food_trace_list, samp_loc_condition, samp_surf_moisture, pool_dna_extracts, samp_stor_dur, avg_occup, timepoint, air_temp, bacterial_density, food_pack_capacity, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, samp_room_id, area_samp_size, room_dim, lot_number, intended_consumer, genetic_mod, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, env_monitoring_zone, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", hygienic_area, samp_source_mat_cat, samp_type, spec_intended_cons, cult_result_org, cult_target, biocide_used, food_prod_char, food_prod_synonym, freq_clean, "IFSAC_category", library_prep_kit, nucl_acid_ext_kit, prod_label_claims, repository_name, samp_collect_device, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, misc_param, organism_count, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, ster_meth_samp_room, study_design, study_tmnt, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, subspecf_gen_lin, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_food_production_facility_MIGS_plant" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	samp_floor VARCHAR(9), 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	surf_material VARCHAR(15), 
	facility_type VARCHAR(24), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	indoor_surf VARCHAR(11), 
	food_trace_list VARCHAR(45), 
	samp_loc_condition VARCHAR(28), 
	samp_surf_moisture VARCHAR(21), 
	pool_dna_extracts TEXT, 
	samp_stor_dur TEXT, 
	avg_occup FLOAT, 
	timepoint FLOAT, 
	air_temp TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	samp_room_id INTEGER, 
	area_samp_size TEXT, 
	room_dim TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	env_monitoring_zone TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT NOT NULL, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	hygienic_area TEXT, 
	samp_source_mat_cat TEXT NOT NULL, 
	samp_type TEXT NOT NULL, 
	spec_intended_cons TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	biocide_used TEXT, 
	food_prod_char TEXT, 
	food_prod_synonym TEXT, 
	freq_clean VARCHAR(9), 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	nucl_acid_ext_kit TEXT, 
	prod_label_claims TEXT, 
	repository_name TEXT, 
	samp_collect_device TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT NOT NULL, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT NOT NULL, 
	samp_stor_media TEXT NOT NULL, 
	ster_meth_samp_room TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	subspecf_gen_lin TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_floor, cult_result, samp_purpose, surf_material, facility_type, food_quality_date, samp_transport_cont, indoor_surf, food_trace_list, samp_loc_condition, samp_surf_moisture, pool_dna_extracts, samp_stor_dur, avg_occup, timepoint, air_temp, bacterial_density, food_pack_capacity, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, samp_room_id, area_samp_size, room_dim, lot_number, intended_consumer, genetic_mod, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, env_monitoring_zone, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", hygienic_area, samp_source_mat_cat, samp_type, spec_intended_cons, cult_result_org, cult_target, biocide_used, food_prod_char, food_prod_synonym, freq_clean, "IFSAC_category", library_prep_kit, nucl_acid_ext_kit, prod_label_claims, repository_name, samp_collect_device, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, misc_param, organism_count, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, ster_meth_samp_room, study_design, study_tmnt, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, subspecf_gen_lin, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_food_production_facility_MIGS_virus" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	samp_floor VARCHAR(9), 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	surf_material VARCHAR(15), 
	facility_type VARCHAR(24), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	indoor_surf VARCHAR(11), 
	food_trace_list VARCHAR(45), 
	samp_loc_condition VARCHAR(28), 
	samp_surf_moisture VARCHAR(21), 
	pool_dna_extracts TEXT, 
	samp_stor_dur TEXT, 
	avg_occup FLOAT, 
	timepoint FLOAT, 
	air_temp TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	samp_room_id INTEGER, 
	area_samp_size TEXT, 
	room_dim TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	env_monitoring_zone TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT NOT NULL, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	hygienic_area TEXT, 
	samp_source_mat_cat TEXT NOT NULL, 
	samp_type TEXT NOT NULL, 
	spec_intended_cons TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	biocide_used TEXT, 
	food_prod_char TEXT, 
	food_prod_synonym TEXT, 
	freq_clean VARCHAR(9), 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	nucl_acid_ext_kit TEXT, 
	prod_label_claims TEXT, 
	repository_name TEXT, 
	samp_collect_device TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT NOT NULL, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT NOT NULL, 
	samp_stor_media TEXT NOT NULL, 
	ster_meth_samp_room TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	subspecf_gen_lin TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_floor, cult_result, samp_purpose, surf_material, facility_type, food_quality_date, samp_transport_cont, indoor_surf, food_trace_list, samp_loc_condition, samp_surf_moisture, pool_dna_extracts, samp_stor_dur, avg_occup, timepoint, air_temp, bacterial_density, food_pack_capacity, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, samp_room_id, area_samp_size, room_dim, lot_number, intended_consumer, genetic_mod, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, env_monitoring_zone, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", hygienic_area, samp_source_mat_cat, samp_type, spec_intended_cons, cult_result_org, cult_target, biocide_used, food_prod_char, food_prod_synonym, freq_clean, "IFSAC_category", library_prep_kit, nucl_acid_ext_kit, prod_label_claims, repository_name, samp_collect_device, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, misc_param, organism_count, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, ster_meth_samp_room, study_design, study_tmnt, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, subspecf_gen_lin, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_food_production_facility_MIMAG" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	samp_floor VARCHAR(9), 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	surf_material VARCHAR(15), 
	facility_type VARCHAR(24), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	indoor_surf VARCHAR(11), 
	food_trace_list VARCHAR(45), 
	samp_loc_condition VARCHAR(28), 
	samp_surf_moisture VARCHAR(21), 
	pool_dna_extracts TEXT, 
	samp_stor_dur TEXT, 
	avg_occup FLOAT, 
	timepoint FLOAT, 
	air_temp TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	samp_room_id INTEGER, 
	area_samp_size TEXT, 
	room_dim TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	env_monitoring_zone TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT NOT NULL, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	hygienic_area TEXT, 
	samp_source_mat_cat TEXT NOT NULL, 
	samp_type TEXT NOT NULL, 
	spec_intended_cons TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	biocide_used TEXT, 
	food_prod_char TEXT, 
	food_prod_synonym TEXT, 
	freq_clean VARCHAR(9), 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	nucl_acid_ext_kit TEXT, 
	prod_label_claims TEXT, 
	repository_name TEXT, 
	samp_collect_device TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT NOT NULL, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT NOT NULL, 
	samp_stor_media TEXT NOT NULL, 
	ster_meth_samp_room TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	subspecf_gen_lin TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_floor, cult_result, samp_purpose, surf_material, facility_type, food_quality_date, samp_transport_cont, indoor_surf, food_trace_list, samp_loc_condition, samp_surf_moisture, pool_dna_extracts, samp_stor_dur, avg_occup, timepoint, air_temp, bacterial_density, food_pack_capacity, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, samp_room_id, area_samp_size, room_dim, lot_number, intended_consumer, genetic_mod, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, env_monitoring_zone, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", hygienic_area, samp_source_mat_cat, samp_type, spec_intended_cons, cult_result_org, cult_target, biocide_used, food_prod_char, food_prod_synonym, freq_clean, "IFSAC_category", library_prep_kit, nucl_acid_ext_kit, prod_label_claims, repository_name, samp_collect_device, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, misc_param, organism_count, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, ster_meth_samp_room, study_design, study_tmnt, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, subspecf_gen_lin, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_food_production_facility_MIMARKS_specimen" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	samp_floor VARCHAR(9), 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	surf_material VARCHAR(15), 
	facility_type VARCHAR(24), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	indoor_surf VARCHAR(11), 
	food_trace_list VARCHAR(45), 
	samp_loc_condition VARCHAR(28), 
	samp_surf_moisture VARCHAR(21), 
	pool_dna_extracts TEXT, 
	samp_stor_dur TEXT, 
	avg_occup FLOAT, 
	timepoint FLOAT, 
	air_temp TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	samp_room_id INTEGER, 
	area_samp_size TEXT, 
	room_dim TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	env_monitoring_zone TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT NOT NULL, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	hygienic_area TEXT, 
	samp_source_mat_cat TEXT NOT NULL, 
	samp_type TEXT NOT NULL, 
	spec_intended_cons TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	biocide_used TEXT, 
	food_prod_char TEXT, 
	food_prod_synonym TEXT, 
	freq_clean VARCHAR(9), 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	nucl_acid_ext_kit TEXT, 
	prod_label_claims TEXT, 
	repository_name TEXT, 
	samp_collect_device TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT NOT NULL, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT NOT NULL, 
	samp_stor_media TEXT NOT NULL, 
	ster_meth_samp_room TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	subspecf_gen_lin TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_floor, cult_result, samp_purpose, surf_material, facility_type, food_quality_date, samp_transport_cont, indoor_surf, food_trace_list, samp_loc_condition, samp_surf_moisture, pool_dna_extracts, samp_stor_dur, avg_occup, timepoint, air_temp, bacterial_density, food_pack_capacity, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, samp_room_id, area_samp_size, room_dim, lot_number, intended_consumer, genetic_mod, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, env_monitoring_zone, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", hygienic_area, samp_source_mat_cat, samp_type, spec_intended_cons, cult_result_org, cult_target, biocide_used, food_prod_char, food_prod_synonym, freq_clean, "IFSAC_category", library_prep_kit, nucl_acid_ext_kit, prod_label_claims, repository_name, samp_collect_device, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, misc_param, organism_count, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, ster_meth_samp_room, study_design, study_tmnt, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, subspecf_gen_lin, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_food_production_facility_MIMARKS_survey" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	samp_floor VARCHAR(9), 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	surf_material VARCHAR(15), 
	facility_type VARCHAR(24), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	indoor_surf VARCHAR(11), 
	food_trace_list VARCHAR(45), 
	samp_loc_condition VARCHAR(28), 
	samp_surf_moisture VARCHAR(21), 
	pool_dna_extracts TEXT, 
	samp_stor_dur TEXT, 
	avg_occup FLOAT, 
	timepoint FLOAT, 
	air_temp TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	samp_room_id INTEGER, 
	area_samp_size TEXT, 
	room_dim TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	env_monitoring_zone TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT NOT NULL, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	hygienic_area TEXT, 
	samp_source_mat_cat TEXT NOT NULL, 
	samp_type TEXT NOT NULL, 
	spec_intended_cons TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	biocide_used TEXT, 
	food_prod_char TEXT, 
	food_prod_synonym TEXT, 
	freq_clean VARCHAR(9), 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	nucl_acid_ext_kit TEXT, 
	prod_label_claims TEXT, 
	repository_name TEXT, 
	samp_collect_device TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT NOT NULL, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT NOT NULL, 
	samp_stor_media TEXT NOT NULL, 
	ster_meth_samp_room TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	subspecf_gen_lin TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_floor, cult_result, samp_purpose, surf_material, facility_type, food_quality_date, samp_transport_cont, indoor_surf, food_trace_list, samp_loc_condition, samp_surf_moisture, pool_dna_extracts, samp_stor_dur, avg_occup, timepoint, air_temp, bacterial_density, food_pack_capacity, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, samp_room_id, area_samp_size, room_dim, lot_number, intended_consumer, genetic_mod, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, env_monitoring_zone, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", hygienic_area, samp_source_mat_cat, samp_type, spec_intended_cons, cult_result_org, cult_target, biocide_used, food_prod_char, food_prod_synonym, freq_clean, "IFSAC_category", library_prep_kit, nucl_acid_ext_kit, prod_label_claims, repository_name, samp_collect_device, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, misc_param, organism_count, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, ster_meth_samp_room, study_design, study_tmnt, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, subspecf_gen_lin, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_food_production_facility_MIMS" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	samp_floor VARCHAR(9), 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	surf_material VARCHAR(15), 
	facility_type VARCHAR(24), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	indoor_surf VARCHAR(11), 
	food_trace_list VARCHAR(45), 
	samp_loc_condition VARCHAR(28), 
	samp_surf_moisture VARCHAR(21), 
	pool_dna_extracts TEXT, 
	samp_stor_dur TEXT, 
	avg_occup FLOAT, 
	timepoint FLOAT, 
	air_temp TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	samp_room_id INTEGER, 
	area_samp_size TEXT, 
	room_dim TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	env_monitoring_zone TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT NOT NULL, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	hygienic_area TEXT, 
	samp_source_mat_cat TEXT NOT NULL, 
	samp_type TEXT NOT NULL, 
	spec_intended_cons TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	biocide_used TEXT, 
	food_prod_char TEXT, 
	food_prod_synonym TEXT, 
	freq_clean VARCHAR(9), 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	nucl_acid_ext_kit TEXT, 
	prod_label_claims TEXT, 
	repository_name TEXT, 
	samp_collect_device TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT NOT NULL, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT NOT NULL, 
	samp_stor_media TEXT NOT NULL, 
	ster_meth_samp_room TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	subspecf_gen_lin TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_floor, cult_result, samp_purpose, surf_material, facility_type, food_quality_date, samp_transport_cont, indoor_surf, food_trace_list, samp_loc_condition, samp_surf_moisture, pool_dna_extracts, samp_stor_dur, avg_occup, timepoint, air_temp, bacterial_density, food_pack_capacity, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, samp_room_id, area_samp_size, room_dim, lot_number, intended_consumer, genetic_mod, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, env_monitoring_zone, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", hygienic_area, samp_source_mat_cat, samp_type, spec_intended_cons, cult_result_org, cult_target, biocide_used, food_prod_char, food_prod_synonym, freq_clean, "IFSAC_category", library_prep_kit, nucl_acid_ext_kit, prod_label_claims, repository_name, samp_collect_device, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, misc_param, organism_count, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, ster_meth_samp_room, study_design, study_tmnt, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, subspecf_gen_lin, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_food_production_facility_MISAG" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	samp_floor VARCHAR(9), 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	surf_material VARCHAR(15), 
	facility_type VARCHAR(24), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	indoor_surf VARCHAR(11), 
	food_trace_list VARCHAR(45), 
	samp_loc_condition VARCHAR(28), 
	samp_surf_moisture VARCHAR(21), 
	pool_dna_extracts TEXT, 
	samp_stor_dur TEXT, 
	avg_occup FLOAT, 
	timepoint FLOAT, 
	air_temp TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	samp_room_id INTEGER, 
	area_samp_size TEXT, 
	room_dim TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	env_monitoring_zone TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT NOT NULL, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	hygienic_area TEXT, 
	samp_source_mat_cat TEXT NOT NULL, 
	samp_type TEXT NOT NULL, 
	spec_intended_cons TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	biocide_used TEXT, 
	food_prod_char TEXT, 
	food_prod_synonym TEXT, 
	freq_clean VARCHAR(9), 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	nucl_acid_ext_kit TEXT, 
	prod_label_claims TEXT, 
	repository_name TEXT, 
	samp_collect_device TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT NOT NULL, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT NOT NULL, 
	samp_stor_media TEXT NOT NULL, 
	ster_meth_samp_room TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	subspecf_gen_lin TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_floor, cult_result, samp_purpose, surf_material, facility_type, food_quality_date, samp_transport_cont, indoor_surf, food_trace_list, samp_loc_condition, samp_surf_moisture, pool_dna_extracts, samp_stor_dur, avg_occup, timepoint, air_temp, bacterial_density, food_pack_capacity, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, samp_room_id, area_samp_size, room_dim, lot_number, intended_consumer, genetic_mod, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, env_monitoring_zone, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", hygienic_area, samp_source_mat_cat, samp_type, spec_intended_cons, cult_result_org, cult_target, biocide_used, food_prod_char, food_prod_synonym, freq_clean, "IFSAC_category", library_prep_kit, nucl_acid_ext_kit, prod_label_claims, repository_name, samp_collect_device, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, misc_param, organism_count, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, ster_meth_samp_room, study_design, study_tmnt, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, subspecf_gen_lin, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_food_production_facility_MIUVIG" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	samp_floor VARCHAR(9), 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	surf_material VARCHAR(15), 
	facility_type VARCHAR(24), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	indoor_surf VARCHAR(11), 
	food_trace_list VARCHAR(45), 
	samp_loc_condition VARCHAR(28), 
	samp_surf_moisture VARCHAR(21), 
	pool_dna_extracts TEXT, 
	samp_stor_dur TEXT, 
	avg_occup FLOAT, 
	timepoint FLOAT, 
	air_temp TEXT, 
	bacterial_density TEXT, 
	food_pack_capacity TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	samp_room_id INTEGER, 
	area_samp_size TEXT, 
	room_dim TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	samp_collect_method TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	env_monitoring_zone TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT NOT NULL, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	hygienic_area TEXT, 
	samp_source_mat_cat TEXT NOT NULL, 
	samp_type TEXT NOT NULL, 
	spec_intended_cons TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	biocide_used TEXT, 
	food_prod_char TEXT, 
	food_prod_synonym TEXT, 
	freq_clean VARCHAR(9), 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	nucl_acid_ext_kit TEXT, 
	prod_label_claims TEXT, 
	repository_name TEXT, 
	samp_collect_device TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT NOT NULL, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_dis_point_city TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT NOT NULL, 
	samp_stor_media TEXT NOT NULL, 
	ster_meth_samp_room TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	subspecf_gen_lin TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_floor, cult_result, samp_purpose, surf_material, facility_type, food_quality_date, samp_transport_cont, indoor_surf, food_trace_list, samp_loc_condition, samp_surf_moisture, pool_dna_extracts, samp_stor_dur, avg_occup, timepoint, air_temp, bacterial_density, food_pack_capacity, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, samp_room_id, area_samp_size, room_dim, lot_number, intended_consumer, genetic_mod, enrichment_protocol, samp_collect_method, food_dis_point, food_origin, env_monitoring_zone, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", hygienic_area, samp_source_mat_cat, samp_type, spec_intended_cons, cult_result_org, cult_target, biocide_used, food_prod_char, food_prod_synonym, freq_clean, "IFSAC_category", library_prep_kit, nucl_acid_ext_kit, prod_label_claims, repository_name, samp_collect_device, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, misc_param, organism_count, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_dis_point_city, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, ster_meth_samp_room, study_design, study_tmnt, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, subspecf_gen_lin, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE food_human_foods (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	food_trace_list VARCHAR(45), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	ferm_time TEXT, 
	samp_stor_dur TEXT, 
	"ferm_pH" FLOAT, 
	timepoint FLOAT, 
	bacterial_density TEXT, 
	ferm_chem_add TEXT, 
	ferm_rel_humidity TEXT, 
	ferm_temp TEXT, 
	food_pack_capacity TEXT, 
	microb_start_inoc TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	ferm_chem_add_perc TEXT, 
	ferm_headspace_oxy TEXT, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	sample_collec_method TEXT, 
	microb_start TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	"microb_start_taxID" TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	ferm_medium TEXT, 
	ferm_vessel TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	microb_start_prep TEXT, 
	microb_start_source TEXT, 
	nucl_acid_ext_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	microb_start_count VARCHAR(12), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, food_trace_list, pool_dna_extracts, cons_food_stor_dur, ferm_time, samp_stor_dur, "ferm_pH", timepoint, bacterial_density, ferm_chem_add, ferm_rel_humidity, ferm_temp, food_pack_capacity, microb_start_inoc, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, ferm_chem_add_perc, ferm_headspace_oxy, cons_qty_purchased, lot_number, intended_consumer, genetic_mod, enrichment_protocol, sample_collec_method, microb_start, food_dis_point, food_origin, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, "microb_start_taxID", serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, ferm_medium, ferm_vessel, food_prod_synonym, "IFSAC_category", library_prep_kit, microb_start_prep, microb_start_source, nucl_acid_ext_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, microb_start_count, perturbation, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name)
);

CREATE TABLE "food_human_foods_MIGS_bacteria" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	food_trace_list VARCHAR(45), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	ferm_time TEXT, 
	samp_stor_dur TEXT, 
	"ferm_pH" FLOAT, 
	timepoint FLOAT, 
	bacterial_density TEXT, 
	ferm_chem_add TEXT, 
	ferm_rel_humidity TEXT, 
	ferm_temp TEXT, 
	food_pack_capacity TEXT, 
	microb_start_inoc TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	ferm_chem_add_perc TEXT, 
	ferm_headspace_oxy TEXT, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	sample_collec_method TEXT, 
	microb_start TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	"microb_start_taxID" TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	ferm_medium TEXT, 
	ferm_vessel TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	microb_start_prep TEXT, 
	microb_start_source TEXT, 
	nucl_acid_ext_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	microb_start_count VARCHAR(12), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, food_trace_list, pool_dna_extracts, cons_food_stor_dur, ferm_time, samp_stor_dur, "ferm_pH", timepoint, bacterial_density, ferm_chem_add, ferm_rel_humidity, ferm_temp, food_pack_capacity, microb_start_inoc, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, ferm_chem_add_perc, ferm_headspace_oxy, cons_qty_purchased, lot_number, intended_consumer, genetic_mod, enrichment_protocol, sample_collec_method, microb_start, food_dis_point, food_origin, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, "microb_start_taxID", serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, ferm_medium, ferm_vessel, food_prod_synonym, "IFSAC_category", library_prep_kit, microb_start_prep, microb_start_source, nucl_acid_ext_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, microb_start_count, perturbation, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_human_foods_MIGS_eukaryote" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	food_trace_list VARCHAR(45), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	ferm_time TEXT, 
	samp_stor_dur TEXT, 
	"ferm_pH" FLOAT, 
	timepoint FLOAT, 
	bacterial_density TEXT, 
	ferm_chem_add TEXT, 
	ferm_rel_humidity TEXT, 
	ferm_temp TEXT, 
	food_pack_capacity TEXT, 
	microb_start_inoc TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	ferm_chem_add_perc TEXT, 
	ferm_headspace_oxy TEXT, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	sample_collec_method TEXT, 
	microb_start TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	"microb_start_taxID" TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	ferm_medium TEXT, 
	ferm_vessel TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	microb_start_prep TEXT, 
	microb_start_source TEXT, 
	nucl_acid_ext_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	microb_start_count VARCHAR(12), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, food_trace_list, pool_dna_extracts, cons_food_stor_dur, ferm_time, samp_stor_dur, "ferm_pH", timepoint, bacterial_density, ferm_chem_add, ferm_rel_humidity, ferm_temp, food_pack_capacity, microb_start_inoc, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, ferm_chem_add_perc, ferm_headspace_oxy, cons_qty_purchased, lot_number, intended_consumer, genetic_mod, enrichment_protocol, sample_collec_method, microb_start, food_dis_point, food_origin, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, "microb_start_taxID", serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, ferm_medium, ferm_vessel, food_prod_synonym, "IFSAC_category", library_prep_kit, microb_start_prep, microb_start_source, nucl_acid_ext_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, microb_start_count, perturbation, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_human_foods_MIGS_org" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	food_trace_list VARCHAR(45), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	ferm_time TEXT, 
	samp_stor_dur TEXT, 
	"ferm_pH" FLOAT, 
	timepoint FLOAT, 
	bacterial_density TEXT, 
	ferm_chem_add TEXT, 
	ferm_rel_humidity TEXT, 
	ferm_temp TEXT, 
	food_pack_capacity TEXT, 
	microb_start_inoc TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	ferm_chem_add_perc TEXT, 
	ferm_headspace_oxy TEXT, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	sample_collec_method TEXT, 
	microb_start TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	"microb_start_taxID" TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	ferm_medium TEXT, 
	ferm_vessel TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	microb_start_prep TEXT, 
	microb_start_source TEXT, 
	nucl_acid_ext_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	microb_start_count VARCHAR(12), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, food_trace_list, pool_dna_extracts, cons_food_stor_dur, ferm_time, samp_stor_dur, "ferm_pH", timepoint, bacterial_density, ferm_chem_add, ferm_rel_humidity, ferm_temp, food_pack_capacity, microb_start_inoc, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, ferm_chem_add_perc, ferm_headspace_oxy, cons_qty_purchased, lot_number, intended_consumer, genetic_mod, enrichment_protocol, sample_collec_method, microb_start, food_dis_point, food_origin, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, "microb_start_taxID", serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, ferm_medium, ferm_vessel, food_prod_synonym, "IFSAC_category", library_prep_kit, microb_start_prep, microb_start_source, nucl_acid_ext_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, microb_start_count, perturbation, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_human_foods_MIGS_plant" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	food_trace_list VARCHAR(45), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	ferm_time TEXT, 
	samp_stor_dur TEXT, 
	"ferm_pH" FLOAT, 
	timepoint FLOAT, 
	bacterial_density TEXT, 
	ferm_chem_add TEXT, 
	ferm_rel_humidity TEXT, 
	ferm_temp TEXT, 
	food_pack_capacity TEXT, 
	microb_start_inoc TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	ferm_chem_add_perc TEXT, 
	ferm_headspace_oxy TEXT, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	sample_collec_method TEXT, 
	microb_start TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	"microb_start_taxID" TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	ferm_medium TEXT, 
	ferm_vessel TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	microb_start_prep TEXT, 
	microb_start_source TEXT, 
	nucl_acid_ext_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	microb_start_count VARCHAR(12), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, food_trace_list, pool_dna_extracts, cons_food_stor_dur, ferm_time, samp_stor_dur, "ferm_pH", timepoint, bacterial_density, ferm_chem_add, ferm_rel_humidity, ferm_temp, food_pack_capacity, microb_start_inoc, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, ferm_chem_add_perc, ferm_headspace_oxy, cons_qty_purchased, lot_number, intended_consumer, genetic_mod, enrichment_protocol, sample_collec_method, microb_start, food_dis_point, food_origin, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, "microb_start_taxID", serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, ferm_medium, ferm_vessel, food_prod_synonym, "IFSAC_category", library_prep_kit, microb_start_prep, microb_start_source, nucl_acid_ext_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, microb_start_count, perturbation, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_human_foods_MIGS_virus" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	food_trace_list VARCHAR(45), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	ferm_time TEXT, 
	samp_stor_dur TEXT, 
	"ferm_pH" FLOAT, 
	timepoint FLOAT, 
	bacterial_density TEXT, 
	ferm_chem_add TEXT, 
	ferm_rel_humidity TEXT, 
	ferm_temp TEXT, 
	food_pack_capacity TEXT, 
	microb_start_inoc TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	ferm_chem_add_perc TEXT, 
	ferm_headspace_oxy TEXT, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	sample_collec_method TEXT, 
	microb_start TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	"microb_start_taxID" TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	ferm_medium TEXT, 
	ferm_vessel TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	microb_start_prep TEXT, 
	microb_start_source TEXT, 
	nucl_acid_ext_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	microb_start_count VARCHAR(12), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, food_trace_list, pool_dna_extracts, cons_food_stor_dur, ferm_time, samp_stor_dur, "ferm_pH", timepoint, bacterial_density, ferm_chem_add, ferm_rel_humidity, ferm_temp, food_pack_capacity, microb_start_inoc, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, ferm_chem_add_perc, ferm_headspace_oxy, cons_qty_purchased, lot_number, intended_consumer, genetic_mod, enrichment_protocol, sample_collec_method, microb_start, food_dis_point, food_origin, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, "microb_start_taxID", serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, ferm_medium, ferm_vessel, food_prod_synonym, "IFSAC_category", library_prep_kit, microb_start_prep, microb_start_source, nucl_acid_ext_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, microb_start_count, perturbation, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_human_foods_MIMAG" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	food_trace_list VARCHAR(45), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	ferm_time TEXT, 
	samp_stor_dur TEXT, 
	"ferm_pH" FLOAT, 
	timepoint FLOAT, 
	bacterial_density TEXT, 
	ferm_chem_add TEXT, 
	ferm_rel_humidity TEXT, 
	ferm_temp TEXT, 
	food_pack_capacity TEXT, 
	microb_start_inoc TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	ferm_chem_add_perc TEXT, 
	ferm_headspace_oxy TEXT, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	sample_collec_method TEXT, 
	microb_start TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	"microb_start_taxID" TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	ferm_medium TEXT, 
	ferm_vessel TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	microb_start_prep TEXT, 
	microb_start_source TEXT, 
	nucl_acid_ext_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	microb_start_count VARCHAR(12), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, food_trace_list, pool_dna_extracts, cons_food_stor_dur, ferm_time, samp_stor_dur, "ferm_pH", timepoint, bacterial_density, ferm_chem_add, ferm_rel_humidity, ferm_temp, food_pack_capacity, microb_start_inoc, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, ferm_chem_add_perc, ferm_headspace_oxy, cons_qty_purchased, lot_number, intended_consumer, genetic_mod, enrichment_protocol, sample_collec_method, microb_start, food_dis_point, food_origin, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, "microb_start_taxID", serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, ferm_medium, ferm_vessel, food_prod_synonym, "IFSAC_category", library_prep_kit, microb_start_prep, microb_start_source, nucl_acid_ext_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, microb_start_count, perturbation, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_human_foods_MIMARKS_specimen" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	food_trace_list VARCHAR(45), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	ferm_time TEXT, 
	samp_stor_dur TEXT, 
	"ferm_pH" FLOAT, 
	timepoint FLOAT, 
	bacterial_density TEXT, 
	ferm_chem_add TEXT, 
	ferm_rel_humidity TEXT, 
	ferm_temp TEXT, 
	food_pack_capacity TEXT, 
	microb_start_inoc TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	ferm_chem_add_perc TEXT, 
	ferm_headspace_oxy TEXT, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	sample_collec_method TEXT, 
	microb_start TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	"microb_start_taxID" TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	ferm_medium TEXT, 
	ferm_vessel TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	microb_start_prep TEXT, 
	microb_start_source TEXT, 
	nucl_acid_ext_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	microb_start_count VARCHAR(12), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, food_trace_list, pool_dna_extracts, cons_food_stor_dur, ferm_time, samp_stor_dur, "ferm_pH", timepoint, bacterial_density, ferm_chem_add, ferm_rel_humidity, ferm_temp, food_pack_capacity, microb_start_inoc, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, ferm_chem_add_perc, ferm_headspace_oxy, cons_qty_purchased, lot_number, intended_consumer, genetic_mod, enrichment_protocol, sample_collec_method, microb_start, food_dis_point, food_origin, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, "microb_start_taxID", serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, ferm_medium, ferm_vessel, food_prod_synonym, "IFSAC_category", library_prep_kit, microb_start_prep, microb_start_source, nucl_acid_ext_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, microb_start_count, perturbation, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_human_foods_MIMARKS_survey" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	food_trace_list VARCHAR(45), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	ferm_time TEXT, 
	samp_stor_dur TEXT, 
	"ferm_pH" FLOAT, 
	timepoint FLOAT, 
	bacterial_density TEXT, 
	ferm_chem_add TEXT, 
	ferm_rel_humidity TEXT, 
	ferm_temp TEXT, 
	food_pack_capacity TEXT, 
	microb_start_inoc TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	ferm_chem_add_perc TEXT, 
	ferm_headspace_oxy TEXT, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	sample_collec_method TEXT, 
	microb_start TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	"microb_start_taxID" TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	ferm_medium TEXT, 
	ferm_vessel TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	microb_start_prep TEXT, 
	microb_start_source TEXT, 
	nucl_acid_ext_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	microb_start_count VARCHAR(12), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, food_trace_list, pool_dna_extracts, cons_food_stor_dur, ferm_time, samp_stor_dur, "ferm_pH", timepoint, bacterial_density, ferm_chem_add, ferm_rel_humidity, ferm_temp, food_pack_capacity, microb_start_inoc, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, ferm_chem_add_perc, ferm_headspace_oxy, cons_qty_purchased, lot_number, intended_consumer, genetic_mod, enrichment_protocol, sample_collec_method, microb_start, food_dis_point, food_origin, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, "microb_start_taxID", serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, ferm_medium, ferm_vessel, food_prod_synonym, "IFSAC_category", library_prep_kit, microb_start_prep, microb_start_source, nucl_acid_ext_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, microb_start_count, perturbation, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_human_foods_MIMS" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	food_trace_list VARCHAR(45), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	ferm_time TEXT, 
	samp_stor_dur TEXT, 
	"ferm_pH" FLOAT, 
	timepoint FLOAT, 
	bacterial_density TEXT, 
	ferm_chem_add TEXT, 
	ferm_rel_humidity TEXT, 
	ferm_temp TEXT, 
	food_pack_capacity TEXT, 
	microb_start_inoc TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	ferm_chem_add_perc TEXT, 
	ferm_headspace_oxy TEXT, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	sample_collec_method TEXT, 
	microb_start TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	"microb_start_taxID" TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	ferm_medium TEXT, 
	ferm_vessel TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	microb_start_prep TEXT, 
	microb_start_source TEXT, 
	nucl_acid_ext_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	microb_start_count VARCHAR(12), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, food_trace_list, pool_dna_extracts, cons_food_stor_dur, ferm_time, samp_stor_dur, "ferm_pH", timepoint, bacterial_density, ferm_chem_add, ferm_rel_humidity, ferm_temp, food_pack_capacity, microb_start_inoc, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, ferm_chem_add_perc, ferm_headspace_oxy, cons_qty_purchased, lot_number, intended_consumer, genetic_mod, enrichment_protocol, sample_collec_method, microb_start, food_dis_point, food_origin, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, "microb_start_taxID", serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, ferm_medium, ferm_vessel, food_prod_synonym, "IFSAC_category", library_prep_kit, microb_start_prep, microb_start_source, nucl_acid_ext_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, microb_start_count, perturbation, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_human_foods_MISAG" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	food_trace_list VARCHAR(45), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	ferm_time TEXT, 
	samp_stor_dur TEXT, 
	"ferm_pH" FLOAT, 
	timepoint FLOAT, 
	bacterial_density TEXT, 
	ferm_chem_add TEXT, 
	ferm_rel_humidity TEXT, 
	ferm_temp TEXT, 
	food_pack_capacity TEXT, 
	microb_start_inoc TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	ferm_chem_add_perc TEXT, 
	ferm_headspace_oxy TEXT, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	sample_collec_method TEXT, 
	microb_start TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	"microb_start_taxID" TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	ferm_medium TEXT, 
	ferm_vessel TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	microb_start_prep TEXT, 
	microb_start_source TEXT, 
	nucl_acid_ext_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	microb_start_count VARCHAR(12), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, food_trace_list, pool_dna_extracts, cons_food_stor_dur, ferm_time, samp_stor_dur, "ferm_pH", timepoint, bacterial_density, ferm_chem_add, ferm_rel_humidity, ferm_temp, food_pack_capacity, microb_start_inoc, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, ferm_chem_add_perc, ferm_headspace_oxy, cons_qty_purchased, lot_number, intended_consumer, genetic_mod, enrichment_protocol, sample_collec_method, microb_start, food_dis_point, food_origin, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, "microb_start_taxID", serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, ferm_medium, ferm_vessel, food_prod_synonym, "IFSAC_category", library_prep_kit, microb_start_prep, microb_start_source, nucl_acid_ext_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, microb_start_count, perturbation, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "food_human_foods_MIUVIG" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	cult_result VARCHAR(8), 
	samp_purpose VARCHAR(48), 
	food_quality_date VARCHAR(15), 
	samp_transport_cont VARCHAR(25), 
	food_trace_list VARCHAR(45), 
	pool_dna_extracts TEXT, 
	cons_food_stor_dur TEXT, 
	ferm_time TEXT, 
	samp_stor_dur TEXT, 
	"ferm_pH" FLOAT, 
	timepoint FLOAT, 
	bacterial_density TEXT, 
	ferm_chem_add TEXT, 
	ferm_rel_humidity TEXT, 
	ferm_temp TEXT, 
	food_pack_capacity TEXT, 
	microb_start_inoc TEXT, 
	num_samp_collect TEXT, 
	samp_rep_biol TEXT, 
	samp_rep_tech TEXT, 
	samp_stor_temp TEXT, 
	samp_transport_dur TEXT, 
	study_inc_dur TEXT, 
	study_inc_temp TEXT, 
	study_timecourse TEXT, 
	samp_transport_temp TEXT, 
	"spikein_AMR" TEXT, 
	cons_food_stor_temp TEXT, 
	ferm_chem_add_perc TEXT, 
	ferm_headspace_oxy TEXT, 
	cons_qty_purchased TEXT, 
	lot_number TEXT, 
	intended_consumer TEXT, 
	genetic_mod TEXT, 
	enrichment_protocol TEXT, 
	sample_collec_method TEXT, 
	microb_start TEXT, 
	food_dis_point TEXT, 
	food_origin TEXT, 
	food_name_status TEXT, 
	food_prod TEXT, 
	food_product_qual TEXT, 
	food_source TEXT, 
	food_trav_mode TEXT, 
	"HACCP_term" TEXT, 
	samp_source_mat_cat TEXT, 
	cult_result_org TEXT, 
	cult_target TEXT, 
	"microb_start_taxID" TEXT, 
	serovar_or_serotype TEXT, 
	spikein_org TEXT, 
	spikein_serovar TEXT, 
	spikein_strain TEXT, 
	samp_collect_device TEXT, 
	ferm_medium TEXT, 
	ferm_vessel TEXT, 
	food_prod_synonym TEXT, 
	"IFSAC_category" TEXT NOT NULL, 
	library_prep_kit TEXT, 
	microb_start_prep TEXT, 
	microb_start_source TEXT, 
	nucl_acid_ext_kit TEXT, 
	repository_name TEXT, 
	samp_pooling TEXT, 
	samp_stor_loc TEXT, 
	sequencing_kit TEXT, 
	sequencing_location TEXT, 
	spikein_antibiotic TEXT, 
	spikein_metal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	spikein_count VARCHAR(5), 
	microb_start_count VARCHAR(12), 
	perturbation TEXT, 
	coll_site_geo_feat TEXT NOT NULL, 
	dietary_claim_use TEXT, 
	food_additive TEXT, 
	food_allergen_label TEXT, 
	food_contact_surf TEXT, 
	food_contain_wrap TEXT, 
	food_cooking_proc TEXT, 
	food_ingredient TEXT, 
	food_pack_integrity TEXT, 
	food_pack_medium TEXT, 
	food_preserv_proc TEXT, 
	food_prior_contact TEXT, 
	food_product_type TEXT NOT NULL, 
	food_trav_vehic TEXT, 
	food_treat_proc TEXT, 
	microb_cult_med TEXT, 
	part_plant_animal TEXT, 
	samp_stor_device TEXT, 
	samp_stor_media TEXT, 
	spikein_growth_med TEXT, 
	study_design TEXT, 
	study_tmnt TEXT, 
	cons_purch_date DATE, 
	cult_isol_date DATE, 
	samp_size TEXT, 
	samp_vol_we_dna_ext TEXT, 
	nucl_acid_ext TEXT, 
	experimental_factor TEXT, 
	seq_meth TEXT NOT NULL, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, cult_result, samp_purpose, food_quality_date, samp_transport_cont, food_trace_list, pool_dna_extracts, cons_food_stor_dur, ferm_time, samp_stor_dur, "ferm_pH", timepoint, bacterial_density, ferm_chem_add, ferm_rel_humidity, ferm_temp, food_pack_capacity, microb_start_inoc, num_samp_collect, samp_rep_biol, samp_rep_tech, samp_stor_temp, samp_transport_dur, study_inc_dur, study_inc_temp, study_timecourse, samp_transport_temp, "spikein_AMR", cons_food_stor_temp, ferm_chem_add_perc, ferm_headspace_oxy, cons_qty_purchased, lot_number, intended_consumer, genetic_mod, enrichment_protocol, sample_collec_method, microb_start, food_dis_point, food_origin, food_name_status, food_prod, food_product_qual, food_source, food_trav_mode, "HACCP_term", samp_source_mat_cat, cult_result_org, cult_target, "microb_start_taxID", serovar_or_serotype, spikein_org, spikein_serovar, spikein_strain, samp_collect_device, ferm_medium, ferm_vessel, food_prod_synonym, "IFSAC_category", library_prep_kit, microb_start_prep, microb_start_source, nucl_acid_ext_kit, repository_name, samp_pooling, samp_stor_loc, sequencing_kit, sequencing_location, spikein_antibiotic, spikein_metal, misc_param, organism_count, spikein_count, microb_start_count, perturbation, coll_site_geo_feat, dietary_claim_use, food_additive, food_allergen_label, food_contact_surf, food_contain_wrap, food_cooking_proc, food_ingredient, food_pack_integrity, food_pack_medium, food_preserv_proc, food_prior_contact, food_product_type, food_trav_vehic, food_treat_proc, microb_cult_med, part_plant_animal, samp_stor_device, samp_stor_media, spikein_growth_med, study_design, study_tmnt, cons_purch_date, cult_isol_date, samp_size, samp_vol_we_dna_ext, nucl_acid_ext, experimental_factor, seq_meth, project_name, samp_name, samp_taxon_id, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, source_uvig, virus_enrich_appr, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE host_associated (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	samp_dis_stage VARCHAR(23), 
	host_sex VARCHAR(28), 
	biol_stat VARCHAR(16), 
	gravidity TEXT, 
	samp_store_dur TEXT, 
	blood_press_diast TEXT, 
	blood_press_syst TEXT, 
	host_age TEXT, 
	host_body_temp TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	host_growth_cond TEXT, 
	host_subspecf_genlin TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ances_data TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, samp_dis_stage, host_sex, biol_stat, gravidity, samp_store_dur, blood_press_diast, blood_press_syst, host_age, host_body_temp, host_dry_mass, host_height, host_length, host_tot_mass, salinity, samp_store_temp, host_taxid, genetic_mod, host_growth_cond, host_subspecf_genlin, host_body_product, host_body_site, host_phenotype, chem_administration, ances_data, host_body_habitat, host_color, host_common_name, host_diet, host_genotype, host_life_stage, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name)
);

CREATE TABLE "host_associated_MIGS_bacteria" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	samp_dis_stage VARCHAR(23), 
	host_sex VARCHAR(28), 
	biol_stat VARCHAR(16), 
	gravidity TEXT, 
	samp_store_dur TEXT, 
	blood_press_diast TEXT, 
	blood_press_syst TEXT, 
	host_age TEXT, 
	host_body_temp TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	host_growth_cond TEXT, 
	host_subspecf_genlin TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ances_data TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, samp_dis_stage, host_sex, biol_stat, gravidity, samp_store_dur, blood_press_diast, blood_press_syst, host_age, host_body_temp, host_dry_mass, host_height, host_length, host_tot_mass, salinity, samp_store_temp, host_taxid, genetic_mod, host_growth_cond, host_subspecf_genlin, host_body_product, host_body_site, host_phenotype, chem_administration, ances_data, host_body_habitat, host_color, host_common_name, host_diet, host_genotype, host_life_stage, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "host_associated_MIGS_eukaryote" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	samp_dis_stage VARCHAR(23), 
	host_sex VARCHAR(28), 
	biol_stat VARCHAR(16), 
	gravidity TEXT, 
	samp_store_dur TEXT, 
	blood_press_diast TEXT, 
	blood_press_syst TEXT, 
	host_age TEXT, 
	host_body_temp TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	host_growth_cond TEXT, 
	host_subspecf_genlin TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ances_data TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, samp_dis_stage, host_sex, biol_stat, gravidity, samp_store_dur, blood_press_diast, blood_press_syst, host_age, host_body_temp, host_dry_mass, host_height, host_length, host_tot_mass, salinity, samp_store_temp, host_taxid, genetic_mod, host_growth_cond, host_subspecf_genlin, host_body_product, host_body_site, host_phenotype, chem_administration, ances_data, host_body_habitat, host_color, host_common_name, host_diet, host_genotype, host_life_stage, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "host_associated_MIGS_org" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	samp_dis_stage VARCHAR(23), 
	host_sex VARCHAR(28), 
	biol_stat VARCHAR(16), 
	gravidity TEXT, 
	samp_store_dur TEXT, 
	blood_press_diast TEXT, 
	blood_press_syst TEXT, 
	host_age TEXT, 
	host_body_temp TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	host_growth_cond TEXT, 
	host_subspecf_genlin TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ances_data TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, samp_dis_stage, host_sex, biol_stat, gravidity, samp_store_dur, blood_press_diast, blood_press_syst, host_age, host_body_temp, host_dry_mass, host_height, host_length, host_tot_mass, salinity, samp_store_temp, host_taxid, genetic_mod, host_growth_cond, host_subspecf_genlin, host_body_product, host_body_site, host_phenotype, chem_administration, ances_data, host_body_habitat, host_color, host_common_name, host_diet, host_genotype, host_life_stage, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "host_associated_MIGS_plant" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	samp_dis_stage VARCHAR(23), 
	host_sex VARCHAR(28), 
	biol_stat VARCHAR(16), 
	gravidity TEXT, 
	samp_store_dur TEXT, 
	blood_press_diast TEXT, 
	blood_press_syst TEXT, 
	host_age TEXT, 
	host_body_temp TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	host_growth_cond TEXT, 
	host_subspecf_genlin TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ances_data TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, samp_dis_stage, host_sex, biol_stat, gravidity, samp_store_dur, blood_press_diast, blood_press_syst, host_age, host_body_temp, host_dry_mass, host_height, host_length, host_tot_mass, salinity, samp_store_temp, host_taxid, genetic_mod, host_growth_cond, host_subspecf_genlin, host_body_product, host_body_site, host_phenotype, chem_administration, ances_data, host_body_habitat, host_color, host_common_name, host_diet, host_genotype, host_life_stage, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "host_associated_MIGS_virus" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	samp_dis_stage VARCHAR(23), 
	host_sex VARCHAR(28), 
	biol_stat VARCHAR(16), 
	gravidity TEXT, 
	samp_store_dur TEXT, 
	blood_press_diast TEXT, 
	blood_press_syst TEXT, 
	host_age TEXT, 
	host_body_temp TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	host_growth_cond TEXT, 
	host_subspecf_genlin TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ances_data TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, samp_dis_stage, host_sex, biol_stat, gravidity, samp_store_dur, blood_press_diast, blood_press_syst, host_age, host_body_temp, host_dry_mass, host_height, host_length, host_tot_mass, salinity, samp_store_temp, host_taxid, genetic_mod, host_growth_cond, host_subspecf_genlin, host_body_product, host_body_site, host_phenotype, chem_administration, ances_data, host_body_habitat, host_color, host_common_name, host_diet, host_genotype, host_life_stage, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "host_associated_MIMAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	samp_dis_stage VARCHAR(23), 
	host_sex VARCHAR(28), 
	biol_stat VARCHAR(16), 
	gravidity TEXT, 
	samp_store_dur TEXT, 
	blood_press_diast TEXT, 
	blood_press_syst TEXT, 
	host_age TEXT, 
	host_body_temp TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	host_growth_cond TEXT, 
	host_subspecf_genlin TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ances_data TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, samp_dis_stage, host_sex, biol_stat, gravidity, samp_store_dur, blood_press_diast, blood_press_syst, host_age, host_body_temp, host_dry_mass, host_height, host_length, host_tot_mass, salinity, samp_store_temp, host_taxid, genetic_mod, host_growth_cond, host_subspecf_genlin, host_body_product, host_body_site, host_phenotype, chem_administration, ances_data, host_body_habitat, host_color, host_common_name, host_diet, host_genotype, host_life_stage, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "host_associated_MIMARKS_specimen" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	samp_dis_stage VARCHAR(23), 
	host_sex VARCHAR(28), 
	biol_stat VARCHAR(16), 
	gravidity TEXT, 
	samp_store_dur TEXT, 
	blood_press_diast TEXT, 
	blood_press_syst TEXT, 
	host_age TEXT, 
	host_body_temp TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	host_growth_cond TEXT, 
	host_subspecf_genlin TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ances_data TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, samp_dis_stage, host_sex, biol_stat, gravidity, samp_store_dur, blood_press_diast, blood_press_syst, host_age, host_body_temp, host_dry_mass, host_height, host_length, host_tot_mass, salinity, samp_store_temp, host_taxid, genetic_mod, host_growth_cond, host_subspecf_genlin, host_body_product, host_body_site, host_phenotype, chem_administration, ances_data, host_body_habitat, host_color, host_common_name, host_diet, host_genotype, host_life_stage, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "host_associated_MIMARKS_survey" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	samp_dis_stage VARCHAR(23), 
	host_sex VARCHAR(28), 
	biol_stat VARCHAR(16), 
	gravidity TEXT, 
	samp_store_dur TEXT, 
	blood_press_diast TEXT, 
	blood_press_syst TEXT, 
	host_age TEXT, 
	host_body_temp TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	host_growth_cond TEXT, 
	host_subspecf_genlin TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ances_data TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, samp_dis_stage, host_sex, biol_stat, gravidity, samp_store_dur, blood_press_diast, blood_press_syst, host_age, host_body_temp, host_dry_mass, host_height, host_length, host_tot_mass, salinity, samp_store_temp, host_taxid, genetic_mod, host_growth_cond, host_subspecf_genlin, host_body_product, host_body_site, host_phenotype, chem_administration, ances_data, host_body_habitat, host_color, host_common_name, host_diet, host_genotype, host_life_stage, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "host_associated_MIMS" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	samp_dis_stage VARCHAR(23), 
	host_sex VARCHAR(28), 
	biol_stat VARCHAR(16), 
	gravidity TEXT, 
	samp_store_dur TEXT, 
	blood_press_diast TEXT, 
	blood_press_syst TEXT, 
	host_age TEXT, 
	host_body_temp TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	host_growth_cond TEXT, 
	host_subspecf_genlin TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ances_data TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, samp_dis_stage, host_sex, biol_stat, gravidity, samp_store_dur, blood_press_diast, blood_press_syst, host_age, host_body_temp, host_dry_mass, host_height, host_length, host_tot_mass, salinity, samp_store_temp, host_taxid, genetic_mod, host_growth_cond, host_subspecf_genlin, host_body_product, host_body_site, host_phenotype, chem_administration, ances_data, host_body_habitat, host_color, host_common_name, host_diet, host_genotype, host_life_stage, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "host_associated_MISAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	samp_dis_stage VARCHAR(23), 
	host_sex VARCHAR(28), 
	biol_stat VARCHAR(16), 
	gravidity TEXT, 
	samp_store_dur TEXT, 
	blood_press_diast TEXT, 
	blood_press_syst TEXT, 
	host_age TEXT, 
	host_body_temp TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	host_growth_cond TEXT, 
	host_subspecf_genlin TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ances_data TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, samp_dis_stage, host_sex, biol_stat, gravidity, samp_store_dur, blood_press_diast, blood_press_syst, host_age, host_body_temp, host_dry_mass, host_height, host_length, host_tot_mass, salinity, samp_store_temp, host_taxid, genetic_mod, host_growth_cond, host_subspecf_genlin, host_body_product, host_body_site, host_phenotype, chem_administration, ances_data, host_body_habitat, host_color, host_common_name, host_diet, host_genotype, host_life_stage, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "host_associated_MIUVIG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	samp_dis_stage VARCHAR(23), 
	host_sex VARCHAR(28), 
	biol_stat VARCHAR(16), 
	gravidity TEXT, 
	samp_store_dur TEXT, 
	blood_press_diast TEXT, 
	blood_press_syst TEXT, 
	host_age TEXT, 
	host_body_temp TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	host_growth_cond TEXT, 
	host_subspecf_genlin TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ances_data TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, samp_dis_stage, host_sex, biol_stat, gravidity, samp_store_dur, blood_press_diast, blood_press_syst, host_age, host_body_temp, host_dry_mass, host_height, host_length, host_tot_mass, salinity, samp_store_temp, host_taxid, genetic_mod, host_growth_cond, host_subspecf_genlin, host_body_product, host_body_site, host_phenotype, chem_administration, ances_data, host_body_habitat, host_color, host_common_name, host_diet, host_genotype, host_life_stage, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE human_associated (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	urine_collect_meth TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	smoker TEXT, 
	twin_sibling TEXT, 
	study_complt_stat VARCHAR(17), 
	host_hiv_stat TEXT, 
	weight_loss_3_month TEXT, 
	diet_last_six_month TEXT, 
	pet_farm_animal TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	amniotic_fluid_color TEXT, 
	blood_blood_disord TEXT, 
	ethnicity TEXT, 
	foetal_health_stat TEXT, 
	gestation_state TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	kidney_disord TEXT, 
	maternal_health_stat TEXT, 
	nose_throat_disord TEXT, 
	pulmonary_disord TEXT, 
	samp_store_loc TEXT, 
	travel_out_six_month TEXT, 
	urogenit_tract_disor TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	drug_usage VARCHAR(5), 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, urine_collect_meth, host_sex, medic_hist_perform, smoker, twin_sibling, study_complt_stat, host_hiv_stat, weight_loss_3_month, diet_last_six_month, pet_farm_animal, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, amniotic_fluid_color, blood_blood_disord, ethnicity, foetal_health_stat, gestation_state, host_diet, host_genotype, host_subject_id, host_symbiont, kidney_disord, maternal_health_stat, nose_throat_disord, pulmonary_disord, samp_store_loc, travel_out_six_month, urogenit_tract_disor, host_last_meal, misc_param, organism_count, drug_usage, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name)
);

CREATE TABLE "human_associated_MIGS_bacteria" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	urine_collect_meth TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	smoker TEXT, 
	twin_sibling TEXT, 
	study_complt_stat VARCHAR(17), 
	host_hiv_stat TEXT, 
	weight_loss_3_month TEXT, 
	diet_last_six_month TEXT, 
	pet_farm_animal TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	amniotic_fluid_color TEXT, 
	blood_blood_disord TEXT, 
	ethnicity TEXT, 
	foetal_health_stat TEXT, 
	gestation_state TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	kidney_disord TEXT, 
	maternal_health_stat TEXT, 
	nose_throat_disord TEXT, 
	pulmonary_disord TEXT, 
	samp_store_loc TEXT, 
	travel_out_six_month TEXT, 
	urogenit_tract_disor TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	drug_usage VARCHAR(5), 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, urine_collect_meth, host_sex, medic_hist_perform, smoker, twin_sibling, study_complt_stat, host_hiv_stat, weight_loss_3_month, diet_last_six_month, pet_farm_animal, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, amniotic_fluid_color, blood_blood_disord, ethnicity, foetal_health_stat, gestation_state, host_diet, host_genotype, host_subject_id, host_symbiont, kidney_disord, maternal_health_stat, nose_throat_disord, pulmonary_disord, samp_store_loc, travel_out_six_month, urogenit_tract_disor, host_last_meal, misc_param, organism_count, drug_usage, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_associated_MIGS_eukaryote" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	urine_collect_meth TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	smoker TEXT, 
	twin_sibling TEXT, 
	study_complt_stat VARCHAR(17), 
	host_hiv_stat TEXT, 
	weight_loss_3_month TEXT, 
	diet_last_six_month TEXT, 
	pet_farm_animal TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	amniotic_fluid_color TEXT, 
	blood_blood_disord TEXT, 
	ethnicity TEXT, 
	foetal_health_stat TEXT, 
	gestation_state TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	kidney_disord TEXT, 
	maternal_health_stat TEXT, 
	nose_throat_disord TEXT, 
	pulmonary_disord TEXT, 
	samp_store_loc TEXT, 
	travel_out_six_month TEXT, 
	urogenit_tract_disor TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	drug_usage VARCHAR(5), 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, urine_collect_meth, host_sex, medic_hist_perform, smoker, twin_sibling, study_complt_stat, host_hiv_stat, weight_loss_3_month, diet_last_six_month, pet_farm_animal, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, amniotic_fluid_color, blood_blood_disord, ethnicity, foetal_health_stat, gestation_state, host_diet, host_genotype, host_subject_id, host_symbiont, kidney_disord, maternal_health_stat, nose_throat_disord, pulmonary_disord, samp_store_loc, travel_out_six_month, urogenit_tract_disor, host_last_meal, misc_param, organism_count, drug_usage, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_associated_MIGS_org" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	urine_collect_meth TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	smoker TEXT, 
	twin_sibling TEXT, 
	study_complt_stat VARCHAR(17), 
	host_hiv_stat TEXT, 
	weight_loss_3_month TEXT, 
	diet_last_six_month TEXT, 
	pet_farm_animal TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	amniotic_fluid_color TEXT, 
	blood_blood_disord TEXT, 
	ethnicity TEXT, 
	foetal_health_stat TEXT, 
	gestation_state TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	kidney_disord TEXT, 
	maternal_health_stat TEXT, 
	nose_throat_disord TEXT, 
	pulmonary_disord TEXT, 
	samp_store_loc TEXT, 
	travel_out_six_month TEXT, 
	urogenit_tract_disor TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	drug_usage VARCHAR(5), 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, urine_collect_meth, host_sex, medic_hist_perform, smoker, twin_sibling, study_complt_stat, host_hiv_stat, weight_loss_3_month, diet_last_six_month, pet_farm_animal, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, amniotic_fluid_color, blood_blood_disord, ethnicity, foetal_health_stat, gestation_state, host_diet, host_genotype, host_subject_id, host_symbiont, kidney_disord, maternal_health_stat, nose_throat_disord, pulmonary_disord, samp_store_loc, travel_out_six_month, urogenit_tract_disor, host_last_meal, misc_param, organism_count, drug_usage, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_associated_MIGS_plant" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	urine_collect_meth TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	smoker TEXT, 
	twin_sibling TEXT, 
	study_complt_stat VARCHAR(17), 
	host_hiv_stat TEXT, 
	weight_loss_3_month TEXT, 
	diet_last_six_month TEXT, 
	pet_farm_animal TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	amniotic_fluid_color TEXT, 
	blood_blood_disord TEXT, 
	ethnicity TEXT, 
	foetal_health_stat TEXT, 
	gestation_state TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	kidney_disord TEXT, 
	maternal_health_stat TEXT, 
	nose_throat_disord TEXT, 
	pulmonary_disord TEXT, 
	samp_store_loc TEXT, 
	travel_out_six_month TEXT, 
	urogenit_tract_disor TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	drug_usage VARCHAR(5), 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, urine_collect_meth, host_sex, medic_hist_perform, smoker, twin_sibling, study_complt_stat, host_hiv_stat, weight_loss_3_month, diet_last_six_month, pet_farm_animal, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, amniotic_fluid_color, blood_blood_disord, ethnicity, foetal_health_stat, gestation_state, host_diet, host_genotype, host_subject_id, host_symbiont, kidney_disord, maternal_health_stat, nose_throat_disord, pulmonary_disord, samp_store_loc, travel_out_six_month, urogenit_tract_disor, host_last_meal, misc_param, organism_count, drug_usage, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_associated_MIGS_virus" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	urine_collect_meth TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	smoker TEXT, 
	twin_sibling TEXT, 
	study_complt_stat VARCHAR(17), 
	host_hiv_stat TEXT, 
	weight_loss_3_month TEXT, 
	diet_last_six_month TEXT, 
	pet_farm_animal TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	amniotic_fluid_color TEXT, 
	blood_blood_disord TEXT, 
	ethnicity TEXT, 
	foetal_health_stat TEXT, 
	gestation_state TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	kidney_disord TEXT, 
	maternal_health_stat TEXT, 
	nose_throat_disord TEXT, 
	pulmonary_disord TEXT, 
	samp_store_loc TEXT, 
	travel_out_six_month TEXT, 
	urogenit_tract_disor TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	drug_usage VARCHAR(5), 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, urine_collect_meth, host_sex, medic_hist_perform, smoker, twin_sibling, study_complt_stat, host_hiv_stat, weight_loss_3_month, diet_last_six_month, pet_farm_animal, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, amniotic_fluid_color, blood_blood_disord, ethnicity, foetal_health_stat, gestation_state, host_diet, host_genotype, host_subject_id, host_symbiont, kidney_disord, maternal_health_stat, nose_throat_disord, pulmonary_disord, samp_store_loc, travel_out_six_month, urogenit_tract_disor, host_last_meal, misc_param, organism_count, drug_usage, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_associated_MIMAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	urine_collect_meth TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	smoker TEXT, 
	twin_sibling TEXT, 
	study_complt_stat VARCHAR(17), 
	host_hiv_stat TEXT, 
	weight_loss_3_month TEXT, 
	diet_last_six_month TEXT, 
	pet_farm_animal TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	amniotic_fluid_color TEXT, 
	blood_blood_disord TEXT, 
	ethnicity TEXT, 
	foetal_health_stat TEXT, 
	gestation_state TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	kidney_disord TEXT, 
	maternal_health_stat TEXT, 
	nose_throat_disord TEXT, 
	pulmonary_disord TEXT, 
	samp_store_loc TEXT, 
	travel_out_six_month TEXT, 
	urogenit_tract_disor TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	drug_usage VARCHAR(5), 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, urine_collect_meth, host_sex, medic_hist_perform, smoker, twin_sibling, study_complt_stat, host_hiv_stat, weight_loss_3_month, diet_last_six_month, pet_farm_animal, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, amniotic_fluid_color, blood_blood_disord, ethnicity, foetal_health_stat, gestation_state, host_diet, host_genotype, host_subject_id, host_symbiont, kidney_disord, maternal_health_stat, nose_throat_disord, pulmonary_disord, samp_store_loc, travel_out_six_month, urogenit_tract_disor, host_last_meal, misc_param, organism_count, drug_usage, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_associated_MIMARKS_specimen" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	urine_collect_meth TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	smoker TEXT, 
	twin_sibling TEXT, 
	study_complt_stat VARCHAR(17), 
	host_hiv_stat TEXT, 
	weight_loss_3_month TEXT, 
	diet_last_six_month TEXT, 
	pet_farm_animal TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	amniotic_fluid_color TEXT, 
	blood_blood_disord TEXT, 
	ethnicity TEXT, 
	foetal_health_stat TEXT, 
	gestation_state TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	kidney_disord TEXT, 
	maternal_health_stat TEXT, 
	nose_throat_disord TEXT, 
	pulmonary_disord TEXT, 
	samp_store_loc TEXT, 
	travel_out_six_month TEXT, 
	urogenit_tract_disor TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	drug_usage VARCHAR(5), 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, urine_collect_meth, host_sex, medic_hist_perform, smoker, twin_sibling, study_complt_stat, host_hiv_stat, weight_loss_3_month, diet_last_six_month, pet_farm_animal, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, amniotic_fluid_color, blood_blood_disord, ethnicity, foetal_health_stat, gestation_state, host_diet, host_genotype, host_subject_id, host_symbiont, kidney_disord, maternal_health_stat, nose_throat_disord, pulmonary_disord, samp_store_loc, travel_out_six_month, urogenit_tract_disor, host_last_meal, misc_param, organism_count, drug_usage, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_associated_MIMARKS_survey" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	urine_collect_meth TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	smoker TEXT, 
	twin_sibling TEXT, 
	study_complt_stat VARCHAR(17), 
	host_hiv_stat TEXT, 
	weight_loss_3_month TEXT, 
	diet_last_six_month TEXT, 
	pet_farm_animal TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	amniotic_fluid_color TEXT, 
	blood_blood_disord TEXT, 
	ethnicity TEXT, 
	foetal_health_stat TEXT, 
	gestation_state TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	kidney_disord TEXT, 
	maternal_health_stat TEXT, 
	nose_throat_disord TEXT, 
	pulmonary_disord TEXT, 
	samp_store_loc TEXT, 
	travel_out_six_month TEXT, 
	urogenit_tract_disor TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	drug_usage VARCHAR(5), 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, urine_collect_meth, host_sex, medic_hist_perform, smoker, twin_sibling, study_complt_stat, host_hiv_stat, weight_loss_3_month, diet_last_six_month, pet_farm_animal, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, amniotic_fluid_color, blood_blood_disord, ethnicity, foetal_health_stat, gestation_state, host_diet, host_genotype, host_subject_id, host_symbiont, kidney_disord, maternal_health_stat, nose_throat_disord, pulmonary_disord, samp_store_loc, travel_out_six_month, urogenit_tract_disor, host_last_meal, misc_param, organism_count, drug_usage, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_associated_MIMS" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	urine_collect_meth TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	smoker TEXT, 
	twin_sibling TEXT, 
	study_complt_stat VARCHAR(17), 
	host_hiv_stat TEXT, 
	weight_loss_3_month TEXT, 
	diet_last_six_month TEXT, 
	pet_farm_animal TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	amniotic_fluid_color TEXT, 
	blood_blood_disord TEXT, 
	ethnicity TEXT, 
	foetal_health_stat TEXT, 
	gestation_state TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	kidney_disord TEXT, 
	maternal_health_stat TEXT, 
	nose_throat_disord TEXT, 
	pulmonary_disord TEXT, 
	samp_store_loc TEXT, 
	travel_out_six_month TEXT, 
	urogenit_tract_disor TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	drug_usage VARCHAR(5), 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, urine_collect_meth, host_sex, medic_hist_perform, smoker, twin_sibling, study_complt_stat, host_hiv_stat, weight_loss_3_month, diet_last_six_month, pet_farm_animal, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, amniotic_fluid_color, blood_blood_disord, ethnicity, foetal_health_stat, gestation_state, host_diet, host_genotype, host_subject_id, host_symbiont, kidney_disord, maternal_health_stat, nose_throat_disord, pulmonary_disord, samp_store_loc, travel_out_six_month, urogenit_tract_disor, host_last_meal, misc_param, organism_count, drug_usage, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_associated_MISAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	urine_collect_meth TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	smoker TEXT, 
	twin_sibling TEXT, 
	study_complt_stat VARCHAR(17), 
	host_hiv_stat TEXT, 
	weight_loss_3_month TEXT, 
	diet_last_six_month TEXT, 
	pet_farm_animal TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	amniotic_fluid_color TEXT, 
	blood_blood_disord TEXT, 
	ethnicity TEXT, 
	foetal_health_stat TEXT, 
	gestation_state TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	kidney_disord TEXT, 
	maternal_health_stat TEXT, 
	nose_throat_disord TEXT, 
	pulmonary_disord TEXT, 
	samp_store_loc TEXT, 
	travel_out_six_month TEXT, 
	urogenit_tract_disor TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	drug_usage VARCHAR(5), 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, urine_collect_meth, host_sex, medic_hist_perform, smoker, twin_sibling, study_complt_stat, host_hiv_stat, weight_loss_3_month, diet_last_six_month, pet_farm_animal, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, amniotic_fluid_color, blood_blood_disord, ethnicity, foetal_health_stat, gestation_state, host_diet, host_genotype, host_subject_id, host_symbiont, kidney_disord, maternal_health_stat, nose_throat_disord, pulmonary_disord, samp_store_loc, travel_out_six_month, urogenit_tract_disor, host_last_meal, misc_param, organism_count, drug_usage, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_associated_MIUVIG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	urine_collect_meth TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	smoker TEXT, 
	twin_sibling TEXT, 
	study_complt_stat VARCHAR(17), 
	host_hiv_stat TEXT, 
	weight_loss_3_month TEXT, 
	diet_last_six_month TEXT, 
	pet_farm_animal TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	amniotic_fluid_color TEXT, 
	blood_blood_disord TEXT, 
	ethnicity TEXT, 
	foetal_health_stat TEXT, 
	gestation_state TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	kidney_disord TEXT, 
	maternal_health_stat TEXT, 
	nose_throat_disord TEXT, 
	pulmonary_disord TEXT, 
	samp_store_loc TEXT, 
	travel_out_six_month TEXT, 
	urogenit_tract_disor TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	drug_usage VARCHAR(5), 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, urine_collect_meth, host_sex, medic_hist_perform, smoker, twin_sibling, study_complt_stat, host_hiv_stat, weight_loss_3_month, diet_last_six_month, pet_farm_animal, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, amniotic_fluid_color, blood_blood_disord, ethnicity, foetal_health_stat, gestation_state, host_diet, host_genotype, host_subject_id, host_symbiont, kidney_disord, maternal_health_stat, nose_throat_disord, pulmonary_disord, samp_store_loc, travel_out_six_month, urogenit_tract_disor, host_last_meal, misc_param, organism_count, drug_usage, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE human_gut (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	special_diet VARCHAR(22), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	gastrointest_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	liver_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, special_diet, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, gastrointest_disord, host_diet, host_genotype, host_subject_id, host_symbiont, liver_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name)
);

CREATE TABLE "human_gut_MIGS_bacteria" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	special_diet VARCHAR(22), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	gastrointest_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	liver_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, special_diet, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, gastrointest_disord, host_diet, host_genotype, host_subject_id, host_symbiont, liver_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_gut_MIGS_eukaryote" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	special_diet VARCHAR(22), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	gastrointest_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	liver_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, special_diet, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, gastrointest_disord, host_diet, host_genotype, host_subject_id, host_symbiont, liver_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_gut_MIGS_org" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	special_diet VARCHAR(22), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	gastrointest_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	liver_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, special_diet, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, gastrointest_disord, host_diet, host_genotype, host_subject_id, host_symbiont, liver_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_gut_MIGS_plant" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	special_diet VARCHAR(22), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	gastrointest_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	liver_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, special_diet, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, gastrointest_disord, host_diet, host_genotype, host_subject_id, host_symbiont, liver_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_gut_MIGS_virus" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	special_diet VARCHAR(22), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	gastrointest_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	liver_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, special_diet, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, gastrointest_disord, host_diet, host_genotype, host_subject_id, host_symbiont, liver_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_gut_MIMAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	special_diet VARCHAR(22), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	gastrointest_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	liver_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, special_diet, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, gastrointest_disord, host_diet, host_genotype, host_subject_id, host_symbiont, liver_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_gut_MIMARKS_specimen" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	special_diet VARCHAR(22), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	gastrointest_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	liver_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, special_diet, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, gastrointest_disord, host_diet, host_genotype, host_subject_id, host_symbiont, liver_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_gut_MIMARKS_survey" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	special_diet VARCHAR(22), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	gastrointest_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	liver_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, special_diet, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, gastrointest_disord, host_diet, host_genotype, host_subject_id, host_symbiont, liver_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_gut_MIMS" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	special_diet VARCHAR(22), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	gastrointest_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	liver_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, special_diet, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, gastrointest_disord, host_diet, host_genotype, host_subject_id, host_symbiont, liver_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_gut_MISAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	special_diet VARCHAR(22), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	gastrointest_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	liver_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, special_diet, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, gastrointest_disord, host_diet, host_genotype, host_subject_id, host_symbiont, liver_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_gut_MIUVIG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	special_diet VARCHAR(22), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	gastrointest_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	liver_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, special_diet, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, gastrointest_disord, host_diet, host_genotype, host_subject_id, host_symbiont, liver_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE human_oral (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_last_toothbrush TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	nose_throat_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, medic_hist_perform, samp_store_dur, time_last_toothbrush, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, nose_throat_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name)
);

CREATE TABLE "human_oral_MIGS_bacteria" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_last_toothbrush TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	nose_throat_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, medic_hist_perform, samp_store_dur, time_last_toothbrush, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, nose_throat_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_oral_MIGS_eukaryote" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_last_toothbrush TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	nose_throat_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, medic_hist_perform, samp_store_dur, time_last_toothbrush, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, nose_throat_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_oral_MIGS_org" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_last_toothbrush TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	nose_throat_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, medic_hist_perform, samp_store_dur, time_last_toothbrush, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, nose_throat_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_oral_MIGS_plant" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_last_toothbrush TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	nose_throat_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, medic_hist_perform, samp_store_dur, time_last_toothbrush, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, nose_throat_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_oral_MIGS_virus" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_last_toothbrush TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	nose_throat_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, medic_hist_perform, samp_store_dur, time_last_toothbrush, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, nose_throat_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_oral_MIMAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_last_toothbrush TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	nose_throat_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, medic_hist_perform, samp_store_dur, time_last_toothbrush, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, nose_throat_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_oral_MIMARKS_specimen" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_last_toothbrush TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	nose_throat_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, medic_hist_perform, samp_store_dur, time_last_toothbrush, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, nose_throat_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_oral_MIMARKS_survey" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_last_toothbrush TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	nose_throat_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, medic_hist_perform, samp_store_dur, time_last_toothbrush, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, nose_throat_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_oral_MIMS" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_last_toothbrush TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	nose_throat_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, medic_hist_perform, samp_store_dur, time_last_toothbrush, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, nose_throat_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_oral_MISAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_last_toothbrush TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	nose_throat_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, medic_hist_perform, samp_store_dur, time_last_toothbrush, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, nose_throat_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_oral_MIUVIG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_last_toothbrush TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	nose_throat_disord TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, medic_hist_perform, samp_store_dur, time_last_toothbrush, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, nose_throat_disord, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE human_skin (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	dominant_hand VARCHAR(12), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_since_last_wash TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	dermatology_disord TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, dominant_hand, medic_hist_perform, samp_store_dur, time_since_last_wash, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, dermatology_disord, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name)
);

CREATE TABLE "human_skin_MIGS_bacteria" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	dominant_hand VARCHAR(12), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_since_last_wash TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	dermatology_disord TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, dominant_hand, medic_hist_perform, samp_store_dur, time_since_last_wash, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, dermatology_disord, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_skin_MIGS_eukaryote" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	dominant_hand VARCHAR(12), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_since_last_wash TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	dermatology_disord TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, dominant_hand, medic_hist_perform, samp_store_dur, time_since_last_wash, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, dermatology_disord, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_skin_MIGS_org" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	dominant_hand VARCHAR(12), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_since_last_wash TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	dermatology_disord TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, dominant_hand, medic_hist_perform, samp_store_dur, time_since_last_wash, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, dermatology_disord, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_skin_MIGS_plant" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	dominant_hand VARCHAR(12), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_since_last_wash TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	dermatology_disord TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, dominant_hand, medic_hist_perform, samp_store_dur, time_since_last_wash, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, dermatology_disord, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_skin_MIGS_virus" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	dominant_hand VARCHAR(12), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_since_last_wash TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	dermatology_disord TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, dominant_hand, medic_hist_perform, samp_store_dur, time_since_last_wash, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, dermatology_disord, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_skin_MIMAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	dominant_hand VARCHAR(12), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_since_last_wash TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	dermatology_disord TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, dominant_hand, medic_hist_perform, samp_store_dur, time_since_last_wash, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, dermatology_disord, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_skin_MIMARKS_specimen" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	dominant_hand VARCHAR(12), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_since_last_wash TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	dermatology_disord TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, dominant_hand, medic_hist_perform, samp_store_dur, time_since_last_wash, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, dermatology_disord, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_skin_MIMARKS_survey" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	dominant_hand VARCHAR(12), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_since_last_wash TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	dermatology_disord TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, dominant_hand, medic_hist_perform, samp_store_dur, time_since_last_wash, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, dermatology_disord, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_skin_MIMS" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	dominant_hand VARCHAR(12), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_since_last_wash TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	dermatology_disord TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, dominant_hand, medic_hist_perform, samp_store_dur, time_since_last_wash, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, dermatology_disord, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_skin_MISAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	dominant_hand VARCHAR(12), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_since_last_wash TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	dermatology_disord TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, dominant_hand, medic_hist_perform, samp_store_dur, time_since_last_wash, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, dermatology_disord, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_skin_MIUVIG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	dominant_hand VARCHAR(12), 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	time_since_last_wash TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	dermatology_disord TEXT, 
	ethnicity TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, dominant_hand, medic_hist_perform, samp_store_dur, time_since_last_wash, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, dermatology_disord, ethnicity, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE human_vaginal (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	hysterectomy TEXT, 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	birth_control TEXT, 
	ethnicity TEXT, 
	gynecologic_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	sexual_act TEXT, 
	urogenit_disord TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	douche DATE, 
	hrt DATE, 
	menarche DATE, 
	menopause DATE, 
	pregnancy DATE, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, hysterectomy, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, birth_control, ethnicity, gynecologic_disord, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, sexual_act, urogenit_disord, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, douche, hrt, menarche, menopause, pregnancy, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name)
);

CREATE TABLE "human_vaginal_MIGS_bacteria" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	hysterectomy TEXT, 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	birth_control TEXT, 
	ethnicity TEXT, 
	gynecologic_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	sexual_act TEXT, 
	urogenit_disord TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	douche DATE, 
	hrt DATE, 
	menarche DATE, 
	menopause DATE, 
	pregnancy DATE, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, hysterectomy, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, birth_control, ethnicity, gynecologic_disord, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, sexual_act, urogenit_disord, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, douche, hrt, menarche, menopause, pregnancy, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_vaginal_MIGS_eukaryote" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	hysterectomy TEXT, 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	birth_control TEXT, 
	ethnicity TEXT, 
	gynecologic_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	sexual_act TEXT, 
	urogenit_disord TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	douche DATE, 
	hrt DATE, 
	menarche DATE, 
	menopause DATE, 
	pregnancy DATE, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, hysterectomy, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, birth_control, ethnicity, gynecologic_disord, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, sexual_act, urogenit_disord, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, douche, hrt, menarche, menopause, pregnancy, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_vaginal_MIGS_org" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	hysterectomy TEXT, 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	birth_control TEXT, 
	ethnicity TEXT, 
	gynecologic_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	sexual_act TEXT, 
	urogenit_disord TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	douche DATE, 
	hrt DATE, 
	menarche DATE, 
	menopause DATE, 
	pregnancy DATE, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, hysterectomy, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, birth_control, ethnicity, gynecologic_disord, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, sexual_act, urogenit_disord, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, douche, hrt, menarche, menopause, pregnancy, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_vaginal_MIGS_plant" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	hysterectomy TEXT, 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	birth_control TEXT, 
	ethnicity TEXT, 
	gynecologic_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	sexual_act TEXT, 
	urogenit_disord TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	douche DATE, 
	hrt DATE, 
	menarche DATE, 
	menopause DATE, 
	pregnancy DATE, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, hysterectomy, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, birth_control, ethnicity, gynecologic_disord, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, sexual_act, urogenit_disord, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, douche, hrt, menarche, menopause, pregnancy, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_vaginal_MIGS_virus" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	hysterectomy TEXT, 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	birth_control TEXT, 
	ethnicity TEXT, 
	gynecologic_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	sexual_act TEXT, 
	urogenit_disord TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	douche DATE, 
	hrt DATE, 
	menarche DATE, 
	menopause DATE, 
	pregnancy DATE, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, hysterectomy, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, birth_control, ethnicity, gynecologic_disord, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, sexual_act, urogenit_disord, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, douche, hrt, menarche, menopause, pregnancy, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_vaginal_MIMAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	hysterectomy TEXT, 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	birth_control TEXT, 
	ethnicity TEXT, 
	gynecologic_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	sexual_act TEXT, 
	urogenit_disord TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	douche DATE, 
	hrt DATE, 
	menarche DATE, 
	menopause DATE, 
	pregnancy DATE, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, hysterectomy, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, birth_control, ethnicity, gynecologic_disord, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, sexual_act, urogenit_disord, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, douche, hrt, menarche, menopause, pregnancy, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_vaginal_MIMARKS_specimen" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	hysterectomy TEXT, 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	birth_control TEXT, 
	ethnicity TEXT, 
	gynecologic_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	sexual_act TEXT, 
	urogenit_disord TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	douche DATE, 
	hrt DATE, 
	menarche DATE, 
	menopause DATE, 
	pregnancy DATE, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, hysterectomy, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, birth_control, ethnicity, gynecologic_disord, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, sexual_act, urogenit_disord, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, douche, hrt, menarche, menopause, pregnancy, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_vaginal_MIMARKS_survey" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	hysterectomy TEXT, 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	birth_control TEXT, 
	ethnicity TEXT, 
	gynecologic_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	sexual_act TEXT, 
	urogenit_disord TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	douche DATE, 
	hrt DATE, 
	menarche DATE, 
	menopause DATE, 
	pregnancy DATE, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, hysterectomy, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, birth_control, ethnicity, gynecologic_disord, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, sexual_act, urogenit_disord, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, douche, hrt, menarche, menopause, pregnancy, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_vaginal_MIMS" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	hysterectomy TEXT, 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	birth_control TEXT, 
	ethnicity TEXT, 
	gynecologic_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	sexual_act TEXT, 
	urogenit_disord TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	douche DATE, 
	hrt DATE, 
	menarche DATE, 
	menopause DATE, 
	pregnancy DATE, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, hysterectomy, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, birth_control, ethnicity, gynecologic_disord, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, sexual_act, urogenit_disord, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, douche, hrt, menarche, menopause, pregnancy, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_vaginal_MISAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	hysterectomy TEXT, 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	birth_control TEXT, 
	ethnicity TEXT, 
	gynecologic_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	sexual_act TEXT, 
	urogenit_disord TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	douche DATE, 
	hrt DATE, 
	menarche DATE, 
	menopause DATE, 
	pregnancy DATE, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, hysterectomy, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, birth_control, ethnicity, gynecologic_disord, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, sexual_act, urogenit_disord, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, douche, hrt, menarche, menopause, pregnancy, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "human_vaginal_MIUVIG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	host_sex VARCHAR(28), 
	hysterectomy TEXT, 
	medic_hist_perform TEXT, 
	samp_store_dur TEXT, 
	host_age TEXT, 
	host_body_mass_index TEXT, 
	host_body_temp TEXT, 
	host_height TEXT, 
	host_pulse TEXT, 
	host_tot_mass TEXT, 
	salinity TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_occupation INTEGER, 
	ihmc_medication_code INTEGER, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_phenotype TEXT, 
	chem_administration TEXT, 
	birth_control TEXT, 
	ethnicity TEXT, 
	gynecologic_disord TEXT, 
	host_diet TEXT, 
	host_genotype TEXT, 
	host_subject_id TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	sexual_act TEXT, 
	urogenit_disord TEXT, 
	host_last_meal TEXT, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relation TEXT, 
	douche DATE, 
	hrt DATE, 
	menarche DATE, 
	menopause DATE, 
	pregnancy DATE, 
	samp_vol_we_dna_ext TEXT, 
	host_disease_stat TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, host_sex, hysterectomy, medic_hist_perform, samp_store_dur, host_age, host_body_mass_index, host_body_temp, host_height, host_pulse, host_tot_mass, salinity, samp_salinity, samp_store_temp, host_occupation, ihmc_medication_code, host_body_product, host_body_site, host_phenotype, chem_administration, birth_control, ethnicity, gynecologic_disord, host_diet, host_genotype, host_subject_id, host_symbiont, samp_store_loc, sexual_act, urogenit_disord, host_last_meal, misc_param, organism_count, perturbation, host_family_relation, douche, hrt, menarche, menopause, pregnancy, samp_vol_we_dna_ext, host_disease_stat, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE hydrocarbon_resources_cores (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	sr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	sr_lithology VARCHAR(13), 
	depos_env VARCHAR(24), 
	sr_dep_env VARCHAR(14), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	sr_kerog_type VARCHAR(8), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT NOT NULL, 
	porosity TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	magnesium TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	owc_tvdss TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfate_fw TEXT NOT NULL, 
	sulfide TEXT, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT NOT NULL, 
	xylene TEXT, 
	samp_md VARCHAR(5), 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	samp_tvdss TEXT, 
	permeability TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	org_count_qpcr_info TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, sr_geol_age, lithology, sr_lithology, depos_env, sr_dep_env, samp_subtype, hcr, hc_produced, sr_kerog_type, samp_store_dur, ph, hcr_pressure, hcr_temp, porosity, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, magnesium, nitrate, nitrite, owc_tvdss, potassium, pour_point, pressure, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, xylene, samp_md, samp_transport_cond, viscosity, samp_tvdss, permeability, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, saturates_pc, suspend_solids, organism_count, org_count_qpcr_info, samp_vol_we_dna_ext, project_name, samp_name)
);

CREATE TABLE "hydrocarbon_resources_cores_MIGS_bacteria" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	sr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	sr_lithology VARCHAR(13), 
	depos_env VARCHAR(24), 
	sr_dep_env VARCHAR(14), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	sr_kerog_type VARCHAR(8), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT NOT NULL, 
	porosity TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	magnesium TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	owc_tvdss TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfate_fw TEXT NOT NULL, 
	sulfide TEXT, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT NOT NULL, 
	xylene TEXT, 
	samp_md VARCHAR(5), 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	samp_tvdss TEXT, 
	permeability TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	org_count_qpcr_info TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, sr_geol_age, lithology, sr_lithology, depos_env, sr_dep_env, samp_subtype, hcr, hc_produced, sr_kerog_type, samp_store_dur, ph, hcr_pressure, hcr_temp, porosity, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, magnesium, nitrate, nitrite, owc_tvdss, potassium, pour_point, pressure, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, xylene, samp_md, samp_transport_cond, viscosity, samp_tvdss, permeability, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, saturates_pc, suspend_solids, organism_count, org_count_qpcr_info, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_cores_MIGS_eukaryote" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	sr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	sr_lithology VARCHAR(13), 
	depos_env VARCHAR(24), 
	sr_dep_env VARCHAR(14), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	sr_kerog_type VARCHAR(8), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT NOT NULL, 
	porosity TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	magnesium TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	owc_tvdss TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfate_fw TEXT NOT NULL, 
	sulfide TEXT, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT NOT NULL, 
	xylene TEXT, 
	samp_md VARCHAR(5), 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	samp_tvdss TEXT, 
	permeability TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	org_count_qpcr_info TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, sr_geol_age, lithology, sr_lithology, depos_env, sr_dep_env, samp_subtype, hcr, hc_produced, sr_kerog_type, samp_store_dur, ph, hcr_pressure, hcr_temp, porosity, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, magnesium, nitrate, nitrite, owc_tvdss, potassium, pour_point, pressure, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, xylene, samp_md, samp_transport_cond, viscosity, samp_tvdss, permeability, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, saturates_pc, suspend_solids, organism_count, org_count_qpcr_info, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_cores_MIGS_org" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	sr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	sr_lithology VARCHAR(13), 
	depos_env VARCHAR(24), 
	sr_dep_env VARCHAR(14), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	sr_kerog_type VARCHAR(8), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT NOT NULL, 
	porosity TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	magnesium TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	owc_tvdss TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfate_fw TEXT NOT NULL, 
	sulfide TEXT, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT NOT NULL, 
	xylene TEXT, 
	samp_md VARCHAR(5), 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	samp_tvdss TEXT, 
	permeability TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	org_count_qpcr_info TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, sr_geol_age, lithology, sr_lithology, depos_env, sr_dep_env, samp_subtype, hcr, hc_produced, sr_kerog_type, samp_store_dur, ph, hcr_pressure, hcr_temp, porosity, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, magnesium, nitrate, nitrite, owc_tvdss, potassium, pour_point, pressure, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, xylene, samp_md, samp_transport_cond, viscosity, samp_tvdss, permeability, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, saturates_pc, suspend_solids, organism_count, org_count_qpcr_info, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_cores_MIGS_plant" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	sr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	sr_lithology VARCHAR(13), 
	depos_env VARCHAR(24), 
	sr_dep_env VARCHAR(14), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	sr_kerog_type VARCHAR(8), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT NOT NULL, 
	porosity TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	magnesium TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	owc_tvdss TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfate_fw TEXT NOT NULL, 
	sulfide TEXT, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT NOT NULL, 
	xylene TEXT, 
	samp_md VARCHAR(5), 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	samp_tvdss TEXT, 
	permeability TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	org_count_qpcr_info TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, sr_geol_age, lithology, sr_lithology, depos_env, sr_dep_env, samp_subtype, hcr, hc_produced, sr_kerog_type, samp_store_dur, ph, hcr_pressure, hcr_temp, porosity, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, magnesium, nitrate, nitrite, owc_tvdss, potassium, pour_point, pressure, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, xylene, samp_md, samp_transport_cond, viscosity, samp_tvdss, permeability, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, saturates_pc, suspend_solids, organism_count, org_count_qpcr_info, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_cores_MIGS_virus" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	sr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	sr_lithology VARCHAR(13), 
	depos_env VARCHAR(24), 
	sr_dep_env VARCHAR(14), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	sr_kerog_type VARCHAR(8), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT NOT NULL, 
	porosity TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	magnesium TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	owc_tvdss TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfate_fw TEXT NOT NULL, 
	sulfide TEXT, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT NOT NULL, 
	xylene TEXT, 
	samp_md VARCHAR(5), 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	samp_tvdss TEXT, 
	permeability TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	org_count_qpcr_info TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, sr_geol_age, lithology, sr_lithology, depos_env, sr_dep_env, samp_subtype, hcr, hc_produced, sr_kerog_type, samp_store_dur, ph, hcr_pressure, hcr_temp, porosity, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, magnesium, nitrate, nitrite, owc_tvdss, potassium, pour_point, pressure, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, xylene, samp_md, samp_transport_cond, viscosity, samp_tvdss, permeability, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, saturates_pc, suspend_solids, organism_count, org_count_qpcr_info, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_cores_MIMAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	sr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	sr_lithology VARCHAR(13), 
	depos_env VARCHAR(24), 
	sr_dep_env VARCHAR(14), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	sr_kerog_type VARCHAR(8), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT NOT NULL, 
	porosity TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	magnesium TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	owc_tvdss TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfate_fw TEXT NOT NULL, 
	sulfide TEXT, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT NOT NULL, 
	xylene TEXT, 
	samp_md VARCHAR(5), 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	samp_tvdss TEXT, 
	permeability TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	org_count_qpcr_info TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, sr_geol_age, lithology, sr_lithology, depos_env, sr_dep_env, samp_subtype, hcr, hc_produced, sr_kerog_type, samp_store_dur, ph, hcr_pressure, hcr_temp, porosity, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, magnesium, nitrate, nitrite, owc_tvdss, potassium, pour_point, pressure, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, xylene, samp_md, samp_transport_cond, viscosity, samp_tvdss, permeability, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, saturates_pc, suspend_solids, organism_count, org_count_qpcr_info, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_cores_MIMARKS_specimen" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	sr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	sr_lithology VARCHAR(13), 
	depos_env VARCHAR(24), 
	sr_dep_env VARCHAR(14), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	sr_kerog_type VARCHAR(8), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT NOT NULL, 
	porosity TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	magnesium TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	owc_tvdss TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfate_fw TEXT NOT NULL, 
	sulfide TEXT, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT NOT NULL, 
	xylene TEXT, 
	samp_md VARCHAR(5), 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	samp_tvdss TEXT, 
	permeability TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	org_count_qpcr_info TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, sr_geol_age, lithology, sr_lithology, depos_env, sr_dep_env, samp_subtype, hcr, hc_produced, sr_kerog_type, samp_store_dur, ph, hcr_pressure, hcr_temp, porosity, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, magnesium, nitrate, nitrite, owc_tvdss, potassium, pour_point, pressure, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, xylene, samp_md, samp_transport_cond, viscosity, samp_tvdss, permeability, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, saturates_pc, suspend_solids, organism_count, org_count_qpcr_info, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_cores_MIMARKS_survey" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	sr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	sr_lithology VARCHAR(13), 
	depos_env VARCHAR(24), 
	sr_dep_env VARCHAR(14), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	sr_kerog_type VARCHAR(8), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT NOT NULL, 
	porosity TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	magnesium TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	owc_tvdss TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfate_fw TEXT NOT NULL, 
	sulfide TEXT, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT NOT NULL, 
	xylene TEXT, 
	samp_md VARCHAR(5), 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	samp_tvdss TEXT, 
	permeability TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	org_count_qpcr_info TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, sr_geol_age, lithology, sr_lithology, depos_env, sr_dep_env, samp_subtype, hcr, hc_produced, sr_kerog_type, samp_store_dur, ph, hcr_pressure, hcr_temp, porosity, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, magnesium, nitrate, nitrite, owc_tvdss, potassium, pour_point, pressure, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, xylene, samp_md, samp_transport_cond, viscosity, samp_tvdss, permeability, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, saturates_pc, suspend_solids, organism_count, org_count_qpcr_info, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_cores_MIMS" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	sr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	sr_lithology VARCHAR(13), 
	depos_env VARCHAR(24), 
	sr_dep_env VARCHAR(14), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	sr_kerog_type VARCHAR(8), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT NOT NULL, 
	porosity TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	magnesium TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	owc_tvdss TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfate_fw TEXT NOT NULL, 
	sulfide TEXT, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT NOT NULL, 
	xylene TEXT, 
	samp_md VARCHAR(5), 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	samp_tvdss TEXT, 
	permeability TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	org_count_qpcr_info TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, sr_geol_age, lithology, sr_lithology, depos_env, sr_dep_env, samp_subtype, hcr, hc_produced, sr_kerog_type, samp_store_dur, ph, hcr_pressure, hcr_temp, porosity, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, magnesium, nitrate, nitrite, owc_tvdss, potassium, pour_point, pressure, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, xylene, samp_md, samp_transport_cond, viscosity, samp_tvdss, permeability, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, saturates_pc, suspend_solids, organism_count, org_count_qpcr_info, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_cores_MISAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	sr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	sr_lithology VARCHAR(13), 
	depos_env VARCHAR(24), 
	sr_dep_env VARCHAR(14), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	sr_kerog_type VARCHAR(8), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT NOT NULL, 
	porosity TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	magnesium TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	owc_tvdss TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfate_fw TEXT NOT NULL, 
	sulfide TEXT, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT NOT NULL, 
	xylene TEXT, 
	samp_md VARCHAR(5), 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	samp_tvdss TEXT, 
	permeability TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	org_count_qpcr_info TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, sr_geol_age, lithology, sr_lithology, depos_env, sr_dep_env, samp_subtype, hcr, hc_produced, sr_kerog_type, samp_store_dur, ph, hcr_pressure, hcr_temp, porosity, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, magnesium, nitrate, nitrite, owc_tvdss, potassium, pour_point, pressure, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, xylene, samp_md, samp_transport_cond, viscosity, samp_tvdss, permeability, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, saturates_pc, suspend_solids, organism_count, org_count_qpcr_info, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_cores_MIUVIG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	sr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	sr_lithology VARCHAR(13), 
	depos_env VARCHAR(24), 
	sr_dep_env VARCHAR(14), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	sr_kerog_type VARCHAR(8), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT NOT NULL, 
	porosity TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	magnesium TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	owc_tvdss TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfate_fw TEXT NOT NULL, 
	sulfide TEXT, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT NOT NULL, 
	xylene TEXT, 
	samp_md VARCHAR(5), 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	samp_tvdss TEXT, 
	permeability TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	org_count_qpcr_info TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, sr_geol_age, lithology, sr_lithology, depos_env, sr_dep_env, samp_subtype, hcr, hc_produced, sr_kerog_type, samp_store_dur, ph, hcr_pressure, hcr_temp, porosity, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, magnesium, nitrate, nitrite, owc_tvdss, potassium, pour_point, pressure, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, xylene, samp_md, samp_transport_cond, viscosity, samp_tvdss, permeability, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, saturates_pc, suspend_solids, organism_count, org_count_qpcr_info, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE hydrocarbon_resources_fluids_swabs (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	depos_env VARCHAR(24), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	add_recov_method VARCHAR(24) NOT NULL, 
	samp_collect_point VARCHAR(12) NOT NULL, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT, 
	samp_loc_corr_rate TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	iwf TEXT NOT NULL, 
	magnesium TEXT, 
	nitrate TEXT NOT NULL, 
	nitrite TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	prod_rate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT NOT NULL, 
	sulfate_fw TEXT, 
	sulfide TEXT NOT NULL, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT, 
	water_cut TEXT NOT NULL, 
	water_prod_rate TEXT, 
	xylene TEXT, 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	biocide_admin_method TEXT, 
	chem_treat_method TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	samp_preserv TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	biocide TEXT, 
	chem_treatment TEXT, 
	org_count_qpcr_info TEXT, 
	iw_bt_date_well DATE, 
	prod_start_date DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, lithology, depos_env, samp_subtype, hcr, hc_produced, add_recov_method, samp_collect_point, samp_store_dur, ph, hcr_pressure, hcr_temp, samp_loc_corr_rate, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, iwf, magnesium, nitrate, nitrite, potassium, pour_point, pressure, prod_rate, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, water_cut, water_prod_rate, xylene, samp_transport_cond, viscosity, biocide_admin_method, chem_treat_method, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, samp_preserv, saturates_pc, suspend_solids, organism_count, biocide, chem_treatment, org_count_qpcr_info, iw_bt_date_well, prod_start_date, samp_vol_we_dna_ext, project_name, samp_name)
);

CREATE TABLE "hydrocarbon_resources_fluids_swabs_MIGS_bacteria" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	depos_env VARCHAR(24), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	add_recov_method VARCHAR(24) NOT NULL, 
	samp_collect_point VARCHAR(12) NOT NULL, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT, 
	samp_loc_corr_rate TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	iwf TEXT NOT NULL, 
	magnesium TEXT, 
	nitrate TEXT NOT NULL, 
	nitrite TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	prod_rate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT NOT NULL, 
	sulfate_fw TEXT, 
	sulfide TEXT NOT NULL, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT, 
	water_cut TEXT NOT NULL, 
	water_prod_rate TEXT, 
	xylene TEXT, 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	biocide_admin_method TEXT, 
	chem_treat_method TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	samp_preserv TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	biocide TEXT, 
	chem_treatment TEXT, 
	org_count_qpcr_info TEXT, 
	iw_bt_date_well DATE, 
	prod_start_date DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, lithology, depos_env, samp_subtype, hcr, hc_produced, add_recov_method, samp_collect_point, samp_store_dur, ph, hcr_pressure, hcr_temp, samp_loc_corr_rate, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, iwf, magnesium, nitrate, nitrite, potassium, pour_point, pressure, prod_rate, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, water_cut, water_prod_rate, xylene, samp_transport_cond, viscosity, biocide_admin_method, chem_treat_method, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, samp_preserv, saturates_pc, suspend_solids, organism_count, biocide, chem_treatment, org_count_qpcr_info, iw_bt_date_well, prod_start_date, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_fluids_swabs_MIGS_eukaryote" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	depos_env VARCHAR(24), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	add_recov_method VARCHAR(24) NOT NULL, 
	samp_collect_point VARCHAR(12) NOT NULL, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT, 
	samp_loc_corr_rate TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	iwf TEXT NOT NULL, 
	magnesium TEXT, 
	nitrate TEXT NOT NULL, 
	nitrite TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	prod_rate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT NOT NULL, 
	sulfate_fw TEXT, 
	sulfide TEXT NOT NULL, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT, 
	water_cut TEXT NOT NULL, 
	water_prod_rate TEXT, 
	xylene TEXT, 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	biocide_admin_method TEXT, 
	chem_treat_method TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	samp_preserv TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	biocide TEXT, 
	chem_treatment TEXT, 
	org_count_qpcr_info TEXT, 
	iw_bt_date_well DATE, 
	prod_start_date DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, lithology, depos_env, samp_subtype, hcr, hc_produced, add_recov_method, samp_collect_point, samp_store_dur, ph, hcr_pressure, hcr_temp, samp_loc_corr_rate, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, iwf, magnesium, nitrate, nitrite, potassium, pour_point, pressure, prod_rate, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, water_cut, water_prod_rate, xylene, samp_transport_cond, viscosity, biocide_admin_method, chem_treat_method, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, samp_preserv, saturates_pc, suspend_solids, organism_count, biocide, chem_treatment, org_count_qpcr_info, iw_bt_date_well, prod_start_date, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_fluids_swabs_MIGS_org" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	depos_env VARCHAR(24), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	add_recov_method VARCHAR(24) NOT NULL, 
	samp_collect_point VARCHAR(12) NOT NULL, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT, 
	samp_loc_corr_rate TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	iwf TEXT NOT NULL, 
	magnesium TEXT, 
	nitrate TEXT NOT NULL, 
	nitrite TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	prod_rate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT NOT NULL, 
	sulfate_fw TEXT, 
	sulfide TEXT NOT NULL, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT, 
	water_cut TEXT NOT NULL, 
	water_prod_rate TEXT, 
	xylene TEXT, 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	biocide_admin_method TEXT, 
	chem_treat_method TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	samp_preserv TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	biocide TEXT, 
	chem_treatment TEXT, 
	org_count_qpcr_info TEXT, 
	iw_bt_date_well DATE, 
	prod_start_date DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, lithology, depos_env, samp_subtype, hcr, hc_produced, add_recov_method, samp_collect_point, samp_store_dur, ph, hcr_pressure, hcr_temp, samp_loc_corr_rate, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, iwf, magnesium, nitrate, nitrite, potassium, pour_point, pressure, prod_rate, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, water_cut, water_prod_rate, xylene, samp_transport_cond, viscosity, biocide_admin_method, chem_treat_method, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, samp_preserv, saturates_pc, suspend_solids, organism_count, biocide, chem_treatment, org_count_qpcr_info, iw_bt_date_well, prod_start_date, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_fluids_swabs_MIGS_plant" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	depos_env VARCHAR(24), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	add_recov_method VARCHAR(24) NOT NULL, 
	samp_collect_point VARCHAR(12) NOT NULL, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT, 
	samp_loc_corr_rate TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	iwf TEXT NOT NULL, 
	magnesium TEXT, 
	nitrate TEXT NOT NULL, 
	nitrite TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	prod_rate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT NOT NULL, 
	sulfate_fw TEXT, 
	sulfide TEXT NOT NULL, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT, 
	water_cut TEXT NOT NULL, 
	water_prod_rate TEXT, 
	xylene TEXT, 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	biocide_admin_method TEXT, 
	chem_treat_method TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	samp_preserv TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	biocide TEXT, 
	chem_treatment TEXT, 
	org_count_qpcr_info TEXT, 
	iw_bt_date_well DATE, 
	prod_start_date DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, lithology, depos_env, samp_subtype, hcr, hc_produced, add_recov_method, samp_collect_point, samp_store_dur, ph, hcr_pressure, hcr_temp, samp_loc_corr_rate, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, iwf, magnesium, nitrate, nitrite, potassium, pour_point, pressure, prod_rate, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, water_cut, water_prod_rate, xylene, samp_transport_cond, viscosity, biocide_admin_method, chem_treat_method, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, samp_preserv, saturates_pc, suspend_solids, organism_count, biocide, chem_treatment, org_count_qpcr_info, iw_bt_date_well, prod_start_date, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_fluids_swabs_MIGS_virus" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	depos_env VARCHAR(24), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	add_recov_method VARCHAR(24) NOT NULL, 
	samp_collect_point VARCHAR(12) NOT NULL, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT, 
	samp_loc_corr_rate TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	iwf TEXT NOT NULL, 
	magnesium TEXT, 
	nitrate TEXT NOT NULL, 
	nitrite TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	prod_rate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT NOT NULL, 
	sulfate_fw TEXT, 
	sulfide TEXT NOT NULL, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT, 
	water_cut TEXT NOT NULL, 
	water_prod_rate TEXT, 
	xylene TEXT, 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	biocide_admin_method TEXT, 
	chem_treat_method TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	samp_preserv TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	biocide TEXT, 
	chem_treatment TEXT, 
	org_count_qpcr_info TEXT, 
	iw_bt_date_well DATE, 
	prod_start_date DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, lithology, depos_env, samp_subtype, hcr, hc_produced, add_recov_method, samp_collect_point, samp_store_dur, ph, hcr_pressure, hcr_temp, samp_loc_corr_rate, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, iwf, magnesium, nitrate, nitrite, potassium, pour_point, pressure, prod_rate, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, water_cut, water_prod_rate, xylene, samp_transport_cond, viscosity, biocide_admin_method, chem_treat_method, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, samp_preserv, saturates_pc, suspend_solids, organism_count, biocide, chem_treatment, org_count_qpcr_info, iw_bt_date_well, prod_start_date, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_fluids_swabs_MIMAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	depos_env VARCHAR(24), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	add_recov_method VARCHAR(24) NOT NULL, 
	samp_collect_point VARCHAR(12) NOT NULL, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT, 
	samp_loc_corr_rate TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	iwf TEXT NOT NULL, 
	magnesium TEXT, 
	nitrate TEXT NOT NULL, 
	nitrite TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	prod_rate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT NOT NULL, 
	sulfate_fw TEXT, 
	sulfide TEXT NOT NULL, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT, 
	water_cut TEXT NOT NULL, 
	water_prod_rate TEXT, 
	xylene TEXT, 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	biocide_admin_method TEXT, 
	chem_treat_method TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	samp_preserv TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	biocide TEXT, 
	chem_treatment TEXT, 
	org_count_qpcr_info TEXT, 
	iw_bt_date_well DATE, 
	prod_start_date DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, lithology, depos_env, samp_subtype, hcr, hc_produced, add_recov_method, samp_collect_point, samp_store_dur, ph, hcr_pressure, hcr_temp, samp_loc_corr_rate, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, iwf, magnesium, nitrate, nitrite, potassium, pour_point, pressure, prod_rate, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, water_cut, water_prod_rate, xylene, samp_transport_cond, viscosity, biocide_admin_method, chem_treat_method, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, samp_preserv, saturates_pc, suspend_solids, organism_count, biocide, chem_treatment, org_count_qpcr_info, iw_bt_date_well, prod_start_date, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_fluids_swabs_MIMARKS_specimen" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	depos_env VARCHAR(24), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	add_recov_method VARCHAR(24) NOT NULL, 
	samp_collect_point VARCHAR(12) NOT NULL, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT, 
	samp_loc_corr_rate TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	iwf TEXT NOT NULL, 
	magnesium TEXT, 
	nitrate TEXT NOT NULL, 
	nitrite TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	prod_rate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT NOT NULL, 
	sulfate_fw TEXT, 
	sulfide TEXT NOT NULL, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT, 
	water_cut TEXT NOT NULL, 
	water_prod_rate TEXT, 
	xylene TEXT, 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	biocide_admin_method TEXT, 
	chem_treat_method TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	samp_preserv TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	biocide TEXT, 
	chem_treatment TEXT, 
	org_count_qpcr_info TEXT, 
	iw_bt_date_well DATE, 
	prod_start_date DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, lithology, depos_env, samp_subtype, hcr, hc_produced, add_recov_method, samp_collect_point, samp_store_dur, ph, hcr_pressure, hcr_temp, samp_loc_corr_rate, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, iwf, magnesium, nitrate, nitrite, potassium, pour_point, pressure, prod_rate, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, water_cut, water_prod_rate, xylene, samp_transport_cond, viscosity, biocide_admin_method, chem_treat_method, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, samp_preserv, saturates_pc, suspend_solids, organism_count, biocide, chem_treatment, org_count_qpcr_info, iw_bt_date_well, prod_start_date, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_fluids_swabs_MIMARKS_survey" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	depos_env VARCHAR(24), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	add_recov_method VARCHAR(24) NOT NULL, 
	samp_collect_point VARCHAR(12) NOT NULL, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT, 
	samp_loc_corr_rate TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	iwf TEXT NOT NULL, 
	magnesium TEXT, 
	nitrate TEXT NOT NULL, 
	nitrite TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	prod_rate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT NOT NULL, 
	sulfate_fw TEXT, 
	sulfide TEXT NOT NULL, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT, 
	water_cut TEXT NOT NULL, 
	water_prod_rate TEXT, 
	xylene TEXT, 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	biocide_admin_method TEXT, 
	chem_treat_method TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	samp_preserv TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	biocide TEXT, 
	chem_treatment TEXT, 
	org_count_qpcr_info TEXT, 
	iw_bt_date_well DATE, 
	prod_start_date DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, lithology, depos_env, samp_subtype, hcr, hc_produced, add_recov_method, samp_collect_point, samp_store_dur, ph, hcr_pressure, hcr_temp, samp_loc_corr_rate, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, iwf, magnesium, nitrate, nitrite, potassium, pour_point, pressure, prod_rate, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, water_cut, water_prod_rate, xylene, samp_transport_cond, viscosity, biocide_admin_method, chem_treat_method, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, samp_preserv, saturates_pc, suspend_solids, organism_count, biocide, chem_treatment, org_count_qpcr_info, iw_bt_date_well, prod_start_date, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_fluids_swabs_MIMS" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	depos_env VARCHAR(24), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	add_recov_method VARCHAR(24) NOT NULL, 
	samp_collect_point VARCHAR(12) NOT NULL, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT, 
	samp_loc_corr_rate TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	iwf TEXT NOT NULL, 
	magnesium TEXT, 
	nitrate TEXT NOT NULL, 
	nitrite TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	prod_rate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT NOT NULL, 
	sulfate_fw TEXT, 
	sulfide TEXT NOT NULL, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT, 
	water_cut TEXT NOT NULL, 
	water_prod_rate TEXT, 
	xylene TEXT, 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	biocide_admin_method TEXT, 
	chem_treat_method TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	samp_preserv TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	biocide TEXT, 
	chem_treatment TEXT, 
	org_count_qpcr_info TEXT, 
	iw_bt_date_well DATE, 
	prod_start_date DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, lithology, depos_env, samp_subtype, hcr, hc_produced, add_recov_method, samp_collect_point, samp_store_dur, ph, hcr_pressure, hcr_temp, samp_loc_corr_rate, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, iwf, magnesium, nitrate, nitrite, potassium, pour_point, pressure, prod_rate, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, water_cut, water_prod_rate, xylene, samp_transport_cond, viscosity, biocide_admin_method, chem_treat_method, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, samp_preserv, saturates_pc, suspend_solids, organism_count, biocide, chem_treatment, org_count_qpcr_info, iw_bt_date_well, prod_start_date, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_fluids_swabs_MISAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	depos_env VARCHAR(24), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	add_recov_method VARCHAR(24) NOT NULL, 
	samp_collect_point VARCHAR(12) NOT NULL, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT, 
	samp_loc_corr_rate TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	iwf TEXT NOT NULL, 
	magnesium TEXT, 
	nitrate TEXT NOT NULL, 
	nitrite TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	prod_rate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT NOT NULL, 
	sulfate_fw TEXT, 
	sulfide TEXT NOT NULL, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT, 
	water_cut TEXT NOT NULL, 
	water_prod_rate TEXT, 
	xylene TEXT, 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	biocide_admin_method TEXT, 
	chem_treat_method TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	samp_preserv TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	biocide TEXT, 
	chem_treatment TEXT, 
	org_count_qpcr_info TEXT, 
	iw_bt_date_well DATE, 
	prod_start_date DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, lithology, depos_env, samp_subtype, hcr, hc_produced, add_recov_method, samp_collect_point, samp_store_dur, ph, hcr_pressure, hcr_temp, samp_loc_corr_rate, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, iwf, magnesium, nitrate, nitrite, potassium, pour_point, pressure, prod_rate, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, water_cut, water_prod_rate, xylene, samp_transport_cond, viscosity, biocide_admin_method, chem_treat_method, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, samp_preserv, saturates_pc, suspend_solids, organism_count, biocide, chem_treatment, org_count_qpcr_info, iw_bt_date_well, prod_start_date, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "hydrocarbon_resources_fluids_swabs_MIUVIG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT NOT NULL, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	hcr_geol_age VARCHAR(13), 
	lithology VARCHAR(12), 
	depos_env VARCHAR(24), 
	samp_subtype VARCHAR(14), 
	hcr VARCHAR(19) NOT NULL, 
	hc_produced VARCHAR(15) NOT NULL, 
	add_recov_method VARCHAR(24) NOT NULL, 
	samp_collect_point VARCHAR(12) NOT NULL, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	hcr_pressure TEXT, 
	hcr_temp TEXT, 
	samp_loc_corr_rate TEXT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	api TEXT NOT NULL, 
	benzene TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_phosp TEXT, 
	diss_iron TEXT, 
	diss_org_carb TEXT, 
	diss_oxygen_fluid TEXT, 
	ethylbenzene TEXT, 
	hcr_fw_salinity TEXT, 
	iwf TEXT NOT NULL, 
	magnesium TEXT, 
	nitrate TEXT NOT NULL, 
	nitrite TEXT, 
	potassium TEXT, 
	pour_point TEXT, 
	pressure TEXT, 
	prod_rate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sodium TEXT, 
	sulfate TEXT NOT NULL, 
	sulfate_fw TEXT, 
	sulfide TEXT NOT NULL, 
	tan TEXT, 
	toluene TEXT, 
	tot_iron TEXT, 
	tot_nitro TEXT, 
	tot_phosp TEXT, 
	tot_sulfur TEXT, 
	tvdss_of_hcr_press TEXT, 
	tvdss_of_hcr_temp TEXT, 
	vfa TEXT, 
	vfa_fw TEXT, 
	water_cut TEXT NOT NULL, 
	water_prod_rate TEXT, 
	xylene TEXT, 
	samp_transport_cond TEXT, 
	viscosity TEXT, 
	biocide_admin_method TEXT, 
	chem_treat_method TEXT, 
	samp_type TEXT NOT NULL, 
	additional_info TEXT, 
	alkalinity_method TEXT, 
	basin TEXT NOT NULL, 
	field TEXT, 
	reservoir TEXT, 
	samp_store_loc TEXT, 
	samp_well_name TEXT, 
	win TEXT, 
	aromatics_pc TEXT, 
	asphaltenes_pc TEXT, 
	misc_param TEXT, 
	resins_pc TEXT, 
	samp_preserv TEXT, 
	saturates_pc TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	biocide TEXT, 
	chem_treatment TEXT, 
	org_count_qpcr_info TEXT, 
	iw_bt_date_well DATE, 
	prod_start_date DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, hcr_geol_age, lithology, depos_env, samp_subtype, hcr, hc_produced, add_recov_method, samp_collect_point, samp_store_dur, ph, hcr_pressure, hcr_temp, samp_loc_corr_rate, alkalinity, ammonium, api, benzene, calcium, chloride, density, diss_carb_dioxide, diss_inorg_carb, diss_inorg_phosp, diss_iron, diss_org_carb, diss_oxygen_fluid, ethylbenzene, hcr_fw_salinity, iwf, magnesium, nitrate, nitrite, potassium, pour_point, pressure, prod_rate, salinity, samp_store_temp, sodium, sulfate, sulfate_fw, sulfide, tan, toluene, tot_iron, tot_nitro, tot_phosp, tot_sulfur, tvdss_of_hcr_press, tvdss_of_hcr_temp, vfa, vfa_fw, water_cut, water_prod_rate, xylene, samp_transport_cond, viscosity, biocide_admin_method, chem_treat_method, samp_type, additional_info, alkalinity_method, basin, field, reservoir, samp_store_loc, samp_well_name, win, aromatics_pc, asphaltenes_pc, misc_param, resins_pc, samp_preserv, saturates_pc, suspend_solids, organism_count, biocide, chem_treatment, org_count_qpcr_info, iw_bt_date_well, prod_start_date, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE microbial_mat_biofilm (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name)
);

CREATE TABLE "microbial_mat_biofilm_MIGS_bacteria" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "microbial_mat_biofilm_MIGS_eukaryote" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "microbial_mat_biofilm_MIGS_org" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "microbial_mat_biofilm_MIGS_plant" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "microbial_mat_biofilm_MIGS_virus" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "microbial_mat_biofilm_MIMAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "microbial_mat_biofilm_MIMARKS_specimen" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "microbial_mat_biofilm_MIMARKS_survey" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "microbial_mat_biofilm_MIMS" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "microbial_mat_biofilm_MISAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "microbial_mat_biofilm_MIUVIG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE miscellaneous_natural_or_artificial_environment (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	water_current TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, ammonium, bromide, calcium, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_nitro, diss_oxygen, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, phosphate, potassium, pressure, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, water_current, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name)
);

CREATE TABLE "miscellaneous_natural_or_artificial_environment_MIGS_bacteria" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	water_current TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, ammonium, bromide, calcium, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_nitro, diss_oxygen, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, phosphate, potassium, pressure, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, water_current, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "miscellaneous_natural_or_artificial_environment_MIGS_eukaryote" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	water_current TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, ammonium, bromide, calcium, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_nitro, diss_oxygen, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, phosphate, potassium, pressure, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, water_current, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "miscellaneous_natural_or_artificial_environment_MIGS_org" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	water_current TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, ammonium, bromide, calcium, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_nitro, diss_oxygen, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, phosphate, potassium, pressure, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, water_current, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "miscellaneous_natural_or_artificial_environment_MIGS_plant" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	water_current TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, ammonium, bromide, calcium, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_nitro, diss_oxygen, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, phosphate, potassium, pressure, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, water_current, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "miscellaneous_natural_or_artificial_environment_MIGS_virus" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	water_current TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, ammonium, bromide, calcium, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_nitro, diss_oxygen, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, phosphate, potassium, pressure, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, water_current, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "miscellaneous_natural_or_artificial_environment_MIMAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	water_current TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, ammonium, bromide, calcium, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_nitro, diss_oxygen, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, phosphate, potassium, pressure, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, water_current, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "miscellaneous_natural_or_artificial_environment_MIMARKS_specimen" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	water_current TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, ammonium, bromide, calcium, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_nitro, diss_oxygen, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, phosphate, potassium, pressure, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, water_current, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "miscellaneous_natural_or_artificial_environment_MIMARKS_survey" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	water_current TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, ammonium, bromide, calcium, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_nitro, diss_oxygen, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, phosphate, potassium, pressure, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, water_current, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "miscellaneous_natural_or_artificial_environment_MIMS" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	water_current TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, ammonium, bromide, calcium, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_nitro, diss_oxygen, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, phosphate, potassium, pressure, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, water_current, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "miscellaneous_natural_or_artificial_environment_MISAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	water_current TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, ammonium, bromide, calcium, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_nitro, diss_oxygen, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, phosphate, potassium, pressure, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, water_current, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "miscellaneous_natural_or_artificial_environment_MIUVIG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	ammonium TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	water_current TEXT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, ammonium, bromide, calcium, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_nitro, diss_oxygen, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, phosphate, potassium, pressure, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, water_current, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE plant_associated (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	plant_sex VARCHAR(19), 
	samp_dis_stage VARCHAR(23), 
	growth_habit VARCHAR(10), 
	biol_stat VARCHAR(16), 
	samp_store_dur TEXT, 
	root_med_ph FLOAT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	host_wet_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	air_temp_regm TEXT, 
	gravity TEXT, 
	humidity_regm TEXT, 
	rainfall_regm TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	root_cond TEXT, 
	tiss_cult_growth_med TEXT, 
	host_subspecf_genlin TEXT, 
	plant_struc TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT, 
	ances_data TEXT, 
	biotic_regm TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_symbiont TEXT, 
	plant_product TEXT, 
	root_med_solid TEXT, 
	samp_store_loc TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	root_med_regl TEXT, 
	root_med_suppl TEXT, 
	organism_count TEXT, 
	light_regm TEXT, 
	antibiotic_regm TEXT, 
	chem_mutagen TEXT, 
	fertilizer_regm TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	growth_hormone_regm TEXT, 
	herbicide_regm TEXT, 
	mineral_nutr_regm TEXT, 
	non_min_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	radiation_regm TEXT, 
	salt_regm TEXT, 
	climate_environment TEXT, 
	perturbation TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	cult_root_med TEXT, 
	growth_facil TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, plant_sex, samp_dis_stage, growth_habit, biol_stat, samp_store_dur, root_med_ph, host_age, host_dry_mass, host_height, host_length, host_tot_mass, host_wet_mass, salinity, samp_store_temp, air_temp_regm, gravity, humidity_regm, rainfall_regm, water_temp_regm, watering_regm, ph_regm, host_taxid, genetic_mod, root_cond, tiss_cult_growth_med, host_subspecf_genlin, plant_struc, plant_growth_med, chem_administration, ances_data, biotic_regm, host_common_name, host_genotype, host_life_stage, host_symbiont, plant_product, root_med_solid, samp_store_loc, misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, root_med_regl, root_med_suppl, organism_count, light_regm, antibiotic_regm, chem_mutagen, fertilizer_regm, fungicide_regm, gaseous_environment, growth_hormone_regm, herbicide_regm, mineral_nutr_regm, non_min_nutr_regm, pesticide_regm, radiation_regm, salt_regm, climate_environment, perturbation, season_environment, standing_water_regm, host_phenotype, mechanical_damage, cult_root_med, growth_facil, samp_vol_we_dna_ext, project_name, samp_name, host_disease_stat)
);

CREATE TABLE "plant_associated_MIGS_bacteria" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	plant_sex VARCHAR(19), 
	samp_dis_stage VARCHAR(23), 
	growth_habit VARCHAR(10), 
	biol_stat VARCHAR(16), 
	samp_store_dur TEXT, 
	root_med_ph FLOAT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	host_wet_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	air_temp_regm TEXT, 
	gravity TEXT, 
	humidity_regm TEXT, 
	rainfall_regm TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	root_cond TEXT, 
	tiss_cult_growth_med TEXT, 
	host_subspecf_genlin TEXT, 
	plant_struc TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT, 
	ances_data TEXT, 
	biotic_regm TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_symbiont TEXT, 
	plant_product TEXT, 
	root_med_solid TEXT, 
	samp_store_loc TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	root_med_regl TEXT, 
	root_med_suppl TEXT, 
	organism_count TEXT, 
	light_regm TEXT, 
	antibiotic_regm TEXT, 
	chem_mutagen TEXT, 
	fertilizer_regm TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	growth_hormone_regm TEXT, 
	herbicide_regm TEXT, 
	mineral_nutr_regm TEXT, 
	non_min_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	radiation_regm TEXT, 
	salt_regm TEXT, 
	climate_environment TEXT, 
	perturbation TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	cult_root_med TEXT, 
	growth_facil TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, plant_sex, samp_dis_stage, growth_habit, biol_stat, samp_store_dur, root_med_ph, host_age, host_dry_mass, host_height, host_length, host_tot_mass, host_wet_mass, salinity, samp_store_temp, air_temp_regm, gravity, humidity_regm, rainfall_regm, water_temp_regm, watering_regm, ph_regm, host_taxid, genetic_mod, root_cond, tiss_cult_growth_med, host_subspecf_genlin, plant_struc, plant_growth_med, chem_administration, ances_data, biotic_regm, host_common_name, host_genotype, host_life_stage, host_symbiont, plant_product, root_med_solid, samp_store_loc, misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, root_med_regl, root_med_suppl, organism_count, light_regm, antibiotic_regm, chem_mutagen, fertilizer_regm, fungicide_regm, gaseous_environment, growth_hormone_regm, herbicide_regm, mineral_nutr_regm, non_min_nutr_regm, pesticide_regm, radiation_regm, salt_regm, climate_environment, perturbation, season_environment, standing_water_regm, host_phenotype, mechanical_damage, cult_root_med, growth_facil, samp_vol_we_dna_ext, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "plant_associated_MIGS_eukaryote" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	plant_sex VARCHAR(19), 
	samp_dis_stage VARCHAR(23), 
	growth_habit VARCHAR(10), 
	biol_stat VARCHAR(16), 
	samp_store_dur TEXT, 
	root_med_ph FLOAT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	host_wet_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	air_temp_regm TEXT, 
	gravity TEXT, 
	humidity_regm TEXT, 
	rainfall_regm TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	root_cond TEXT, 
	tiss_cult_growth_med TEXT, 
	host_subspecf_genlin TEXT, 
	plant_struc TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT, 
	ances_data TEXT, 
	biotic_regm TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_symbiont TEXT, 
	plant_product TEXT, 
	root_med_solid TEXT, 
	samp_store_loc TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	root_med_regl TEXT, 
	root_med_suppl TEXT, 
	organism_count TEXT, 
	light_regm TEXT, 
	antibiotic_regm TEXT, 
	chem_mutagen TEXT, 
	fertilizer_regm TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	growth_hormone_regm TEXT, 
	herbicide_regm TEXT, 
	mineral_nutr_regm TEXT, 
	non_min_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	radiation_regm TEXT, 
	salt_regm TEXT, 
	climate_environment TEXT, 
	perturbation TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	cult_root_med TEXT, 
	growth_facil TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, plant_sex, samp_dis_stage, growth_habit, biol_stat, samp_store_dur, root_med_ph, host_age, host_dry_mass, host_height, host_length, host_tot_mass, host_wet_mass, salinity, samp_store_temp, air_temp_regm, gravity, humidity_regm, rainfall_regm, water_temp_regm, watering_regm, ph_regm, host_taxid, genetic_mod, root_cond, tiss_cult_growth_med, host_subspecf_genlin, plant_struc, plant_growth_med, chem_administration, ances_data, biotic_regm, host_common_name, host_genotype, host_life_stage, host_symbiont, plant_product, root_med_solid, samp_store_loc, misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, root_med_regl, root_med_suppl, organism_count, light_regm, antibiotic_regm, chem_mutagen, fertilizer_regm, fungicide_regm, gaseous_environment, growth_hormone_regm, herbicide_regm, mineral_nutr_regm, non_min_nutr_regm, pesticide_regm, radiation_regm, salt_regm, climate_environment, perturbation, season_environment, standing_water_regm, host_phenotype, mechanical_damage, cult_root_med, growth_facil, samp_vol_we_dna_ext, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "plant_associated_MIGS_org" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	plant_sex VARCHAR(19), 
	samp_dis_stage VARCHAR(23), 
	growth_habit VARCHAR(10), 
	biol_stat VARCHAR(16), 
	samp_store_dur TEXT, 
	root_med_ph FLOAT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	host_wet_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	air_temp_regm TEXT, 
	gravity TEXT, 
	humidity_regm TEXT, 
	rainfall_regm TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	root_cond TEXT, 
	tiss_cult_growth_med TEXT, 
	host_subspecf_genlin TEXT, 
	plant_struc TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT, 
	ances_data TEXT, 
	biotic_regm TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_symbiont TEXT, 
	plant_product TEXT, 
	root_med_solid TEXT, 
	samp_store_loc TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	root_med_regl TEXT, 
	root_med_suppl TEXT, 
	organism_count TEXT, 
	light_regm TEXT, 
	antibiotic_regm TEXT, 
	chem_mutagen TEXT, 
	fertilizer_regm TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	growth_hormone_regm TEXT, 
	herbicide_regm TEXT, 
	mineral_nutr_regm TEXT, 
	non_min_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	radiation_regm TEXT, 
	salt_regm TEXT, 
	climate_environment TEXT, 
	perturbation TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	cult_root_med TEXT, 
	growth_facil TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, plant_sex, samp_dis_stage, growth_habit, biol_stat, samp_store_dur, root_med_ph, host_age, host_dry_mass, host_height, host_length, host_tot_mass, host_wet_mass, salinity, samp_store_temp, air_temp_regm, gravity, humidity_regm, rainfall_regm, water_temp_regm, watering_regm, ph_regm, host_taxid, genetic_mod, root_cond, tiss_cult_growth_med, host_subspecf_genlin, plant_struc, plant_growth_med, chem_administration, ances_data, biotic_regm, host_common_name, host_genotype, host_life_stage, host_symbiont, plant_product, root_med_solid, samp_store_loc, misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, root_med_regl, root_med_suppl, organism_count, light_regm, antibiotic_regm, chem_mutagen, fertilizer_regm, fungicide_regm, gaseous_environment, growth_hormone_regm, herbicide_regm, mineral_nutr_regm, non_min_nutr_regm, pesticide_regm, radiation_regm, salt_regm, climate_environment, perturbation, season_environment, standing_water_regm, host_phenotype, mechanical_damage, cult_root_med, growth_facil, samp_vol_we_dna_ext, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "plant_associated_MIGS_plant" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	plant_sex VARCHAR(19), 
	samp_dis_stage VARCHAR(23), 
	growth_habit VARCHAR(10), 
	biol_stat VARCHAR(16), 
	samp_store_dur TEXT, 
	root_med_ph FLOAT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	host_wet_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	air_temp_regm TEXT, 
	gravity TEXT, 
	humidity_regm TEXT, 
	rainfall_regm TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	root_cond TEXT, 
	tiss_cult_growth_med TEXT, 
	host_subspecf_genlin TEXT, 
	plant_struc TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT, 
	ances_data TEXT, 
	biotic_regm TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_symbiont TEXT, 
	plant_product TEXT, 
	root_med_solid TEXT, 
	samp_store_loc TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	root_med_regl TEXT, 
	root_med_suppl TEXT, 
	organism_count TEXT, 
	light_regm TEXT, 
	antibiotic_regm TEXT, 
	chem_mutagen TEXT, 
	fertilizer_regm TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	growth_hormone_regm TEXT, 
	herbicide_regm TEXT, 
	mineral_nutr_regm TEXT, 
	non_min_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	radiation_regm TEXT, 
	salt_regm TEXT, 
	climate_environment TEXT, 
	perturbation TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	cult_root_med TEXT, 
	growth_facil TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, plant_sex, samp_dis_stage, growth_habit, biol_stat, samp_store_dur, root_med_ph, host_age, host_dry_mass, host_height, host_length, host_tot_mass, host_wet_mass, salinity, samp_store_temp, air_temp_regm, gravity, humidity_regm, rainfall_regm, water_temp_regm, watering_regm, ph_regm, host_taxid, genetic_mod, root_cond, tiss_cult_growth_med, host_subspecf_genlin, plant_struc, plant_growth_med, chem_administration, ances_data, biotic_regm, host_common_name, host_genotype, host_life_stage, host_symbiont, plant_product, root_med_solid, samp_store_loc, misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, root_med_regl, root_med_suppl, organism_count, light_regm, antibiotic_regm, chem_mutagen, fertilizer_regm, fungicide_regm, gaseous_environment, growth_hormone_regm, herbicide_regm, mineral_nutr_regm, non_min_nutr_regm, pesticide_regm, radiation_regm, salt_regm, climate_environment, perturbation, season_environment, standing_water_regm, host_phenotype, mechanical_damage, cult_root_med, growth_facil, samp_vol_we_dna_ext, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "plant_associated_MIGS_virus" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	plant_sex VARCHAR(19), 
	samp_dis_stage VARCHAR(23), 
	growth_habit VARCHAR(10), 
	biol_stat VARCHAR(16), 
	samp_store_dur TEXT, 
	root_med_ph FLOAT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	host_wet_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	air_temp_regm TEXT, 
	gravity TEXT, 
	humidity_regm TEXT, 
	rainfall_regm TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	root_cond TEXT, 
	tiss_cult_growth_med TEXT, 
	host_subspecf_genlin TEXT, 
	plant_struc TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT, 
	ances_data TEXT, 
	biotic_regm TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_symbiont TEXT, 
	plant_product TEXT, 
	root_med_solid TEXT, 
	samp_store_loc TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	root_med_regl TEXT, 
	root_med_suppl TEXT, 
	organism_count TEXT, 
	light_regm TEXT, 
	antibiotic_regm TEXT, 
	chem_mutagen TEXT, 
	fertilizer_regm TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	growth_hormone_regm TEXT, 
	herbicide_regm TEXT, 
	mineral_nutr_regm TEXT, 
	non_min_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	radiation_regm TEXT, 
	salt_regm TEXT, 
	climate_environment TEXT, 
	perturbation TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	cult_root_med TEXT, 
	growth_facil TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, plant_sex, samp_dis_stage, growth_habit, biol_stat, samp_store_dur, root_med_ph, host_age, host_dry_mass, host_height, host_length, host_tot_mass, host_wet_mass, salinity, samp_store_temp, air_temp_regm, gravity, humidity_regm, rainfall_regm, water_temp_regm, watering_regm, ph_regm, host_taxid, genetic_mod, root_cond, tiss_cult_growth_med, host_subspecf_genlin, plant_struc, plant_growth_med, chem_administration, ances_data, biotic_regm, host_common_name, host_genotype, host_life_stage, host_symbiont, plant_product, root_med_solid, samp_store_loc, misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, root_med_regl, root_med_suppl, organism_count, light_regm, antibiotic_regm, chem_mutagen, fertilizer_regm, fungicide_regm, gaseous_environment, growth_hormone_regm, herbicide_regm, mineral_nutr_regm, non_min_nutr_regm, pesticide_regm, radiation_regm, salt_regm, climate_environment, perturbation, season_environment, standing_water_regm, host_phenotype, mechanical_damage, cult_root_med, growth_facil, samp_vol_we_dna_ext, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "plant_associated_MIMAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	plant_sex VARCHAR(19), 
	samp_dis_stage VARCHAR(23), 
	growth_habit VARCHAR(10), 
	biol_stat VARCHAR(16), 
	samp_store_dur TEXT, 
	root_med_ph FLOAT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	host_wet_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	air_temp_regm TEXT, 
	gravity TEXT, 
	humidity_regm TEXT, 
	rainfall_regm TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	root_cond TEXT, 
	tiss_cult_growth_med TEXT, 
	host_subspecf_genlin TEXT, 
	plant_struc TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT, 
	ances_data TEXT, 
	biotic_regm TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_symbiont TEXT, 
	plant_product TEXT, 
	root_med_solid TEXT, 
	samp_store_loc TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	root_med_regl TEXT, 
	root_med_suppl TEXT, 
	organism_count TEXT, 
	light_regm TEXT, 
	antibiotic_regm TEXT, 
	chem_mutagen TEXT, 
	fertilizer_regm TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	growth_hormone_regm TEXT, 
	herbicide_regm TEXT, 
	mineral_nutr_regm TEXT, 
	non_min_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	radiation_regm TEXT, 
	salt_regm TEXT, 
	climate_environment TEXT, 
	perturbation TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	cult_root_med TEXT, 
	growth_facil TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, plant_sex, samp_dis_stage, growth_habit, biol_stat, samp_store_dur, root_med_ph, host_age, host_dry_mass, host_height, host_length, host_tot_mass, host_wet_mass, salinity, samp_store_temp, air_temp_regm, gravity, humidity_regm, rainfall_regm, water_temp_regm, watering_regm, ph_regm, host_taxid, genetic_mod, root_cond, tiss_cult_growth_med, host_subspecf_genlin, plant_struc, plant_growth_med, chem_administration, ances_data, biotic_regm, host_common_name, host_genotype, host_life_stage, host_symbiont, plant_product, root_med_solid, samp_store_loc, misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, root_med_regl, root_med_suppl, organism_count, light_regm, antibiotic_regm, chem_mutagen, fertilizer_regm, fungicide_regm, gaseous_environment, growth_hormone_regm, herbicide_regm, mineral_nutr_regm, non_min_nutr_regm, pesticide_regm, radiation_regm, salt_regm, climate_environment, perturbation, season_environment, standing_water_regm, host_phenotype, mechanical_damage, cult_root_med, growth_facil, samp_vol_we_dna_ext, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "plant_associated_MIMARKS_specimen" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	plant_sex VARCHAR(19), 
	samp_dis_stage VARCHAR(23), 
	growth_habit VARCHAR(10), 
	biol_stat VARCHAR(16), 
	samp_store_dur TEXT, 
	root_med_ph FLOAT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	host_wet_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	air_temp_regm TEXT, 
	gravity TEXT, 
	humidity_regm TEXT, 
	rainfall_regm TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	root_cond TEXT, 
	tiss_cult_growth_med TEXT, 
	host_subspecf_genlin TEXT, 
	plant_struc TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT, 
	ances_data TEXT, 
	biotic_regm TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_symbiont TEXT, 
	plant_product TEXT, 
	root_med_solid TEXT, 
	samp_store_loc TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	root_med_regl TEXT, 
	root_med_suppl TEXT, 
	organism_count TEXT, 
	light_regm TEXT, 
	antibiotic_regm TEXT, 
	chem_mutagen TEXT, 
	fertilizer_regm TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	growth_hormone_regm TEXT, 
	herbicide_regm TEXT, 
	mineral_nutr_regm TEXT, 
	non_min_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	radiation_regm TEXT, 
	salt_regm TEXT, 
	climate_environment TEXT, 
	perturbation TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	cult_root_med TEXT, 
	growth_facil TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, plant_sex, samp_dis_stage, growth_habit, biol_stat, samp_store_dur, root_med_ph, host_age, host_dry_mass, host_height, host_length, host_tot_mass, host_wet_mass, salinity, samp_store_temp, air_temp_regm, gravity, humidity_regm, rainfall_regm, water_temp_regm, watering_regm, ph_regm, host_taxid, genetic_mod, root_cond, tiss_cult_growth_med, host_subspecf_genlin, plant_struc, plant_growth_med, chem_administration, ances_data, biotic_regm, host_common_name, host_genotype, host_life_stage, host_symbiont, plant_product, root_med_solid, samp_store_loc, misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, root_med_regl, root_med_suppl, organism_count, light_regm, antibiotic_regm, chem_mutagen, fertilizer_regm, fungicide_regm, gaseous_environment, growth_hormone_regm, herbicide_regm, mineral_nutr_regm, non_min_nutr_regm, pesticide_regm, radiation_regm, salt_regm, climate_environment, perturbation, season_environment, standing_water_regm, host_phenotype, mechanical_damage, cult_root_med, growth_facil, samp_vol_we_dna_ext, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "plant_associated_MIMARKS_survey" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	plant_sex VARCHAR(19), 
	samp_dis_stage VARCHAR(23), 
	growth_habit VARCHAR(10), 
	biol_stat VARCHAR(16), 
	samp_store_dur TEXT, 
	root_med_ph FLOAT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	host_wet_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	air_temp_regm TEXT, 
	gravity TEXT, 
	humidity_regm TEXT, 
	rainfall_regm TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	root_cond TEXT, 
	tiss_cult_growth_med TEXT, 
	host_subspecf_genlin TEXT, 
	plant_struc TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT, 
	ances_data TEXT, 
	biotic_regm TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_symbiont TEXT, 
	plant_product TEXT, 
	root_med_solid TEXT, 
	samp_store_loc TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	root_med_regl TEXT, 
	root_med_suppl TEXT, 
	organism_count TEXT, 
	light_regm TEXT, 
	antibiotic_regm TEXT, 
	chem_mutagen TEXT, 
	fertilizer_regm TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	growth_hormone_regm TEXT, 
	herbicide_regm TEXT, 
	mineral_nutr_regm TEXT, 
	non_min_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	radiation_regm TEXT, 
	salt_regm TEXT, 
	climate_environment TEXT, 
	perturbation TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	cult_root_med TEXT, 
	growth_facil TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, plant_sex, samp_dis_stage, growth_habit, biol_stat, samp_store_dur, root_med_ph, host_age, host_dry_mass, host_height, host_length, host_tot_mass, host_wet_mass, salinity, samp_store_temp, air_temp_regm, gravity, humidity_regm, rainfall_regm, water_temp_regm, watering_regm, ph_regm, host_taxid, genetic_mod, root_cond, tiss_cult_growth_med, host_subspecf_genlin, plant_struc, plant_growth_med, chem_administration, ances_data, biotic_regm, host_common_name, host_genotype, host_life_stage, host_symbiont, plant_product, root_med_solid, samp_store_loc, misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, root_med_regl, root_med_suppl, organism_count, light_regm, antibiotic_regm, chem_mutagen, fertilizer_regm, fungicide_regm, gaseous_environment, growth_hormone_regm, herbicide_regm, mineral_nutr_regm, non_min_nutr_regm, pesticide_regm, radiation_regm, salt_regm, climate_environment, perturbation, season_environment, standing_water_regm, host_phenotype, mechanical_damage, cult_root_med, growth_facil, samp_vol_we_dna_ext, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "plant_associated_MIMS" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	plant_sex VARCHAR(19), 
	samp_dis_stage VARCHAR(23), 
	growth_habit VARCHAR(10), 
	biol_stat VARCHAR(16), 
	samp_store_dur TEXT, 
	root_med_ph FLOAT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	host_wet_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	air_temp_regm TEXT, 
	gravity TEXT, 
	humidity_regm TEXT, 
	rainfall_regm TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	root_cond TEXT, 
	tiss_cult_growth_med TEXT, 
	host_subspecf_genlin TEXT, 
	plant_struc TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT, 
	ances_data TEXT, 
	biotic_regm TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_symbiont TEXT, 
	plant_product TEXT, 
	root_med_solid TEXT, 
	samp_store_loc TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	root_med_regl TEXT, 
	root_med_suppl TEXT, 
	organism_count TEXT, 
	light_regm TEXT, 
	antibiotic_regm TEXT, 
	chem_mutagen TEXT, 
	fertilizer_regm TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	growth_hormone_regm TEXT, 
	herbicide_regm TEXT, 
	mineral_nutr_regm TEXT, 
	non_min_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	radiation_regm TEXT, 
	salt_regm TEXT, 
	climate_environment TEXT, 
	perturbation TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	cult_root_med TEXT, 
	growth_facil TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, plant_sex, samp_dis_stage, growth_habit, biol_stat, samp_store_dur, root_med_ph, host_age, host_dry_mass, host_height, host_length, host_tot_mass, host_wet_mass, salinity, samp_store_temp, air_temp_regm, gravity, humidity_regm, rainfall_regm, water_temp_regm, watering_regm, ph_regm, host_taxid, genetic_mod, root_cond, tiss_cult_growth_med, host_subspecf_genlin, plant_struc, plant_growth_med, chem_administration, ances_data, biotic_regm, host_common_name, host_genotype, host_life_stage, host_symbiont, plant_product, root_med_solid, samp_store_loc, misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, root_med_regl, root_med_suppl, organism_count, light_regm, antibiotic_regm, chem_mutagen, fertilizer_regm, fungicide_regm, gaseous_environment, growth_hormone_regm, herbicide_regm, mineral_nutr_regm, non_min_nutr_regm, pesticide_regm, radiation_regm, salt_regm, climate_environment, perturbation, season_environment, standing_water_regm, host_phenotype, mechanical_damage, cult_root_med, growth_facil, samp_vol_we_dna_ext, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "plant_associated_MISAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	plant_sex VARCHAR(19), 
	samp_dis_stage VARCHAR(23), 
	growth_habit VARCHAR(10), 
	biol_stat VARCHAR(16), 
	samp_store_dur TEXT, 
	root_med_ph FLOAT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	host_wet_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	air_temp_regm TEXT, 
	gravity TEXT, 
	humidity_regm TEXT, 
	rainfall_regm TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	root_cond TEXT, 
	tiss_cult_growth_med TEXT, 
	host_subspecf_genlin TEXT, 
	plant_struc TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT, 
	ances_data TEXT, 
	biotic_regm TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_symbiont TEXT, 
	plant_product TEXT, 
	root_med_solid TEXT, 
	samp_store_loc TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	root_med_regl TEXT, 
	root_med_suppl TEXT, 
	organism_count TEXT, 
	light_regm TEXT, 
	antibiotic_regm TEXT, 
	chem_mutagen TEXT, 
	fertilizer_regm TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	growth_hormone_regm TEXT, 
	herbicide_regm TEXT, 
	mineral_nutr_regm TEXT, 
	non_min_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	radiation_regm TEXT, 
	salt_regm TEXT, 
	climate_environment TEXT, 
	perturbation TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	cult_root_med TEXT, 
	growth_facil TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, plant_sex, samp_dis_stage, growth_habit, biol_stat, samp_store_dur, root_med_ph, host_age, host_dry_mass, host_height, host_length, host_tot_mass, host_wet_mass, salinity, samp_store_temp, air_temp_regm, gravity, humidity_regm, rainfall_regm, water_temp_regm, watering_regm, ph_regm, host_taxid, genetic_mod, root_cond, tiss_cult_growth_med, host_subspecf_genlin, plant_struc, plant_growth_med, chem_administration, ances_data, biotic_regm, host_common_name, host_genotype, host_life_stage, host_symbiont, plant_product, root_med_solid, samp_store_loc, misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, root_med_regl, root_med_suppl, organism_count, light_regm, antibiotic_regm, chem_mutagen, fertilizer_regm, fungicide_regm, gaseous_environment, growth_hormone_regm, herbicide_regm, mineral_nutr_regm, non_min_nutr_regm, pesticide_regm, radiation_regm, salt_regm, climate_environment, perturbation, season_environment, standing_water_regm, host_phenotype, mechanical_damage, cult_root_med, growth_facil, samp_vol_we_dna_ext, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "plant_associated_MIUVIG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	samp_capt_status VARCHAR(48), 
	oxy_stat_samp TEXT, 
	plant_sex VARCHAR(19), 
	samp_dis_stage VARCHAR(23), 
	growth_habit VARCHAR(10), 
	biol_stat VARCHAR(16), 
	samp_store_dur TEXT, 
	root_med_ph FLOAT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_tot_mass TEXT, 
	host_wet_mass TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	air_temp_regm TEXT, 
	gravity TEXT, 
	humidity_regm TEXT, 
	rainfall_regm TEXT, 
	water_temp_regm TEXT, 
	watering_regm TEXT, 
	ph_regm TEXT, 
	host_taxid INTEGER, 
	genetic_mod TEXT, 
	root_cond TEXT, 
	tiss_cult_growth_med TEXT, 
	host_subspecf_genlin TEXT, 
	plant_struc TEXT, 
	plant_growth_med VARCHAR(30), 
	chem_administration TEXT, 
	ances_data TEXT, 
	biotic_regm TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_life_stage TEXT, 
	host_symbiont TEXT, 
	plant_product TEXT, 
	root_med_solid TEXT, 
	samp_store_loc TEXT, 
	misc_param TEXT, 
	root_med_carbon TEXT, 
	root_med_macronutr TEXT, 
	root_med_micronutr TEXT, 
	root_med_regl TEXT, 
	root_med_suppl TEXT, 
	organism_count TEXT, 
	light_regm TEXT, 
	antibiotic_regm TEXT, 
	chem_mutagen TEXT, 
	fertilizer_regm TEXT, 
	fungicide_regm TEXT, 
	gaseous_environment TEXT, 
	growth_hormone_regm TEXT, 
	herbicide_regm TEXT, 
	mineral_nutr_regm TEXT, 
	non_min_nutr_regm TEXT, 
	pesticide_regm TEXT, 
	radiation_regm TEXT, 
	salt_regm TEXT, 
	climate_environment TEXT, 
	perturbation TEXT, 
	season_environment TEXT, 
	standing_water_regm TEXT, 
	host_phenotype TEXT, 
	mechanical_damage TEXT, 
	cult_root_med TEXT, 
	growth_facil TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	host_disease_stat TEXT, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, samp_capt_status, oxy_stat_samp, plant_sex, samp_dis_stage, growth_habit, biol_stat, samp_store_dur, root_med_ph, host_age, host_dry_mass, host_height, host_length, host_tot_mass, host_wet_mass, salinity, samp_store_temp, air_temp_regm, gravity, humidity_regm, rainfall_regm, water_temp_regm, watering_regm, ph_regm, host_taxid, genetic_mod, root_cond, tiss_cult_growth_med, host_subspecf_genlin, plant_struc, plant_growth_med, chem_administration, ances_data, biotic_regm, host_common_name, host_genotype, host_life_stage, host_symbiont, plant_product, root_med_solid, samp_store_loc, misc_param, root_med_carbon, root_med_macronutr, root_med_micronutr, root_med_regl, root_med_suppl, organism_count, light_regm, antibiotic_regm, chem_mutagen, fertilizer_regm, fungicide_regm, gaseous_environment, growth_hormone_regm, herbicide_regm, mineral_nutr_regm, non_min_nutr_regm, pesticide_regm, radiation_regm, salt_regm, climate_environment, perturbation, season_environment, standing_water_regm, host_phenotype, mechanical_damage, cult_root_med, growth_facil, samp_vol_we_dna_ext, project_name, samp_name, host_disease_stat, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE quantity_value (
	has_unit TEXT, 
	has_numeric_value FLOAT, 
	has_raw_value TEXT, 
	PRIMARY KEY (has_unit, has_numeric_value, has_raw_value)
);

CREATE TABLE sediment (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	sediment_type VARCHAR(11), 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	porosity TEXT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_depth_water_col TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	particle_class TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, sediment_type, tidal_stage, samp_store_dur, ph, porosity, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_depth_water_col, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, particle_class, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name)
);

CREATE TABLE "sediment_MIGS_bacteria" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	sediment_type VARCHAR(11), 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	porosity TEXT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_depth_water_col TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	particle_class TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, sediment_type, tidal_stage, samp_store_dur, ph, porosity, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_depth_water_col, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, particle_class, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "sediment_MIGS_eukaryote" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	sediment_type VARCHAR(11), 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	porosity TEXT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_depth_water_col TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	particle_class TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, sediment_type, tidal_stage, samp_store_dur, ph, porosity, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_depth_water_col, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, particle_class, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "sediment_MIGS_org" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	sediment_type VARCHAR(11), 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	porosity TEXT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_depth_water_col TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	particle_class TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, sediment_type, tidal_stage, samp_store_dur, ph, porosity, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_depth_water_col, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, particle_class, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "sediment_MIGS_plant" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	sediment_type VARCHAR(11), 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	porosity TEXT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_depth_water_col TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	particle_class TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, sediment_type, tidal_stage, samp_store_dur, ph, porosity, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_depth_water_col, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, particle_class, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "sediment_MIGS_virus" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	sediment_type VARCHAR(11), 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	porosity TEXT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_depth_water_col TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	particle_class TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, sediment_type, tidal_stage, samp_store_dur, ph, porosity, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_depth_water_col, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, particle_class, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "sediment_MIMAG" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	sediment_type VARCHAR(11), 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	porosity TEXT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_depth_water_col TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	particle_class TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, sediment_type, tidal_stage, samp_store_dur, ph, porosity, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_depth_water_col, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, particle_class, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "sediment_MIMARKS_specimen" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	sediment_type VARCHAR(11), 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	porosity TEXT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_depth_water_col TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	particle_class TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, sediment_type, tidal_stage, samp_store_dur, ph, porosity, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_depth_water_col, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, particle_class, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "sediment_MIMARKS_survey" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	sediment_type VARCHAR(11), 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	porosity TEXT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_depth_water_col TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	particle_class TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, sediment_type, tidal_stage, samp_store_dur, ph, porosity, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_depth_water_col, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, particle_class, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "sediment_MIMS" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	sediment_type VARCHAR(11), 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	porosity TEXT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_depth_water_col TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	particle_class TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, sediment_type, tidal_stage, samp_store_dur, ph, porosity, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_depth_water_col, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, particle_class, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "sediment_MISAG" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	sediment_type VARCHAR(11), 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	porosity TEXT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_depth_water_col TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	particle_class TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, sediment_type, tidal_stage, samp_store_dur, ph, porosity, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_depth_water_col, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, particle_class, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "sediment_MIUVIG" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	sediment_type VARCHAR(11), 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	porosity TEXT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	carb_nitro_ratio TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	glucosidase_act TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	methane TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	sodium TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	tot_carb TEXT, 
	tot_depth_water_col TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	turbidity TEXT, 
	water_content FLOAT, 
	chem_administration TEXT, 
	samp_store_loc TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	particle_class TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, sediment_type, tidal_stage, samp_store_dur, ph, porosity, alkalinity, alkyl_diethers, aminopept_act, ammonium, bacteria_carb_prod, bishomohopanol, bromide, calcium, carb_nitro_ratio, chloride, chlorophyll, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_org_carb, diss_org_nitro, diss_oxygen, glucosidase_act, magnesium, mean_frict_vel, mean_peak_frict_vel, methane, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, petroleum_hydrocarb, phosphate, potassium, pressure, redox_potential, salinity, samp_store_temp, silicate, sodium, sulfate, sulfide, tot_carb, tot_depth_water_col, tot_nitro_content, tot_org_carb, turbidity, water_content, chem_administration, samp_store_loc, biomass, diether_lipids, misc_param, n_alkanes, particle_class, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE soil (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT NOT NULL, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	fao_class VARCHAR(13), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	soil_horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT, 
	pool_dna_extracts TEXT, 
	crop_rotation TEXT, 
	ph FLOAT, 
	water_content FLOAT, 
	al_sat TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	microbial_biomass TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	soil_text_measure TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	al_sat_meth TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_addit_analys TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	micro_biomass_meth TEXT, 
	ph_meth TEXT, 
	prev_land_use_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	water_cont_soil_meth TEXT, 
	soil_type TEXT, 
	cur_vegetation TEXT, 
	local_class TEXT, 
	store_cond TEXT, 
	heavy_metals TEXT, 
	misc_param TEXT, 
	agrochem_addition TEXT, 
	previous_land_use TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, fao_class, cur_land_use, tillage, soil_horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, ph, water_content, al_sat, annual_precpt, annual_temp, microbial_biomass, org_matter, org_nitro, season_precpt, season_temp, slope_aspect, slope_gradient, soil_text_measure, tot_nitro_content, tot_org_carb, al_sat_meth, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_addit_analys, link_class_info, link_climate_info, local_class_meth, micro_biomass_meth, ph_meth, prev_land_use_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, water_cont_soil_meth, soil_type, cur_vegetation, local_class, store_cond, heavy_metals, misc_param, agrochem_addition, previous_land_use, extreme_event, fire, flooding, samp_vol_we_dna_ext, project_name, samp_name)
);

CREATE TABLE "soil_MIGS_bacteria" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT NOT NULL, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	fao_class VARCHAR(13), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	soil_horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT, 
	pool_dna_extracts TEXT, 
	crop_rotation TEXT, 
	ph FLOAT, 
	water_content FLOAT, 
	al_sat TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	microbial_biomass TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	soil_text_measure TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	al_sat_meth TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_addit_analys TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	micro_biomass_meth TEXT, 
	ph_meth TEXT, 
	prev_land_use_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	water_cont_soil_meth TEXT, 
	soil_type TEXT, 
	cur_vegetation TEXT, 
	local_class TEXT, 
	store_cond TEXT, 
	heavy_metals TEXT, 
	misc_param TEXT, 
	agrochem_addition TEXT, 
	previous_land_use TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, fao_class, cur_land_use, tillage, soil_horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, ph, water_content, al_sat, annual_precpt, annual_temp, microbial_biomass, org_matter, org_nitro, season_precpt, season_temp, slope_aspect, slope_gradient, soil_text_measure, tot_nitro_content, tot_org_carb, al_sat_meth, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_addit_analys, link_class_info, link_climate_info, local_class_meth, micro_biomass_meth, ph_meth, prev_land_use_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, water_cont_soil_meth, soil_type, cur_vegetation, local_class, store_cond, heavy_metals, misc_param, agrochem_addition, previous_land_use, extreme_event, fire, flooding, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "soil_MIGS_eukaryote" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT NOT NULL, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	fao_class VARCHAR(13), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	soil_horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT, 
	pool_dna_extracts TEXT, 
	crop_rotation TEXT, 
	ph FLOAT, 
	water_content FLOAT, 
	al_sat TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	microbial_biomass TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	soil_text_measure TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	al_sat_meth TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_addit_analys TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	micro_biomass_meth TEXT, 
	ph_meth TEXT, 
	prev_land_use_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	water_cont_soil_meth TEXT, 
	soil_type TEXT, 
	cur_vegetation TEXT, 
	local_class TEXT, 
	store_cond TEXT, 
	heavy_metals TEXT, 
	misc_param TEXT, 
	agrochem_addition TEXT, 
	previous_land_use TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, fao_class, cur_land_use, tillage, soil_horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, ph, water_content, al_sat, annual_precpt, annual_temp, microbial_biomass, org_matter, org_nitro, season_precpt, season_temp, slope_aspect, slope_gradient, soil_text_measure, tot_nitro_content, tot_org_carb, al_sat_meth, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_addit_analys, link_class_info, link_climate_info, local_class_meth, micro_biomass_meth, ph_meth, prev_land_use_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, water_cont_soil_meth, soil_type, cur_vegetation, local_class, store_cond, heavy_metals, misc_param, agrochem_addition, previous_land_use, extreme_event, fire, flooding, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "soil_MIGS_org" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT NOT NULL, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	fao_class VARCHAR(13), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	soil_horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT, 
	pool_dna_extracts TEXT, 
	crop_rotation TEXT, 
	ph FLOAT, 
	water_content FLOAT, 
	al_sat TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	microbial_biomass TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	soil_text_measure TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	al_sat_meth TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_addit_analys TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	micro_biomass_meth TEXT, 
	ph_meth TEXT, 
	prev_land_use_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	water_cont_soil_meth TEXT, 
	soil_type TEXT, 
	cur_vegetation TEXT, 
	local_class TEXT, 
	store_cond TEXT, 
	heavy_metals TEXT, 
	misc_param TEXT, 
	agrochem_addition TEXT, 
	previous_land_use TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, fao_class, cur_land_use, tillage, soil_horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, ph, water_content, al_sat, annual_precpt, annual_temp, microbial_biomass, org_matter, org_nitro, season_precpt, season_temp, slope_aspect, slope_gradient, soil_text_measure, tot_nitro_content, tot_org_carb, al_sat_meth, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_addit_analys, link_class_info, link_climate_info, local_class_meth, micro_biomass_meth, ph_meth, prev_land_use_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, water_cont_soil_meth, soil_type, cur_vegetation, local_class, store_cond, heavy_metals, misc_param, agrochem_addition, previous_land_use, extreme_event, fire, flooding, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "soil_MIGS_plant" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT NOT NULL, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	fao_class VARCHAR(13), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	soil_horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT, 
	pool_dna_extracts TEXT, 
	crop_rotation TEXT, 
	ph FLOAT, 
	water_content FLOAT, 
	al_sat TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	microbial_biomass TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	soil_text_measure TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	al_sat_meth TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_addit_analys TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	micro_biomass_meth TEXT, 
	ph_meth TEXT, 
	prev_land_use_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	water_cont_soil_meth TEXT, 
	soil_type TEXT, 
	cur_vegetation TEXT, 
	local_class TEXT, 
	store_cond TEXT, 
	heavy_metals TEXT, 
	misc_param TEXT, 
	agrochem_addition TEXT, 
	previous_land_use TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, fao_class, cur_land_use, tillage, soil_horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, ph, water_content, al_sat, annual_precpt, annual_temp, microbial_biomass, org_matter, org_nitro, season_precpt, season_temp, slope_aspect, slope_gradient, soil_text_measure, tot_nitro_content, tot_org_carb, al_sat_meth, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_addit_analys, link_class_info, link_climate_info, local_class_meth, micro_biomass_meth, ph_meth, prev_land_use_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, water_cont_soil_meth, soil_type, cur_vegetation, local_class, store_cond, heavy_metals, misc_param, agrochem_addition, previous_land_use, extreme_event, fire, flooding, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "soil_MIGS_virus" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT NOT NULL, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	fao_class VARCHAR(13), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	soil_horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT, 
	pool_dna_extracts TEXT, 
	crop_rotation TEXT, 
	ph FLOAT, 
	water_content FLOAT, 
	al_sat TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	microbial_biomass TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	soil_text_measure TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	al_sat_meth TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_addit_analys TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	micro_biomass_meth TEXT, 
	ph_meth TEXT, 
	prev_land_use_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	water_cont_soil_meth TEXT, 
	soil_type TEXT, 
	cur_vegetation TEXT, 
	local_class TEXT, 
	store_cond TEXT, 
	heavy_metals TEXT, 
	misc_param TEXT, 
	agrochem_addition TEXT, 
	previous_land_use TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, fao_class, cur_land_use, tillage, soil_horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, ph, water_content, al_sat, annual_precpt, annual_temp, microbial_biomass, org_matter, org_nitro, season_precpt, season_temp, slope_aspect, slope_gradient, soil_text_measure, tot_nitro_content, tot_org_carb, al_sat_meth, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_addit_analys, link_class_info, link_climate_info, local_class_meth, micro_biomass_meth, ph_meth, prev_land_use_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, water_cont_soil_meth, soil_type, cur_vegetation, local_class, store_cond, heavy_metals, misc_param, agrochem_addition, previous_land_use, extreme_event, fire, flooding, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "soil_MIMAG" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT NOT NULL, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	fao_class VARCHAR(13), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	soil_horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT, 
	pool_dna_extracts TEXT, 
	crop_rotation TEXT, 
	ph FLOAT, 
	water_content FLOAT, 
	al_sat TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	microbial_biomass TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	soil_text_measure TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	al_sat_meth TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_addit_analys TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	micro_biomass_meth TEXT, 
	ph_meth TEXT, 
	prev_land_use_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	water_cont_soil_meth TEXT, 
	soil_type TEXT, 
	cur_vegetation TEXT, 
	local_class TEXT, 
	store_cond TEXT, 
	heavy_metals TEXT, 
	misc_param TEXT, 
	agrochem_addition TEXT, 
	previous_land_use TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, fao_class, cur_land_use, tillage, soil_horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, ph, water_content, al_sat, annual_precpt, annual_temp, microbial_biomass, org_matter, org_nitro, season_precpt, season_temp, slope_aspect, slope_gradient, soil_text_measure, tot_nitro_content, tot_org_carb, al_sat_meth, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_addit_analys, link_class_info, link_climate_info, local_class_meth, micro_biomass_meth, ph_meth, prev_land_use_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, water_cont_soil_meth, soil_type, cur_vegetation, local_class, store_cond, heavy_metals, misc_param, agrochem_addition, previous_land_use, extreme_event, fire, flooding, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "soil_MIMARKS_specimen" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT NOT NULL, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	fao_class VARCHAR(13), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	soil_horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT, 
	pool_dna_extracts TEXT, 
	crop_rotation TEXT, 
	ph FLOAT, 
	water_content FLOAT, 
	al_sat TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	microbial_biomass TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	soil_text_measure TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	al_sat_meth TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_addit_analys TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	micro_biomass_meth TEXT, 
	ph_meth TEXT, 
	prev_land_use_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	water_cont_soil_meth TEXT, 
	soil_type TEXT, 
	cur_vegetation TEXT, 
	local_class TEXT, 
	store_cond TEXT, 
	heavy_metals TEXT, 
	misc_param TEXT, 
	agrochem_addition TEXT, 
	previous_land_use TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, fao_class, cur_land_use, tillage, soil_horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, ph, water_content, al_sat, annual_precpt, annual_temp, microbial_biomass, org_matter, org_nitro, season_precpt, season_temp, slope_aspect, slope_gradient, soil_text_measure, tot_nitro_content, tot_org_carb, al_sat_meth, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_addit_analys, link_class_info, link_climate_info, local_class_meth, micro_biomass_meth, ph_meth, prev_land_use_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, water_cont_soil_meth, soil_type, cur_vegetation, local_class, store_cond, heavy_metals, misc_param, agrochem_addition, previous_land_use, extreme_event, fire, flooding, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "soil_MIMARKS_survey" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT NOT NULL, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	fao_class VARCHAR(13), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	soil_horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT, 
	pool_dna_extracts TEXT, 
	crop_rotation TEXT, 
	ph FLOAT, 
	water_content FLOAT, 
	al_sat TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	microbial_biomass TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	soil_text_measure TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	al_sat_meth TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_addit_analys TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	micro_biomass_meth TEXT, 
	ph_meth TEXT, 
	prev_land_use_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	water_cont_soil_meth TEXT, 
	soil_type TEXT, 
	cur_vegetation TEXT, 
	local_class TEXT, 
	store_cond TEXT, 
	heavy_metals TEXT, 
	misc_param TEXT, 
	agrochem_addition TEXT, 
	previous_land_use TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, fao_class, cur_land_use, tillage, soil_horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, ph, water_content, al_sat, annual_precpt, annual_temp, microbial_biomass, org_matter, org_nitro, season_precpt, season_temp, slope_aspect, slope_gradient, soil_text_measure, tot_nitro_content, tot_org_carb, al_sat_meth, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_addit_analys, link_class_info, link_climate_info, local_class_meth, micro_biomass_meth, ph_meth, prev_land_use_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, water_cont_soil_meth, soil_type, cur_vegetation, local_class, store_cond, heavy_metals, misc_param, agrochem_addition, previous_land_use, extreme_event, fire, flooding, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "soil_MIMS" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT NOT NULL, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	fao_class VARCHAR(13), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	soil_horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT, 
	pool_dna_extracts TEXT, 
	crop_rotation TEXT, 
	ph FLOAT, 
	water_content FLOAT, 
	al_sat TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	microbial_biomass TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	soil_text_measure TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	al_sat_meth TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_addit_analys TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	micro_biomass_meth TEXT, 
	ph_meth TEXT, 
	prev_land_use_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	water_cont_soil_meth TEXT, 
	soil_type TEXT, 
	cur_vegetation TEXT, 
	local_class TEXT, 
	store_cond TEXT, 
	heavy_metals TEXT, 
	misc_param TEXT, 
	agrochem_addition TEXT, 
	previous_land_use TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, fao_class, cur_land_use, tillage, soil_horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, ph, water_content, al_sat, annual_precpt, annual_temp, microbial_biomass, org_matter, org_nitro, season_precpt, season_temp, slope_aspect, slope_gradient, soil_text_measure, tot_nitro_content, tot_org_carb, al_sat_meth, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_addit_analys, link_class_info, link_climate_info, local_class_meth, micro_biomass_meth, ph_meth, prev_land_use_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, water_cont_soil_meth, soil_type, cur_vegetation, local_class, store_cond, heavy_metals, misc_param, agrochem_addition, previous_land_use, extreme_event, fire, flooding, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "soil_MISAG" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT NOT NULL, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	fao_class VARCHAR(13), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	soil_horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT, 
	pool_dna_extracts TEXT, 
	crop_rotation TEXT, 
	ph FLOAT, 
	water_content FLOAT, 
	al_sat TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	microbial_biomass TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	soil_text_measure TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	al_sat_meth TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_addit_analys TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	micro_biomass_meth TEXT, 
	ph_meth TEXT, 
	prev_land_use_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	water_cont_soil_meth TEXT, 
	soil_type TEXT, 
	cur_vegetation TEXT, 
	local_class TEXT, 
	store_cond TEXT, 
	heavy_metals TEXT, 
	misc_param TEXT, 
	agrochem_addition TEXT, 
	previous_land_use TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, fao_class, cur_land_use, tillage, soil_horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, ph, water_content, al_sat, annual_precpt, annual_temp, microbial_biomass, org_matter, org_nitro, season_precpt, season_temp, slope_aspect, slope_gradient, soil_text_measure, tot_nitro_content, tot_org_carb, al_sat_meth, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_addit_analys, link_class_info, link_climate_info, local_class_meth, micro_biomass_meth, ph_meth, prev_land_use_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, water_cont_soil_meth, soil_type, cur_vegetation, local_class, store_cond, heavy_metals, misc_param, agrochem_addition, previous_land_use, extreme_event, fire, flooding, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "soil_MIUVIG" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT NOT NULL, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	fao_class VARCHAR(13), 
	cur_land_use VARCHAR(93), 
	tillage VARCHAR(13), 
	soil_horizon VARCHAR(10), 
	profile_position VARCHAR(9), 
	drainage_class VARCHAR(19), 
	sieving TEXT, 
	pool_dna_extracts TEXT, 
	crop_rotation TEXT, 
	ph FLOAT, 
	water_content FLOAT, 
	al_sat TEXT, 
	annual_precpt TEXT, 
	annual_temp TEXT, 
	microbial_biomass TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	season_precpt TEXT, 
	season_temp TEXT, 
	slope_aspect TEXT, 
	slope_gradient TEXT, 
	soil_text_measure TEXT, 
	tot_nitro_content TEXT, 
	tot_org_carb TEXT, 
	al_sat_meth TEXT, 
	cur_vegetation_meth TEXT, 
	heavy_metals_meth TEXT, 
	horizon_meth TEXT, 
	link_addit_analys TEXT, 
	link_class_info TEXT, 
	link_climate_info TEXT, 
	local_class_meth TEXT, 
	micro_biomass_meth TEXT, 
	ph_meth TEXT, 
	prev_land_use_meth TEXT, 
	salinity_meth TEXT, 
	soil_texture_meth TEXT, 
	soil_type_meth TEXT, 
	tot_nitro_cont_meth TEXT, 
	tot_org_c_meth TEXT, 
	water_cont_soil_meth TEXT, 
	soil_type TEXT, 
	cur_vegetation TEXT, 
	local_class TEXT, 
	store_cond TEXT, 
	heavy_metals TEXT, 
	misc_param TEXT, 
	agrochem_addition TEXT, 
	previous_land_use TEXT, 
	extreme_event DATE, 
	fire DATE, 
	flooding DATE, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, fao_class, cur_land_use, tillage, soil_horizon, profile_position, drainage_class, sieving, pool_dna_extracts, crop_rotation, ph, water_content, al_sat, annual_precpt, annual_temp, microbial_biomass, org_matter, org_nitro, season_precpt, season_temp, slope_aspect, slope_gradient, soil_text_measure, tot_nitro_content, tot_org_carb, al_sat_meth, cur_vegetation_meth, heavy_metals_meth, horizon_meth, link_addit_analys, link_class_info, link_climate_info, local_class_meth, micro_biomass_meth, ph_meth, prev_land_use_meth, salinity_meth, soil_texture_meth, soil_type_meth, tot_nitro_cont_meth, tot_org_c_meth, water_cont_soil_meth, soil_type, cur_vegetation, local_class, store_cond, heavy_metals, misc_param, agrochem_addition, previous_land_use, extreme_event, fire, flooding, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE symbiont_associated (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	symbiont_host_role VARCHAR(12), 
	oxy_stat_samp TEXT, 
	type_of_symbiosis VARCHAR(14), 
	sym_life_cycle_type TEXT NOT NULL, 
	route_transmission VARCHAR(31), 
	host_dependence TEXT NOT NULL, 
	host_specificity VARCHAR(16), 
	host_sex VARCHAR(28), 
	mode_transmission VARCHAR(34), 
	host_cellular_loc VARCHAR(14), 
	gravidity TEXT, 
	host_of_host_gravid TEXT, 
	samp_store_dur TEXT, 
	association_duration TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_of_host_totmass TEXT, 
	host_tot_mass TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_number TEXT, 
	host_of_host_taxid INTEGER, 
	host_taxid INTEGER, 
	host_growth_cond TEXT, 
	host_of_host_pheno TEXT, 
	host_phenotype TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_of_host_env_loc TEXT, 
	host_of_host_env_med TEXT, 
	chem_administration TEXT, 
	host_of_host_disease TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_infra_specific_name TEXT, 
	host_infra_specific_rank TEXT, 
	host_life_stage TEXT NOT NULL, 
	host_of_host_coinf TEXT, 
	host_of_host_geno TEXT, 
	host_of_host_infname TEXT, 
	host_of_host_infrank TEXT, 
	host_of_host_name TEXT, 
	host_of_host_sub_id TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	samp_store_sol TEXT, 
	sample_name TEXT NOT NULL, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relationship TEXT, 
	host_of_host_fam_rel TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	source_mat_id TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, symbiont_host_role, oxy_stat_samp, type_of_symbiosis, sym_life_cycle_type, route_transmission, host_dependence, host_specificity, host_sex, mode_transmission, host_cellular_loc, gravidity, host_of_host_gravid, samp_store_dur, association_duration, host_age, host_dry_mass, host_height, host_length, host_of_host_totmass, host_tot_mass, samp_salinity, samp_store_temp, host_number, host_of_host_taxid, host_taxid, host_growth_cond, host_of_host_pheno, host_phenotype, host_body_product, host_body_site, host_of_host_env_loc, host_of_host_env_med, chem_administration, host_of_host_disease, host_body_habitat, host_color, host_common_name, host_genotype, host_infra_specific_name, host_infra_specific_rank, host_life_stage, host_of_host_coinf, host_of_host_geno, host_of_host_infname, host_of_host_infrank, host_of_host_name, host_of_host_sub_id, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, samp_store_sol, sample_name, misc_param, organism_count, perturbation, host_family_relationship, host_of_host_fam_rel, samp_vol_we_dna_ext, project_name, source_mat_id)
);

CREATE TABLE "symbiont_associated_MIGS_bacteria" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	symbiont_host_role VARCHAR(12), 
	oxy_stat_samp TEXT, 
	type_of_symbiosis VARCHAR(14), 
	sym_life_cycle_type TEXT NOT NULL, 
	route_transmission VARCHAR(31), 
	host_dependence TEXT NOT NULL, 
	host_specificity VARCHAR(16), 
	host_sex VARCHAR(28), 
	mode_transmission VARCHAR(34), 
	host_cellular_loc VARCHAR(14), 
	gravidity TEXT, 
	host_of_host_gravid TEXT, 
	samp_store_dur TEXT, 
	association_duration TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_of_host_totmass TEXT, 
	host_tot_mass TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_number TEXT, 
	host_of_host_taxid INTEGER, 
	host_taxid INTEGER, 
	host_growth_cond TEXT, 
	host_of_host_pheno TEXT, 
	host_phenotype TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_of_host_env_loc TEXT, 
	host_of_host_env_med TEXT, 
	chem_administration TEXT, 
	host_of_host_disease TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_infra_specific_name TEXT, 
	host_infra_specific_rank TEXT, 
	host_life_stage TEXT NOT NULL, 
	host_of_host_coinf TEXT, 
	host_of_host_geno TEXT, 
	host_of_host_infname TEXT, 
	host_of_host_infrank TEXT, 
	host_of_host_name TEXT, 
	host_of_host_sub_id TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	samp_store_sol TEXT, 
	sample_name TEXT NOT NULL, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relationship TEXT, 
	host_of_host_fam_rel TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	source_mat_id TEXT, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, symbiont_host_role, oxy_stat_samp, type_of_symbiosis, sym_life_cycle_type, route_transmission, host_dependence, host_specificity, host_sex, mode_transmission, host_cellular_loc, gravidity, host_of_host_gravid, samp_store_dur, association_duration, host_age, host_dry_mass, host_height, host_length, host_of_host_totmass, host_tot_mass, samp_salinity, samp_store_temp, host_number, host_of_host_taxid, host_taxid, host_growth_cond, host_of_host_pheno, host_phenotype, host_body_product, host_body_site, host_of_host_env_loc, host_of_host_env_med, chem_administration, host_of_host_disease, host_body_habitat, host_color, host_common_name, host_genotype, host_infra_specific_name, host_infra_specific_rank, host_life_stage, host_of_host_coinf, host_of_host_geno, host_of_host_infname, host_of_host_infrank, host_of_host_name, host_of_host_sub_id, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, samp_store_sol, sample_name, misc_param, organism_count, perturbation, host_family_relationship, host_of_host_fam_rel, samp_vol_we_dna_ext, project_name, source_mat_id, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "symbiont_associated_MIGS_eukaryote" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	symbiont_host_role VARCHAR(12), 
	oxy_stat_samp TEXT, 
	type_of_symbiosis VARCHAR(14), 
	sym_life_cycle_type TEXT NOT NULL, 
	route_transmission VARCHAR(31), 
	host_dependence TEXT NOT NULL, 
	host_specificity VARCHAR(16), 
	host_sex VARCHAR(28), 
	mode_transmission VARCHAR(34), 
	host_cellular_loc VARCHAR(14), 
	gravidity TEXT, 
	host_of_host_gravid TEXT, 
	samp_store_dur TEXT, 
	association_duration TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_of_host_totmass TEXT, 
	host_tot_mass TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_number TEXT, 
	host_of_host_taxid INTEGER, 
	host_taxid INTEGER, 
	host_growth_cond TEXT, 
	host_of_host_pheno TEXT, 
	host_phenotype TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_of_host_env_loc TEXT, 
	host_of_host_env_med TEXT, 
	chem_administration TEXT, 
	host_of_host_disease TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_infra_specific_name TEXT, 
	host_infra_specific_rank TEXT, 
	host_life_stage TEXT NOT NULL, 
	host_of_host_coinf TEXT, 
	host_of_host_geno TEXT, 
	host_of_host_infname TEXT, 
	host_of_host_infrank TEXT, 
	host_of_host_name TEXT, 
	host_of_host_sub_id TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	samp_store_sol TEXT, 
	sample_name TEXT NOT NULL, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relationship TEXT, 
	host_of_host_fam_rel TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	source_mat_id TEXT, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, symbiont_host_role, oxy_stat_samp, type_of_symbiosis, sym_life_cycle_type, route_transmission, host_dependence, host_specificity, host_sex, mode_transmission, host_cellular_loc, gravidity, host_of_host_gravid, samp_store_dur, association_duration, host_age, host_dry_mass, host_height, host_length, host_of_host_totmass, host_tot_mass, samp_salinity, samp_store_temp, host_number, host_of_host_taxid, host_taxid, host_growth_cond, host_of_host_pheno, host_phenotype, host_body_product, host_body_site, host_of_host_env_loc, host_of_host_env_med, chem_administration, host_of_host_disease, host_body_habitat, host_color, host_common_name, host_genotype, host_infra_specific_name, host_infra_specific_rank, host_life_stage, host_of_host_coinf, host_of_host_geno, host_of_host_infname, host_of_host_infrank, host_of_host_name, host_of_host_sub_id, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, samp_store_sol, sample_name, misc_param, organism_count, perturbation, host_family_relationship, host_of_host_fam_rel, samp_vol_we_dna_ext, project_name, source_mat_id, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "symbiont_associated_MIGS_org" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	symbiont_host_role VARCHAR(12), 
	oxy_stat_samp TEXT, 
	type_of_symbiosis VARCHAR(14), 
	sym_life_cycle_type TEXT NOT NULL, 
	route_transmission VARCHAR(31), 
	host_dependence TEXT NOT NULL, 
	host_specificity VARCHAR(16), 
	host_sex VARCHAR(28), 
	mode_transmission VARCHAR(34), 
	host_cellular_loc VARCHAR(14), 
	gravidity TEXT, 
	host_of_host_gravid TEXT, 
	samp_store_dur TEXT, 
	association_duration TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_of_host_totmass TEXT, 
	host_tot_mass TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_number TEXT, 
	host_of_host_taxid INTEGER, 
	host_taxid INTEGER, 
	host_growth_cond TEXT, 
	host_of_host_pheno TEXT, 
	host_phenotype TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_of_host_env_loc TEXT, 
	host_of_host_env_med TEXT, 
	chem_administration TEXT, 
	host_of_host_disease TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_infra_specific_name TEXT, 
	host_infra_specific_rank TEXT, 
	host_life_stage TEXT NOT NULL, 
	host_of_host_coinf TEXT, 
	host_of_host_geno TEXT, 
	host_of_host_infname TEXT, 
	host_of_host_infrank TEXT, 
	host_of_host_name TEXT, 
	host_of_host_sub_id TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	samp_store_sol TEXT, 
	sample_name TEXT NOT NULL, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relationship TEXT, 
	host_of_host_fam_rel TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	source_mat_id TEXT, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, symbiont_host_role, oxy_stat_samp, type_of_symbiosis, sym_life_cycle_type, route_transmission, host_dependence, host_specificity, host_sex, mode_transmission, host_cellular_loc, gravidity, host_of_host_gravid, samp_store_dur, association_duration, host_age, host_dry_mass, host_height, host_length, host_of_host_totmass, host_tot_mass, samp_salinity, samp_store_temp, host_number, host_of_host_taxid, host_taxid, host_growth_cond, host_of_host_pheno, host_phenotype, host_body_product, host_body_site, host_of_host_env_loc, host_of_host_env_med, chem_administration, host_of_host_disease, host_body_habitat, host_color, host_common_name, host_genotype, host_infra_specific_name, host_infra_specific_rank, host_life_stage, host_of_host_coinf, host_of_host_geno, host_of_host_infname, host_of_host_infrank, host_of_host_name, host_of_host_sub_id, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, samp_store_sol, sample_name, misc_param, organism_count, perturbation, host_family_relationship, host_of_host_fam_rel, samp_vol_we_dna_ext, project_name, source_mat_id, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "symbiont_associated_MIGS_plant" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	symbiont_host_role VARCHAR(12), 
	oxy_stat_samp TEXT, 
	type_of_symbiosis VARCHAR(14), 
	sym_life_cycle_type TEXT NOT NULL, 
	route_transmission VARCHAR(31), 
	host_dependence TEXT NOT NULL, 
	host_specificity VARCHAR(16), 
	host_sex VARCHAR(28), 
	mode_transmission VARCHAR(34), 
	host_cellular_loc VARCHAR(14), 
	gravidity TEXT, 
	host_of_host_gravid TEXT, 
	samp_store_dur TEXT, 
	association_duration TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_of_host_totmass TEXT, 
	host_tot_mass TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_number TEXT, 
	host_of_host_taxid INTEGER, 
	host_taxid INTEGER, 
	host_growth_cond TEXT, 
	host_of_host_pheno TEXT, 
	host_phenotype TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_of_host_env_loc TEXT, 
	host_of_host_env_med TEXT, 
	chem_administration TEXT, 
	host_of_host_disease TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_infra_specific_name TEXT, 
	host_infra_specific_rank TEXT, 
	host_life_stage TEXT NOT NULL, 
	host_of_host_coinf TEXT, 
	host_of_host_geno TEXT, 
	host_of_host_infname TEXT, 
	host_of_host_infrank TEXT, 
	host_of_host_name TEXT, 
	host_of_host_sub_id TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	samp_store_sol TEXT, 
	sample_name TEXT NOT NULL, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relationship TEXT, 
	host_of_host_fam_rel TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	source_mat_id TEXT, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, symbiont_host_role, oxy_stat_samp, type_of_symbiosis, sym_life_cycle_type, route_transmission, host_dependence, host_specificity, host_sex, mode_transmission, host_cellular_loc, gravidity, host_of_host_gravid, samp_store_dur, association_duration, host_age, host_dry_mass, host_height, host_length, host_of_host_totmass, host_tot_mass, samp_salinity, samp_store_temp, host_number, host_of_host_taxid, host_taxid, host_growth_cond, host_of_host_pheno, host_phenotype, host_body_product, host_body_site, host_of_host_env_loc, host_of_host_env_med, chem_administration, host_of_host_disease, host_body_habitat, host_color, host_common_name, host_genotype, host_infra_specific_name, host_infra_specific_rank, host_life_stage, host_of_host_coinf, host_of_host_geno, host_of_host_infname, host_of_host_infrank, host_of_host_name, host_of_host_sub_id, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, samp_store_sol, sample_name, misc_param, organism_count, perturbation, host_family_relationship, host_of_host_fam_rel, samp_vol_we_dna_ext, project_name, source_mat_id, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "symbiont_associated_MIGS_virus" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	symbiont_host_role VARCHAR(12), 
	oxy_stat_samp TEXT, 
	type_of_symbiosis VARCHAR(14), 
	sym_life_cycle_type TEXT NOT NULL, 
	route_transmission VARCHAR(31), 
	host_dependence TEXT NOT NULL, 
	host_specificity VARCHAR(16), 
	host_sex VARCHAR(28), 
	mode_transmission VARCHAR(34), 
	host_cellular_loc VARCHAR(14), 
	gravidity TEXT, 
	host_of_host_gravid TEXT, 
	samp_store_dur TEXT, 
	association_duration TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_of_host_totmass TEXT, 
	host_tot_mass TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_number TEXT, 
	host_of_host_taxid INTEGER, 
	host_taxid INTEGER, 
	host_growth_cond TEXT, 
	host_of_host_pheno TEXT, 
	host_phenotype TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_of_host_env_loc TEXT, 
	host_of_host_env_med TEXT, 
	chem_administration TEXT, 
	host_of_host_disease TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_infra_specific_name TEXT, 
	host_infra_specific_rank TEXT, 
	host_life_stage TEXT NOT NULL, 
	host_of_host_coinf TEXT, 
	host_of_host_geno TEXT, 
	host_of_host_infname TEXT, 
	host_of_host_infrank TEXT, 
	host_of_host_name TEXT, 
	host_of_host_sub_id TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	samp_store_sol TEXT, 
	sample_name TEXT NOT NULL, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relationship TEXT, 
	host_of_host_fam_rel TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	source_mat_id TEXT, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, symbiont_host_role, oxy_stat_samp, type_of_symbiosis, sym_life_cycle_type, route_transmission, host_dependence, host_specificity, host_sex, mode_transmission, host_cellular_loc, gravidity, host_of_host_gravid, samp_store_dur, association_duration, host_age, host_dry_mass, host_height, host_length, host_of_host_totmass, host_tot_mass, samp_salinity, samp_store_temp, host_number, host_of_host_taxid, host_taxid, host_growth_cond, host_of_host_pheno, host_phenotype, host_body_product, host_body_site, host_of_host_env_loc, host_of_host_env_med, chem_administration, host_of_host_disease, host_body_habitat, host_color, host_common_name, host_genotype, host_infra_specific_name, host_infra_specific_rank, host_life_stage, host_of_host_coinf, host_of_host_geno, host_of_host_infname, host_of_host_infrank, host_of_host_name, host_of_host_sub_id, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, samp_store_sol, sample_name, misc_param, organism_count, perturbation, host_family_relationship, host_of_host_fam_rel, samp_vol_we_dna_ext, project_name, source_mat_id, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "symbiont_associated_MIMAG" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	symbiont_host_role VARCHAR(12), 
	oxy_stat_samp TEXT, 
	type_of_symbiosis VARCHAR(14), 
	sym_life_cycle_type TEXT NOT NULL, 
	route_transmission VARCHAR(31), 
	host_dependence TEXT NOT NULL, 
	host_specificity VARCHAR(16), 
	host_sex VARCHAR(28), 
	mode_transmission VARCHAR(34), 
	host_cellular_loc VARCHAR(14), 
	gravidity TEXT, 
	host_of_host_gravid TEXT, 
	samp_store_dur TEXT, 
	association_duration TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_of_host_totmass TEXT, 
	host_tot_mass TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_number TEXT, 
	host_of_host_taxid INTEGER, 
	host_taxid INTEGER, 
	host_growth_cond TEXT, 
	host_of_host_pheno TEXT, 
	host_phenotype TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_of_host_env_loc TEXT, 
	host_of_host_env_med TEXT, 
	chem_administration TEXT, 
	host_of_host_disease TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_infra_specific_name TEXT, 
	host_infra_specific_rank TEXT, 
	host_life_stage TEXT NOT NULL, 
	host_of_host_coinf TEXT, 
	host_of_host_geno TEXT, 
	host_of_host_infname TEXT, 
	host_of_host_infrank TEXT, 
	host_of_host_name TEXT, 
	host_of_host_sub_id TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	samp_store_sol TEXT, 
	sample_name TEXT NOT NULL, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relationship TEXT, 
	host_of_host_fam_rel TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	source_mat_id TEXT, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, symbiont_host_role, oxy_stat_samp, type_of_symbiosis, sym_life_cycle_type, route_transmission, host_dependence, host_specificity, host_sex, mode_transmission, host_cellular_loc, gravidity, host_of_host_gravid, samp_store_dur, association_duration, host_age, host_dry_mass, host_height, host_length, host_of_host_totmass, host_tot_mass, samp_salinity, samp_store_temp, host_number, host_of_host_taxid, host_taxid, host_growth_cond, host_of_host_pheno, host_phenotype, host_body_product, host_body_site, host_of_host_env_loc, host_of_host_env_med, chem_administration, host_of_host_disease, host_body_habitat, host_color, host_common_name, host_genotype, host_infra_specific_name, host_infra_specific_rank, host_life_stage, host_of_host_coinf, host_of_host_geno, host_of_host_infname, host_of_host_infrank, host_of_host_name, host_of_host_sub_id, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, samp_store_sol, sample_name, misc_param, organism_count, perturbation, host_family_relationship, host_of_host_fam_rel, samp_vol_we_dna_ext, project_name, source_mat_id, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "symbiont_associated_MIMARKS_specimen" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	symbiont_host_role VARCHAR(12), 
	oxy_stat_samp TEXT, 
	type_of_symbiosis VARCHAR(14), 
	sym_life_cycle_type TEXT NOT NULL, 
	route_transmission VARCHAR(31), 
	host_dependence TEXT NOT NULL, 
	host_specificity VARCHAR(16), 
	host_sex VARCHAR(28), 
	mode_transmission VARCHAR(34), 
	host_cellular_loc VARCHAR(14), 
	gravidity TEXT, 
	host_of_host_gravid TEXT, 
	samp_store_dur TEXT, 
	association_duration TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_of_host_totmass TEXT, 
	host_tot_mass TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_number TEXT, 
	host_of_host_taxid INTEGER, 
	host_taxid INTEGER, 
	host_growth_cond TEXT, 
	host_of_host_pheno TEXT, 
	host_phenotype TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_of_host_env_loc TEXT, 
	host_of_host_env_med TEXT, 
	chem_administration TEXT, 
	host_of_host_disease TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_infra_specific_name TEXT, 
	host_infra_specific_rank TEXT, 
	host_life_stage TEXT NOT NULL, 
	host_of_host_coinf TEXT, 
	host_of_host_geno TEXT, 
	host_of_host_infname TEXT, 
	host_of_host_infrank TEXT, 
	host_of_host_name TEXT, 
	host_of_host_sub_id TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	samp_store_sol TEXT, 
	sample_name TEXT NOT NULL, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relationship TEXT, 
	host_of_host_fam_rel TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	source_mat_id TEXT, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, symbiont_host_role, oxy_stat_samp, type_of_symbiosis, sym_life_cycle_type, route_transmission, host_dependence, host_specificity, host_sex, mode_transmission, host_cellular_loc, gravidity, host_of_host_gravid, samp_store_dur, association_duration, host_age, host_dry_mass, host_height, host_length, host_of_host_totmass, host_tot_mass, samp_salinity, samp_store_temp, host_number, host_of_host_taxid, host_taxid, host_growth_cond, host_of_host_pheno, host_phenotype, host_body_product, host_body_site, host_of_host_env_loc, host_of_host_env_med, chem_administration, host_of_host_disease, host_body_habitat, host_color, host_common_name, host_genotype, host_infra_specific_name, host_infra_specific_rank, host_life_stage, host_of_host_coinf, host_of_host_geno, host_of_host_infname, host_of_host_infrank, host_of_host_name, host_of_host_sub_id, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, samp_store_sol, sample_name, misc_param, organism_count, perturbation, host_family_relationship, host_of_host_fam_rel, samp_vol_we_dna_ext, project_name, source_mat_id, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "symbiont_associated_MIMARKS_survey" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	symbiont_host_role VARCHAR(12), 
	oxy_stat_samp TEXT, 
	type_of_symbiosis VARCHAR(14), 
	sym_life_cycle_type TEXT NOT NULL, 
	route_transmission VARCHAR(31), 
	host_dependence TEXT NOT NULL, 
	host_specificity VARCHAR(16), 
	host_sex VARCHAR(28), 
	mode_transmission VARCHAR(34), 
	host_cellular_loc VARCHAR(14), 
	gravidity TEXT, 
	host_of_host_gravid TEXT, 
	samp_store_dur TEXT, 
	association_duration TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_of_host_totmass TEXT, 
	host_tot_mass TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_number TEXT, 
	host_of_host_taxid INTEGER, 
	host_taxid INTEGER, 
	host_growth_cond TEXT, 
	host_of_host_pheno TEXT, 
	host_phenotype TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_of_host_env_loc TEXT, 
	host_of_host_env_med TEXT, 
	chem_administration TEXT, 
	host_of_host_disease TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_infra_specific_name TEXT, 
	host_infra_specific_rank TEXT, 
	host_life_stage TEXT NOT NULL, 
	host_of_host_coinf TEXT, 
	host_of_host_geno TEXT, 
	host_of_host_infname TEXT, 
	host_of_host_infrank TEXT, 
	host_of_host_name TEXT, 
	host_of_host_sub_id TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	samp_store_sol TEXT, 
	sample_name TEXT NOT NULL, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relationship TEXT, 
	host_of_host_fam_rel TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	source_mat_id TEXT, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, symbiont_host_role, oxy_stat_samp, type_of_symbiosis, sym_life_cycle_type, route_transmission, host_dependence, host_specificity, host_sex, mode_transmission, host_cellular_loc, gravidity, host_of_host_gravid, samp_store_dur, association_duration, host_age, host_dry_mass, host_height, host_length, host_of_host_totmass, host_tot_mass, samp_salinity, samp_store_temp, host_number, host_of_host_taxid, host_taxid, host_growth_cond, host_of_host_pheno, host_phenotype, host_body_product, host_body_site, host_of_host_env_loc, host_of_host_env_med, chem_administration, host_of_host_disease, host_body_habitat, host_color, host_common_name, host_genotype, host_infra_specific_name, host_infra_specific_rank, host_life_stage, host_of_host_coinf, host_of_host_geno, host_of_host_infname, host_of_host_infrank, host_of_host_name, host_of_host_sub_id, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, samp_store_sol, sample_name, misc_param, organism_count, perturbation, host_family_relationship, host_of_host_fam_rel, samp_vol_we_dna_ext, project_name, source_mat_id, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "symbiont_associated_MIMS" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	symbiont_host_role VARCHAR(12), 
	oxy_stat_samp TEXT, 
	type_of_symbiosis VARCHAR(14), 
	sym_life_cycle_type TEXT NOT NULL, 
	route_transmission VARCHAR(31), 
	host_dependence TEXT NOT NULL, 
	host_specificity VARCHAR(16), 
	host_sex VARCHAR(28), 
	mode_transmission VARCHAR(34), 
	host_cellular_loc VARCHAR(14), 
	gravidity TEXT, 
	host_of_host_gravid TEXT, 
	samp_store_dur TEXT, 
	association_duration TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_of_host_totmass TEXT, 
	host_tot_mass TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_number TEXT, 
	host_of_host_taxid INTEGER, 
	host_taxid INTEGER, 
	host_growth_cond TEXT, 
	host_of_host_pheno TEXT, 
	host_phenotype TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_of_host_env_loc TEXT, 
	host_of_host_env_med TEXT, 
	chem_administration TEXT, 
	host_of_host_disease TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_infra_specific_name TEXT, 
	host_infra_specific_rank TEXT, 
	host_life_stage TEXT NOT NULL, 
	host_of_host_coinf TEXT, 
	host_of_host_geno TEXT, 
	host_of_host_infname TEXT, 
	host_of_host_infrank TEXT, 
	host_of_host_name TEXT, 
	host_of_host_sub_id TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	samp_store_sol TEXT, 
	sample_name TEXT NOT NULL, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relationship TEXT, 
	host_of_host_fam_rel TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	source_mat_id TEXT, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, symbiont_host_role, oxy_stat_samp, type_of_symbiosis, sym_life_cycle_type, route_transmission, host_dependence, host_specificity, host_sex, mode_transmission, host_cellular_loc, gravidity, host_of_host_gravid, samp_store_dur, association_duration, host_age, host_dry_mass, host_height, host_length, host_of_host_totmass, host_tot_mass, samp_salinity, samp_store_temp, host_number, host_of_host_taxid, host_taxid, host_growth_cond, host_of_host_pheno, host_phenotype, host_body_product, host_body_site, host_of_host_env_loc, host_of_host_env_med, chem_administration, host_of_host_disease, host_body_habitat, host_color, host_common_name, host_genotype, host_infra_specific_name, host_infra_specific_rank, host_life_stage, host_of_host_coinf, host_of_host_geno, host_of_host_infname, host_of_host_infrank, host_of_host_name, host_of_host_sub_id, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, samp_store_sol, sample_name, misc_param, organism_count, perturbation, host_family_relationship, host_of_host_fam_rel, samp_vol_we_dna_ext, project_name, source_mat_id, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "symbiont_associated_MISAG" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	symbiont_host_role VARCHAR(12), 
	oxy_stat_samp TEXT, 
	type_of_symbiosis VARCHAR(14), 
	sym_life_cycle_type TEXT NOT NULL, 
	route_transmission VARCHAR(31), 
	host_dependence TEXT NOT NULL, 
	host_specificity VARCHAR(16), 
	host_sex VARCHAR(28), 
	mode_transmission VARCHAR(34), 
	host_cellular_loc VARCHAR(14), 
	gravidity TEXT, 
	host_of_host_gravid TEXT, 
	samp_store_dur TEXT, 
	association_duration TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_of_host_totmass TEXT, 
	host_tot_mass TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_number TEXT, 
	host_of_host_taxid INTEGER, 
	host_taxid INTEGER, 
	host_growth_cond TEXT, 
	host_of_host_pheno TEXT, 
	host_phenotype TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_of_host_env_loc TEXT, 
	host_of_host_env_med TEXT, 
	chem_administration TEXT, 
	host_of_host_disease TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_infra_specific_name TEXT, 
	host_infra_specific_rank TEXT, 
	host_life_stage TEXT NOT NULL, 
	host_of_host_coinf TEXT, 
	host_of_host_geno TEXT, 
	host_of_host_infname TEXT, 
	host_of_host_infrank TEXT, 
	host_of_host_name TEXT, 
	host_of_host_sub_id TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	samp_store_sol TEXT, 
	sample_name TEXT NOT NULL, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relationship TEXT, 
	host_of_host_fam_rel TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	source_mat_id TEXT, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, symbiont_host_role, oxy_stat_samp, type_of_symbiosis, sym_life_cycle_type, route_transmission, host_dependence, host_specificity, host_sex, mode_transmission, host_cellular_loc, gravidity, host_of_host_gravid, samp_store_dur, association_duration, host_age, host_dry_mass, host_height, host_length, host_of_host_totmass, host_tot_mass, samp_salinity, samp_store_temp, host_number, host_of_host_taxid, host_taxid, host_growth_cond, host_of_host_pheno, host_phenotype, host_body_product, host_body_site, host_of_host_env_loc, host_of_host_env_med, chem_administration, host_of_host_disease, host_body_habitat, host_color, host_common_name, host_genotype, host_infra_specific_name, host_infra_specific_rank, host_life_stage, host_of_host_coinf, host_of_host_geno, host_of_host_infname, host_of_host_infrank, host_of_host_name, host_of_host_sub_id, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, samp_store_sol, sample_name, misc_param, organism_count, perturbation, host_family_relationship, host_of_host_fam_rel, samp_vol_we_dna_ext, project_name, source_mat_id, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "symbiont_associated_MIUVIG" (
	lat_lon TEXT NOT NULL, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT NOT NULL, 
	collection_date DATE NOT NULL, 
	env_broad_scale TEXT NOT NULL, 
	env_local_scale TEXT NOT NULL, 
	env_medium TEXT NOT NULL, 
	symbiont_host_role VARCHAR(12), 
	oxy_stat_samp TEXT, 
	type_of_symbiosis VARCHAR(14), 
	sym_life_cycle_type TEXT NOT NULL, 
	route_transmission VARCHAR(31), 
	host_dependence TEXT NOT NULL, 
	host_specificity VARCHAR(16), 
	host_sex VARCHAR(28), 
	mode_transmission VARCHAR(34), 
	host_cellular_loc VARCHAR(14), 
	gravidity TEXT, 
	host_of_host_gravid TEXT, 
	samp_store_dur TEXT, 
	association_duration TEXT, 
	host_age TEXT, 
	host_dry_mass TEXT, 
	host_height TEXT, 
	host_length TEXT, 
	host_of_host_totmass TEXT, 
	host_tot_mass TEXT, 
	samp_salinity TEXT, 
	samp_store_temp TEXT, 
	host_number TEXT, 
	host_of_host_taxid INTEGER, 
	host_taxid INTEGER, 
	host_growth_cond TEXT, 
	host_of_host_pheno TEXT, 
	host_phenotype TEXT, 
	host_body_product TEXT, 
	host_body_site TEXT, 
	host_of_host_env_loc TEXT, 
	host_of_host_env_med TEXT, 
	chem_administration TEXT, 
	host_of_host_disease TEXT, 
	host_body_habitat TEXT, 
	host_color TEXT, 
	host_common_name TEXT, 
	host_genotype TEXT, 
	host_infra_specific_name TEXT, 
	host_infra_specific_rank TEXT, 
	host_life_stage TEXT NOT NULL, 
	host_of_host_coinf TEXT, 
	host_of_host_geno TEXT, 
	host_of_host_infname TEXT, 
	host_of_host_infrank TEXT, 
	host_of_host_name TEXT, 
	host_of_host_sub_id TEXT, 
	host_shape TEXT, 
	host_subject_id TEXT, 
	host_substrate TEXT, 
	host_symbiont TEXT, 
	samp_store_loc TEXT, 
	samp_store_sol TEXT, 
	sample_name TEXT NOT NULL, 
	misc_param TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	host_family_relationship TEXT, 
	host_of_host_fam_rel TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	source_mat_id TEXT, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, symbiont_host_role, oxy_stat_samp, type_of_symbiosis, sym_life_cycle_type, route_transmission, host_dependence, host_specificity, host_sex, mode_transmission, host_cellular_loc, gravidity, host_of_host_gravid, samp_store_dur, association_duration, host_age, host_dry_mass, host_height, host_length, host_of_host_totmass, host_tot_mass, samp_salinity, samp_store_temp, host_number, host_of_host_taxid, host_taxid, host_growth_cond, host_of_host_pheno, host_phenotype, host_body_product, host_body_site, host_of_host_env_loc, host_of_host_env_med, chem_administration, host_of_host_disease, host_body_habitat, host_color, host_common_name, host_genotype, host_infra_specific_name, host_infra_specific_rank, host_life_stage, host_of_host_coinf, host_of_host_geno, host_of_host_infname, host_of_host_infrank, host_of_host_name, host_of_host_sub_id, host_shape, host_subject_id, host_substrate, host_symbiont, samp_store_loc, samp_store_sol, sample_name, misc_param, organism_count, perturbation, host_family_relationship, host_of_host_fam_rel, samp_vol_we_dna_ext, project_name, source_mat_id, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE wastewater_sludge (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	biochem_oxygen_dem TEXT, 
	chem_oxygen_dem TEXT, 
	efficiency_percent TEXT, 
	indust_eff_percent TEXT, 
	nitrate TEXT, 
	phosphate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sludge_retent_time TEXT, 
	sodium TEXT, 
	tot_nitro TEXT, 
	tot_phosphate TEXT, 
	chem_administration TEXT, 
	pre_treatment TEXT, 
	primary_treatment TEXT, 
	reactor_type TEXT, 
	samp_store_loc TEXT, 
	secondary_treatment TEXT, 
	sewage_type TEXT, 
	tertiary_treatment TEXT, 
	wastewater_type TEXT, 
	emulsions TEXT, 
	gaseous_substances TEXT, 
	inorg_particles TEXT, 
	misc_param TEXT, 
	org_particles TEXT, 
	soluble_inorg_mat TEXT, 
	soluble_org_mat TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, biochem_oxygen_dem, chem_oxygen_dem, efficiency_percent, indust_eff_percent, nitrate, phosphate, salinity, samp_store_temp, sludge_retent_time, sodium, tot_nitro, tot_phosphate, chem_administration, pre_treatment, primary_treatment, reactor_type, samp_store_loc, secondary_treatment, sewage_type, tertiary_treatment, wastewater_type, emulsions, gaseous_substances, inorg_particles, misc_param, org_particles, soluble_inorg_mat, soluble_org_mat, suspend_solids, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name)
);

CREATE TABLE "wastewater_sludge_MIGS_bacteria" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	biochem_oxygen_dem TEXT, 
	chem_oxygen_dem TEXT, 
	efficiency_percent TEXT, 
	indust_eff_percent TEXT, 
	nitrate TEXT, 
	phosphate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sludge_retent_time TEXT, 
	sodium TEXT, 
	tot_nitro TEXT, 
	tot_phosphate TEXT, 
	chem_administration TEXT, 
	pre_treatment TEXT, 
	primary_treatment TEXT, 
	reactor_type TEXT, 
	samp_store_loc TEXT, 
	secondary_treatment TEXT, 
	sewage_type TEXT, 
	tertiary_treatment TEXT, 
	wastewater_type TEXT, 
	emulsions TEXT, 
	gaseous_substances TEXT, 
	inorg_particles TEXT, 
	misc_param TEXT, 
	org_particles TEXT, 
	soluble_inorg_mat TEXT, 
	soluble_org_mat TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, biochem_oxygen_dem, chem_oxygen_dem, efficiency_percent, indust_eff_percent, nitrate, phosphate, salinity, samp_store_temp, sludge_retent_time, sodium, tot_nitro, tot_phosphate, chem_administration, pre_treatment, primary_treatment, reactor_type, samp_store_loc, secondary_treatment, sewage_type, tertiary_treatment, wastewater_type, emulsions, gaseous_substances, inorg_particles, misc_param, org_particles, soluble_inorg_mat, soluble_org_mat, suspend_solids, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "wastewater_sludge_MIGS_eukaryote" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	biochem_oxygen_dem TEXT, 
	chem_oxygen_dem TEXT, 
	efficiency_percent TEXT, 
	indust_eff_percent TEXT, 
	nitrate TEXT, 
	phosphate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sludge_retent_time TEXT, 
	sodium TEXT, 
	tot_nitro TEXT, 
	tot_phosphate TEXT, 
	chem_administration TEXT, 
	pre_treatment TEXT, 
	primary_treatment TEXT, 
	reactor_type TEXT, 
	samp_store_loc TEXT, 
	secondary_treatment TEXT, 
	sewage_type TEXT, 
	tertiary_treatment TEXT, 
	wastewater_type TEXT, 
	emulsions TEXT, 
	gaseous_substances TEXT, 
	inorg_particles TEXT, 
	misc_param TEXT, 
	org_particles TEXT, 
	soluble_inorg_mat TEXT, 
	soluble_org_mat TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, biochem_oxygen_dem, chem_oxygen_dem, efficiency_percent, indust_eff_percent, nitrate, phosphate, salinity, samp_store_temp, sludge_retent_time, sodium, tot_nitro, tot_phosphate, chem_administration, pre_treatment, primary_treatment, reactor_type, samp_store_loc, secondary_treatment, sewage_type, tertiary_treatment, wastewater_type, emulsions, gaseous_substances, inorg_particles, misc_param, org_particles, soluble_inorg_mat, soluble_org_mat, suspend_solids, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "wastewater_sludge_MIGS_org" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	biochem_oxygen_dem TEXT, 
	chem_oxygen_dem TEXT, 
	efficiency_percent TEXT, 
	indust_eff_percent TEXT, 
	nitrate TEXT, 
	phosphate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sludge_retent_time TEXT, 
	sodium TEXT, 
	tot_nitro TEXT, 
	tot_phosphate TEXT, 
	chem_administration TEXT, 
	pre_treatment TEXT, 
	primary_treatment TEXT, 
	reactor_type TEXT, 
	samp_store_loc TEXT, 
	secondary_treatment TEXT, 
	sewage_type TEXT, 
	tertiary_treatment TEXT, 
	wastewater_type TEXT, 
	emulsions TEXT, 
	gaseous_substances TEXT, 
	inorg_particles TEXT, 
	misc_param TEXT, 
	org_particles TEXT, 
	soluble_inorg_mat TEXT, 
	soluble_org_mat TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, biochem_oxygen_dem, chem_oxygen_dem, efficiency_percent, indust_eff_percent, nitrate, phosphate, salinity, samp_store_temp, sludge_retent_time, sodium, tot_nitro, tot_phosphate, chem_administration, pre_treatment, primary_treatment, reactor_type, samp_store_loc, secondary_treatment, sewage_type, tertiary_treatment, wastewater_type, emulsions, gaseous_substances, inorg_particles, misc_param, org_particles, soluble_inorg_mat, soluble_org_mat, suspend_solids, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "wastewater_sludge_MIGS_plant" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	biochem_oxygen_dem TEXT, 
	chem_oxygen_dem TEXT, 
	efficiency_percent TEXT, 
	indust_eff_percent TEXT, 
	nitrate TEXT, 
	phosphate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sludge_retent_time TEXT, 
	sodium TEXT, 
	tot_nitro TEXT, 
	tot_phosphate TEXT, 
	chem_administration TEXT, 
	pre_treatment TEXT, 
	primary_treatment TEXT, 
	reactor_type TEXT, 
	samp_store_loc TEXT, 
	secondary_treatment TEXT, 
	sewage_type TEXT, 
	tertiary_treatment TEXT, 
	wastewater_type TEXT, 
	emulsions TEXT, 
	gaseous_substances TEXT, 
	inorg_particles TEXT, 
	misc_param TEXT, 
	org_particles TEXT, 
	soluble_inorg_mat TEXT, 
	soluble_org_mat TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, biochem_oxygen_dem, chem_oxygen_dem, efficiency_percent, indust_eff_percent, nitrate, phosphate, salinity, samp_store_temp, sludge_retent_time, sodium, tot_nitro, tot_phosphate, chem_administration, pre_treatment, primary_treatment, reactor_type, samp_store_loc, secondary_treatment, sewage_type, tertiary_treatment, wastewater_type, emulsions, gaseous_substances, inorg_particles, misc_param, org_particles, soluble_inorg_mat, soluble_org_mat, suspend_solids, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "wastewater_sludge_MIGS_virus" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	biochem_oxygen_dem TEXT, 
	chem_oxygen_dem TEXT, 
	efficiency_percent TEXT, 
	indust_eff_percent TEXT, 
	nitrate TEXT, 
	phosphate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sludge_retent_time TEXT, 
	sodium TEXT, 
	tot_nitro TEXT, 
	tot_phosphate TEXT, 
	chem_administration TEXT, 
	pre_treatment TEXT, 
	primary_treatment TEXT, 
	reactor_type TEXT, 
	samp_store_loc TEXT, 
	secondary_treatment TEXT, 
	sewage_type TEXT, 
	tertiary_treatment TEXT, 
	wastewater_type TEXT, 
	emulsions TEXT, 
	gaseous_substances TEXT, 
	inorg_particles TEXT, 
	misc_param TEXT, 
	org_particles TEXT, 
	soluble_inorg_mat TEXT, 
	soluble_org_mat TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, biochem_oxygen_dem, chem_oxygen_dem, efficiency_percent, indust_eff_percent, nitrate, phosphate, salinity, samp_store_temp, sludge_retent_time, sodium, tot_nitro, tot_phosphate, chem_administration, pre_treatment, primary_treatment, reactor_type, samp_store_loc, secondary_treatment, sewage_type, tertiary_treatment, wastewater_type, emulsions, gaseous_substances, inorg_particles, misc_param, org_particles, soluble_inorg_mat, soluble_org_mat, suspend_solids, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "wastewater_sludge_MIMAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	biochem_oxygen_dem TEXT, 
	chem_oxygen_dem TEXT, 
	efficiency_percent TEXT, 
	indust_eff_percent TEXT, 
	nitrate TEXT, 
	phosphate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sludge_retent_time TEXT, 
	sodium TEXT, 
	tot_nitro TEXT, 
	tot_phosphate TEXT, 
	chem_administration TEXT, 
	pre_treatment TEXT, 
	primary_treatment TEXT, 
	reactor_type TEXT, 
	samp_store_loc TEXT, 
	secondary_treatment TEXT, 
	sewage_type TEXT, 
	tertiary_treatment TEXT, 
	wastewater_type TEXT, 
	emulsions TEXT, 
	gaseous_substances TEXT, 
	inorg_particles TEXT, 
	misc_param TEXT, 
	org_particles TEXT, 
	soluble_inorg_mat TEXT, 
	soluble_org_mat TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, biochem_oxygen_dem, chem_oxygen_dem, efficiency_percent, indust_eff_percent, nitrate, phosphate, salinity, samp_store_temp, sludge_retent_time, sodium, tot_nitro, tot_phosphate, chem_administration, pre_treatment, primary_treatment, reactor_type, samp_store_loc, secondary_treatment, sewage_type, tertiary_treatment, wastewater_type, emulsions, gaseous_substances, inorg_particles, misc_param, org_particles, soluble_inorg_mat, soluble_org_mat, suspend_solids, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "wastewater_sludge_MIMARKS_specimen" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	biochem_oxygen_dem TEXT, 
	chem_oxygen_dem TEXT, 
	efficiency_percent TEXT, 
	indust_eff_percent TEXT, 
	nitrate TEXT, 
	phosphate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sludge_retent_time TEXT, 
	sodium TEXT, 
	tot_nitro TEXT, 
	tot_phosphate TEXT, 
	chem_administration TEXT, 
	pre_treatment TEXT, 
	primary_treatment TEXT, 
	reactor_type TEXT, 
	samp_store_loc TEXT, 
	secondary_treatment TEXT, 
	sewage_type TEXT, 
	tertiary_treatment TEXT, 
	wastewater_type TEXT, 
	emulsions TEXT, 
	gaseous_substances TEXT, 
	inorg_particles TEXT, 
	misc_param TEXT, 
	org_particles TEXT, 
	soluble_inorg_mat TEXT, 
	soluble_org_mat TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, biochem_oxygen_dem, chem_oxygen_dem, efficiency_percent, indust_eff_percent, nitrate, phosphate, salinity, samp_store_temp, sludge_retent_time, sodium, tot_nitro, tot_phosphate, chem_administration, pre_treatment, primary_treatment, reactor_type, samp_store_loc, secondary_treatment, sewage_type, tertiary_treatment, wastewater_type, emulsions, gaseous_substances, inorg_particles, misc_param, org_particles, soluble_inorg_mat, soluble_org_mat, suspend_solids, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "wastewater_sludge_MIMARKS_survey" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	biochem_oxygen_dem TEXT, 
	chem_oxygen_dem TEXT, 
	efficiency_percent TEXT, 
	indust_eff_percent TEXT, 
	nitrate TEXT, 
	phosphate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sludge_retent_time TEXT, 
	sodium TEXT, 
	tot_nitro TEXT, 
	tot_phosphate TEXT, 
	chem_administration TEXT, 
	pre_treatment TEXT, 
	primary_treatment TEXT, 
	reactor_type TEXT, 
	samp_store_loc TEXT, 
	secondary_treatment TEXT, 
	sewage_type TEXT, 
	tertiary_treatment TEXT, 
	wastewater_type TEXT, 
	emulsions TEXT, 
	gaseous_substances TEXT, 
	inorg_particles TEXT, 
	misc_param TEXT, 
	org_particles TEXT, 
	soluble_inorg_mat TEXT, 
	soluble_org_mat TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, biochem_oxygen_dem, chem_oxygen_dem, efficiency_percent, indust_eff_percent, nitrate, phosphate, salinity, samp_store_temp, sludge_retent_time, sodium, tot_nitro, tot_phosphate, chem_administration, pre_treatment, primary_treatment, reactor_type, samp_store_loc, secondary_treatment, sewage_type, tertiary_treatment, wastewater_type, emulsions, gaseous_substances, inorg_particles, misc_param, org_particles, soluble_inorg_mat, soluble_org_mat, suspend_solids, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "wastewater_sludge_MIMS" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	biochem_oxygen_dem TEXT, 
	chem_oxygen_dem TEXT, 
	efficiency_percent TEXT, 
	indust_eff_percent TEXT, 
	nitrate TEXT, 
	phosphate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sludge_retent_time TEXT, 
	sodium TEXT, 
	tot_nitro TEXT, 
	tot_phosphate TEXT, 
	chem_administration TEXT, 
	pre_treatment TEXT, 
	primary_treatment TEXT, 
	reactor_type TEXT, 
	samp_store_loc TEXT, 
	secondary_treatment TEXT, 
	sewage_type TEXT, 
	tertiary_treatment TEXT, 
	wastewater_type TEXT, 
	emulsions TEXT, 
	gaseous_substances TEXT, 
	inorg_particles TEXT, 
	misc_param TEXT, 
	org_particles TEXT, 
	soluble_inorg_mat TEXT, 
	soluble_org_mat TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, biochem_oxygen_dem, chem_oxygen_dem, efficiency_percent, indust_eff_percent, nitrate, phosphate, salinity, samp_store_temp, sludge_retent_time, sodium, tot_nitro, tot_phosphate, chem_administration, pre_treatment, primary_treatment, reactor_type, samp_store_loc, secondary_treatment, sewage_type, tertiary_treatment, wastewater_type, emulsions, gaseous_substances, inorg_particles, misc_param, org_particles, soluble_inorg_mat, soluble_org_mat, suspend_solids, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "wastewater_sludge_MISAG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	biochem_oxygen_dem TEXT, 
	chem_oxygen_dem TEXT, 
	efficiency_percent TEXT, 
	indust_eff_percent TEXT, 
	nitrate TEXT, 
	phosphate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sludge_retent_time TEXT, 
	sodium TEXT, 
	tot_nitro TEXT, 
	tot_phosphate TEXT, 
	chem_administration TEXT, 
	pre_treatment TEXT, 
	primary_treatment TEXT, 
	reactor_type TEXT, 
	samp_store_loc TEXT, 
	secondary_treatment TEXT, 
	sewage_type TEXT, 
	tertiary_treatment TEXT, 
	wastewater_type TEXT, 
	emulsions TEXT, 
	gaseous_substances TEXT, 
	inorg_particles TEXT, 
	misc_param TEXT, 
	org_particles TEXT, 
	soluble_inorg_mat TEXT, 
	soluble_org_mat TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, biochem_oxygen_dem, chem_oxygen_dem, efficiency_percent, indust_eff_percent, nitrate, phosphate, salinity, samp_store_temp, sludge_retent_time, sodium, tot_nitro, tot_phosphate, chem_administration, pre_treatment, primary_treatment, reactor_type, samp_store_loc, secondary_treatment, sewage_type, tertiary_treatment, wastewater_type, emulsions, gaseous_substances, inorg_particles, misc_param, org_particles, soluble_inorg_mat, soluble_org_mat, suspend_solids, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "wastewater_sludge_MIUVIG" (
	lat_lon TEXT, 
	depth TEXT, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	biochem_oxygen_dem TEXT, 
	chem_oxygen_dem TEXT, 
	efficiency_percent TEXT, 
	indust_eff_percent TEXT, 
	nitrate TEXT, 
	phosphate TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	sludge_retent_time TEXT, 
	sodium TEXT, 
	tot_nitro TEXT, 
	tot_phosphate TEXT, 
	chem_administration TEXT, 
	pre_treatment TEXT, 
	primary_treatment TEXT, 
	reactor_type TEXT, 
	samp_store_loc TEXT, 
	secondary_treatment TEXT, 
	sewage_type TEXT, 
	tertiary_treatment TEXT, 
	wastewater_type TEXT, 
	emulsions TEXT, 
	gaseous_substances TEXT, 
	inorg_particles TEXT, 
	misc_param TEXT, 
	org_particles TEXT, 
	soluble_inorg_mat TEXT, 
	soluble_org_mat TEXT, 
	suspend_solids TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, samp_store_dur, ph, alkalinity, biochem_oxygen_dem, chem_oxygen_dem, efficiency_percent, indust_eff_percent, nitrate, phosphate, salinity, samp_store_temp, sludge_retent_time, sodium, tot_nitro, tot_phosphate, chem_administration, pre_treatment, primary_treatment, reactor_type, samp_store_loc, secondary_treatment, sewage_type, tertiary_treatment, wastewater_type, emulsions, gaseous_substances, inorg_particles, misc_param, org_particles, soluble_inorg_mat, soluble_org_mat, suspend_solids, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE water (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bac_prod TEXT, 
	bac_resp TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	conduc TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_nitro TEXT, 
	diss_inorg_phosp TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	down_par TEXT, 
	fluor TEXT, 
	glucosidase_act TEXT, 
	light_intensity TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	part_org_nitro TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	photon_flux TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	primary_prod TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	sodium TEXT, 
	soluble_react_phosp TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	suspend_part_matter TEXT, 
	tot_depth_water_col TEXT, 
	tot_diss_nitro TEXT, 
	tot_inorg_nitro TEXT, 
	tot_nitro TEXT, 
	tot_part_carb TEXT, 
	tot_phosp TEXT, 
	turbidity TEXT, 
	water_current TEXT, 
	carb_nitro_ratio TEXT, 
	chem_administration TEXT, 
	alkalinity_method TEXT, 
	samp_store_loc TEXT, 
	atmospheric_data TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, tidal_stage, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bac_prod, bac_resp, bacteria_carb_prod, bishomohopanol, bromide, calcium, chloride, chlorophyll, conduc, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_inorg_nitro, diss_inorg_phosp, diss_org_carb, diss_org_nitro, diss_oxygen, down_par, fluor, glucosidase_act, light_intensity, magnesium, mean_frict_vel, mean_peak_frict_vel, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, part_org_nitro, petroleum_hydrocarb, phosphate, photon_flux, potassium, pressure, primary_prod, redox_potential, salinity, samp_store_temp, silicate, size_frac_low, size_frac_up, sodium, soluble_react_phosp, sulfate, sulfide, suspend_part_matter, tot_depth_water_col, tot_diss_nitro, tot_inorg_nitro, tot_nitro, tot_part_carb, tot_phosp, turbidity, water_current, carb_nitro_ratio, chem_administration, alkalinity_method, samp_store_loc, atmospheric_data, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name)
);

CREATE TABLE "water_MIGS_bacteria" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bac_prod TEXT, 
	bac_resp TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	conduc TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_nitro TEXT, 
	diss_inorg_phosp TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	down_par TEXT, 
	fluor TEXT, 
	glucosidase_act TEXT, 
	light_intensity TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	part_org_nitro TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	photon_flux TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	primary_prod TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	sodium TEXT, 
	soluble_react_phosp TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	suspend_part_matter TEXT, 
	tot_depth_water_col TEXT, 
	tot_diss_nitro TEXT, 
	tot_inorg_nitro TEXT, 
	tot_nitro TEXT, 
	tot_part_carb TEXT, 
	tot_phosp TEXT, 
	turbidity TEXT, 
	water_current TEXT, 
	carb_nitro_ratio TEXT, 
	chem_administration TEXT, 
	alkalinity_method TEXT, 
	samp_store_loc TEXT, 
	atmospheric_data TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER NOT NULL, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT NOT NULL, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, tidal_stage, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bac_prod, bac_resp, bacteria_carb_prod, bishomohopanol, bromide, calcium, chloride, chlorophyll, conduc, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_inorg_nitro, diss_inorg_phosp, diss_org_carb, diss_org_nitro, diss_oxygen, down_par, fluor, glucosidase_act, light_intensity, magnesium, mean_frict_vel, mean_peak_frict_vel, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, part_org_nitro, petroleum_hydrocarb, phosphate, photon_flux, potassium, pressure, primary_prod, redox_potential, salinity, samp_store_temp, silicate, size_frac_low, size_frac_up, sodium, soluble_react_phosp, sulfate, sulfide, suspend_part_matter, tot_depth_water_col, tot_diss_nitro, tot_inorg_nitro, tot_nitro, tot_part_carb, tot_phosp, turbidity, water_current, carb_nitro_ratio, chem_administration, alkalinity_method, samp_store_loc, atmospheric_data, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "water_MIGS_eukaryote" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bac_prod TEXT, 
	bac_resp TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	conduc TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_nitro TEXT, 
	diss_inorg_phosp TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	down_par TEXT, 
	fluor TEXT, 
	glucosidase_act TEXT, 
	light_intensity TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	part_org_nitro TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	photon_flux TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	primary_prod TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	sodium TEXT, 
	soluble_react_phosp TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	suspend_part_matter TEXT, 
	tot_depth_water_col TEXT, 
	tot_diss_nitro TEXT, 
	tot_inorg_nitro TEXT, 
	tot_nitro TEXT, 
	tot_part_carb TEXT, 
	tot_phosp TEXT, 
	turbidity TEXT, 
	water_current TEXT, 
	carb_nitro_ratio TEXT, 
	chem_administration TEXT, 
	alkalinity_method TEXT, 
	samp_store_loc TEXT, 
	atmospheric_data TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, tidal_stage, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bac_prod, bac_resp, bacteria_carb_prod, bishomohopanol, bromide, calcium, chloride, chlorophyll, conduc, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_inorg_nitro, diss_inorg_phosp, diss_org_carb, diss_org_nitro, diss_oxygen, down_par, fluor, glucosidase_act, light_intensity, magnesium, mean_frict_vel, mean_peak_frict_vel, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, part_org_nitro, petroleum_hydrocarb, phosphate, photon_flux, potassium, pressure, primary_prod, redox_potential, salinity, samp_store_temp, silicate, size_frac_low, size_frac_up, sodium, soluble_react_phosp, sulfate, sulfide, suspend_part_matter, tot_depth_water_col, tot_diss_nitro, tot_inorg_nitro, tot_nitro, tot_part_carb, tot_phosp, turbidity, water_current, carb_nitro_ratio, chem_administration, alkalinity_method, samp_store_loc, atmospheric_data, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "water_MIGS_org" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bac_prod TEXT, 
	bac_resp TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	conduc TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_nitro TEXT, 
	diss_inorg_phosp TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	down_par TEXT, 
	fluor TEXT, 
	glucosidase_act TEXT, 
	light_intensity TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	part_org_nitro TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	photon_flux TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	primary_prod TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	sodium TEXT, 
	soluble_react_phosp TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	suspend_part_matter TEXT, 
	tot_depth_water_col TEXT, 
	tot_diss_nitro TEXT, 
	tot_inorg_nitro TEXT, 
	tot_nitro TEXT, 
	tot_part_carb TEXT, 
	tot_phosp TEXT, 
	turbidity TEXT, 
	water_current TEXT, 
	carb_nitro_ratio TEXT, 
	chem_administration TEXT, 
	alkalinity_method TEXT, 
	samp_store_loc TEXT, 
	atmospheric_data TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, tidal_stage, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bac_prod, bac_resp, bacteria_carb_prod, bishomohopanol, bromide, calcium, chloride, chlorophyll, conduc, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_inorg_nitro, diss_inorg_phosp, diss_org_carb, diss_org_nitro, diss_oxygen, down_par, fluor, glucosidase_act, light_intensity, magnesium, mean_frict_vel, mean_peak_frict_vel, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, part_org_nitro, petroleum_hydrocarb, phosphate, photon_flux, potassium, pressure, primary_prod, redox_potential, salinity, samp_store_temp, silicate, size_frac_low, size_frac_up, sodium, soluble_react_phosp, sulfate, sulfide, suspend_part_matter, tot_depth_water_col, tot_diss_nitro, tot_inorg_nitro, tot_nitro, tot_part_carb, tot_phosp, turbidity, water_current, carb_nitro_ratio, chem_administration, alkalinity_method, samp_store_loc, atmospheric_data, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "water_MIGS_plant" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bac_prod TEXT, 
	bac_resp TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	conduc TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_nitro TEXT, 
	diss_inorg_phosp TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	down_par TEXT, 
	fluor TEXT, 
	glucosidase_act TEXT, 
	light_intensity TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	part_org_nitro TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	photon_flux TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	primary_prod TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	sodium TEXT, 
	soluble_react_phosp TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	suspend_part_matter TEXT, 
	tot_depth_water_col TEXT, 
	tot_diss_nitro TEXT, 
	tot_inorg_nitro TEXT, 
	tot_nitro TEXT, 
	tot_part_carb TEXT, 
	tot_phosp TEXT, 
	turbidity TEXT, 
	water_current TEXT, 
	carb_nitro_ratio TEXT, 
	chem_administration TEXT, 
	alkalinity_method TEXT, 
	samp_store_loc TEXT, 
	atmospheric_data TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, tidal_stage, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bac_prod, bac_resp, bacteria_carb_prod, bishomohopanol, bromide, calcium, chloride, chlorophyll, conduc, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_inorg_nitro, diss_inorg_phosp, diss_org_carb, diss_org_nitro, diss_oxygen, down_par, fluor, glucosidase_act, light_intensity, magnesium, mean_frict_vel, mean_peak_frict_vel, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, part_org_nitro, petroleum_hydrocarb, phosphate, photon_flux, potassium, pressure, primary_prod, redox_potential, salinity, samp_store_temp, silicate, size_frac_low, size_frac_up, sodium, soluble_react_phosp, sulfate, sulfide, suspend_part_matter, tot_depth_water_col, tot_diss_nitro, tot_inorg_nitro, tot_nitro, tot_part_carb, tot_phosp, turbidity, water_current, carb_nitro_ratio, chem_administration, alkalinity_method, samp_store_loc, atmospheric_data, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "water_MIGS_virus" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bac_prod TEXT, 
	bac_resp TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	conduc TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_nitro TEXT, 
	diss_inorg_phosp TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	down_par TEXT, 
	fluor TEXT, 
	glucosidase_act TEXT, 
	light_intensity TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	part_org_nitro TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	photon_flux TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	primary_prod TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	sodium TEXT, 
	soluble_react_phosp TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	suspend_part_matter TEXT, 
	tot_depth_water_col TEXT, 
	tot_diss_nitro TEXT, 
	tot_inorg_nitro TEXT, 
	tot_nitro TEXT, 
	tot_part_carb TEXT, 
	tot_phosp TEXT, 
	turbidity TEXT, 
	water_current TEXT, 
	carb_nitro_ratio TEXT, 
	chem_administration TEXT, 
	alkalinity_method TEXT, 
	samp_store_loc TEXT, 
	atmospheric_data TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT NOT NULL, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, tidal_stage, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bac_prod, bac_resp, bacteria_carb_prod, bishomohopanol, bromide, calcium, chloride, chlorophyll, conduc, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_inorg_nitro, diss_inorg_phosp, diss_org_carb, diss_org_nitro, diss_oxygen, down_par, fluor, glucosidase_act, light_intensity, magnesium, mean_frict_vel, mean_peak_frict_vel, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, part_org_nitro, petroleum_hydrocarb, phosphate, photon_flux, potassium, pressure, primary_prod, redox_potential, salinity, samp_store_temp, silicate, size_frac_low, size_frac_up, sodium, soluble_react_phosp, sulfate, sulfide, suspend_part_matter, tot_depth_water_col, tot_diss_nitro, tot_inorg_nitro, tot_nitro, tot_part_carb, tot_phosp, turbidity, water_current, carb_nitro_ratio, chem_administration, alkalinity_method, samp_store_loc, atmospheric_data, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "water_MIMAG" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bac_prod TEXT, 
	bac_resp TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	conduc TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_nitro TEXT, 
	diss_inorg_phosp TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	down_par TEXT, 
	fluor TEXT, 
	glucosidase_act TEXT, 
	light_intensity TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	part_org_nitro TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	photon_flux TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	primary_prod TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	sodium TEXT, 
	soluble_react_phosp TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	suspend_part_matter TEXT, 
	tot_depth_water_col TEXT, 
	tot_diss_nitro TEXT, 
	tot_inorg_nitro TEXT, 
	tot_nitro TEXT, 
	tot_part_carb TEXT, 
	tot_phosp TEXT, 
	turbidity TEXT, 
	water_current TEXT, 
	carb_nitro_ratio TEXT, 
	chem_administration TEXT, 
	alkalinity_method TEXT, 
	samp_store_loc TEXT, 
	atmospheric_data TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15) NOT NULL, 
	bin_software TEXT NOT NULL, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, tidal_stage, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bac_prod, bac_resp, bacteria_carb_prod, bishomohopanol, bromide, calcium, chloride, chlorophyll, conduc, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_inorg_nitro, diss_inorg_phosp, diss_org_carb, diss_org_nitro, diss_oxygen, down_par, fluor, glucosidase_act, light_intensity, magnesium, mean_frict_vel, mean_peak_frict_vel, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, part_org_nitro, petroleum_hydrocarb, phosphate, photon_flux, potassium, pressure, primary_prod, redox_potential, salinity, samp_store_temp, silicate, size_frac_low, size_frac_up, sodium, soluble_react_phosp, sulfate, sulfide, suspend_part_matter, tot_depth_water_col, tot_diss_nitro, tot_inorg_nitro, tot_nitro, tot_part_carb, tot_phosp, turbidity, water_current, carb_nitro_ratio, chem_administration, alkalinity_method, samp_store_loc, atmospheric_data, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "water_MIMARKS_specimen" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bac_prod TEXT, 
	bac_resp TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	conduc TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_nitro TEXT, 
	diss_inorg_phosp TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	down_par TEXT, 
	fluor TEXT, 
	glucosidase_act TEXT, 
	light_intensity TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	part_org_nitro TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	photon_flux TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	primary_prod TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	sodium TEXT, 
	soluble_react_phosp TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	suspend_part_matter TEXT, 
	tot_depth_water_col TEXT, 
	tot_diss_nitro TEXT, 
	tot_inorg_nitro TEXT, 
	tot_nitro TEXT, 
	tot_part_carb TEXT, 
	tot_phosp TEXT, 
	turbidity TEXT, 
	water_current TEXT, 
	carb_nitro_ratio TEXT, 
	chem_administration TEXT, 
	alkalinity_method TEXT, 
	samp_store_loc TEXT, 
	atmospheric_data TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, tidal_stage, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bac_prod, bac_resp, bacteria_carb_prod, bishomohopanol, bromide, calcium, chloride, chlorophyll, conduc, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_inorg_nitro, diss_inorg_phosp, diss_org_carb, diss_org_nitro, diss_oxygen, down_par, fluor, glucosidase_act, light_intensity, magnesium, mean_frict_vel, mean_peak_frict_vel, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, part_org_nitro, petroleum_hydrocarb, phosphate, photon_flux, potassium, pressure, primary_prod, redox_potential, salinity, samp_store_temp, silicate, size_frac_low, size_frac_up, sodium, soluble_react_phosp, sulfate, sulfide, suspend_part_matter, tot_depth_water_col, tot_diss_nitro, tot_inorg_nitro, tot_nitro, tot_part_carb, tot_phosp, turbidity, water_current, carb_nitro_ratio, chem_administration, alkalinity_method, samp_store_loc, atmospheric_data, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "water_MIMARKS_survey" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bac_prod TEXT, 
	bac_resp TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	conduc TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_nitro TEXT, 
	diss_inorg_phosp TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	down_par TEXT, 
	fluor TEXT, 
	glucosidase_act TEXT, 
	light_intensity TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	part_org_nitro TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	photon_flux TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	primary_prod TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	sodium TEXT, 
	soluble_react_phosp TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	suspend_part_matter TEXT, 
	tot_depth_water_col TEXT, 
	tot_diss_nitro TEXT, 
	tot_inorg_nitro TEXT, 
	tot_nitro TEXT, 
	tot_part_carb TEXT, 
	tot_phosp TEXT, 
	turbidity TEXT, 
	water_current TEXT, 
	carb_nitro_ratio TEXT, 
	chem_administration TEXT, 
	alkalinity_method TEXT, 
	samp_store_loc TEXT, 
	atmospheric_data TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT NOT NULL, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT NOT NULL, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, tidal_stage, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bac_prod, bac_resp, bacteria_carb_prod, bishomohopanol, bromide, calcium, chloride, chlorophyll, conduc, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_inorg_nitro, diss_inorg_phosp, diss_org_carb, diss_org_nitro, diss_oxygen, down_par, fluor, glucosidase_act, light_intensity, magnesium, mean_frict_vel, mean_peak_frict_vel, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, part_org_nitro, petroleum_hydrocarb, phosphate, photon_flux, potassium, pressure, primary_prod, redox_potential, salinity, samp_store_temp, silicate, size_frac_low, size_frac_up, sodium, soluble_react_phosp, sulfate, sulfide, suspend_part_matter, tot_depth_water_col, tot_diss_nitro, tot_inorg_nitro, tot_nitro, tot_part_carb, tot_phosp, turbidity, water_current, carb_nitro_ratio, chem_administration, alkalinity_method, samp_store_loc, atmospheric_data, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "water_MIMS" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bac_prod TEXT, 
	bac_resp TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	conduc TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_nitro TEXT, 
	diss_inorg_phosp TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	down_par TEXT, 
	fluor TEXT, 
	glucosidase_act TEXT, 
	light_intensity TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	part_org_nitro TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	photon_flux TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	primary_prod TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	sodium TEXT, 
	soluble_react_phosp TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	suspend_part_matter TEXT, 
	tot_depth_water_col TEXT, 
	tot_diss_nitro TEXT, 
	tot_inorg_nitro TEXT, 
	tot_nitro TEXT, 
	tot_part_carb TEXT, 
	tot_phosp TEXT, 
	turbidity TEXT, 
	water_current TEXT, 
	carb_nitro_ratio TEXT, 
	chem_administration TEXT, 
	alkalinity_method TEXT, 
	samp_store_loc TEXT, 
	atmospheric_data TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27), 
	assembly_name TEXT, 
	assembly_software TEXT, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, tidal_stage, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bac_prod, bac_resp, bacteria_carb_prod, bishomohopanol, bromide, calcium, chloride, chlorophyll, conduc, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_inorg_nitro, diss_inorg_phosp, diss_org_carb, diss_org_nitro, diss_oxygen, down_par, fluor, glucosidase_act, light_intensity, magnesium, mean_frict_vel, mean_peak_frict_vel, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, part_org_nitro, petroleum_hydrocarb, phosphate, photon_flux, potassium, pressure, primary_prod, redox_potential, salinity, samp_store_temp, silicate, size_frac_low, size_frac_up, sodium, soluble_react_phosp, sulfate, sulfide, suspend_part_matter, tot_depth_water_col, tot_diss_nitro, tot_inorg_nitro, tot_nitro, tot_part_carb, tot_phosp, turbidity, water_current, carb_nitro_ratio, chem_administration, alkalinity_method, samp_store_loc, atmospheric_data, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "water_MISAG" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bac_prod TEXT, 
	bac_resp TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	conduc TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_nitro TEXT, 
	diss_inorg_phosp TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	down_par TEXT, 
	fluor TEXT, 
	glucosidase_act TEXT, 
	light_intensity TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	part_org_nitro TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	photon_flux TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	primary_prod TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	sodium TEXT, 
	soluble_react_phosp TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	suspend_part_matter TEXT, 
	tot_depth_water_col TEXT, 
	tot_diss_nitro TEXT, 
	tot_inorg_nitro TEXT, 
	tot_nitro TEXT, 
	tot_part_carb TEXT, 
	tot_phosp TEXT, 
	turbidity TEXT, 
	water_current TEXT, 
	carb_nitro_ratio TEXT, 
	chem_administration TEXT, 
	alkalinity_method TEXT, 
	samp_store_loc TEXT, 
	atmospheric_data TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42), 
	virus_enrich_appr VARCHAR(25), 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21) NOT NULL, 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16) NOT NULL, 
	compl_software TEXT NOT NULL, 
	compl_appr VARCHAR(15), 
	contam_score TEXT NOT NULL, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28) NOT NULL, 
	single_cell_lysis_appr VARCHAR(11) NOT NULL, 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT NOT NULL, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT, 
	pred_genome_type VARCHAR(15), 
	pred_genome_struc VARCHAR(13), 
	detec_type TEXT, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, tidal_stage, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bac_prod, bac_resp, bacteria_carb_prod, bishomohopanol, bromide, calcium, chloride, chlorophyll, conduc, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_inorg_nitro, diss_inorg_phosp, diss_org_carb, diss_org_nitro, diss_oxygen, down_par, fluor, glucosidase_act, light_intensity, magnesium, mean_frict_vel, mean_peak_frict_vel, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, part_org_nitro, petroleum_hydrocarb, phosphate, photon_flux, potassium, pressure, primary_prod, redox_potential, salinity, samp_store_temp, silicate, size_frac_low, size_frac_up, sodium, soluble_react_phosp, sulfate, sulfide, suspend_part_matter, tot_depth_water_col, tot_diss_nitro, tot_inorg_nitro, tot_nitro, tot_part_carb, tot_phosp, turbidity, water_current, carb_nitro_ratio, chem_administration, alkalinity_method, samp_store_loc, atmospheric_data, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);

CREATE TABLE "water_MIUVIG" (
	lat_lon TEXT, 
	depth TEXT NOT NULL, 
	alt TEXT, 
	elev TEXT, 
	"temp" TEXT, 
	geo_loc_name TEXT, 
	collection_date DATE, 
	env_broad_scale TEXT, 
	env_local_scale TEXT, 
	env_medium TEXT, 
	oxy_stat_samp TEXT, 
	tidal_stage VARCHAR(10), 
	samp_store_dur TEXT, 
	ph FLOAT, 
	alkalinity TEXT, 
	alkyl_diethers TEXT, 
	aminopept_act TEXT, 
	ammonium TEXT, 
	bac_prod TEXT, 
	bac_resp TEXT, 
	bacteria_carb_prod TEXT, 
	bishomohopanol TEXT, 
	bromide TEXT, 
	calcium TEXT, 
	chloride TEXT, 
	chlorophyll TEXT, 
	conduc TEXT, 
	density TEXT, 
	diss_carb_dioxide TEXT, 
	diss_hydrogen TEXT, 
	diss_inorg_carb TEXT, 
	diss_inorg_nitro TEXT, 
	diss_inorg_phosp TEXT, 
	diss_org_carb TEXT, 
	diss_org_nitro TEXT, 
	diss_oxygen TEXT, 
	down_par TEXT, 
	fluor TEXT, 
	glucosidase_act TEXT, 
	light_intensity TEXT, 
	magnesium TEXT, 
	mean_frict_vel TEXT, 
	mean_peak_frict_vel TEXT, 
	nitrate TEXT, 
	nitrite TEXT, 
	nitro TEXT, 
	org_carb TEXT, 
	org_matter TEXT, 
	org_nitro TEXT, 
	part_org_carb TEXT, 
	part_org_nitro TEXT, 
	petroleum_hydrocarb TEXT, 
	phosphate TEXT, 
	photon_flux TEXT, 
	potassium TEXT, 
	pressure TEXT, 
	primary_prod TEXT, 
	redox_potential TEXT, 
	salinity TEXT, 
	samp_store_temp TEXT, 
	silicate TEXT, 
	size_frac_low TEXT, 
	size_frac_up TEXT, 
	sodium TEXT, 
	soluble_react_phosp TEXT, 
	sulfate TEXT, 
	sulfide TEXT, 
	suspend_part_matter TEXT, 
	tot_depth_water_col TEXT, 
	tot_diss_nitro TEXT, 
	tot_inorg_nitro TEXT, 
	tot_nitro TEXT, 
	tot_part_carb TEXT, 
	tot_phosp TEXT, 
	turbidity TEXT, 
	water_current TEXT, 
	carb_nitro_ratio TEXT, 
	chem_administration TEXT, 
	alkalinity_method TEXT, 
	samp_store_loc TEXT, 
	atmospheric_data TEXT, 
	biomass TEXT, 
	diether_lipids TEXT, 
	misc_param TEXT, 
	n_alkanes TEXT, 
	phaeopigments TEXT, 
	phosplipid_fatt_acid TEXT, 
	organism_count TEXT, 
	perturbation TEXT, 
	samp_vol_we_dna_ext TEXT, 
	project_name TEXT NOT NULL, 
	samp_name TEXT NOT NULL, 
	samp_taxon_id TEXT NOT NULL, 
	experimental_factor TEXT, 
	neg_cont_type VARCHAR(23), 
	pos_cont_type TEXT, 
	subspecf_gen_lin TEXT, 
	ploidy TEXT, 
	num_replicons INTEGER, 
	extrachrom_elements INTEGER, 
	estimated_size TEXT, 
	ref_biomaterial TEXT, 
	source_mat_id TEXT, 
	pathogenicity TEXT, 
	biotic_relationship VARCHAR(11), 
	specific_host TEXT, 
	host_spec_range INTEGER, 
	host_disease_stat TEXT, 
	trophic_level VARCHAR(22), 
	propagation TEXT, 
	encoded_traits TEXT, 
	rel_to_oxygen VARCHAR(17), 
	isol_growth_condt TEXT, 
	samp_collec_device TEXT, 
	samp_collec_method TEXT, 
	samp_mat_process TEXT, 
	size_frac TEXT, 
	samp_size TEXT, 
	source_uvig VARCHAR(42) NOT NULL, 
	virus_enrich_appr VARCHAR(25) NOT NULL, 
	nucl_acid_ext TEXT, 
	nucl_acid_amp TEXT, 
	lib_size INTEGER, 
	lib_reads_seqd INTEGER, 
	lib_layout VARCHAR(6), 
	lib_vector TEXT, 
	lib_screen TEXT, 
	target_gene TEXT, 
	target_subfragment TEXT, 
	pcr_primers TEXT, 
	mid TEXT, 
	adapters TEXT, 
	pcr_cond TEXT, 
	seq_meth TEXT NOT NULL, 
	seq_quality_check TEXT, 
	chimera_check TEXT, 
	tax_ident VARCHAR(21), 
	assembly_qual VARCHAR(27) NOT NULL, 
	assembly_name TEXT, 
	assembly_software TEXT NOT NULL, 
	annot TEXT, 
	number_contig INTEGER NOT NULL, 
	feat_pred TEXT, 
	ref_db TEXT, 
	sim_search_meth TEXT, 
	tax_class TEXT, 
	x_16s_recover TEXT, 
	x_16s_recover_software TEXT, 
	trnas INTEGER, 
	trna_ext_software TEXT, 
	compl_score VARCHAR(16), 
	compl_software TEXT, 
	compl_appr VARCHAR(15), 
	contam_score TEXT, 
	contam_screen_input TEXT, 
	contam_screen_param VARCHAR(17), 
	decontam_software VARCHAR(24), 
	sort_tech VARCHAR(28), 
	single_cell_lysis_appr VARCHAR(11), 
	single_cell_lysis_prot TEXT, 
	wga_amp_appr TEXT, 
	wga_amp_kit TEXT, 
	bin_param VARCHAR(15), 
	bin_software TEXT, 
	reassembly_bin TEXT, 
	mag_cov_software VARCHAR(6), 
	vir_ident_software TEXT NOT NULL, 
	pred_genome_type VARCHAR(15) NOT NULL, 
	pred_genome_struc VARCHAR(13) NOT NULL, 
	detec_type TEXT NOT NULL, 
	otu_class_appr TEXT, 
	otu_seq_comp_appr TEXT, 
	otu_db TEXT, 
	host_pred_appr VARCHAR(24), 
	host_pred_est_acc TEXT, 
	associated_resource TEXT, 
	sop TEXT, 
	PRIMARY KEY (lat_lon, depth, alt, elev, "temp", geo_loc_name, collection_date, env_broad_scale, env_local_scale, env_medium, oxy_stat_samp, tidal_stage, samp_store_dur, ph, alkalinity, alkyl_diethers, aminopept_act, ammonium, bac_prod, bac_resp, bacteria_carb_prod, bishomohopanol, bromide, calcium, chloride, chlorophyll, conduc, density, diss_carb_dioxide, diss_hydrogen, diss_inorg_carb, diss_inorg_nitro, diss_inorg_phosp, diss_org_carb, diss_org_nitro, diss_oxygen, down_par, fluor, glucosidase_act, light_intensity, magnesium, mean_frict_vel, mean_peak_frict_vel, nitrate, nitrite, nitro, org_carb, org_matter, org_nitro, part_org_carb, part_org_nitro, petroleum_hydrocarb, phosphate, photon_flux, potassium, pressure, primary_prod, redox_potential, salinity, samp_store_temp, silicate, size_frac_low, size_frac_up, sodium, soluble_react_phosp, sulfate, sulfide, suspend_part_matter, tot_depth_water_col, tot_diss_nitro, tot_inorg_nitro, tot_nitro, tot_part_carb, tot_phosp, turbidity, water_current, carb_nitro_ratio, chem_administration, alkalinity_method, samp_store_loc, atmospheric_data, biomass, diether_lipids, misc_param, n_alkanes, phaeopigments, phosplipid_fatt_acid, organism_count, perturbation, samp_vol_we_dna_ext, project_name, samp_name, samp_taxon_id, experimental_factor, neg_cont_type, pos_cont_type, subspecf_gen_lin, ploidy, num_replicons, extrachrom_elements, estimated_size, ref_biomaterial, source_mat_id, pathogenicity, biotic_relationship, specific_host, host_spec_range, host_disease_stat, trophic_level, propagation, encoded_traits, rel_to_oxygen, isol_growth_condt, samp_collec_device, samp_collec_method, samp_mat_process, size_frac, samp_size, source_uvig, virus_enrich_appr, nucl_acid_ext, nucl_acid_amp, lib_size, lib_reads_seqd, lib_layout, lib_vector, lib_screen, target_gene, target_subfragment, pcr_primers, mid, adapters, pcr_cond, seq_meth, seq_quality_check, chimera_check, tax_ident, assembly_qual, assembly_name, assembly_software, annot, number_contig, feat_pred, ref_db, sim_search_meth, tax_class, x_16s_recover, x_16s_recover_software, trnas, trna_ext_software, compl_score, compl_software, compl_appr, contam_score, contam_screen_input, contam_screen_param, decontam_software, sort_tech, single_cell_lysis_appr, single_cell_lysis_prot, wga_amp_appr, wga_amp_kit, bin_param, bin_software, reassembly_bin, mag_cov_software, vir_ident_software, pred_genome_type, pred_genome_struc, detec_type, otu_class_appr, otu_seq_comp_appr, otu_db, host_pred_appr, host_pred_est_acc, associated_resource, sop)
);
