
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://GenomicsStandardsConsortium.github.io/mixs/schema_diff_v6.0.0_to_9a865a63b/">
      
      
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.18">
    
    
      
        <title>MIxS Schema Diff: v6.0.0 (74744ee) to Main Branch (9a865a63b) - mixs</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.7e37652d.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#mixs-schema-diff-v600-74744ee-to-main-branch-9a865a63b" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="mixs" class="md-header__button md-logo" aria-label="mixs" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            mixs
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              MIxS Schema Diff: v6.0.0 (74744ee) to Main Branch (9a865a63b)
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/GenomicsStandardsConsortium/mixs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="mixs" class="md-nav__button md-logo" aria-label="mixs" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    mixs
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GenomicsStandardsConsortium/mixs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Index
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../term_list/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Full term table
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../combinations/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Combinations
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../enumerations/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Enumerations
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#maintenance-note" class="md-nav__link">
    <span class="md-ellipsis">
      Maintenance Note
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#release-timeline" class="md-nav__link">
    <span class="md-ellipsis">
      Release Timeline
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#schema-structure-changes" class="md-nav__link">
    <span class="md-ellipsis">
      Schema Structure Changes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Schema Structure Changes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#v600-structure-modular" class="md-nav__link">
    <span class="md-ellipsis">
      v6.0.0 Structure (Modular)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#main-branch-at-9a865a63b-consolidated" class="md-nav__link">
    <span class="md-ellipsis">
      Main Branch at 9a865a63b (Consolidated)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#root-scalar-differences" class="md-nav__link">
    <span class="md-ellipsis">
      Root Scalar Differences
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Root Scalar Differences">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#added-in-main-branch-9a865a63b" class="md-nav__link">
    <span class="md-ellipsis">
      Added in Main Branch (9a865a63b)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#removed-from-main-branch" class="md-nav__link">
    <span class="md-ellipsis">
      Removed from Main Branch
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#settings-structured-patterns" class="md-nav__link">
    <span class="md-ellipsis">
      Settings (Structured Patterns)
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Settings (Structured Patterns)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#settings-usage-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      Settings Usage Analysis
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prefix-changes" class="md-nav__link">
    <span class="md-ellipsis">
      Prefix Changes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#subset-reorganization" class="md-nav__link">
    <span class="md-ellipsis">
      Subset Reorganization
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Subset Reorganization">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#v600-subsets-structural-all-removed" class="md-nav__link">
    <span class="md-ellipsis">
      v6.0.0 Subsets (Structural - All Removed)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#main-branch-subsets-functionalthematic-all-new" class="md-nav__link">
    <span class="md-ellipsis">
      Main Branch Subsets (Functional/Thematic - All New)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#class-changes" class="md-nav__link">
    <span class="md-ellipsis">
      Class Changes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Class Changes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#naming-transformation" class="md-nav__link">
    <span class="md-ellipsis">
      Naming Transformation
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#new-base-classes-in-main" class="md-nav__link">
    <span class="md-ellipsis">
      New Base Classes in Main
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#removed-classes" class="md-nav__link">
    <span class="md-ellipsis">
      Removed Classes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#class-metadata-fields" class="md-nav__link">
    <span class="md-ellipsis">
      Class Metadata Fields
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inheritance-changes" class="md-nav__link">
    <span class="md-ellipsis">
      Inheritance Changes
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slot-renamingsnormalizations" class="md-nav__link">
    <span class="md-ellipsis">
      Slot Renamings/Normalizations
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Slot Renamings/Normalizations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#new-slot" class="md-nav__link">
    <span class="md-ellipsis">
      New Slot
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#removed-slots" class="md-nav__link">
    <span class="md-ellipsis">
      Removed Slots
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#removed-organizational-fields-refactored-as-subsets" class="md-nav__link">
    <span class="md-ellipsis">
      Removed Organizational Fields (Refactored as Subsets)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slot-to-class-assignment-changes" class="md-nav__link">
    <span class="md-ellipsis">
      Slot-to-Class Assignment Changes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Slot-to-Class Assignment Changes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#general-patterns" class="md-nav__link">
    <span class="md-ellipsis">
      General Patterns
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#checklist-changes" class="md-nav__link">
    <span class="md-ellipsis">
      Checklist Changes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#extension-changes" class="md-nav__link">
    <span class="md-ellipsis">
      Extension Changes
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#extension-use-cases" class="md-nav__link">
    <span class="md-ellipsis">
      Extension Use Cases
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#generating-structured-diffs" class="md-nav__link">
    <span class="md-ellipsis">
      Generating Structured Diffs
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Generating Structured Diffs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#automatic-generation-release-workflow" class="md-nav__link">
    <span class="md-ellipsis">
      Automatic Generation (Release Workflow)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#why-the-formats-differ" class="md-nav__link">
    <span class="md-ellipsis">
      Why the Formats Differ
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#common-pitfall" class="md-nav__link">
    <span class="md-ellipsis">
      Common Pitfall
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#manual-generation-cli" class="md-nav__link">
    <span class="md-ellipsis">
      Manual Generation (CLI)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#helper-commands-for-manual-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      Helper Commands for Manual Analysis
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#related-resources" class="md-nav__link">
    <span class="md-ellipsis">
      Related Resources
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#related-issues" class="md-nav__link">
    <span class="md-ellipsis">
      Related Issues
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="mixs-schema-diff-v600-74744ee-to-main-branch-9a865a63b">MIxS Schema Diff: v6.0.0 (74744ee) to Main Branch (9a865a63b)</h1>
<p>This document compares two specific points in the MIxS schema history:</p>
<ul>
<li><strong>Before:</strong> MIxS v6.0.0, commit <code>74744ee</code>, dated 2022-03-23</li>
<li><strong>After:</strong> Main branch, commit <code>9a865a63b</code>, dated 2025-07-02</li>
</ul>
<p>The "after" commit represents the state of main in anticipation of MIxS v7.0.0. This analysis was originally compiled in <a href="https://github.com/GenomicsStandardsConsortium/mixs/issues/845">GitHub Issue #845</a>. The release workflow now generates structured diffs in <code>assets/diff_results/schema_comparison_results.yaml</code>.</p>
<h2 id="maintenance-note">Maintenance Note</h2>
<p>This document should be regenerated periodically. Consider creating a new diff and curated summary when:</p>
<ul>
<li>A <strong>major release</strong> (e.g., v7.0.0) is published</li>
<li>A <strong>minor release</strong> introduces significant schema changes (new classes, slots, or structural reorganization)</li>
<li>A <strong>significant PR</strong> is merged that substantially changes the schema</li>
</ul>
<p>Patch releases that only fix typos, descriptions, or examples typically do not warrant a new curated summary, though the automated <code>schema_comparison_results.yaml</code> will still capture these changes.</p>
<p>When regenerating, update:
1. The filename to reflect the new comparison endpoints
2. The commit hashes and dates in the header
3. All sections with current analysis</p>
<h2 id="release-timeline">Release Timeline</h2>
<table>
<thead>
<tr>
<th>Version</th>
<th>Date</th>
<th>Tag</th>
</tr>
</thead>
<tbody>
<tr>
<td>MIxS5</td>
<td>Feb 27, 2022</td>
<td>MIxS5</td>
</tr>
<tr>
<td>v6.0.0</td>
<td>Mar 24, 2022</td>
<td>mixs6.0.0</td>
</tr>
<tr>
<td>v6.1.0</td>
<td>Jul 5, 2022</td>
<td>mixs6.1.0</td>
</tr>
<tr>
<td>v6.1.1</td>
<td>Oct 9, 2023</td>
<td>mixs6.1.1</td>
</tr>
<tr>
<td>v6.2.0</td>
<td>Oct 18, 2023</td>
<td>v6.2.0</td>
</tr>
</tbody>
</table>
<h2 id="schema-structure-changes">Schema Structure Changes</h2>
<h3 id="v600-structure-modular">v6.0.0 Structure (Modular)</h3>
<ul>
<li><strong>Schema root:</strong> <code>model/schema/mixs.yaml</code></li>
<li><strong>Commit:</strong> 74744ee (2022-03-23)</li>
<li><strong>Organization:</strong> Individual modules for each environment/checklist</li>
</ul>
<p><strong>Schema files (28 total):</strong>
- agriculture.yaml, air.yaml, built_environment.yaml, checklists.yaml, core.yaml
- food_animal_and_animal_feed.yaml, food_farm_environment.yaml, food_food_production_facility.yaml, food_human_foods.yaml
- host_associated.yaml, human_associated.yaml, human_gut.yaml, human_oral.yaml, human_skin.yaml, human_vaginal.yaml
- hydrocarbon_resources_cores.yaml, hydrocarbon_resources_fluids_swabs.yaml
- microbial_mat_biofilm.yaml, miscellaneous_natural_or_artificial_environment.yaml, mixs.yaml
- plant_associated.yaml, ranges.yaml, sediment.yaml, soil.yaml, symbiont_associated.yaml
- terms.yaml, wastewater_sludge.yaml, water.yaml</p>
<h3 id="main-branch-at-9a865a63b-consolidated">Main Branch at 9a865a63b (Consolidated)</h3>
<ul>
<li><strong>Schema root:</strong> <code>src/mixs/schema/mixs.yaml</code></li>
<li><strong>Commit:</strong> 9a865a63b (2025-07-02)</li>
<li><strong>Organization:</strong> Monolithic schema with deprecated.yaml</li>
</ul>
<p><strong>Schema files (2 total):</strong>
- mixs.yaml
- deprecated.yaml</p>
<h2 id="root-scalar-differences">Root Scalar Differences</h2>
<table>
<thead>
<tr>
<th>Field</th>
<th>v6.0.0 (74744ee)</th>
<th>Main (9a865a63b)</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>default_prefix</code></td>
<td>mixs.vocab</td>
<td>MIXS</td>
</tr>
<tr>
<td><code>description</code></td>
<td>Minimal Information about any Sequence Standard</td>
<td>Extended description with LinkML and GSC references</td>
</tr>
<tr>
<td><code>id</code></td>
<td><a href="http://w3id.org/mixs">http://w3id.org/mixs</a></td>
<td><a href="https://w3id.org/mixs">https://w3id.org/mixs</a></td>
</tr>
<tr>
<td><code>name</code></td>
<td>MIxS</td>
<td>mixs</td>
</tr>
</tbody>
</table>
<h3 id="added-in-main-branch-9a865a63b">Added in Main Branch (9a865a63b)</h3>
<ul>
<li><code>comments</code> - Notes about slot titles associated with multiple slot names</li>
<li><code>source</code> - Reference to mixs_v6.xls spreadsheet</li>
<li><code>version</code> - v6.2.0</li>
<li><code>default_range</code> - string</li>
<li><code>settings</code> - Structured pattern definitions (see below)</li>
</ul>
<h3 id="removed-from-main-branch">Removed from Main Branch</h3>
<ul>
<li><code>types</code> (moved to LinkML defaults)</li>
<li><code>source_file</code></li>
</ul>
<h2 id="settings-structured-patterns">Settings (Structured Patterns)</h2>
<p>Settings were added post-v6.0.0 to build <code>structured_pattern</code>s reminiscent of the "Value syntax" in the MIxS 6 spreadsheet.</p>
<pre><code class="language-yaml"># Frequently used patterns
text: '.*'
float: '[-+]?[0-9]*\.?[0-9]+'
integer: '[1-9][0-9]*'
scientific_float: '[-+]?[0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?'
boolean: '(?:yes|no)'

# Date/time patterns
date_time_stamp: '(\d{4})(-(0[1-9]|1[0-2])(-(0[1-9]|[12]\d|3[01])(T([01]\d|2[0-3]):([0-5]\d):([0-5]\d)(\.\d+)?(Z|([+-][01]\d:[0-5]\d))?)?)?)?$'
duration: 'P(?:(?:\d+D|\d+M(?:\d+D)?|\d+Y(?:\d+M(?:\d+D)?)?)...)'

# Reference patterns
DOI: '^doi:10.\d{2,9}/.*$'
PMID: '^PMID:\d+$'
URL: '^https?:\/\/(?:www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}...'
NCBItaxon_id: 'NCBITaxon:\d+'

# Sequence patterns
dna: '^[ACGT]+$'
ambiguous_nucleotides: '[ACGTRKSYMWBHDVN]+'

# Geographic patterns
lat: '(-?((?:[0-8]?[0-9](?:\.\d{0,8})?)|90))'
lon: '-?[0-9]+(?:\.[0-9]{0,8})?$|^-?(1[0-7]{1,2})'

# Term patterns
termID: '[a-zA-Z]{2,}:[a-zA-Z0-9]\d+'
termLabel: '([^\s-]{1,2}|[^\s-]+.+[^\s-]+)'
unit: '([^\s-]{1,2}|[^\s-]+.+[^\s-]+)'
</code></pre>
<h3 id="settings-usage-analysis">Settings Usage Analysis</h3>
<p><strong>Low usage patterns (1 use each):</strong>
- NCBItaxon_id, adapter_A_DNA_sequence, adapter_B_DNA_sequence
- add_recov_methods, agrochemical_name, amount, boolean
- lat, lon, particulate_matter_name</p>
<p><strong>Unused settings (6 total):</strong>
- adapter, country, dna, region, specific_location, storage_condition_type</p>
<h2 id="prefix-changes">Prefix Changes</h2>
<table>
<thead>
<tr>
<th>Change Type</th>
<th>Prefix</th>
<th>v6.0.0</th>
<th>Main</th>
</tr>
</thead>
<tbody>
<tr>
<td>Removed</td>
<td>mixs.vocab</td>
<td><a href="https://w3id.org/mixs/vocab/">https://w3id.org/mixs/vocab/</a></td>
<td>-</td>
</tr>
<tr>
<td>Removed</td>
<td>MIGS</td>
<td><a href="https://w3id.org/mixs/migs/">https://w3id.org/mixs/migs/</a></td>
<td>-</td>
</tr>
<tr>
<td>Changed</td>
<td>MIXS</td>
<td><a href="https://w3id.org/mixs/terms/">https://w3id.org/mixs/terms/</a></td>
<td><a href="https://w3id.org/mixs/">https://w3id.org/mixs/</a></td>
</tr>
<tr>
<td>Added</td>
<td>SO</td>
<td>-</td>
<td><a href="http://purl.obolibrary.org/obo/SO">http://purl.obolibrary.org/obo/SO</a>_</td>
</tr>
<tr>
<td>Unchanged</td>
<td>linkml, xsd, shex, schema</td>
<td>(same)</td>
<td>(same)</td>
</tr>
</tbody>
</table>
<h2 id="subset-reorganization">Subset Reorganization</h2>
<p>The organizational approach changed completely:</p>
<h3 id="v600-subsets-structural-all-removed">v6.0.0 Subsets (Structural - All Removed)</h3>
<table>
<thead>
<tr>
<th>Subset</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>checklist</td>
<td>A MIxS checklist. These can be combined with packages</td>
</tr>
<tr>
<td>package</td>
<td>A MIxS package. These can be combined with checklists</td>
</tr>
<tr>
<td>checklist_package_combination</td>
<td>A combination of a checklist and a package</td>
</tr>
</tbody>
</table>
<h3 id="main-branch-subsets-functionalthematic-all-new">Main Branch Subsets (Functional/Thematic - All New)</h3>
<table>
<thead>
<tr>
<th>Subset</th>
<th>Usage Count</th>
</tr>
</thead>
<tbody>
<tr>
<td>combination_classes</td>
<td>253</td>
</tr>
<tr>
<td>sequencing</td>
<td>57</td>
</tr>
<tr>
<td>nucleic acid sequence source</td>
<td>25</td>
</tr>
<tr>
<td>environment</td>
<td>10</td>
</tr>
<tr>
<td>investigation</td>
<td>6</td>
</tr>
</tbody>
</table>
<p><strong>Key observation:</strong> The knowledge that a class is a Checklist or Extension is now communicated via <code>is_a</code> relationships, not <code>in_subset</code>.</p>
<h2 id="class-changes">Class Changes</h2>
<h3 id="naming-transformation">Naming Transformation</h3>
<table>
<thead>
<tr>
<th>Aspect</th>
<th>v6.0.0</th>
<th>Main</th>
</tr>
</thead>
<tbody>
<tr>
<td>Total classes</td>
<td>289</td>
<td>290</td>
</tr>
<tr>
<td>Naming style</td>
<td>Space-delimited ("soil MIGS bacteria")</td>
<td>CamelCase ("MigsBaSoil")</td>
</tr>
<tr>
<td>Order</td>
<td>{environment} {checklist}</td>
<td>{checklist}{environment}</td>
</tr>
</tbody>
</table>
<h3 id="new-base-classes-in-main">New Base Classes in Main</h3>
<ul>
<li><code>Checklist</code> - Abstract base for all checklists</li>
<li><code>Extension</code> - Abstract base for all extensions</li>
<li><code>MixsCompliantData</code> - Required for validating collections (CSV files, etc.)</li>
</ul>
<h3 id="removed-classes">Removed Classes</h3>
<ul>
<li><code>core</code></li>
<li><code>quantity value</code></li>
</ul>
<h3 id="class-metadata-fields">Class Metadata Fields</h3>
<table>
<thead>
<tr>
<th>Field</th>
<th>v6.0.0</th>
<th>Main</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>name</code></td>
<td>Yes</td>
<td>Yes</td>
</tr>
<tr>
<td><code>description</code></td>
<td>Yes</td>
<td>Yes</td>
</tr>
<tr>
<td><code>from_schema</code></td>
<td>Yes</td>
<td>Yes</td>
</tr>
<tr>
<td><code>aliases</code></td>
<td>Yes</td>
<td>Yes</td>
</tr>
<tr>
<td><code>mixin</code></td>
<td>Yes</td>
<td>Yes</td>
</tr>
<tr>
<td><code>slot_usage</code></td>
<td>Yes</td>
<td>Yes</td>
</tr>
<tr>
<td><code>title</code></td>
<td>No</td>
<td><strong>Added</strong></td>
</tr>
<tr>
<td><code>is_a</code></td>
<td>No</td>
<td><strong>Added</strong></td>
</tr>
<tr>
<td><code>class_uri</code></td>
<td>No</td>
<td><strong>Added</strong></td>
</tr>
<tr>
<td><code>tree_root</code></td>
<td>No</td>
<td><strong>Added</strong> (to MixsCompliantData)</td>
</tr>
</tbody>
</table>
<h3 id="inheritance-changes">Inheritance Changes</h3>
<ul>
<li><strong>v6.0.0:</strong> Flat structure, no inheritance</li>
<li><strong>Main:</strong> Hierarchical structure with <code>is_a</code> relationships</li>
</ul>
<h2 id="slot-renamingsnormalizations">Slot Renamings/Normalizations</h2>
<table>
<thead>
<tr>
<th>Old Name (v6.0.0)</th>
<th>New Name (Main)</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>air particulate matter concentration</code></td>
<td><code>air_PM_concen</code></td>
</tr>
<tr>
<td><code>assembly_quality</code></td>
<td><code>assembly_qual</code></td>
</tr>
<tr>
<td><code>associated resource</code></td>
<td><code>associated_resource</code></td>
</tr>
<tr>
<td><code>Food_Product_type</code></td>
<td><code>food_product_type</code></td>
</tr>
<tr>
<td><code>Food_source</code></td>
<td><code>food_source</code></td>
</tr>
<tr>
<td><code>host_family_relation</code></td>
<td><code>host_fam_rel</code></td>
</tr>
<tr>
<td><code>host_family_relationship</code></td>
<td><code>host_fam_rel</code></td>
</tr>
<tr>
<td><code>horizon</code></td>
<td><code>soil_horizon</code></td>
</tr>
<tr>
<td><code>host_infra_specific_name</code></td>
<td><code>host_infra_spec_name</code></td>
</tr>
<tr>
<td><code>host_infra_specific_rank</code></td>
<td><code>host_infra_spec_rank</code></td>
</tr>
<tr>
<td><code>microbial_biomass_meth</code></td>
<td><code>micro_biomass_meth</code></td>
</tr>
<tr>
<td><code>non_mineral_nutr_regm</code></td>
<td><code>non_min_nutr_regm</code></td>
</tr>
<tr>
<td><code>previous_land_use_meth</code></td>
<td><code>prev_land_use_meth</code></td>
</tr>
<tr>
<td><code>samp_collec_device</code></td>
<td><code>samp_collect_device</code></td>
</tr>
<tr>
<td><code>samp_collec_method</code></td>
<td><code>samp_collect_method</code></td>
</tr>
<tr>
<td><code>samp_salinity</code></td>
<td><code>salinity</code></td>
</tr>
<tr>
<td><code>samp_stor_dur</code></td>
<td><code>samp_store_dur</code></td>
</tr>
<tr>
<td><code>samp_stor_loc</code></td>
<td><code>samp_store_loc</code></td>
</tr>
<tr>
<td><code>samp_stor_temp</code></td>
<td><code>samp_store_temp</code></td>
</tr>
<tr>
<td><code>sample_name</code></td>
<td><code>samp_name</code></td>
</tr>
<tr>
<td><code>single_cell_lysis_appr</code></td>
<td><code>sc_lysis_approach</code></td>
</tr>
<tr>
<td><code>single_cell_lysis_prot</code></td>
<td><code>sc_lysis_method</code></td>
</tr>
<tr>
<td><code>soil_text_measure</code></td>
<td><code>soil_texture_meth</code></td>
</tr>
<tr>
<td><code>texture_meth</code></td>
<td><code>soil_texture_meth</code></td>
</tr>
<tr>
<td><code>texture</code></td>
<td><code>soil_texture</code></td>
</tr>
<tr>
<td><code>tot_car</code></td>
<td><code>tot_carb</code></td>
</tr>
<tr>
<td><code>tot_n_meth</code></td>
<td><code>tot_nitro_cont_meth</code></td>
</tr>
<tr>
<td><code>tot_nitro</code></td>
<td><code>tot_nitro_content</code></td>
</tr>
<tr>
<td><code>tot_phos</code></td>
<td><code>tot_phosphate</code></td>
</tr>
<tr>
<td><code>x_16s_recover_software</code></td>
<td><code>x16s_recover_software</code></td>
</tr>
<tr>
<td><code>x_16s_recover</code></td>
<td><code>x16s_recover</code></td>
</tr>
</tbody>
</table>
<h3 id="new-slot">New Slot</h3>
<ul>
<li><code>urobiom_sex</code> - New content field</li>
</ul>
<h3 id="removed-slots">Removed Slots</h3>
<ul>
<li><code>salinity_meth</code></li>
<li><code>url</code> (replaced by <code>associated_resource</code>)</li>
</ul>
<h3 id="removed-organizational-fields-refactored-as-subsets">Removed Organizational Fields (Refactored as Subsets)</h3>
<ul>
<li><code>core field</code></li>
<li><code>environment field</code></li>
<li><code>investigation field</code></li>
<li><code>mixs extension field</code></li>
<li><code>nucleic acid sequence source field</code></li>
</ul>
<h2 id="slot-to-class-assignment-changes">Slot-to-Class Assignment Changes</h2>
<h3 id="general-patterns">General Patterns</h3>
<ol>
<li><strong>Core checklists</strong> (MIGS, MIMS, MIMARKS, etc.) all gained: <code>alt</code>, <code>depth</code>, <code>elev</code>, <code>temp</code></li>
<li><strong>Environmental extensions</strong> generally lost core metadata fields (<code>collection_date</code>, <code>env_broad_scale</code>, <code>env_local_scale</code>, <code>env_medium</code>, <code>geo_loc_name</code>, <code>lat_lon</code>) while gaining project-specific fields (<code>project_name</code>, <code>samp_name</code>, <code>samp_vol_we_dna_ext</code>)</li>
<li><strong>Single-cell specific changes</strong> in MISAG/MIUVIG where slot names were updated</li>
</ol>
<h3 id="checklist-changes">Checklist Changes</h3>
<table>
<thead>
<tr>
<th>Class</th>
<th>Added Slots</th>
<th>Removed Slots</th>
</tr>
</thead>
<tbody>
<tr>
<td>MigsEu</td>
<td>alt, depth, elev, temp</td>
<td>-</td>
</tr>
<tr>
<td>MigsBa</td>
<td>alt, depth, elev, temp</td>
<td>-</td>
</tr>
<tr>
<td>MigsPl</td>
<td>alt, depth, elev, temp</td>
<td>-</td>
</tr>
<tr>
<td>MigsVi</td>
<td>alt, depth, elev, temp</td>
<td>-</td>
</tr>
<tr>
<td>MigsOrg</td>
<td>alt, depth, elev, temp</td>
<td>-</td>
</tr>
<tr>
<td>Mims</td>
<td>alt, depth, elev, temp</td>
<td>-</td>
</tr>
<tr>
<td>MimarksS</td>
<td>alt, depth, elev, temp</td>
<td>-</td>
</tr>
<tr>
<td>MimarksC</td>
<td>alt, depth, elev, temp</td>
<td>-</td>
</tr>
<tr>
<td>Misag</td>
<td>alt, depth, elev, sc_lysis_approach, sc_lysis_method, temp, x16s_recover, x16s_recover_software</td>
<td>single_cell_lysis_appr, single_cell_lysis_prot, x_16s_recover, x_16s_recover_software</td>
</tr>
<tr>
<td>Mimag</td>
<td>alt, depth, elev, temp, x16s_recover, x16s_recover_software</td>
<td>x_16s_recover, x_16s_recover_software</td>
</tr>
<tr>
<td>Miuvig</td>
<td>alt, depth, elev, sc_lysis_approach, sc_lysis_method, temp</td>
<td>single_cell_lysis_appr, single_cell_lysis_prot</td>
</tr>
</tbody>
</table>
<h3 id="extension-changes">Extension Changes</h3>
<table>
<thead>
<tr>
<th>Class</th>
<th>Added Slots</th>
<th>Removed Slots</th>
</tr>
</thead>
<tbody>
<tr>
<td>Air</td>
<td>air_PM_concen, project_name, samp_name, samp_vol_we_dna_ext</td>
<td>air_particulate_matter_concentration, collection_date, depth, env_broad_scale, env_local_scale, env_medium, geo_loc_name, lat_lon</td>
</tr>
<tr>
<td>BuiltEnvironment</td>
<td>project_name, samp_name</td>
<td>alt, collection_date, depth, elev, env_broad_scale, env_local_scale, env_medium, geo_loc_name, lat_lon, temp</td>
</tr>
<tr>
<td>HostAssociated</td>
<td>host_disease_stat, host_fam_rel, project_name, samp_name, samp_vol_we_dna_ext</td>
<td>collection_date, env_broad_scale, env_local_scale, env_medium, geo_loc_name, host_family_relation, lat_lon</td>
</tr>
<tr>
<td>Soil</td>
<td>project_name, samp_name, samp_vol_we_dna_ext, soil_texture</td>
<td>alt, collection_date, env_broad_scale, env_local_scale, env_medium, geo_loc_name, lat_lon, salinity_meth, soil_text_measure</td>
</tr>
<tr>
<td>Water</td>
<td>project_name, samp_name, samp_vol_we_dna_ext</td>
<td>alt, collection_date, env_broad_scale, env_local_scale, env_medium, geo_loc_name, lat_lon</td>
</tr>
</tbody>
</table>
<p><em>(See <a href="https://github.com/GenomicsStandardsConsortium/mixs/issues/845">Issue #845</a> for complete extension changes)</em></p>
<h2 id="extension-use-cases">Extension Use Cases</h2>
<p>The main branch adds use case annotations on Extension classes:</p>
<table>
<thead>
<tr>
<th>Extension</th>
<th>Example Use Cases</th>
</tr>
</thead>
<tbody>
<tr>
<td>Agriculture</td>
<td>Agricultural Microbiomes Research Coordination Network; microbiome studies in agricultural sites; eDNA in manure samples</td>
</tr>
<tr>
<td>Air</td>
<td>bioaerosol samples; pathogen load in urban air; aerosols</td>
</tr>
<tr>
<td>BuiltEnvironment</td>
<td>NASA space station sampling; MetaSUB transit system sampling; hospitals; office buildings</td>
</tr>
<tr>
<td>FoodHumanFoods</td>
<td>Microbiome of foods intended for human consumption</td>
</tr>
<tr>
<td>HostAssociated</td>
<td>elephant fecal matter; cat oral cavity</td>
</tr>
<tr>
<td>HumanGut</td>
<td>human stool or fecal samples; samples collected directly from the gut</td>
</tr>
<tr>
<td>PlantAssociated</td>
<td>plant surface swabs; root soil or rhizosphere; plant phenotyping</td>
</tr>
<tr>
<td>Soil</td>
<td>soil collection; island microbiome sampling; farm land or forest floor</td>
</tr>
<tr>
<td>Water</td>
<td>sea or river water; global ocean sampling day</td>
</tr>
</tbody>
</table>
<h2 id="generating-structured-diffs">Generating Structured Diffs</h2>
<h3 id="automatic-generation-release-workflow">Automatic Generation (Release Workflow)</h3>
<p>The structured diff (<code>assets/diff_results/schema_comparison_results.yaml</code>) is <strong>automatically generated</strong> when running the release workflow at <code>.github/workflows/create-release-pr.yaml</code>.</p>
<p><strong>The before/after versions are specified via workflow input parameters:</strong></p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Format</th>
<th>Example</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>version</code></td>
<td>The new version number being released</td>
<td><code>X.Y.Z</code> (no <code>v</code> prefix)</td>
<td><code>6.2.3</code></td>
</tr>
<tr>
<td><code>diff_old</code></td>
<td>The "before" version - a git tag or branch name</td>
<td>Exact git tag or branch</td>
<td><code>v6.2.2</code>, <code>mixs6.0.0</code></td>
</tr>
<tr>
<td><code>diff_new</code></td>
<td>The "after" version - a git tag or branch name</td>
<td>Exact git tag or branch</td>
<td><code>main</code></td>
</tr>
</tbody>
</table>
<p>The workflow automatically constructs the full schema path as:
<code>GenomicsStandardsConsortium/mixs@{diff_old}:src/mixs/schema/mixs.yaml</code></p>
<p><strong>To trigger via GitHub Actions UI:</strong>
1. Go to Actions &gt; "Create Release PR"
2. Click "Run workflow"
3. Fill in the three parameters above</p>
<h3 id="why-the-formats-differ">Why the Formats Differ</h3>
<p>The <code>version</code> and <code>diff_old</code>/<code>diff_new</code> parameters use different formats for different reasons:</p>
<ul>
<li>
<p><strong><code>version</code></strong> is used to update version strings in files like <code>pyproject.toml</code>, <code>CITATION.cff</code>, and <code>.zenodo.json</code>. These files conventionally use bare version numbers without a <code>v</code> prefix (e.g., <code>version = "6.2.3"</code>).</p>
</li>
<li>
<p><strong><code>diff_old</code>/<code>diff_new</code></strong> must match actual git references (tags or branches) exactly. Git tags in this repo use different naming conventions depending on when they were created:</p>
</li>
<li>Pre-v6.2.0 tags: <code>mixs6.0.0</code>, <code>mixs6.1.0</code>, <code>mixs6.1.1</code></li>
<li>Post-v6.2.0 tags: <code>v6.2.0</code>, <code>v6.2.1</code>, <code>v6.2.2</code>, <code>v6.2.3</code></li>
</ul>
<h3 id="common-pitfall">Common Pitfall</h3>
<p>In PR #1102, <code>diff_old</code> was set to <code>6.2.2</code> instead of <code>v6.2.2</code>. Because <code>6.2.2</code> is not a valid git reference, the diff tool fell back to cached data from an older version, producing an incorrect diff.</p>
<p><strong>Always verify:</strong>
1. <code>version</code> = bare number, no <code>v</code> prefix (e.g., <code>6.2.3</code>)
2. <code>diff_old</code> = exact git tag with correct prefix (e.g., <code>v6.2.2</code> not <code>6.2.2</code>)
3. Check that the tag exists: <code>git tag -l "v6.2.2"</code></p>
<p>See <a href="https://github.com/GenomicsStandardsConsortium/mixs/issues/1104">issue #1104</a> for proposed workflow improvements to validate tags before running.</p>
<h3 id="manual-generation-cli">Manual Generation (CLI)</h3>
<p>To regenerate the structured diff locally:</p>
<pre><code class="language-shell">poetry run diff-releases \
    --old &quot;GenomicsStandardsConsortium/mixs@v6.0.0:src/mixs/schema/mixs.yaml&quot; \
    --new &quot;GenomicsStandardsConsortium/mixs@main:src/mixs/schema/mixs.yaml&quot; \
    --output-dir assets/diff_results \
    --mappings-dir assets/between_diff_mappings/6_to_pre_7
</code></pre>
<p><strong>Parameters:</strong>
- <code>--old</code>: Reference to old schema (<code>{repo}@{tag/branch}:{path}</code>)
- <code>--new</code>: Reference to new schema
- <code>--output-dir</code>: Where to write <code>schema_comparison_results.yaml</code> and <code>releases.yaml</code>
- <code>--mappings-dir</code>: Directory containing slot/class/enum name mapping TSVs for renamed elements</p>
<h3 id="helper-commands-for-manual-analysis">Helper Commands for Manual Analysis</h3>
<p><strong>Merge modular v6.0.0 schema files:</strong></p>
<pre><code class="language-shell">poetry run linkml generate linkml \
    --format yaml \
    --no-materialize model/schema/mixs.yaml &gt; mixs_6_0_0_merged.yaml
</code></pre>
<p><strong>Extract element lists for comparison:</strong></p>
<pre><code class="language-shell">yq 'keys' mixs_6_0_0_merged.yaml &gt; mixs_6_0_0_root_keys.txt
yq '.classes | keys' mixs_6_0_0_merged.yaml &gt; mixs_6_0_0_classes.txt
yq '.slots | keys' mixs_6_0_0_merged.yaml &gt; mixs_6_0_0_slots.txt
</code></pre>
<p><strong>Materialize patterns for validation:</strong></p>
<pre><code class="language-shell">poetry run linkml generate linkml \
    --format yaml \
    --no-materialize-attributes \
    --materialize-patterns src/mixs/schema/mixs.yaml &gt; mixs_main_materialized.yaml
</code></pre>
<p><strong>Validate data against schema:</strong></p>
<pre><code class="language-shell">poetry run linkml-validate \
    --schema mixs_main_materialized.yaml \
    --target-class MixsCompliantData your_data.yaml
</code></pre>
<h2 id="related-resources">Related Resources</h2>
<ul>
<li><strong>Structured diff:</strong> <code>assets/diff_results/schema_comparison_results.yaml</code></li>
<li><strong>Release registry:</strong> <code>assets/diff_results/releases.yaml</code></li>
<li><strong>Mapping files:</strong> <code>assets/between_diff_mappings/6_to_pre_7/</code></li>
<li><strong>GitHub Issue:</strong> <a href="https://github.com/GenomicsStandardsConsortium/mixs/issues/845">#845</a></li>
</ul>
<h2 id="related-issues">Related Issues</h2>
<ul>
<li><a href="https://github.com/GenomicsStandardsConsortium/mixs/issues/356">#356</a> - Set up autogenerated release notes</li>
<li><a href="https://github.com/GenomicsStandardsConsortium/mixs/issues/537">#537</a> - Change log to be included with new releases</li>
<li><a href="https://github.com/GenomicsStandardsConsortium/mixs/issues/835">#835</a> - Table of release dates</li>
<li><a href="https://github.com/GenomicsStandardsConsortium/mixs/issues/882">#882</a> - Version update tracking for consumers</li>
<li><a href="https://github.com/GenomicsStandardsConsortium/mixs/issues/1085">#1085</a> - LLM-generated narrative summaries</li>
</ul>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": ["content.tabs.link", "search.suggest"], "search": "../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.92b07e13.min.js"></script>
      
        <script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
      
        <script src="../javascripts/tablesort.js"></script>
      
    
  </body>
</html>